<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>WRAP Data Prep 2025</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/lumen.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.33/datatables.js"></script>
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">My Website</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Build_WRAP_Phenotype_2025.html">WRAP</a>
</li>
<li>
  <a href="Build_PRADI_Phenotype_2025.html">PRADI</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">WRAP Data Prep 2025</h1>

</div>


<p>Author: Mengna Zhang</p>
<p><em>Last updated on:</em> 2025-08-26</p>
<div id="set-path" class="section level1">
<h1>Set Path</h1>
<pre class="r"><code>## directory: can extend to the main CNT folder
directory &lt;- &quot;/Users/&quot;

## your own directory
my_directory &lt;- paste0(directory, &quot;mengnazhang/Desktop/&quot;)

## set WRAP path
wrap_directory &lt;- paste0(my_directory, &quot;ADSP_DataPrep_local/WRAP/Phenotype/2025/Raw/&quot;)

## output path
out_directory &lt;- paste0(my_directory, &quot;ADSP_DataPrep_local/WRAP/Phenotype/2025/Cleaned/&quot;)

## script path
script_directory &lt;- paste0(my_directory, &quot;ADSP_DataPrep_local/WRAP/Phenotype/2025/Scripts/&quot;)</code></pre>
<p><br></p>
</div>
<div id="load-helper-scripts" class="section level1">
<h1>Load Helper Scripts</h1>
<pre class="r"><code>source(paste0(script_directory,&quot;helperScripts_WRAP.R&quot;))</code></pre>
<p><br></p>
</div>
<div id="load-packages" class="section level1">
<h1>Load Packages</h1>
<pre class="r"><code>require(dplyr)
require(tidyr)
require(stringr)</code></pre>
<p><br> <br></p>
</div>
<div id="variable-check-per-file" class="section level1">
<h1>Variable Check Per File</h1>
<div id="abnormal_factor_flags_long" class="section level2">
<h2>Abnormal_Factor_Flags_long</h2>
<p>Indicator variables indicating low scores on FactorScores factors,
adjusting for demographics.</p>
<pre class="r"><code>## load data
Abnormal_Factor_Flags_long &lt;- read.csv(paste0(wrap_directory,&quot;Abnormal_Factor_Flags_long_05Jun2025.csv&quot;))

info(Abnormal_Factor_Flags_long,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:8212, cols:7, inds:1814</code></pre>
<pre class="r"><code>str(Abnormal_Factor_Flags_long)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    8212 obs. of  7 variables:
##  $ wrapnum           : chr  &quot;wrap0609&quot; &quot;wrap0792&quot; &quot;wrap1113&quot; &quot;wrapA0076&quot; ...
##  $ WRAPNo            : chr  &quot;0609&quot; &quot;0792&quot; &quot;1113&quot; &quot;A0076&quot; ...
##  $ VisNo             : int  8 6 1 3 3 3 2 6 5 5 ...
##  $ Low_Mem_Factor_Vis: int  0 9 1 1 1 0 0 0 0 0 ...
##  $ Low_Exe_Factor_Vis: int  9 9 9 9 9 0 0 9 0 0 ...
##  $ Explanation       : chr  &quot;https://kb.wisc.edu/warp/104723&quot; &quot;https://kb.wisc.edu/warp/104723&quot; &quot;https://kb.wisc.edu/warp/104723&quot; &quot;https://kb.wisc.edu/warp/104723&quot; ...
##  $ update_stamp      : chr  &quot;2024-09-13 08:51:30.0&quot; &quot;2024-09-13 08:51:30.0&quot; &quot;2024-09-13 08:51:30.0&quot; &quot;2024-09-13 08:51:30.0&quot; ...</code></pre>
</details>
<pre class="r"><code>## check table frequency for all variables

## drop update_stamp column
Abnormal_Factor_Flags_long &lt;- subset(Abnormal_Factor_Flags_long,select = -c(update_stamp)) ## 6 columns left</code></pre>
<pre class="r"><code>str(Abnormal_Factor_Flags_long)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    8212 obs. of  6 variables:
##  $ wrapnum           : chr  &quot;wrap0609&quot; &quot;wrap0792&quot; &quot;wrap1113&quot; &quot;wrapA0076&quot; ...
##  $ WRAPNo            : chr  &quot;0609&quot; &quot;0792&quot; &quot;1113&quot; &quot;A0076&quot; ...
##  $ VisNo             : int  8 6 1 3 3 3 2 6 5 5 ...
##  $ Low_Mem_Factor_Vis: int  0 9 1 1 1 0 0 0 0 0 ...
##  $ Low_Exe_Factor_Vis: int  9 9 9 9 9 0 0 9 0 0 ...
##  $ Explanation       : chr  &quot;https://kb.wisc.edu/warp/104723&quot; &quot;https://kb.wisc.edu/warp/104723&quot; &quot;https://kb.wisc.edu/warp/104723&quot; &quot;https://kb.wisc.edu/warp/104723&quot; ...</code></pre>
</details>
<p><br> <br></p>
</div>
<div id="abnormal_factor_flags_wide" class="section level2">
<h2>Abnormal_Factor_Flags_wide</h2>
<p>Indicator variables indicating low scores on FactorScores factors,
adjusting for demographics.</p>
<pre class="r"><code>## load data
Abnormal_Factor_Flags_wide &lt;- read.csv(paste0(wrap_directory,&quot;Abnormal_Factor_Flags_wide_05Jun2025.csv&quot;))
info(Abnormal_Factor_Flags_wide,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:1815, cols:6, inds:1815</code></pre>
<pre class="r"><code>str(Abnormal_Factor_Flags_wide)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    1815 obs. of  6 variables:
##  $ wrapnum       : chr  &quot;wrap0543&quot; &quot;wrap0579&quot; &quot;wrap0596&quot; &quot;wrap0671&quot; ...
##  $ WRAPNo        : chr  &quot;0543&quot; &quot;0579&quot; &quot;0596&quot; &quot;0671&quot; ...
##  $ Low_Mem_Factor: int  110019 0 99999 100001 110109 10000 99 99999 1999999 1199 ...
##  $ Low_Exe_Factor: int  1111999 9 99999 9 1101119 9 999 999999 9999999 999 ...
##  $ Explanation   : chr  &quot;https://kb.wisc.edu/warp/104723&quot; &quot;https://kb.wisc.edu/warp/104723&quot; &quot;https://kb.wisc.edu/warp/104723&quot; &quot;https://kb.wisc.edu/warp/104723&quot; ...
##  $ update_stamp  : chr  &quot;2024-09-13 08:51:31.0&quot; &quot;2024-09-13 08:51:31.0&quot; &quot;2024-09-13 08:51:31.0&quot; &quot;2024-09-13 08:51:31.0&quot; ...</code></pre>
</details>
<pre class="r"><code>## check table frequency for all variables

## drop update_stamp column
Abnormal_Factor_Flags_wide &lt;- subset(Abnormal_Factor_Flags_wide,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(Abnormal_Factor_Flags_wide)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    1815 obs. of  5 variables:
##  $ wrapnum       : chr  &quot;wrap0543&quot; &quot;wrap0579&quot; &quot;wrap0596&quot; &quot;wrap0671&quot; ...
##  $ WRAPNo        : chr  &quot;0543&quot; &quot;0579&quot; &quot;0596&quot; &quot;0671&quot; ...
##  $ Low_Mem_Factor: int  110019 0 99999 100001 110109 10000 99 99999 1999999 1199 ...
##  $ Low_Exe_Factor: int  1111999 9 99999 9 1101119 9 999 999999 9999999 999 ...
##  $ Explanation   : chr  &quot;https://kb.wisc.edu/warp/104723&quot; &quot;https://kb.wisc.edu/warp/104723&quot; &quot;https://kb.wisc.edu/warp/104723&quot; &quot;https://kb.wisc.edu/warp/104723&quot; ...</code></pre>
</details>
<p><br> <br></p>
</div>
<div id="animalnaming_itemlevel" class="section level2">
<h2>AnimalNaming_ItemLevel</h2>
<p>Animal Naming Item Level Data</p>
<pre class="r"><code>## load data
AnimalNaming_ItemLevel &lt;- read.csv(paste0(wrap_directory, &quot;AnimalNaming_ItemLevel_05Jun2025.csv&quot;))
info(AnimalNaming_ItemLevel,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:1096, cols:54, inds:944</code></pre>
<pre class="r"><code>str(AnimalNaming_ItemLevel)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    1096 obs. of  54 variables:
##  $ wrapnum     : chr  &quot;wrap0609&quot; &quot;wrap1058&quot; &quot;wrap1113&quot; &quot;wrap1269&quot; ...
##  $ WRAPNo      : chr  &quot;0609&quot; &quot;1058&quot; &quot;1113&quot; &quot;1269&quot; ...
##  $ VisNo       : int  8 3 1 1 7 5 1 6 5 7 ...
##  $ AN_1        : chr  &quot;dog&quot; &quot;dog&quot; &quot;cat&quot; &quot;bear&quot; ...
##  $ AN_2        : chr  &quot;cat&quot; &quot;cat&quot; &quot;dog&quot; &quot;lion&quot; ...
##  $ AN_3        : chr  &quot;kangaroo&quot; &quot;bird&quot; &quot;cow&quot; &quot;tiger&quot; ...
##  $ AN_4        : chr  &quot;llama&quot; &quot;chicken&quot; &quot;pig&quot; &quot;gopher&quot; ...
##  $ AN_5        : chr  &quot;alpaca&quot; &quot;duck&quot; &quot;goat&quot; &quot;squirrel&quot; ...
##  $ AN_6        : chr  &quot;rooster&quot; &quot;goose&quot; &quot;elephant&quot; &quot;giraffe&quot; ...
##  $ AN_7        : chr  &quot;crow&quot; &quot;cow&quot; &quot;giraffe&quot; &quot;monkey&quot; ...
##  $ AN_8        : chr  &quot;tiger&quot; &quot;horse&quot; &quot;tiger&quot; &quot;seal&quot; ...
##  $ AN_9        : chr  &quot;lion&quot; &quot;mule&quot; &quot;lion&quot; &quot;dolphin&quot; ...
##  $ AN_10       : chr  &quot;puma&quot; &quot;weasel&quot; &quot;chimpanzee&quot; &quot;camel&quot; ...
##  $ AN_11       : chr  &quot;leopard&quot; &quot;skunk&quot; &quot;monkey&quot; &quot;dog&quot; ...
##  $ AN_12       : chr  &quot;bear&quot; &quot;rabbit&quot; &quot;gorilla&quot; &quot;cat&quot; ...
##  $ AN_13       : chr  &quot;girzzly bear&quot; &quot;deer&quot; &quot;bear&quot; &quot;chimpanzee&quot; ...
##  $ AN_14       : chr  &quot;brown bear&quot; &quot;moose&quot; &quot;leopard&quot; &quot;aardvark&quot; ...
##  $ AN_15       : chr  &quot;polar bear&quot; &quot;elk&quot; &quot;cougar&quot; &quot;fox&quot; ...
##  $ AN_16       : chr  &quot;spectacled bear&quot; &quot;antelope&quot; &quot;&quot; &quot;&quot; ...
##  $ AN_17       : chr  &quot;ant &quot; &quot;coyote&quot; &quot;&quot; &quot;&quot; ...
##  $ AN_18       : chr  &quot;spider&quot; &quot;bobcat&quot; &quot;&quot; &quot;&quot; ...
##  $ AN_19       : chr  &quot;fly&quot; &quot;cougar&quot; &quot;&quot; &quot;&quot; ...
##  $ AN_20       : chr  &quot;cuckoo&quot; &quot;lion&quot; &quot;&quot; &quot;&quot; ...
##  $ AN_21       : chr  &quot;heron&quot; &quot;tiger&quot; &quot;&quot; &quot;&quot; ...
##  $ AN_22       : chr  &quot;egret&quot; &quot;elephant&quot; &quot;&quot; &quot;&quot; ...
##  $ AN_23       : chr  &quot;crane&quot; &quot;rhinoceros&quot; &quot;&quot; &quot;&quot; ...
##  $ AN_24       : chr  &quot;blue bird&quot; &quot;giraffe&quot; &quot;&quot; &quot;&quot; ...
##  $ AN_25       : chr  &quot;bird&quot; &quot;zebra&quot; &quot;&quot; &quot;&quot; ...
##  $ AN_26       : chr  &quot;robin&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ AN_27       : chr  &quot;sparrow&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ AN_28       : chr  &quot;chickadee&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ AN_29       : chr  &quot;nuthatch&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ AN_30       : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ AN_31       : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ AN_32       : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ AN_33       : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ AN_34       : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ AN_35       : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ AN_36       : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ AN_37       : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ AN_38       : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ AN_39       : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ AN_40       : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ AN_41       : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ AN_42       : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ AN_43       : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ AN_44       : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ AN_45       : logi  NA NA NA NA NA NA ...
##  $ AN_46       : logi  NA NA NA NA NA NA ...
##  $ AN_47       : logi  NA NA NA NA NA NA ...
##  $ AN_48       : logi  NA NA NA NA NA NA ...
##  $ AN_49       : logi  NA NA NA NA NA NA ...
##  $ AN_50       : logi  NA NA NA NA NA NA ...
##  $ update_stamp: chr  &quot;2024-09-13 08:51:33.0&quot; &quot;2024-09-13 08:51:33.0&quot; &quot;2024-09-13 08:51:33.0&quot; &quot;2024-09-13 08:51:33.0&quot; ...</code></pre>
</details>
<pre class="r"><code>## check table frequency for all variables

## convert &quot;&quot; to NA for chr columns
chrcols &lt;- colnames(AnimalNaming_ItemLevel)[sapply(AnimalNaming_ItemLevel, is.character)]
AnimalNaming_ItemLevel[chrcols] &lt;- lapply(AnimalNaming_ItemLevel[chrcols], function(x) ifelse(x == &quot;&quot;, NA, x))

## convert from logi to character for columns: AN_45 - AN_50
## these columns were confirmed to only have NA values in the raw data
logicols &lt;- paste0(&quot;AN_&quot;,seq(45,50))
AnimalNaming_ItemLevel[logicols] &lt;- lapply(AnimalNaming_ItemLevel[logicols], as.character)

## drop update_stamp column
AnimalNaming_ItemLevel &lt;- subset(AnimalNaming_ItemLevel,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(AnimalNaming_ItemLevel)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    1096 obs. of  53 variables:
##  $ wrapnum: chr  &quot;wrap0609&quot; &quot;wrap1058&quot; &quot;wrap1113&quot; &quot;wrap1269&quot; ...
##  $ WRAPNo : chr  &quot;0609&quot; &quot;1058&quot; &quot;1113&quot; &quot;1269&quot; ...
##  $ VisNo  : int  8 3 1 1 7 5 1 6 5 7 ...
##  $ AN_1   : chr  &quot;dog&quot; &quot;dog&quot; &quot;cat&quot; &quot;bear&quot; ...
##  $ AN_2   : chr  &quot;cat&quot; &quot;cat&quot; &quot;dog&quot; &quot;lion&quot; ...
##  $ AN_3   : chr  &quot;kangaroo&quot; &quot;bird&quot; &quot;cow&quot; &quot;tiger&quot; ...
##  $ AN_4   : chr  &quot;llama&quot; &quot;chicken&quot; &quot;pig&quot; &quot;gopher&quot; ...
##  $ AN_5   : chr  &quot;alpaca&quot; &quot;duck&quot; &quot;goat&quot; &quot;squirrel&quot; ...
##  $ AN_6   : chr  &quot;rooster&quot; &quot;goose&quot; &quot;elephant&quot; &quot;giraffe&quot; ...
##  $ AN_7   : chr  &quot;crow&quot; &quot;cow&quot; &quot;giraffe&quot; &quot;monkey&quot; ...
##  $ AN_8   : chr  &quot;tiger&quot; &quot;horse&quot; &quot;tiger&quot; &quot;seal&quot; ...
##  $ AN_9   : chr  &quot;lion&quot; &quot;mule&quot; &quot;lion&quot; &quot;dolphin&quot; ...
##  $ AN_10  : chr  &quot;puma&quot; &quot;weasel&quot; &quot;chimpanzee&quot; &quot;camel&quot; ...
##  $ AN_11  : chr  &quot;leopard&quot; &quot;skunk&quot; &quot;monkey&quot; &quot;dog&quot; ...
##  $ AN_12  : chr  &quot;bear&quot; &quot;rabbit&quot; &quot;gorilla&quot; &quot;cat&quot; ...
##  $ AN_13  : chr  &quot;girzzly bear&quot; &quot;deer&quot; &quot;bear&quot; &quot;chimpanzee&quot; ...
##  $ AN_14  : chr  &quot;brown bear&quot; &quot;moose&quot; &quot;leopard&quot; &quot;aardvark&quot; ...
##  $ AN_15  : chr  &quot;polar bear&quot; &quot;elk&quot; &quot;cougar&quot; &quot;fox&quot; ...
##  $ AN_16  : chr  &quot;spectacled bear&quot; &quot;antelope&quot; NA NA ...
##  $ AN_17  : chr  &quot;ant &quot; &quot;coyote&quot; NA NA ...
##  $ AN_18  : chr  &quot;spider&quot; &quot;bobcat&quot; NA NA ...
##  $ AN_19  : chr  &quot;fly&quot; &quot;cougar&quot; NA NA ...
##  $ AN_20  : chr  &quot;cuckoo&quot; &quot;lion&quot; NA NA ...
##  $ AN_21  : chr  &quot;heron&quot; &quot;tiger&quot; NA NA ...
##  $ AN_22  : chr  &quot;egret&quot; &quot;elephant&quot; NA NA ...
##  $ AN_23  : chr  &quot;crane&quot; &quot;rhinoceros&quot; NA NA ...
##  $ AN_24  : chr  &quot;blue bird&quot; &quot;giraffe&quot; NA NA ...
##  $ AN_25  : chr  &quot;bird&quot; &quot;zebra&quot; NA NA ...
##  $ AN_26  : chr  &quot;robin&quot; NA NA NA ...
##  $ AN_27  : chr  &quot;sparrow&quot; NA NA NA ...
##  $ AN_28  : chr  &quot;chickadee&quot; NA NA NA ...
##  $ AN_29  : chr  &quot;nuthatch&quot; NA NA NA ...
##  $ AN_30  : chr  NA NA NA NA ...
##  $ AN_31  : chr  NA NA NA NA ...
##  $ AN_32  : chr  NA NA NA NA ...
##  $ AN_33  : chr  NA NA NA NA ...
##  $ AN_34  : chr  NA NA NA NA ...
##  $ AN_35  : chr  NA NA NA NA ...
##  $ AN_36  : chr  NA NA NA NA ...
##  $ AN_37  : chr  NA NA NA NA ...
##  $ AN_38  : chr  NA NA NA NA ...
##  $ AN_39  : chr  NA NA NA NA ...
##  $ AN_40  : chr  NA NA NA NA ...
##  $ AN_41  : chr  NA NA NA NA ...
##  $ AN_42  : chr  NA NA NA NA ...
##  $ AN_43  : chr  NA NA NA NA ...
##  $ AN_44  : chr  NA NA NA NA ...
##  $ AN_45  : chr  NA NA NA NA ...
##  $ AN_46  : chr  NA NA NA NA ...
##  $ AN_47  : chr  NA NA NA NA ...
##  $ AN_48  : chr  NA NA NA NA ...
##  $ AN_49  : chr  NA NA NA NA ...
##  $ AN_50  : chr  NA NA NA NA ...</code></pre>
</details>
<p><br> <br></p>
</div>
<div id="apg" class="section level2">
<h2>APG</h2>
<p>APOE genotype</p>
<pre class="r"><code>## load data
APG &lt;- read.csv(paste0(wrap_directory, &quot;APG_05Jun2025.csv&quot;))
info(APG,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:1638, cols:7, inds:1638</code></pre>
<pre class="r"><code>str(APG)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    1638 obs. of  7 variables:
##  $ wrapnum       : chr  &quot;wrap0001&quot; &quot;wrap0002&quot; &quot;wrap0003&quot; &quot;wrap0004&quot; ...
##  $ WRAPNo        : chr  &quot;0001&quot; &quot;0002&quot; &quot;0003&quot; &quot;0004&quot; ...
##  $ all1          : int  2 4 3 3 3 3 3 2 3 3 ...
##  $ all2          : int  4 4 4 4 3 3 4 3 3 4 ...
##  $ APOE_RiskScore: num  1.492 3.252 1.914 1.914 0.695 ...
##  $ Explanation   : chr  &quot;https://kb.wisc.edu/warp/104729&quot; &quot;https://kb.wisc.edu/warp/104729&quot; &quot;https://kb.wisc.edu/warp/104729&quot; &quot;https://kb.wisc.edu/warp/104729&quot; ...
##  $ update_stamp  : chr  &quot;2025-01-24 08:16:30.0&quot; &quot;2025-01-24 08:16:30.0&quot; &quot;2025-01-24 08:16:30.0&quot; &quot;2025-01-24 08:16:30.0&quot; ...</code></pre>
</details>
<pre class="r"><code>## check table frequency for all variables

## drop update_stamp column
APG &lt;- subset(APG,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(APG)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    1638 obs. of  6 variables:
##  $ wrapnum       : chr  &quot;wrap0001&quot; &quot;wrap0002&quot; &quot;wrap0003&quot; &quot;wrap0004&quot; ...
##  $ WRAPNo        : chr  &quot;0001&quot; &quot;0002&quot; &quot;0003&quot; &quot;0004&quot; ...
##  $ all1          : int  2 4 3 3 3 3 3 2 3 3 ...
##  $ all2          : int  4 4 4 4 3 3 4 3 3 4 ...
##  $ APOE_RiskScore: num  1.492 3.252 1.914 1.914 0.695 ...
##  $ Explanation   : chr  &quot;https://kb.wisc.edu/warp/104729&quot; &quot;https://kb.wisc.edu/warp/104729&quot; &quot;https://kb.wisc.edu/warp/104729&quot; &quot;https://kb.wisc.edu/warp/104729&quot; ...</code></pre>
</details>
<p><br> <br></p>
</div>
<div id="avltpositionscores" class="section level2">
<h2>AVLTPositionScores</h2>
<p>Serial position of answers given per trial on Rey AVLT</p>
<pre class="r"><code>## load data
AVLTPositionScores &lt;- read.csv(paste0(wrap_directory, &quot;AVLTPositionScores_05Jun2025.csv&quot;))
info(AVLTPositionScores,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:7946, cols:124, inds:1795</code></pre>
<pre class="r"><code>str(AVLTPositionScores, max.level = 99, list.len = 99999)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    7946 obs. of  124 variables:
##  $ wrapnum     : chr  &quot;wrap0002&quot; &quot;wrap0068&quot; &quot;wrap0146&quot; &quot;wrap0148&quot; ...
##  $ WRAPNo      : chr  &quot;0002&quot; &quot;0068&quot; &quot;0146&quot; &quot;0148&quot; ...
##  $ VisNo       : int  5 2 4 5 5 4 6 4 7 4 ...
##  $ T1.1        : int  15 8 14 1 1 1 15 1 1 1 ...
##  $ T1.2        : int  14 12 15 2 2 5 1 12 10 15 ...
##  $ T1.3        : int  13 14 12 5 9 8 12 8 12 9 ...
##  $ T1.4        : int  12 15 NA 15 12 12 14 9 14 12 ...
##  $ T1.5        : int  8 9 NA 12 11 11 NA 2 8 NA ...
##  $ T1.6        : int  NA 10 NA 14 6 9 NA 3 NA NA ...
##  $ T1.7        : int  NA NA NA NA 5 10 NA NA NA NA ...
##  $ T1.8        : int  NA NA NA NA NA 7 NA NA NA NA ...
##  $ T1.9        : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ T1.10       : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ T1.11       : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ T1.12       : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ T1.13       : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ T1.14       : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ T1.15       : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ T2.1        : int  10 9 15 1 14 14 2 1 2 1 ...
##  $ T2.2        : int  11 10 14 5 15 15 3 12 12 2 ...
##  $ T2.3        : int  8 11 13 15 5 3 15 9 14 11 ...
##  $ T2.4        : int  12 13 12 4 10 8 14 5 8 9 ...
##  $ T2.5        : int  15 15 3 6 12 6 12 4 10 5 ...
##  $ T2.6        : int  1 12 2 NA 9 5 1 3 13 NA ...
##  $ T2.7        : int  NA 4 NA NA 1 10 5 11 NA NA ...
##  $ T2.8        : int  NA 1 NA NA 2 12 NA 2 NA NA ...
##  $ T2.9        : int  NA 2 NA NA 3 11 NA 8 NA NA ...
##  $ T2.10       : int  NA NA NA NA 6 9 NA 15 NA NA ...
##  $ T2.11       : int  NA NA NA NA NA 7 NA 7 NA NA ...
##  $ T2.12       : int  NA NA NA NA NA NA NA 14 NA NA ...
##  $ T2.13       : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ T2.14       : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ T2.15       : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ T3.1        : int  13 15 13 14 4 4 14 1 2 13 ...
##  $ T3.2        : int  14 14 14 15 7 13 15 12 8 14 ...
##  $ T3.3        : int  15 12 15 10 14 14 12 9 9 15 ...
##  $ T3.4        : int  12 9 11 9 15 15 3 3 10 1 ...
##  $ T3.5        : int  11 10 2 1 9 7 11 5 12 2 ...
##  $ T3.6        : int  10 1 3 12 10 12 4 6 13 3 ...
##  $ T3.7        : int  8 2 NA 5 12 10 2 7 14 9 ...
##  $ T3.8        : int  1 4 NA 3 1 5 6 15 15 12 ...
##  $ T3.9        : int  3 6 NA NA 2 3 9 8 NA 11 ...
##  $ T3.10       : int  2 NA NA NA 8 2 NA 2 NA 8 ...
##  $ T3.11       : int  4 NA NA NA 5 6 NA 11 NA NA ...
##  $ T3.12       : int  5 NA NA NA 3 8 NA 13 NA NA ...
##  $ T3.13       : int  NA NA NA NA 6 NA NA NA NA NA ...
##  $ T3.14       : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ T3.15       : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ T4.1        : int  15 9 13 15 13 1 13 1 1 13 ...
##  $ T4.2        : int  14 10 14 14 14 3 3 12 2 14 ...
##  $ T4.3        : int  13 11 15 11 15 2 9 9 4 15 ...
##  $ T4.4        : int  12 13 11 5 9 5 10 3 5 1 ...
##  $ T4.5        : int  10 15 6 3 12 14 12 5 11 2 ...
##  $ T4.6        : int  5 1 3 7 10 4 15 4 10 3 ...
##  $ T4.7        : int  1 2 1 12 11 15 5 10 15 5 ...
##  $ T4.8        : int  3 4 2 8 8 7 6 8 9 6 ...
##  $ T4.9        : int  2 6 7 NA 1 6 2 2 13 10 ...
##  $ T4.10       : int  8 5 10 NA 2 10 4 13 7 11 ...
##  $ T4.11       : int  NA NA 9 NA 3 12 8 6 NA 12 ...
##  $ T4.12       : int  NA NA NA NA 4 11 11 11 NA 8 ...
##  $ T4.13       : int  NA NA NA NA 6 8 NA 15 NA 9 ...
##  $ T4.14       : int  NA NA NA NA 5 NA NA 7 NA NA ...
##  $ T4.15       : int  NA NA NA NA NA NA NA 14 NA NA ...
##  $ T5.1        : int  15 10 13 1 13 13 3 1 1 11 ...
##  $ T5.2        : int  14 12 14 5 14 14 2 12 2 12 ...
##  $ T5.3        : int  13 9 15 12 15 15 1 9 4 13 ...
##  $ T5.4        : int  12 15 12 3 9 2 5 3 3 14 ...
##  $ T5.5        : int  10 2 3 7 12 1 6 5 6 15 ...
##  $ T5.6        : int  9 4 10 11 11 4 12 14 15 1 ...
##  $ T5.7        : int  11 1 11 6 2 10 10 10 9 2 ...
##  $ T5.8        : int  7 6 2 10 1 11 8 11 7 3 ...
##  $ T5.9        : int  1 5 1 15 3 12 13 7 NA 5 ...
##  $ T5.10       : int  3 11 4 14 5 8 NA 6 NA 6 ...
##  $ T5.11       : int  2 NA 7 NA 6 6 NA 15 NA 10 ...
##  $ T5.12       : int  5 NA 6 NA 4 5 NA 4 NA 9 ...
##  $ T5.13       : int  4 NA 5 NA 10 3 NA 8 NA 4 ...
##  $ T5.14       : int  8 NA 9 NA NA NA NA 2 NA NA ...
##  $ T5.15       : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ B.1         : int  15 10 13 14 11 1 13 14 1 13 ...
##  $ B.2         : int  14 11 14 13 12 11 14 15 9 14 ...
##  $ B.3         : int  13 12 15 15 13 13 15 13 NA 15 ...
##  $ B.4         : int  2 14 3 9 14 3 9 12 NA NA ...
##  $ B.5         : int  1 15 2 2 15 15 NA 6 NA NA ...
##  $ B.6         : int  7 6 NA 6 1 9 NA 2 NA NA ...
##  $ B.7         : int  NA 9 NA NA 2 2 NA NA NA NA ...
##  $ B.8         : int  NA 8 NA NA NA 5 NA NA NA NA ...
##  $ B.9         : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ B.10        : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ B.11        : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ B.12        : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ B.13        : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ B.14        : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ B.15        : logi  NA NA NA NA NA NA ...
##  $ SD.1        : int  1 1 14 3 1 3 3 1 1 1 ...
##  $ SD.2        : int  3 2 2 15 2 1 2 12 2 2 ...
##  $ SD.3        : int  2 4 1 12 3 2 5 9 15 3 ...
##  $ SD.4        : int  5 6 6 6 4 5 10 10 7 6 ...
##  $ SD.5        : int  4 5 4 5 5 12 11 5 9 10 ...
##  $ SD.6        : int  8 11 5 4 14 11 8 6 11 7 ...
##  $ SD.7        : int  7 10 9 14 6 10 1 7 6 11 ...
##  $ SD.8        : int  15 15 10 NA 12 6 6 15 NA 4 ...
##  $ SD.9        : int  11 NA 15 NA 10 7 15 13 NA 8 ...
##  $ SD.10       : int  9 NA NA NA 15 15 NA 11 NA NA ...
##  $ SD.11       : int  NA NA NA NA NA 4 NA 3 NA NA ...
##  $ SD.12       : int  NA NA NA NA NA 8 NA 4 NA NA ...
##  $ SD.13       : int  NA NA NA NA NA NA NA 2 NA NA ...
##  $ SD.14       : int  NA NA NA NA NA NA NA 14 NA NA ...
##  $ SD.15       : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ D.1         : int  1 1 1 15 1 2 11 1 1 1 ...
##  $ D.2         : int  3 2 4 5 2 1 3 3 2 2 ...
##  $ D.3         : int  2 4 5 14 3 3 5 10 14 3 ...
##  $ D.4         : int  4 6 10 9 8 5 10 9 9 5 ...
##  $ D.5         : int  7 5 15 2 10 8 8 8 6 4 ...
##  $ D.6         : int  8 11 3 4 14 10 1 15 5 8 ...
##  $ D.7         : int  9 15 12 12 5 6 9 5 NA 10 ...
##  $ D.8         : int  15 10 14 3 6 11 14 14 NA 6 ...
##  $ D.9         : int  10 NA 2 8 4 9 6 2 NA 7 ...
##  $ D.10        : int  11 NA 9 10 12 12 NA 12 NA NA ...
##  $ D.11        : int  NA NA NA NA 15 7 NA 11 NA NA ...
##  $ D.12        : int  NA NA NA NA NA 15 NA 7 NA NA ...
##  $ D.13        : int  NA NA NA NA NA 4 NA 6 NA NA ...
##  $ D.14        : int  NA NA NA NA NA NA NA 4 NA NA ...
##  $ D.15        : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ update_stamp: chr  &quot;2024-09-13 08:51:42.0&quot; &quot;2024-09-13 08:51:42.0&quot; &quot;2024-09-13 08:51:42.0&quot; &quot;2024-09-13 08:51:42.0&quot; ...</code></pre>
</details>
<pre class="r"><code>## convert column B.15 from logistic to integer
## this column was confirmed to only have NA values in the raw data
AVLTPositionScores[,&quot;B.15&quot;] &lt;- as.integer(AVLTPositionScores[,&quot;B.15&quot;])

## check table frequency for all variables

## values in columns [&quot;T1.1&quot; - &quot;D.15&quot;] should be 1-15 (integer)
cols = c(paste0(&quot;T1.&quot;,seq(1:15)),paste0(&quot;T2.&quot;,seq(1:15)),paste0(&quot;T3.&quot;,seq(1:15)),
         paste0(&quot;T4.&quot;,seq(1:15)),paste0(&quot;T5.&quot;,seq(1:15)),paste0(&quot;B.&quot;,seq(1:15)),
         paste0(&quot;SD.&quot;,seq(1:15)),paste0(&quot;D.&quot;,seq(1:15)))

for (col in cols) {
  vals &lt;- unique(AVLTPositionScores[[col]])
  invalid_vals &lt;- vals[!vals %in% c(1:15, NA)]
  
  if (length(invalid_vals) &gt; 0) {
    cat(&quot;Column:&quot;, col, &quot;- Invalid values:&quot;, paste(invalid_vals, collapse = &quot;, &quot;), &quot;\n&quot;)
  }
}

## drop update_stamp column
AVLTPositionScores &lt;- subset(AVLTPositionScores,select = -c(update_stamp)) ## 124 columns left</code></pre>
<pre class="r"><code>str(AVLTPositionScores, max.level = 99, list.len = 99999)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    7946 obs. of  123 variables:
##  $ wrapnum: chr  &quot;wrap0002&quot; &quot;wrap0068&quot; &quot;wrap0146&quot; &quot;wrap0148&quot; ...
##  $ WRAPNo : chr  &quot;0002&quot; &quot;0068&quot; &quot;0146&quot; &quot;0148&quot; ...
##  $ VisNo  : int  5 2 4 5 5 4 6 4 7 4 ...
##  $ T1.1   : int  15 8 14 1 1 1 15 1 1 1 ...
##  $ T1.2   : int  14 12 15 2 2 5 1 12 10 15 ...
##  $ T1.3   : int  13 14 12 5 9 8 12 8 12 9 ...
##  $ T1.4   : int  12 15 NA 15 12 12 14 9 14 12 ...
##  $ T1.5   : int  8 9 NA 12 11 11 NA 2 8 NA ...
##  $ T1.6   : int  NA 10 NA 14 6 9 NA 3 NA NA ...
##  $ T1.7   : int  NA NA NA NA 5 10 NA NA NA NA ...
##  $ T1.8   : int  NA NA NA NA NA 7 NA NA NA NA ...
##  $ T1.9   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ T1.10  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ T1.11  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ T1.12  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ T1.13  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ T1.14  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ T1.15  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ T2.1   : int  10 9 15 1 14 14 2 1 2 1 ...
##  $ T2.2   : int  11 10 14 5 15 15 3 12 12 2 ...
##  $ T2.3   : int  8 11 13 15 5 3 15 9 14 11 ...
##  $ T2.4   : int  12 13 12 4 10 8 14 5 8 9 ...
##  $ T2.5   : int  15 15 3 6 12 6 12 4 10 5 ...
##  $ T2.6   : int  1 12 2 NA 9 5 1 3 13 NA ...
##  $ T2.7   : int  NA 4 NA NA 1 10 5 11 NA NA ...
##  $ T2.8   : int  NA 1 NA NA 2 12 NA 2 NA NA ...
##  $ T2.9   : int  NA 2 NA NA 3 11 NA 8 NA NA ...
##  $ T2.10  : int  NA NA NA NA 6 9 NA 15 NA NA ...
##  $ T2.11  : int  NA NA NA NA NA 7 NA 7 NA NA ...
##  $ T2.12  : int  NA NA NA NA NA NA NA 14 NA NA ...
##  $ T2.13  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ T2.14  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ T2.15  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ T3.1   : int  13 15 13 14 4 4 14 1 2 13 ...
##  $ T3.2   : int  14 14 14 15 7 13 15 12 8 14 ...
##  $ T3.3   : int  15 12 15 10 14 14 12 9 9 15 ...
##  $ T3.4   : int  12 9 11 9 15 15 3 3 10 1 ...
##  $ T3.5   : int  11 10 2 1 9 7 11 5 12 2 ...
##  $ T3.6   : int  10 1 3 12 10 12 4 6 13 3 ...
##  $ T3.7   : int  8 2 NA 5 12 10 2 7 14 9 ...
##  $ T3.8   : int  1 4 NA 3 1 5 6 15 15 12 ...
##  $ T3.9   : int  3 6 NA NA 2 3 9 8 NA 11 ...
##  $ T3.10  : int  2 NA NA NA 8 2 NA 2 NA 8 ...
##  $ T3.11  : int  4 NA NA NA 5 6 NA 11 NA NA ...
##  $ T3.12  : int  5 NA NA NA 3 8 NA 13 NA NA ...
##  $ T3.13  : int  NA NA NA NA 6 NA NA NA NA NA ...
##  $ T3.14  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ T3.15  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ T4.1   : int  15 9 13 15 13 1 13 1 1 13 ...
##  $ T4.2   : int  14 10 14 14 14 3 3 12 2 14 ...
##  $ T4.3   : int  13 11 15 11 15 2 9 9 4 15 ...
##  $ T4.4   : int  12 13 11 5 9 5 10 3 5 1 ...
##  $ T4.5   : int  10 15 6 3 12 14 12 5 11 2 ...
##  $ T4.6   : int  5 1 3 7 10 4 15 4 10 3 ...
##  $ T4.7   : int  1 2 1 12 11 15 5 10 15 5 ...
##  $ T4.8   : int  3 4 2 8 8 7 6 8 9 6 ...
##  $ T4.9   : int  2 6 7 NA 1 6 2 2 13 10 ...
##  $ T4.10  : int  8 5 10 NA 2 10 4 13 7 11 ...
##  $ T4.11  : int  NA NA 9 NA 3 12 8 6 NA 12 ...
##  $ T4.12  : int  NA NA NA NA 4 11 11 11 NA 8 ...
##  $ T4.13  : int  NA NA NA NA 6 8 NA 15 NA 9 ...
##  $ T4.14  : int  NA NA NA NA 5 NA NA 7 NA NA ...
##  $ T4.15  : int  NA NA NA NA NA NA NA 14 NA NA ...
##  $ T5.1   : int  15 10 13 1 13 13 3 1 1 11 ...
##  $ T5.2   : int  14 12 14 5 14 14 2 12 2 12 ...
##  $ T5.3   : int  13 9 15 12 15 15 1 9 4 13 ...
##  $ T5.4   : int  12 15 12 3 9 2 5 3 3 14 ...
##  $ T5.5   : int  10 2 3 7 12 1 6 5 6 15 ...
##  $ T5.6   : int  9 4 10 11 11 4 12 14 15 1 ...
##  $ T5.7   : int  11 1 11 6 2 10 10 10 9 2 ...
##  $ T5.8   : int  7 6 2 10 1 11 8 11 7 3 ...
##  $ T5.9   : int  1 5 1 15 3 12 13 7 NA 5 ...
##  $ T5.10  : int  3 11 4 14 5 8 NA 6 NA 6 ...
##  $ T5.11  : int  2 NA 7 NA 6 6 NA 15 NA 10 ...
##  $ T5.12  : int  5 NA 6 NA 4 5 NA 4 NA 9 ...
##  $ T5.13  : int  4 NA 5 NA 10 3 NA 8 NA 4 ...
##  $ T5.14  : int  8 NA 9 NA NA NA NA 2 NA NA ...
##  $ T5.15  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ B.1    : int  15 10 13 14 11 1 13 14 1 13 ...
##  $ B.2    : int  14 11 14 13 12 11 14 15 9 14 ...
##  $ B.3    : int  13 12 15 15 13 13 15 13 NA 15 ...
##  $ B.4    : int  2 14 3 9 14 3 9 12 NA NA ...
##  $ B.5    : int  1 15 2 2 15 15 NA 6 NA NA ...
##  $ B.6    : int  7 6 NA 6 1 9 NA 2 NA NA ...
##  $ B.7    : int  NA 9 NA NA 2 2 NA NA NA NA ...
##  $ B.8    : int  NA 8 NA NA NA 5 NA NA NA NA ...
##  $ B.9    : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ B.10   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ B.11   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ B.12   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ B.13   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ B.14   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ B.15   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ SD.1   : int  1 1 14 3 1 3 3 1 1 1 ...
##  $ SD.2   : int  3 2 2 15 2 1 2 12 2 2 ...
##  $ SD.3   : int  2 4 1 12 3 2 5 9 15 3 ...
##  $ SD.4   : int  5 6 6 6 4 5 10 10 7 6 ...
##  $ SD.5   : int  4 5 4 5 5 12 11 5 9 10 ...
##  $ SD.6   : int  8 11 5 4 14 11 8 6 11 7 ...
##  $ SD.7   : int  7 10 9 14 6 10 1 7 6 11 ...
##  $ SD.8   : int  15 15 10 NA 12 6 6 15 NA 4 ...
##  $ SD.9   : int  11 NA 15 NA 10 7 15 13 NA 8 ...
##  $ SD.10  : int  9 NA NA NA 15 15 NA 11 NA NA ...
##  $ SD.11  : int  NA NA NA NA NA 4 NA 3 NA NA ...
##  $ SD.12  : int  NA NA NA NA NA 8 NA 4 NA NA ...
##  $ SD.13  : int  NA NA NA NA NA NA NA 2 NA NA ...
##  $ SD.14  : int  NA NA NA NA NA NA NA 14 NA NA ...
##  $ SD.15  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ D.1    : int  1 1 1 15 1 2 11 1 1 1 ...
##  $ D.2    : int  3 2 4 5 2 1 3 3 2 2 ...
##  $ D.3    : int  2 4 5 14 3 3 5 10 14 3 ...
##  $ D.4    : int  4 6 10 9 8 5 10 9 9 5 ...
##  $ D.5    : int  7 5 15 2 10 8 8 8 6 4 ...
##  $ D.6    : int  8 11 3 4 14 10 1 15 5 8 ...
##  $ D.7    : int  9 15 12 12 5 6 9 5 NA 10 ...
##  $ D.8    : int  15 10 14 3 6 11 14 14 NA 6 ...
##  $ D.9    : int  10 NA 2 8 4 9 6 2 NA 7 ...
##  $ D.10   : int  11 NA 9 10 12 12 NA 12 NA NA ...
##  $ D.11   : int  NA NA NA NA 15 7 NA 11 NA NA ...
##  $ D.12   : int  NA NA NA NA NA 15 NA 7 NA NA ...
##  $ D.13   : int  NA NA NA NA NA 4 NA 6 NA NA ...
##  $ D.14   : int  NA NA NA NA NA NA NA 4 NA NA ...
##  $ D.15   : int  NA NA NA NA NA NA NA NA NA NA ...</code></pre>
</details>
<p><br> <br></p>
</div>
<div id="behavioralobservationchecklist" class="section level2">
<h2>BehavioralObservationChecklist</h2>
<p>Behavioral Observation Checklist</p>
<pre class="r"><code>## load data
BehavioralObservationChecklist &lt;- read.csv(paste0(wrap_directory, &quot;BehavioralObservationChecklist_05Jun2025.csv&quot;))

info(BehavioralObservationChecklist,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:2601, cols:85, inds:1322</code></pre>
<pre class="r"><code>str(BehavioralObservationChecklist, max.level = 99, list.len = 99999)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    2601 obs. of  85 variables:
##  $ wrapnum                           : chr  &quot;wrap0401&quot; &quot;wrap0434&quot; &quot;wrap0609&quot; &quot;wrap0624&quot; ...
##  $ WRAPNo                            : chr  &quot;0401&quot; &quot;0434&quot; &quot;0609&quot; &quot;0624&quot; ...
##  $ VisNo                             : int  6 7 8 6 3 1 1 1 3 4 ...
##  $ VisType                           : chr  &quot;In-Person&quot; &quot;T-Cog&quot; &quot;In-Person&quot; &quot;In-Person&quot; ...
##  $ BOC_Rapport_EasilyEstab           : int  1 1 1 1 1 1 1 1 2 1 ...
##  $ BOC_Rapport_DifficultEstab        : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ BOC_Rapport_SlowWarm              : int  2 2 2 2 2 2 2 2 1 2 ...
##  $ BOC_Rapport_QuietHesitant         : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ BOC_Rapport_Oppositional          : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ BOC_Affect_Normal                 : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ BOC_Affect_Restricted             : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ BOC_Affect_FlatBlunted            : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ BOC_Affect_Anxious                : int  2 2 2 2 2 1 2 2 2 2 ...
##  $ BOC_Affect_Pessimistic            : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ BOC_Affect_UnableDeterm           : int  NA 2 NA NA NA NA 2 NA NA NA ...
##  $ BOC_Mood_HappyPositive            : int  1 1 NA 1 NA 1 1 1 1 1 ...
##  $ BOC_Mood_IrritableAngry           : int  2 2 NA 2 NA 2 2 2 2 2 ...
##  $ BOC_Mood_SadTearful               : int  2 2 NA 2 NA 2 2 2 2 2 ...
##  $ BOC_Mood_Depressed                : int  2 2 NA 2 NA 2 2 2 2 2 ...
##  $ BOC_Mood_Frustrated               : int  2 2 NA 2 NA 2 2 2 2 2 ...
##  $ BOC_Mood_Neutral                  : int  NA 2 NA NA NA NA 2 NA NA NA ...
##  $ BOC_Mood_UnableDeterm             : int  NA 2 NA NA NA NA 2 NA NA NA ...
##  $ BOC_EyeContact_Appropriate        : int  1 NA 1 1 1 1 NA 1 1 1 ...
##  $ BOC_EyeContact_Limited            : int  2 NA 2 2 2 2 NA 2 2 2 ...
##  $ BOC_EyeContact_Absent             : int  2 NA 2 2 2 2 NA 2 2 2 ...
##  $ BOC_EyeContact_Intense            : int  2 NA 2 2 2 2 NA 2 2 2 ...
##  $ BOC_EyeContact_Variable           : int  2 NA 2 2 2 2 NA 2 2 2 ...
##  $ BOC_AttitudeToTest_AffectPos      : int  1 1 1 1 NA 1 1 1 1 1 ...
##  $ BOC_AttitudeToTest_AffectNeg      : int  2 2 2 2 NA 2 2 2 2 2 ...
##  $ BOC_AttitudeToTest_PoorFrustrToler: int  2 2 2 2 NA 2 2 2 2 2 ...
##  $ BOC_AttitudeToTest_EagerToSpeak   : int  2 2 2 2 NA 2 2 2 2 2 ...
##  $ BOC_AttitudeToTest_NotedPracTests : int  2 2 2 2 NA 2 2 2 2 2 ...
##  $ BOC_AttitudeToTest_UnableDeterm   : int  NA 2 NA NA NA NA 2 NA NA NA ...
##  $ BOC_EffortCoop_Excellent          : int  1 1 1 1 2 1 2 1 2 1 ...
##  $ BOC_EffortCoop_Good               : int  2 2 2 2 1 2 1 2 1 2 ...
##  $ BOC_EffortCoop_Fair               : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ BOC_EffortCoop_Poor               : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ BOC_EffortCoop_Fluctuating        : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ BOC_AttnConc_Consistent           : int  1 1 1 1 2 1 1 1 1 1 ...
##  $ BOC_AttnConc_Variable             : int  2 2 2 2 1 2 2 2 2 2 ...
##  $ BOC_AttnConc_Impulsive            : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ BOC_AttnConc_TCog_OthrDistract    : int  NA 2 NA NA NA NA 2 NA NA NA ...
##  $ BOC_AttnConc_TCog_BackNoise       : int  NA 2 NA NA NA NA 2 NA NA NA ...
##  $ BOC_AttnConc_OthrDistract         : int  1 NA 2 1 2 2 NA NA 2 2 ...
##  $ BOC_ApproachTasks_Logical         : int  1 2 1 1 1 1 2 1 1 1 ...
##  $ BOC_ApproachTasks_Random          : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ BOC_ApproachTasks_TrialError      : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ BOC_ApproachTasks_Persistent      : int  1 1 1 2 2 1 1 2 2 2 ...
##  $ BOC_ApproachTasks_QuitsEasily     : int  2 2 2 2 1 2 2 2 2 2 ...
##  $ BOC_ApproachTasks_UnableDeterm    : int  NA 2 NA NA NA NA 2 NA NA NA ...
##  $ BOC_Comprehension_Intact          : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ BOC_Comprehension_NeedRepet       : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ BOC_Comprehension_NeedClarif      : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ BOC_Hearing_HearingAids           : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ BOC_Hearing_NeededRepetClarif     : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ BOC_Hearing_SuspectedHearProb     : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ BOC_Hearing_PhoneIssueConnctProb  : int  NA 2 NA NA NA NA 2 NA NA NA ...
##  $ BOC_Vision_Glasses                : int  1 NA 2 1 2 1 NA 1 1 2 ...
##  $ BOC_Vision_ReadingGlasses         : int  2 NA 1 2 2 2 NA 2 1 2 ...
##  $ BOC_Vision_Complaints             : int  2 NA 2 2 2 2 NA 2 2 2 ...
##  $ BOC_Motor_GaitProbs               : int  2 NA 1 2 1 2 NA NA 1 2 ...
##  $ BOC_Motor_Tremors                 : int  2 NA 1 1 2 2 NA NA 2 2 ...
##  $ BOC_Motor_Handedness              : int  1 NA 1 2 1 1 NA 1 1 1 ...
##  $ BOC_LangSpeech_FluentArticulate   : int  1 1 1 1 1 1 1 1 NA 1 ...
##  $ BOC_LangSpeech_NonFluent          : int  2 2 2 2 2 2 2 2 NA 2 ...
##  $ BOC_LangSpeech_PoorVolumeRegulat  : int  2 2 2 2 2 2 2 2 NA 2 ...
##  $ BOC_LangSpeech_StutterExcessPauses: int  2 2 2 2 2 2 2 2 NA 2 ...
##  $ BOC_LangSpeech_DialectDiff        : int  2 2 2 2 2 2 2 NA 2 2 ...
##  $ BOC_LangSpeech_EnglishSecondLang  : int  2 2 2 2 2 2 2 NA 2 2 ...
##  $ BOC_Other_TiredYawning            : int  2 2 2 2 1 2 2 2 2 2 ...
##  $ BOC_Other_Illness                 : int  2 2 1 2 2 2 2 2 2 2 ...
##  $ BOC_Other_TalkTangen              : int  2 2 2 2 1 2 2 2 2 2 ...
##  $ BOC_Other_BatteryAbbrev           : int  1 2 1 2 1 2 2 2 2 2 ...
##  $ BOC_Other_Disinhibited            : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ BOC_Appearance_AsExpected         : int  1 NA 1 1 1 1 NA 1 1 1 ...
##  $ BOC_Appearance_Unkempt            : int  2 NA 2 2 2 2 NA 2 2 2 ...
##  $ BOC_Other_UncomfortWithPhone      : int  NA 2 NA NA NA NA 2 NA NA NA ...
##  $ BOC_Phone_Cellphone               : int  NA 1 NA NA NA NA 1 NA NA NA ...
##  $ BOC_Phone_Landline                : int  NA 2 NA NA NA NA 2 NA NA NA ...
##  $ BOC_Phone_SpeakerPhone            : int  NA 2 NA NA NA NA 2 NA NA NA ...
##  $ BOC_Phone_ConnectIssue            : int  NA 2 NA NA NA NA 2 NA NA NA ...
##  $ BOC_CESD                          : int  NA NA 2 NA NA NA NA NA NA NA ...
##  $ BOC_Clinician_Informed            : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ BOC_Address_Not_PM                : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ update_stamp                      : chr  &quot;2024-09-13 08:51:45.0&quot; &quot;2024-09-13 08:51:45.0&quot; &quot;2024-09-13 08:51:45.0&quot; &quot;2024-09-13 08:51:45.0&quot; ...</code></pre>
</details>
<pre class="r"><code>## check table frequency for all variables

## values in columns [BOC_Rapport_EasilyEstab - BOC_Address_Not_PM] should be {1,2}
## 1=selected;2=other value selected;null=no values in set selected
cols = names(BehavioralObservationChecklist)[5:84]

for (col in cols) {
  vals &lt;- unique(BehavioralObservationChecklist[[col]])
  invalid_vals &lt;- vals[!vals %in% c(1:2, NA)]
  
  if (length(invalid_vals) &gt; 0) {
    cat(&quot;Column:&quot;, col, &quot;- Invalid values:&quot;, paste(invalid_vals, collapse = &quot;, &quot;), &quot;\n&quot;)
  }
}</code></pre>
<pre><code>## Column: BOC_Motor_Handedness - Invalid values: 3</code></pre>
<pre class="r"><code>## Column: BOC_Motor_Handedness - Invalid values: 3 

table(BehavioralObservationChecklist$BOC_Motor_Handedness, useNA = &quot;ifany&quot;)</code></pre>
<pre><code>## 
##    1    2    3 &lt;NA&gt; 
## 2020  214    2  365</code></pre>
<pre class="r"><code>#    1    2    3 &lt;NA&gt; 
# 2020  214    2  365 

## covert value 3 to NA for column BOC_Motor_Handedness
BehavioralObservationChecklist$BOC_Motor_Handedness[BehavioralObservationChecklist$BOC_Motor_Handedness == 3] &lt;- NA
table(BehavioralObservationChecklist$BOC_Motor_Handedness, useNA = &quot;ifany&quot;)</code></pre>
<pre><code>## 
##    1    2 &lt;NA&gt; 
## 2020  214  367</code></pre>
<pre class="r"><code>#    1    2 &lt;NA&gt; 
# 2020  214  367 

## drop update_stamp column
BehavioralObservationChecklist &lt;- subset(BehavioralObservationChecklist,select = -c(update_stamp)) ## 84 columns left</code></pre>
<pre class="r"><code>str(BehavioralObservationChecklist, max.level = 99, list.len = 99999)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    2601 obs. of  84 variables:
##  $ wrapnum                           : chr  &quot;wrap0401&quot; &quot;wrap0434&quot; &quot;wrap0609&quot; &quot;wrap0624&quot; ...
##  $ WRAPNo                            : chr  &quot;0401&quot; &quot;0434&quot; &quot;0609&quot; &quot;0624&quot; ...
##  $ VisNo                             : int  6 7 8 6 3 1 1 1 3 4 ...
##  $ VisType                           : chr  &quot;In-Person&quot; &quot;T-Cog&quot; &quot;In-Person&quot; &quot;In-Person&quot; ...
##  $ BOC_Rapport_EasilyEstab           : int  1 1 1 1 1 1 1 1 2 1 ...
##  $ BOC_Rapport_DifficultEstab        : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ BOC_Rapport_SlowWarm              : int  2 2 2 2 2 2 2 2 1 2 ...
##  $ BOC_Rapport_QuietHesitant         : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ BOC_Rapport_Oppositional          : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ BOC_Affect_Normal                 : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ BOC_Affect_Restricted             : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ BOC_Affect_FlatBlunted            : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ BOC_Affect_Anxious                : int  2 2 2 2 2 1 2 2 2 2 ...
##  $ BOC_Affect_Pessimistic            : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ BOC_Affect_UnableDeterm           : int  NA 2 NA NA NA NA 2 NA NA NA ...
##  $ BOC_Mood_HappyPositive            : int  1 1 NA 1 NA 1 1 1 1 1 ...
##  $ BOC_Mood_IrritableAngry           : int  2 2 NA 2 NA 2 2 2 2 2 ...
##  $ BOC_Mood_SadTearful               : int  2 2 NA 2 NA 2 2 2 2 2 ...
##  $ BOC_Mood_Depressed                : int  2 2 NA 2 NA 2 2 2 2 2 ...
##  $ BOC_Mood_Frustrated               : int  2 2 NA 2 NA 2 2 2 2 2 ...
##  $ BOC_Mood_Neutral                  : int  NA 2 NA NA NA NA 2 NA NA NA ...
##  $ BOC_Mood_UnableDeterm             : int  NA 2 NA NA NA NA 2 NA NA NA ...
##  $ BOC_EyeContact_Appropriate        : int  1 NA 1 1 1 1 NA 1 1 1 ...
##  $ BOC_EyeContact_Limited            : int  2 NA 2 2 2 2 NA 2 2 2 ...
##  $ BOC_EyeContact_Absent             : int  2 NA 2 2 2 2 NA 2 2 2 ...
##  $ BOC_EyeContact_Intense            : int  2 NA 2 2 2 2 NA 2 2 2 ...
##  $ BOC_EyeContact_Variable           : int  2 NA 2 2 2 2 NA 2 2 2 ...
##  $ BOC_AttitudeToTest_AffectPos      : int  1 1 1 1 NA 1 1 1 1 1 ...
##  $ BOC_AttitudeToTest_AffectNeg      : int  2 2 2 2 NA 2 2 2 2 2 ...
##  $ BOC_AttitudeToTest_PoorFrustrToler: int  2 2 2 2 NA 2 2 2 2 2 ...
##  $ BOC_AttitudeToTest_EagerToSpeak   : int  2 2 2 2 NA 2 2 2 2 2 ...
##  $ BOC_AttitudeToTest_NotedPracTests : int  2 2 2 2 NA 2 2 2 2 2 ...
##  $ BOC_AttitudeToTest_UnableDeterm   : int  NA 2 NA NA NA NA 2 NA NA NA ...
##  $ BOC_EffortCoop_Excellent          : int  1 1 1 1 2 1 2 1 2 1 ...
##  $ BOC_EffortCoop_Good               : int  2 2 2 2 1 2 1 2 1 2 ...
##  $ BOC_EffortCoop_Fair               : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ BOC_EffortCoop_Poor               : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ BOC_EffortCoop_Fluctuating        : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ BOC_AttnConc_Consistent           : int  1 1 1 1 2 1 1 1 1 1 ...
##  $ BOC_AttnConc_Variable             : int  2 2 2 2 1 2 2 2 2 2 ...
##  $ BOC_AttnConc_Impulsive            : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ BOC_AttnConc_TCog_OthrDistract    : int  NA 2 NA NA NA NA 2 NA NA NA ...
##  $ BOC_AttnConc_TCog_BackNoise       : int  NA 2 NA NA NA NA 2 NA NA NA ...
##  $ BOC_AttnConc_OthrDistract         : int  1 NA 2 1 2 2 NA NA 2 2 ...
##  $ BOC_ApproachTasks_Logical         : int  1 2 1 1 1 1 2 1 1 1 ...
##  $ BOC_ApproachTasks_Random          : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ BOC_ApproachTasks_TrialError      : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ BOC_ApproachTasks_Persistent      : int  1 1 1 2 2 1 1 2 2 2 ...
##  $ BOC_ApproachTasks_QuitsEasily     : int  2 2 2 2 1 2 2 2 2 2 ...
##  $ BOC_ApproachTasks_UnableDeterm    : int  NA 2 NA NA NA NA 2 NA NA NA ...
##  $ BOC_Comprehension_Intact          : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ BOC_Comprehension_NeedRepet       : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ BOC_Comprehension_NeedClarif      : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ BOC_Hearing_HearingAids           : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ BOC_Hearing_NeededRepetClarif     : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ BOC_Hearing_SuspectedHearProb     : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ BOC_Hearing_PhoneIssueConnctProb  : int  NA 2 NA NA NA NA 2 NA NA NA ...
##  $ BOC_Vision_Glasses                : int  1 NA 2 1 2 1 NA 1 1 2 ...
##  $ BOC_Vision_ReadingGlasses         : int  2 NA 1 2 2 2 NA 2 1 2 ...
##  $ BOC_Vision_Complaints             : int  2 NA 2 2 2 2 NA 2 2 2 ...
##  $ BOC_Motor_GaitProbs               : int  2 NA 1 2 1 2 NA NA 1 2 ...
##  $ BOC_Motor_Tremors                 : int  2 NA 1 1 2 2 NA NA 2 2 ...
##  $ BOC_Motor_Handedness              : int  1 NA 1 2 1 1 NA 1 1 1 ...
##  $ BOC_LangSpeech_FluentArticulate   : int  1 1 1 1 1 1 1 1 NA 1 ...
##  $ BOC_LangSpeech_NonFluent          : int  2 2 2 2 2 2 2 2 NA 2 ...
##  $ BOC_LangSpeech_PoorVolumeRegulat  : int  2 2 2 2 2 2 2 2 NA 2 ...
##  $ BOC_LangSpeech_StutterExcessPauses: int  2 2 2 2 2 2 2 2 NA 2 ...
##  $ BOC_LangSpeech_DialectDiff        : int  2 2 2 2 2 2 2 NA 2 2 ...
##  $ BOC_LangSpeech_EnglishSecondLang  : int  2 2 2 2 2 2 2 NA 2 2 ...
##  $ BOC_Other_TiredYawning            : int  2 2 2 2 1 2 2 2 2 2 ...
##  $ BOC_Other_Illness                 : int  2 2 1 2 2 2 2 2 2 2 ...
##  $ BOC_Other_TalkTangen              : int  2 2 2 2 1 2 2 2 2 2 ...
##  $ BOC_Other_BatteryAbbrev           : int  1 2 1 2 1 2 2 2 2 2 ...
##  $ BOC_Other_Disinhibited            : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ BOC_Appearance_AsExpected         : int  1 NA 1 1 1 1 NA 1 1 1 ...
##  $ BOC_Appearance_Unkempt            : int  2 NA 2 2 2 2 NA 2 2 2 ...
##  $ BOC_Other_UncomfortWithPhone      : int  NA 2 NA NA NA NA 2 NA NA NA ...
##  $ BOC_Phone_Cellphone               : int  NA 1 NA NA NA NA 1 NA NA NA ...
##  $ BOC_Phone_Landline                : int  NA 2 NA NA NA NA 2 NA NA NA ...
##  $ BOC_Phone_SpeakerPhone            : int  NA 2 NA NA NA NA 2 NA NA NA ...
##  $ BOC_Phone_ConnectIssue            : int  NA 2 NA NA NA NA 2 NA NA NA ...
##  $ BOC_CESD                          : int  NA NA 2 NA NA NA NA NA NA NA ...
##  $ BOC_Clinician_Informed            : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ BOC_Address_Not_PM                : int  NA NA NA NA NA NA NA NA NA NA ...</code></pre>
</details>
<p><br> <br></p>
</div>
<div id="ca_genome" class="section level2">
<h2>CA_Genome</h2>
<p>TOMM40 genotype</p>
<pre class="r"><code>## load data
CA_Genome &lt;- read.csv(paste0(wrap_directory, &quot;CA_Genome_05Jun2025.csv&quot;))
info(CA_Genome,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:1405, cols:5, inds:1405</code></pre>
<pre class="r"><code>str(CA_Genome)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    1405 obs. of  5 variables:
##  $ wrapnum     : chr  &quot;wrap0047&quot; &quot;wrap0101&quot; &quot;wrap0100&quot; &quot;wrap0109&quot; ...
##  $ WRAPNo      : chr  &quot;0047&quot; &quot;0101&quot; &quot;0100&quot; &quot;0109&quot; ...
##  $ T40T16All1  : int  16 16 16 16 16 16 16 16 16 29 ...
##  $ T40T16All2  : int  34 34 16 38 16 16 28 47 16 34 ...
##  $ update_stamp: chr  &quot;2024-09-13 08:51:45.0&quot; &quot;2024-09-13 08:51:45.0&quot; &quot;2024-09-13 08:51:45.0&quot; &quot;2024-09-13 08:51:45.0&quot; ...</code></pre>
</details>
<pre class="r"><code>## check table frequency for all variables

## drop update_stamp column
CA_Genome &lt;- subset(CA_Genome,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(CA_Genome)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    1405 obs. of  4 variables:
##  $ wrapnum   : chr  &quot;wrap0047&quot; &quot;wrap0101&quot; &quot;wrap0100&quot; &quot;wrap0109&quot; ...
##  $ WRAPNo    : chr  &quot;0047&quot; &quot;0101&quot; &quot;0100&quot; &quot;0109&quot; ...
##  $ T40T16All1: int  16 16 16 16 16 16 16 16 16 29 ...
##  $ T40T16All2: int  34 34 16 38 16 16 28 47 16 34 ...</code></pre>
</details>
<p><br> <br></p>
</div>
<div id="cdr" class="section level2">
<h2>CDR</h2>
<p>Clinical Dementia Rating</p>
<pre class="r"><code>## load data
CDR &lt;- read.csv(paste0(wrap_directory, &quot;CDR_05Jun2025.csv&quot;))
info(CDR,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:1361, cols:16, inds:886</code></pre>
<pre class="r"><code>str(CDR)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    1361 obs. of  16 variables:
##  $ wrapnum                                    : chr  &quot;wrap0148&quot; &quot;wrap0705&quot; &quot;wrap0332&quot; &quot;wrap0146&quot; ...
##  $ WRAPNo                                     : chr  &quot;0148&quot; &quot;0705&quot; &quot;0332&quot; &quot;0146&quot; ...
##  $ VisNo                                      : int  5 5 4 4 4 4 4 4 5 6 ...
##  $ Memory                                     : num  0 0 0 0 0 0.5 0.5 0 0.5 NA ...
##  $ Orientation                                : num  0 0 0 0 0 0 0 0 0 NA ...
##  $ JudgePS                                    : num  0 0 0 0 0 0 0.5 0 0 NA ...
##  $ CommAffairs                                : num  0 0 0 0 0 0 0 0 0 NA ...
##  $ HomeHob                                    : num  0 0 0 0 0 0 0 0 0 NA ...
##  $ PersCare                                   : int  0 0 0 0 0 0 0 0 0 NA ...
##  $ CDRRating                                  : num  0 0 0 0 0 0.5 0.5 0 0.5 NA ...
##  $ SumOfBoxes                                 : num  0 0 0 0 0 0.5 1 0 0.5 NA ...
##  $ estimated_questionnaire_days_after_baseline: int  4787 3789 3288 3574 3862 3374 3629 3788 3942 NA ...
##  $ Returned_Days_From_Visit                   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ RelPPT                                     : int  1 6 1 2 1 1 4 1 2 4 ...
##  $ RelLive                                    : int  1 NA 1 1 1 1 0 1 0 NA ...
##  $ update_stamp                               : chr  &quot;2024-09-13 08:51:46.0&quot; &quot;2024-09-13 08:51:46.0&quot; &quot;2024-09-13 08:51:46.0&quot; &quot;2024-09-13 08:51:46.0&quot; ...</code></pre>
</details>
<pre class="r"><code>## check table frequency for all variables

## DD has 1=yes,2=no coding but data has 0/1 coding. 
## Confirmed with other 0/1 variables that 0 is no and 1 stays yes, so will update the dataset to match the DD
CDR$RelLive[CDR$RelLive==0] &lt;- 2
CDR$RelLive &lt;- as.integer(CDR$RelLive)

## drop update_stamp column
CDR &lt;- subset(CDR,select = -c(update_stamp)) ## 15 columns left</code></pre>
<pre class="r"><code>str(CDR)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    1361 obs. of  15 variables:
##  $ wrapnum                                    : chr  &quot;wrap0148&quot; &quot;wrap0705&quot; &quot;wrap0332&quot; &quot;wrap0146&quot; ...
##  $ WRAPNo                                     : chr  &quot;0148&quot; &quot;0705&quot; &quot;0332&quot; &quot;0146&quot; ...
##  $ VisNo                                      : int  5 5 4 4 4 4 4 4 5 6 ...
##  $ Memory                                     : num  0 0 0 0 0 0.5 0.5 0 0.5 NA ...
##  $ Orientation                                : num  0 0 0 0 0 0 0 0 0 NA ...
##  $ JudgePS                                    : num  0 0 0 0 0 0 0.5 0 0 NA ...
##  $ CommAffairs                                : num  0 0 0 0 0 0 0 0 0 NA ...
##  $ HomeHob                                    : num  0 0 0 0 0 0 0 0 0 NA ...
##  $ PersCare                                   : int  0 0 0 0 0 0 0 0 0 NA ...
##  $ CDRRating                                  : num  0 0 0 0 0 0.5 0.5 0 0.5 NA ...
##  $ SumOfBoxes                                 : num  0 0 0 0 0 0.5 1 0 0.5 NA ...
##  $ estimated_questionnaire_days_after_baseline: int  4787 3789 3288 3574 3862 3374 3629 3788 3942 NA ...
##  $ Returned_Days_From_Visit                   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ RelPPT                                     : int  1 6 1 2 1 1 4 1 2 4 ...
##  $ RelLive                                    : int  1 NA 1 1 1 1 2 1 2 NA ...</code></pre>
</details>
<p><br> <br></p>
</div>
<div id="cerad" class="section level2">
<h2>CERAD</h2>
<p>Consortium to Establish a Registry for Alzheimer’s Disease Data
(CERAD)</p>
<pre class="r"><code>## load data
CERAD &lt;- read.csv(paste0(wrap_directory, &quot;CERAD_05Jun2025.csv&quot;))
info(CERAD,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:372, cols:20, inds:372</code></pre>
<pre class="r"><code>str(CERAD)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    372 obs. of  20 variables:
##  $ wrapnum     : chr  &quot;wrap0176&quot; &quot;wrap0346&quot; &quot;wrap0422&quot; &quot;wrap0042&quot; ...
##  $ WRAPNo      : chr  &quot;0176&quot; &quot;0346&quot; &quot;0422&quot; &quot;0042&quot; ...
##  $ OverImp     : chr  &quot;Normal&quot; &quot;Normal&quot; &quot;Normal&quot; &quot;Normal&quot; ...
##  $ MPI         : int  65 68 64 73 59 80 72 72 67 60 ...
##  $ IR1         : int  6 7 5 9 6 7 6 7 6 5 ...
##  $ IR2         : int  8 8 5 9 7 10 8 8 8 6 ...
##  $ IR3         : int  8 9 6 10 8 10 9 9 9 6 ...
##  $ IRTotal     : int  22 24 16 28 21 27 23 24 23 17 ...
##  $ IRStand     : num  0.6 0 -2.3 1.6 -0.9 0.9 -0.3 0 -0.3 -1 ...
##  $ DFREst      : num  5 7 4 6 5 8 7 5 6 4.5 ...
##  $ DFRAct      : int  8 8 7 9 3 10 10 8 7 7 ...
##  $ DFRDiff     : num  -3 -1 -3 -3 2 -2 -3 -3 -1 -2.5 ...
##  $ DFRStand    : num  0.9 -0.1 -0.9 0.8 -3.7 1.3 1.3 -0.1 -0.9 0.1 ...
##  $ DCRYes      : int  10 10 10 10 10 10 10 10 9 10 ...
##  $ DCRYStand   : num  0.4 0.5 0.5 0.6 0.5 0.5 0.5 0.5 -2 0.4 ...
##  $ DCRNo       : int  10 10 10 10 10 10 10 10 10 10 ...
##  $ DCRNStand   : num  0.6 1 1 1 1 1 1 1 1 0.4 ...
##  $ DCRTotal    : int  20 20 20 20 20 20 20 20 19 20 ...
##  $ AnimRecall  : int  9 8 8 8 9 7 7 8 8 7 ...
##  $ update_stamp: chr  &quot;2025-01-24 08:16:43.0&quot; &quot;2025-01-24 08:16:43.0&quot; &quot;2025-01-24 08:16:43.0&quot; &quot;2025-01-24 08:16:43.0&quot; ...</code></pre>
</details>
<pre class="r"><code>## convert &quot;&quot; to NA for column OverImp
CERAD$OverImp[CERAD$OverImp == &quot;&quot;] &lt;- NA

## drop update_stamp column
CERAD &lt;- subset(CERAD,select = -c(update_stamp)) ## 19 columns left</code></pre>
<pre class="r"><code>str(CERAD)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    372 obs. of  19 variables:
##  $ wrapnum   : chr  &quot;wrap0176&quot; &quot;wrap0346&quot; &quot;wrap0422&quot; &quot;wrap0042&quot; ...
##  $ WRAPNo    : chr  &quot;0176&quot; &quot;0346&quot; &quot;0422&quot; &quot;0042&quot; ...
##  $ OverImp   : chr  &quot;Normal&quot; &quot;Normal&quot; &quot;Normal&quot; &quot;Normal&quot; ...
##  $ MPI       : int  65 68 64 73 59 80 72 72 67 60 ...
##  $ IR1       : int  6 7 5 9 6 7 6 7 6 5 ...
##  $ IR2       : int  8 8 5 9 7 10 8 8 8 6 ...
##  $ IR3       : int  8 9 6 10 8 10 9 9 9 6 ...
##  $ IRTotal   : int  22 24 16 28 21 27 23 24 23 17 ...
##  $ IRStand   : num  0.6 0 -2.3 1.6 -0.9 0.9 -0.3 0 -0.3 -1 ...
##  $ DFREst    : num  5 7 4 6 5 8 7 5 6 4.5 ...
##  $ DFRAct    : int  8 8 7 9 3 10 10 8 7 7 ...
##  $ DFRDiff   : num  -3 -1 -3 -3 2 -2 -3 -3 -1 -2.5 ...
##  $ DFRStand  : num  0.9 -0.1 -0.9 0.8 -3.7 1.3 1.3 -0.1 -0.9 0.1 ...
##  $ DCRYes    : int  10 10 10 10 10 10 10 10 9 10 ...
##  $ DCRYStand : num  0.4 0.5 0.5 0.6 0.5 0.5 0.5 0.5 -2 0.4 ...
##  $ DCRNo     : int  10 10 10 10 10 10 10 10 10 10 ...
##  $ DCRNStand : num  0.6 1 1 1 1 1 1 1 1 0.4 ...
##  $ DCRTotal  : int  20 20 20 20 20 20 20 20 19 20 ...
##  $ AnimRecall: int  9 8 8 8 9 7 7 8 8 7 ...</code></pre>
</details>
<p><br> <br></p>
</div>
<div id="ces_d_scale" class="section level2">
<h2>CES_D_Scale</h2>
<p>Center for Epidemiologic Studies-Depression Scale</p>
<pre class="r"><code>## load data
CES_D_Scale &lt;- read.csv(paste0(wrap_directory, &quot;CES_D_Scale_05Jun2025.csv&quot;))
info(CES_D_Scale,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:6566, cols:25, inds:1590</code></pre>
<pre class="r"><code>str(CES_D_Scale)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    6566 obs. of  25 variables:
##  $ wrapnum     : chr  &quot;wrap0002&quot; &quot;wrap0068&quot; &quot;wrap0146&quot; &quot;wrap0148&quot; ...
##  $ WRAPNo      : chr  &quot;0002&quot; &quot;0068&quot; &quot;0146&quot; &quot;0148&quot; ...
##  $ VisNo       : int  5 2 4 5 5 4 6 4 7 4 ...
##  $ CESDBother  : int  0 0 0 0 0 1 1 0 0 0 ...
##  $ CESDEat     : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ CESDShake   : int  0 0 0 0 0 0 1 0 0 0 ...
##  $ CESDGood    : int  1 0 0 0 0 0 0 0 0 0 ...
##  $ CESDKeep    : int  0 0 0 0 0 0 1 0 0 0 ...
##  $ CESDDep     : int  1 0 0 0 0 0 1 0 0 1 ...
##  $ CESDEff     : int  0 0 0 0 0 0 1 0 0 0 ...
##  $ CESDFuture  : int  0 0 0 0 0 0 1 0 0 0 ...
##  $ CESDFail    : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ CESDFear    : int  0 0 0 0 0 0 1 0 0 0 ...
##  $ CESDRest    : int  0 0 0 0 0 1 1 2 0 1 ...
##  $ CESDHappy   : int  1 0 0 0 1 0 1 0 0 0 ...
##  $ CESDLessUs  : int  1 0 0 0 1 0 1 0 0 0 ...
##  $ CESDLone    : int  1 0 0 0 0 0 0 0 0 0 ...
##  $ CESDUnfriend: int  0 0 0 0 0 0 0 0 0 0 ...
##  $ CESDEnjoy   : int  1 0 0 0 1 0 1 0 0 0 ...
##  $ CESDCry     : int  0 0 0 0 0 1 0 0 0 0 ...
##  $ CESDSad     : int  0 2 0 0 0 0 0 1 0 0 ...
##  $ CESDDislike : int  1 0 0 0 0 0 0 0 0 0 ...
##  $ CESDNoGo    : int  0 0 0 0 0 0 0 0 0 1 ...
##  $ Typical3    : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ update_stamp: chr  &quot;2024-09-13 08:51:49.0&quot; &quot;2024-09-13 08:51:49.0&quot; &quot;2024-09-13 08:51:49.0&quot; &quot;2024-09-13 08:51:49.0&quot; ...</code></pre>
</details>
<pre class="r"><code>## check table frequency for all variables

## drop update_stamp column
CES_D_Scale &lt;- subset(CES_D_Scale,select = -c(update_stamp)) ## 24 columns left</code></pre>
<pre class="r"><code>str(CES_D_Scale)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    6566 obs. of  24 variables:
##  $ wrapnum     : chr  &quot;wrap0002&quot; &quot;wrap0068&quot; &quot;wrap0146&quot; &quot;wrap0148&quot; ...
##  $ WRAPNo      : chr  &quot;0002&quot; &quot;0068&quot; &quot;0146&quot; &quot;0148&quot; ...
##  $ VisNo       : int  5 2 4 5 5 4 6 4 7 4 ...
##  $ CESDBother  : int  0 0 0 0 0 1 1 0 0 0 ...
##  $ CESDEat     : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ CESDShake   : int  0 0 0 0 0 0 1 0 0 0 ...
##  $ CESDGood    : int  1 0 0 0 0 0 0 0 0 0 ...
##  $ CESDKeep    : int  0 0 0 0 0 0 1 0 0 0 ...
##  $ CESDDep     : int  1 0 0 0 0 0 1 0 0 1 ...
##  $ CESDEff     : int  0 0 0 0 0 0 1 0 0 0 ...
##  $ CESDFuture  : int  0 0 0 0 0 0 1 0 0 0 ...
##  $ CESDFail    : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ CESDFear    : int  0 0 0 0 0 0 1 0 0 0 ...
##  $ CESDRest    : int  0 0 0 0 0 1 1 2 0 1 ...
##  $ CESDHappy   : int  1 0 0 0 1 0 1 0 0 0 ...
##  $ CESDLessUs  : int  1 0 0 0 1 0 1 0 0 0 ...
##  $ CESDLone    : int  1 0 0 0 0 0 0 0 0 0 ...
##  $ CESDUnfriend: int  0 0 0 0 0 0 0 0 0 0 ...
##  $ CESDEnjoy   : int  1 0 0 0 1 0 1 0 0 0 ...
##  $ CESDCry     : int  0 0 0 0 0 1 0 0 0 0 ...
##  $ CESDSad     : int  0 2 0 0 0 0 0 1 0 0 ...
##  $ CESDDislike : int  1 0 0 0 0 0 0 0 0 0 ...
##  $ CESDNoGo    : int  0 0 0 0 0 0 0 0 0 1 ...
##  $ Typical3    : int  1 1 1 1 1 1 1 1 1 1 ...</code></pre>
</details>
<p><br> <br></p>
</div>
<div id="champs" class="section level2">
<h2>CHAMPS</h2>
<p>Community Healthy Activities Model Program for Seniors (CHAMPS)</p>
<pre class="r"><code>## load data
CHAMPS &lt;- read.csv(paste0(wrap_directory, &quot;CHAMPS_05Jun2025.csv&quot;))
info(CHAMPS,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:1888, cols:127, inds:1181</code></pre>
<pre class="r"><code>str(CHAMPS, max.level = 99, list.len = 99999)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    1888 obs. of  127 variables:
##  $ wrapnum                : chr  &quot;wrap0002&quot; &quot;wrap0148&quot; &quot;wrap0305&quot; &quot;wrap0401&quot; ...
##  $ WRAPNo                 : chr  &quot;0002&quot; &quot;0148&quot; &quot;0305&quot; &quot;0401&quot; ...
##  $ VisNo                  : int  5 5 5 6 6 5 3 4 5 5 ...
##  $ CHAMPS_FrndFam         : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ CHAMPS_FrndFam_Fq      : int  12 7 2 5 15 2 5 1 5 3 ...
##  $ CHAMPS_FrndFam_HW      : int  5 3 2 6 4 4 2 1 5 2 ...
##  $ CHAMPS_SnrCtr          : int  2 2 2 2 2 1 1 1 2 2 ...
##  $ CHAMPS_SnrCtr_Fq       : int  NA NA NA NA NA 1 4 3 NA NA ...
##  $ CHAMPS_SnrCtr_HW       : int  NA NA NA NA NA 2 4 5 NA NA ...
##  $ CHAMPS_Volun           : int  1 2 1 2 1 2 1 1 1 2 ...
##  $ CHAMPS_Volun_Fq        : int  2 NA 1 NA 3 NA 4 5 NA NA ...
##  $ CHAMPS_Volun_HW        : int  4 NA 2 NA 6 NA 3 5 3 NA ...
##  $ CHAMPS_Church          : int  2 1 1 1 1 2 2 2 1 1 ...
##  $ CHAMPS_Church_Fq       : int  NA 1 1 1 2 NA NA NA 1 2 ...
##  $ CHAMPS_Church_HW       : int  NA 2 3 2 3 NA NA NA 2 2 ...
##  $ CHAMPS_ClubMtg         : int  1 2 2 2 1 1 2 2 1 2 ...
##  $ CHAMPS_ClubMtg_Fq      : int  0 NA NA NA 1 1 NA NA 2 NA ...
##  $ CHAMPS_ClubMtg_HW      : int  4 NA NA NA 2 2 NA NA 4 NA ...
##  $ CHAMPS_UseComp         : int  1 1 1 1 1 1 2 2 1 1 ...
##  $ CHAMPS_UseComp_Fq      : int  7 5 7 5 25 7 NA NA 5 28 ...
##  $ CHAMPS_UseComp_HW      : int  6 5 6 6 6 6 NA NA 2 1 ...
##  $ CHAMPS_Dance           : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ CHAMPS_Dance_Fq        : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_Dance_HW        : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_ArtCraft        : int  1 1 2 2 1 1 2 2 2 2 ...
##  $ CHAMPS_ArtCraft_Fq     : int  3 3 NA NA 20 2 NA NA NA NA ...
##  $ CHAMPS_ArtCraft_HW     : int  4 2 NA NA 3 1 NA NA NA NA ...
##  $ CHAMPS_GolfCrryEqp     : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ CHAMPS_GolfCrryEqp_Fq  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_GolfCrryEqp_HW  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_GolfRde         : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ CHAMPS_GolfRde_Fq      : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_GolfRde_HW      : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_Concert         : int  1 2 1 1 2 1 2 2 1 2 ...
##  $ CHAMPS_Concert_Fq      : int  1 NA 1 1 NA 1 NA NA 2 NA ...
##  $ CHAMPS_Concert_HW      : int  2 NA 2 2 NA 2 NA NA 5 NA ...
##  $ CHAMPS_Games           : int  2 2 1 1 1 2 1 1 1 2 ...
##  $ CHAMPS_Games_Fq        : int  NA NA 0 1 3 NA 4 3 2 NA ...
##  $ CHAMPS_Games_HW        : int  NA NA 1 2 2 NA 2 3 3 NA ...
##  $ CHAMPS_Pool            : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ CHAMPS_Pool_Fq         : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_Pool_HW         : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_TennisSng       : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ CHAMPS_TennisSng_Fq    : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_TennisSng_HW    : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_TennisDbl       : int  2 2 2 2 2 2 2 2 2 NA ...
##  $ CHAMPS_TennisDbl_Fq    : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_TennisDbl_HW    : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_Skate           : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ CHAMPS_Skate_Fq        : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_Skate_HW        : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_PlyMuscIns      : int  2 2 1 2 1 1 2 2 2 2 ...
##  $ CHAMPS_PlyMuscIns_Fq   : int  NA NA 5 NA 20 7 NA NA NA NA ...
##  $ CHAMPS_PlyMuscIns_HW   : int  NA NA 2 NA 4 3 NA NA NA NA ...
##  $ CHAMPS_Read            : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ CHAMPS_Read_Fq         : int  7 6 7 4 20 7 4 7 4 NA ...
##  $ CHAMPS_Read_HW         : int  3 2 3 2 3 3 1 3 4 6 ...
##  $ CHAMPS_HseHvyWrk       : int  1 2 1 2 1 1 2 2 2 1 ...
##  $ CHAMPS_HseHvyWrk_Fq    : int  1 NA 1 NA 2 1 NA NA NA 1 ...
##  $ CHAMPS_HseHvyWrk_HW    : int  2 NA 1 NA 1 2 NA NA NA 1 ...
##  $ CHAMPS_HseLtWrk        : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ CHAMPS_HseLtWrk_Fq     : int  1 2 3 1 2 2 1 3 1 6 ...
##  $ CHAMPS_HseLtWrk_HW     : int  1 2 2 1 2 3 1 1 2 1 ...
##  $ CHAMPS_GrdnHvyWrk      : int  1 2 2 2 2 1 2 2 1 2 ...
##  $ CHAMPS_GrdnHvyWrk_Fq   : int  2 NA NA NA NA 0 NA NA 1 NA ...
##  $ CHAMPS_GrdnHvyWrk_HW   : int  2 NA NA NA NA 1 NA NA 2 NA ...
##  $ CHAMPS_GrdnLtWrk       : int  1 2 2 2 1 1 2 2 1 1 ...
##  $ CHAMPS_GrdnLtWrk_Fq    : int  4 NA NA NA 4 1 NA NA 1 2 ...
##  $ CHAMPS_GrdnLtWrk_HW    : int  4 NA NA NA 2 2 NA NA 1 1 ...
##  $ CHAMPS_WrkOnMchn       : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ CHAMPS_WrkOnMchn_Fq    : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_WrkOnMchn_HW    : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_Run             : int  2 2 2 1 2 2 2 2 2 2 ...
##  $ CHAMPS_Run_Fq          : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_Run_HW          : int  NA NA NA 2 NA NA NA NA NA NA ...
##  $ CHAMPS_WalkUphll       : int  2 2 2 1 2 1 1 2 2 2 ...
##  $ CHAMPS_WalkUphll_Fq    : int  NA NA NA 3 NA 0 1 NA NA NA ...
##  $ CHAMPS_WalkUphll_HW    : int  NA NA NA 1 NA 2 1 NA NA NA ...
##  $ CHAMPS_WalkFast        : int  2 2 1 1 1 1 2 2 1 1 ...
##  $ CHAMPS_WalkFast_Fq     : int  NA NA 10 3 3 2 NA NA 6 2 ...
##  $ CHAMPS_WalkFast_HW     : int  NA NA 2 3 3 1 NA NA 5 1 ...
##  $ CHAMPS_WalkErrnd       : int  1 2 1 1 2 1 1 1 1 1 ...
##  $ CHAMPS_WalkErrnd_Fq    : int  3 NA 3 4 NA 1 2 2 2 1 ...
##  $ CHAMPS_WalkErrnd_HW    : int  2 NA 1 2 NA 2 1 1 3 1 ...
##  $ CHAMPS_WalkLeisure     : int  2 1 1 1 1 1 2 1 2 1 ...
##  $ CHAMPS_WalkLeisure_Fq  : int  NA 2 2 4 2 3 NA 2 NA 5 ...
##  $ CHAMPS_WalkLeisure_HW  : int  NA 1 2 3 1 1 NA 1 NA 1 ...
##  $ CHAMPS_RideBike        : int  2 2 2 1 2 2 2 2 2 1 ...
##  $ CHAMPS_RideBike_Fq     : int  NA NA NA 4 NA NA NA NA NA 5 ...
##  $ CHAMPS_RideBike_HW     : int  NA NA NA 4 NA NA NA NA NA 1 ...
##  $ CHAMPS_OthAerobic      : int  2 2 2 1 2 2 2 2 2 2 ...
##  $ CHAMPS_OthAerobic_Fq   : int  NA NA NA 3 NA NA NA NA NA NA ...
##  $ CHAMPS_OthAerobic_HW   : int  NA NA NA 2 NA NA NA NA NA NA ...
##  $ CHAMPS_WaterEx         : int  1 2 2 2 2 2 2 2 2 2 ...
##  $ CHAMPS_WaterEx_Fq      : int  3 NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_WaterEx_HW      : int  3 NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_SwimModFast     : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ CHAMPS_SwimModFast_Fq  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_SwimModFast_HW  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_SwimGent        : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ CHAMPS_SwimGent_Fq     : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_SwimGent_HW     : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_Stretch         : int  2 2 2 1 2 1 2 2 1 1 ...
##  $ CHAMPS_Stretch_Fq      : int  NA NA NA 4 NA 6 NA NA 7 7 ...
##  $ CHAMPS_Stretch_HW      : int  NA NA NA 1 NA 2 NA NA 4 1 ...
##  $ CHAMPS_Yoga            : int  1 2 2 2 2 2 2 2 2 1 ...
##  $ CHAMPS_Yoga_Fq         : int  1 NA NA NA NA NA NA NA NA 7 ...
##  $ CHAMPS_Yoga_HW         : int  2 NA NA NA NA NA NA NA NA 1 ...
##  $ CHAMPS_Aerobics        : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ CHAMPS_Aerobics_Fq     : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_Aerobics_HW     : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_StrngTrnModHv   : int  2 2 2 1 1 2 2 2 2 2 ...
##  $ CHAMPS_StrngTrnModHv_Fq: int  NA NA NA 2 3 NA NA NA NA NA ...
##  $ CHAMPS_StrngTrnModHv_HW: int  NA NA NA 1 2 NA NA NA NA NA ...
##  $ CHAMPS_StrngTrnLt      : int  2 2 2 2 2 1 2 2 1 1 ...
##  $ CHAMPS_StrngTrnLt_Fq   : int  NA NA NA NA NA 6 NA NA 2 4 ...
##  $ CHAMPS_StrngTrnLt_HW   : int  NA NA NA NA NA 2 NA NA 2 1 ...
##  $ CHAMPS_GenConditn      : int  2 2 2 2 2 1 2 1 2 1 ...
##  $ CHAMPS_GenConditn_Fq   : int  NA NA NA NA NA 3 NA 3 NA 5 ...
##  $ CHAMPS_GenConditn_HW   : int  NA NA NA NA NA 1 NA 1 NA 1 ...
##  $ CHAMPS_PlyBBScRB       : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ CHAMPS_PlyBBScRB_Fq    : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_PlyBBScRB_HW    : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_OthPhysAct      : int  2 2 2 2 2 2 2 2 2 1 ...
##  $ CHAMPS_OthPhysAct_Fq   : int  NA NA NA NA NA NA NA NA NA 7 ...
##  $ CHAMPS_OthPhysAct_HW   : int  NA NA NA NA NA NA NA NA NA 2 ...
##  $ update_stamp           : chr  &quot;2024-09-13 08:51:52.0&quot; &quot;2024-09-13 08:51:52.0&quot; &quot;2024-09-13 08:51:52.0&quot; &quot;2024-09-13 08:51:52.0&quot; ...</code></pre>
</details>
<pre class="r"><code>## check table frequency for all variables

## drop update_stamp column
CHAMPS &lt;- subset(CHAMPS,select = -c(update_stamp)) ## 126 columns left</code></pre>
<pre class="r"><code>str(CHAMPS, max.level = 99, list.len = 99999)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    1888 obs. of  126 variables:
##  $ wrapnum                : chr  &quot;wrap0002&quot; &quot;wrap0148&quot; &quot;wrap0305&quot; &quot;wrap0401&quot; ...
##  $ WRAPNo                 : chr  &quot;0002&quot; &quot;0148&quot; &quot;0305&quot; &quot;0401&quot; ...
##  $ VisNo                  : int  5 5 5 6 6 5 3 4 5 5 ...
##  $ CHAMPS_FrndFam         : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ CHAMPS_FrndFam_Fq      : int  12 7 2 5 15 2 5 1 5 3 ...
##  $ CHAMPS_FrndFam_HW      : int  5 3 2 6 4 4 2 1 5 2 ...
##  $ CHAMPS_SnrCtr          : int  2 2 2 2 2 1 1 1 2 2 ...
##  $ CHAMPS_SnrCtr_Fq       : int  NA NA NA NA NA 1 4 3 NA NA ...
##  $ CHAMPS_SnrCtr_HW       : int  NA NA NA NA NA 2 4 5 NA NA ...
##  $ CHAMPS_Volun           : int  1 2 1 2 1 2 1 1 1 2 ...
##  $ CHAMPS_Volun_Fq        : int  2 NA 1 NA 3 NA 4 5 NA NA ...
##  $ CHAMPS_Volun_HW        : int  4 NA 2 NA 6 NA 3 5 3 NA ...
##  $ CHAMPS_Church          : int  2 1 1 1 1 2 2 2 1 1 ...
##  $ CHAMPS_Church_Fq       : int  NA 1 1 1 2 NA NA NA 1 2 ...
##  $ CHAMPS_Church_HW       : int  NA 2 3 2 3 NA NA NA 2 2 ...
##  $ CHAMPS_ClubMtg         : int  1 2 2 2 1 1 2 2 1 2 ...
##  $ CHAMPS_ClubMtg_Fq      : int  0 NA NA NA 1 1 NA NA 2 NA ...
##  $ CHAMPS_ClubMtg_HW      : int  4 NA NA NA 2 2 NA NA 4 NA ...
##  $ CHAMPS_UseComp         : int  1 1 1 1 1 1 2 2 1 1 ...
##  $ CHAMPS_UseComp_Fq      : int  7 5 7 5 25 7 NA NA 5 28 ...
##  $ CHAMPS_UseComp_HW      : int  6 5 6 6 6 6 NA NA 2 1 ...
##  $ CHAMPS_Dance           : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ CHAMPS_Dance_Fq        : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_Dance_HW        : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_ArtCraft        : int  1 1 2 2 1 1 2 2 2 2 ...
##  $ CHAMPS_ArtCraft_Fq     : int  3 3 NA NA 20 2 NA NA NA NA ...
##  $ CHAMPS_ArtCraft_HW     : int  4 2 NA NA 3 1 NA NA NA NA ...
##  $ CHAMPS_GolfCrryEqp     : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ CHAMPS_GolfCrryEqp_Fq  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_GolfCrryEqp_HW  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_GolfRde         : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ CHAMPS_GolfRde_Fq      : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_GolfRde_HW      : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_Concert         : int  1 2 1 1 2 1 2 2 1 2 ...
##  $ CHAMPS_Concert_Fq      : int  1 NA 1 1 NA 1 NA NA 2 NA ...
##  $ CHAMPS_Concert_HW      : int  2 NA 2 2 NA 2 NA NA 5 NA ...
##  $ CHAMPS_Games           : int  2 2 1 1 1 2 1 1 1 2 ...
##  $ CHAMPS_Games_Fq        : int  NA NA 0 1 3 NA 4 3 2 NA ...
##  $ CHAMPS_Games_HW        : int  NA NA 1 2 2 NA 2 3 3 NA ...
##  $ CHAMPS_Pool            : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ CHAMPS_Pool_Fq         : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_Pool_HW         : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_TennisSng       : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ CHAMPS_TennisSng_Fq    : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_TennisSng_HW    : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_TennisDbl       : int  2 2 2 2 2 2 2 2 2 NA ...
##  $ CHAMPS_TennisDbl_Fq    : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_TennisDbl_HW    : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_Skate           : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ CHAMPS_Skate_Fq        : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_Skate_HW        : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_PlyMuscIns      : int  2 2 1 2 1 1 2 2 2 2 ...
##  $ CHAMPS_PlyMuscIns_Fq   : int  NA NA 5 NA 20 7 NA NA NA NA ...
##  $ CHAMPS_PlyMuscIns_HW   : int  NA NA 2 NA 4 3 NA NA NA NA ...
##  $ CHAMPS_Read            : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ CHAMPS_Read_Fq         : int  7 6 7 4 20 7 4 7 4 NA ...
##  $ CHAMPS_Read_HW         : int  3 2 3 2 3 3 1 3 4 6 ...
##  $ CHAMPS_HseHvyWrk       : int  1 2 1 2 1 1 2 2 2 1 ...
##  $ CHAMPS_HseHvyWrk_Fq    : int  1 NA 1 NA 2 1 NA NA NA 1 ...
##  $ CHAMPS_HseHvyWrk_HW    : int  2 NA 1 NA 1 2 NA NA NA 1 ...
##  $ CHAMPS_HseLtWrk        : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ CHAMPS_HseLtWrk_Fq     : int  1 2 3 1 2 2 1 3 1 6 ...
##  $ CHAMPS_HseLtWrk_HW     : int  1 2 2 1 2 3 1 1 2 1 ...
##  $ CHAMPS_GrdnHvyWrk      : int  1 2 2 2 2 1 2 2 1 2 ...
##  $ CHAMPS_GrdnHvyWrk_Fq   : int  2 NA NA NA NA 0 NA NA 1 NA ...
##  $ CHAMPS_GrdnHvyWrk_HW   : int  2 NA NA NA NA 1 NA NA 2 NA ...
##  $ CHAMPS_GrdnLtWrk       : int  1 2 2 2 1 1 2 2 1 1 ...
##  $ CHAMPS_GrdnLtWrk_Fq    : int  4 NA NA NA 4 1 NA NA 1 2 ...
##  $ CHAMPS_GrdnLtWrk_HW    : int  4 NA NA NA 2 2 NA NA 1 1 ...
##  $ CHAMPS_WrkOnMchn       : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ CHAMPS_WrkOnMchn_Fq    : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_WrkOnMchn_HW    : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_Run             : int  2 2 2 1 2 2 2 2 2 2 ...
##  $ CHAMPS_Run_Fq          : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_Run_HW          : int  NA NA NA 2 NA NA NA NA NA NA ...
##  $ CHAMPS_WalkUphll       : int  2 2 2 1 2 1 1 2 2 2 ...
##  $ CHAMPS_WalkUphll_Fq    : int  NA NA NA 3 NA 0 1 NA NA NA ...
##  $ CHAMPS_WalkUphll_HW    : int  NA NA NA 1 NA 2 1 NA NA NA ...
##  $ CHAMPS_WalkFast        : int  2 2 1 1 1 1 2 2 1 1 ...
##  $ CHAMPS_WalkFast_Fq     : int  NA NA 10 3 3 2 NA NA 6 2 ...
##  $ CHAMPS_WalkFast_HW     : int  NA NA 2 3 3 1 NA NA 5 1 ...
##  $ CHAMPS_WalkErrnd       : int  1 2 1 1 2 1 1 1 1 1 ...
##  $ CHAMPS_WalkErrnd_Fq    : int  3 NA 3 4 NA 1 2 2 2 1 ...
##  $ CHAMPS_WalkErrnd_HW    : int  2 NA 1 2 NA 2 1 1 3 1 ...
##  $ CHAMPS_WalkLeisure     : int  2 1 1 1 1 1 2 1 2 1 ...
##  $ CHAMPS_WalkLeisure_Fq  : int  NA 2 2 4 2 3 NA 2 NA 5 ...
##  $ CHAMPS_WalkLeisure_HW  : int  NA 1 2 3 1 1 NA 1 NA 1 ...
##  $ CHAMPS_RideBike        : int  2 2 2 1 2 2 2 2 2 1 ...
##  $ CHAMPS_RideBike_Fq     : int  NA NA NA 4 NA NA NA NA NA 5 ...
##  $ CHAMPS_RideBike_HW     : int  NA NA NA 4 NA NA NA NA NA 1 ...
##  $ CHAMPS_OthAerobic      : int  2 2 2 1 2 2 2 2 2 2 ...
##  $ CHAMPS_OthAerobic_Fq   : int  NA NA NA 3 NA NA NA NA NA NA ...
##  $ CHAMPS_OthAerobic_HW   : int  NA NA NA 2 NA NA NA NA NA NA ...
##  $ CHAMPS_WaterEx         : int  1 2 2 2 2 2 2 2 2 2 ...
##  $ CHAMPS_WaterEx_Fq      : int  3 NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_WaterEx_HW      : int  3 NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_SwimModFast     : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ CHAMPS_SwimModFast_Fq  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_SwimModFast_HW  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_SwimGent        : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ CHAMPS_SwimGent_Fq     : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_SwimGent_HW     : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_Stretch         : int  2 2 2 1 2 1 2 2 1 1 ...
##  $ CHAMPS_Stretch_Fq      : int  NA NA NA 4 NA 6 NA NA 7 7 ...
##  $ CHAMPS_Stretch_HW      : int  NA NA NA 1 NA 2 NA NA 4 1 ...
##  $ CHAMPS_Yoga            : int  1 2 2 2 2 2 2 2 2 1 ...
##  $ CHAMPS_Yoga_Fq         : int  1 NA NA NA NA NA NA NA NA 7 ...
##  $ CHAMPS_Yoga_HW         : int  2 NA NA NA NA NA NA NA NA 1 ...
##  $ CHAMPS_Aerobics        : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ CHAMPS_Aerobics_Fq     : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_Aerobics_HW     : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_StrngTrnModHv   : int  2 2 2 1 1 2 2 2 2 2 ...
##  $ CHAMPS_StrngTrnModHv_Fq: int  NA NA NA 2 3 NA NA NA NA NA ...
##  $ CHAMPS_StrngTrnModHv_HW: int  NA NA NA 1 2 NA NA NA NA NA ...
##  $ CHAMPS_StrngTrnLt      : int  2 2 2 2 2 1 2 2 1 1 ...
##  $ CHAMPS_StrngTrnLt_Fq   : int  NA NA NA NA NA 6 NA NA 2 4 ...
##  $ CHAMPS_StrngTrnLt_HW   : int  NA NA NA NA NA 2 NA NA 2 1 ...
##  $ CHAMPS_GenConditn      : int  2 2 2 2 2 1 2 1 2 1 ...
##  $ CHAMPS_GenConditn_Fq   : int  NA NA NA NA NA 3 NA 3 NA 5 ...
##  $ CHAMPS_GenConditn_HW   : int  NA NA NA NA NA 1 NA 1 NA 1 ...
##  $ CHAMPS_PlyBBScRB       : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ CHAMPS_PlyBBScRB_Fq    : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_PlyBBScRB_HW    : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CHAMPS_OthPhysAct      : int  2 2 2 2 2 2 2 2 2 1 ...
##  $ CHAMPS_OthPhysAct_Fq   : int  NA NA NA NA NA NA NA NA NA 7 ...
##  $ CHAMPS_OthPhysAct_HW   : int  NA NA NA NA NA NA NA NA NA 2 ...</code></pre>
</details>
<p><br> <br></p>
</div>
<div id="cogstatedata" class="section level2">
<h2>CogStateData</h2>
<p>Cogstate Testing Data</p>
<pre class="r"><code>## load data
CogStateData &lt;- read.csv(paste0(wrap_directory, &quot;CogStateData_05Jun2025.csv&quot;))
info(CogStateData,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:2283, cols:228, inds:1224</code></pre>
<pre class="r"><code>str(CogStateData, max.level = 99, list.len = 99999)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    2283 obs. of  228 variables:
##  $ wrapnum         : chr  &quot;wrap0002&quot; &quot;wrap0148&quot; &quot;wrap0305&quot; &quot;wrap0401&quot; ...
##  $ WRAPNo          : chr  &quot;0002&quot; &quot;0148&quot; &quot;0305&quot; &quot;0401&quot; ...
##  $ VisNo           : int  5 5 5 6 6 5 1 3 4 1 ...
##  $ Hand            : chr  &quot;Right&quot; &quot;Right&quot; &quot;Right&quot; &quot;&quot; ...
##  $ Sex             : chr  &quot;Female&quot; &quot;Female&quot; &quot;Female&quot; &quot;&quot; ...
##  $ CPAL_dur        : int  360688 370426 531773 NA 343473 232056 483750 NA NA NA ...
##  $ CPAL_lmn        : num  3.51 3.48 3.61 NA 3.43 ...
##  $ CPAL_lsd        : num  0.299 0.375 0.44 NA 0.301 ...
##  $ CPAL_acc        : num  0.572 0.647 0.772 NA 0.857 ...
##  $ CPAL_cor        : int  56 56 56 NA 56 56 56 NA NA NA ...
##  $ CPAL_err        : int  135 98 59 NA 42 2 74 NA NA NA ...
##  $ CPAL_presnt     : int  191 154 115 NA 98 58 130 NA NA NA ...
##  $ CPAL_sti        : int  56 56 56 NA 56 56 56 NA NA NA ...
##  $ CPAL_res        : int  191 154 115 NA 98 58 130 NA NA NA ...
##  $ CPAL_Completion : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ CPAL_Integrity  : logi  NA NA NA NA NA NA ...
##  $ CPAL1_dur       : int  34209 33274 45506 NA 30696 34671 35425 NA NA NA ...
##  $ CPAL1_lmn       : num  3.35 3.27 3.53 NA 3.27 ...
##  $ CPAL1_lsd       : num  0.1367 0.2606 0.2196 NA 0.0963 ...
##  $ CPAL1_acc       : num  1.57 1.02 1.23 NA 1.57 ...
##  $ CPAL1_cor       : int  8 8 8 NA 8 8 8 NA NA NA ...
##  $ CPAL1_err       : int  0 3 1 NA 0 0 0 NA NA NA ...
##  $ CPAL1_presnt    : int  8 11 9 NA 8 8 8 NA NA NA ...
##  $ CPAL1_sti       : int  8 8 8 NA 8 8 8 NA NA NA ...
##  $ CPAL1_res       : int  8 11 9 NA 8 8 8 NA NA NA ...
##  $ CPAL1_Completion: logi  NA NA NA NA NA NA ...
##  $ CPAL1_Integrity : logi  NA NA NA NA NA NA ...
##  $ CPAL2_dur       : int  47155 72612 115482 NA 71387 32087 75817 NA NA NA ...
##  $ CPAL2_lmn       : num  3.52 3.67 3.91 NA 3.75 ...
##  $ CPAL2_lsd       : num  0.259 0.467 0.48 NA 0.335 ...
##  $ CPAL2_acc       : num  0.543 0.553 0.498 NA 0.615 ...
##  $ CPAL2_cor       : int  8 8 8 NA 8 8 8 NA NA NA ...
##  $ CPAL2_err       : int  22 21 27 NA 16 2 13 NA NA NA ...
##  $ CPAL2_presnt    : int  30 29 35 NA 24 10 21 NA NA NA ...
##  $ CPAL2_sti       : int  8 8 8 NA 8 8 8 NA NA NA ...
##  $ CPAL2_res       : int  30 29 35 NA 24 10 21 NA NA NA ...
##  $ CPAL2_Completion: logi  NA NA NA NA NA NA ...
##  $ CPAL2_Integrity : logi  NA NA NA NA NA NA ...
##  $ CPAL3_dur       : int  39352 61915 86112 NA 52415 26305 76206 NA NA NA ...
##  $ CPAL3_lmn       : num  3.37 3.6 3.67 NA 3.52 ...
##  $ CPAL3_lsd       : num  0.317 0.438 0.51 NA 0.343 ...
##  $ CPAL3_acc       : num  0.588 0.553 0.665 NA 0.706 ...
##  $ CPAL3_cor       : int  8 8 8 NA 8 8 8 NA NA NA ...
##  $ CPAL3_err       : int  18 21 13 NA 11 0 18 NA NA NA ...
##  $ CPAL3_presnt    : int  26 29 21 NA 19 8 26 NA NA NA ...
##  $ CPAL3_sti       : int  8 8 8 NA 8 8 8 NA NA NA ...
##  $ CPAL3_res       : int  26 29 21 NA 19 8 26 NA NA NA ...
##  $ CPAL3_Completion: logi  NA NA NA NA NA NA ...
##  $ CPAL3_Integrity : logi  NA NA NA NA NA NA ...
##  $ CPAL4_dur       : int  40612 38835 71408 NA 49569 26061 69068 NA NA NA ...
##  $ CPAL4_lmn       : num  3.45 3.34 3.64 NA 3.5 ...
##  $ CPAL4_lsd       : num  0.227 0.334 0.444 NA 0.356 ...
##  $ CPAL4_acc       : num  0.601 0.73 0.756 NA 0.785 ...
##  $ CPAL4_cor       : int  8 8 8 NA 8 8 8 NA NA NA ...
##  $ CPAL4_err       : int  17 10 9 NA 8 0 9 NA NA NA ...
##  $ CPAL4_presnt    : int  25 18 17 NA 16 8 17 NA NA NA ...
##  $ CPAL4_sti       : int  8 8 8 NA 8 8 8 NA NA NA ...
##  $ CPAL4_res       : int  25 18 17 NA 16 8 17 NA NA NA ...
##  $ CPAL4_Completion: logi  NA NA NA NA NA NA ...
##  $ CPAL4_Integrity : logi  NA NA NA NA NA NA ...
##  $ CPAL5_dur       : int  54466 48133 51421 NA 34614 25035 81750 NA NA NA ...
##  $ CPAL5_lmn       : num  3.6 3.51 3.49 NA 3.31 ...
##  $ CPAL5_lsd       : num  0.315 0.333 0.362 NA 0.252 ...
##  $ CPAL5_acc       : num  0.484 0.631 1.021 NA 1.021 ...
##  $ CPAL5_cor       : int  8 8 8 NA 8 8 8 NA NA NA ...
##  $ CPAL5_err       : int  29 15 3 NA 3 0 18 NA NA NA ...
##  $ CPAL5_presnt    : int  37 23 11 NA 11 8 26 NA NA NA ...
##  $ CPAL5_sti       : int  8 8 8 NA 8 8 8 NA NA NA ...
##  $ CPAL5_res       : int  37 23 11 NA 11 8 26 NA NA NA ...
##  $ CPAL5_Completion: logi  NA NA NA NA NA NA ...
##  $ CPAL5_Integrity : logi  NA NA NA NA NA NA ...
##  $ CPAL6_dur       : int  58755 54312 73568 NA 31100 24328 64003 NA NA NA ...
##  $ CPAL6_lmn       : num  3.64 3.58 3.55 NA 3.27 ...
##  $ CPAL6_lsd       : num  0.335 0.336 0.518 NA 0.137 ...
##  $ CPAL6_acc       : num  0.498 0.588 0.857 NA 1.571 ...
##  $ CPAL6_cor       : int  8 8 8 NA 8 8 8 NA NA NA ...
##  $ CPAL6_err       : int  27 18 6 NA 0 0 8 NA NA NA ...
##  $ CPAL6_presnt    : int  35 26 14 NA 8 8 16 NA NA NA ...
##  $ CPAL6_sti       : int  8 8 8 NA 8 8 8 NA NA NA ...
##  $ CPAL6_res       : int  35 26 14 NA 8 8 16 NA NA NA ...
##  $ CPAL6_Completion: logi  NA NA NA NA NA NA ...
##  $ CPAL6_Integrity : logi  NA NA NA NA NA NA ...
##  $ CPAL7_dur       : int  63087 43512 60684 NA 38679 23943 55046 NA NA NA ...
##  $ CPAL7_lmn       : num  3.62 3.4 3.51 NA 3.4 ...
##  $ CPAL7_lsd       : num  0.396 0.372 0.486 NA 0.253 ...
##  $ CPAL7_acc       : num  0.543 0.73 1.571 NA 0.955 ...
##  $ CPAL7_cor       : int  8 8 8 NA 8 8 8 NA NA NA ...
##  $ CPAL7_err       : int  22 10 0 NA 4 0 8 NA NA NA ...
##  $ CPAL7_presnt    : int  30 18 8 NA 12 8 16 NA NA NA ...
##  $ CPAL7_sti       : int  8 8 8 NA 8 8 8 NA NA NA ...
##  $ CPAL7_res       : int  30 18 8 NA 12 8 16 NA NA NA ...
##  $ CPAL7_Completion: logi  NA NA NA NA NA NA ...
##  $ CPAL7_Integrity : logi  NA NA NA NA NA NA ...
##  $ DET_lmn         : num  NA NA NA NA NA NA NA NA NA NA ...
##  $ DET_lsd         : num  NA NA NA NA NA NA NA NA NA NA ...
##  $ DET_acc         : num  NA NA NA NA NA NA NA NA NA NA ...
##  $ DET_cor         : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ DET_err         : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ DET_presnt      : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ DET_sti         : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ DET_Completion  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ DET_Integrity   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ GMCT_GMLidx     : int  -1 -1 -1 NA -1 -1 -1 NA NA NA ...
##  $ GMCT_mps        : num  1.33 1.4 1.3 NA 1.1 ...
##  $ GMCT_dur        : int  30011 30005 30002 NA 30004 30013 30001 NA NA NA ...
##  $ GMCT_ter        : int  0 0 0 NA 0 0 3 NA NA NA ...
##  $ GMCT_ler        : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ GMCT_rer        : int  0 0 0 NA 0 0 3 NA NA NA ...
##  $ GMCT_per        : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ GMCT_cmv        : int  40 42 39 NA 33 35 31 NA NA NA ...
##  $ GMCT_rth        : int  0 0 0 NA 0 0 1 NA NA NA ...
##  $ GMCT_Completion : logi  NA NA NA NA NA NA ...
##  $ GMCT_Integrity  : logi  NA NA NA NA NA NA ...
##  $ GML_GMLidx      : int  5 17 11 NA 18 11 11 NA NA NA ...
##  $ GML_mps         : num  0.498 0.71 0.508 NA 0.552 ...
##  $ GML_dur         : int  280889 197144 275833 NA 253652 206585 303546 NA NA NA ...
##  $ GML_ter         : int  50 62 54 NA 34 42 67 NA NA NA ...
##  $ GML_ler         : int  49 56 42 NA 33 38 61 NA NA NA ...
##  $ GML_rer         : int  1 6 12 NA 1 4 6 NA NA NA ...
##  $ GML_per         : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ GML_cmv         : int  140 140 140 NA 140 140 140 NA NA NA ...
##  $ GML_rth         : int  50 59 49 NA 34 39 65 NA NA NA ...
##  $ GML_Completion  : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ GML_Integrity   : logi  NA NA NA NA NA NA ...
##  $ GML1_GMLidx     : int  5 17 11 NA 18 11 11 NA NA NA ...
##  $ GML1_mps        : num  0.475 0.627 0.514 NA 0.513 ...
##  $ GML1_dur        : int  58969 44641 54513 NA 54528 49329 57109 NA NA NA ...
##  $ GML1_ter        : int  18 17 19 NA 10 12 17 NA NA NA ...
##  $ GML1_ler        : int  18 13 17 NA 10 12 16 NA NA NA ...
##  $ GML1_rer        : int  0 4 2 NA 0 0 1 NA NA NA ...
##  $ GML1_per        : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ GML1_cmv        : int  28 28 28 NA 28 28 28 NA NA NA ...
##  $ GML1_rth        : int  18 15 18 NA 10 12 16 NA NA NA ...
##  $ GML1_Completion : logi  NA NA NA NA NA NA ...
##  $ GML1_Integrity  : logi  NA NA NA NA NA NA ...
##  $ GML2_GMLidx     : int  5 17 11 NA 18 11 11 NA NA NA ...
##  $ GML2_mps        : num  0.634 0.853 0.619 NA 0.549 ...
##  $ GML2_dur        : int  44184 32825 45265 NA 51030 37921 56949 NA NA NA ...
##  $ GML2_ter        : int  9 12 11 NA 9 9 16 NA NA NA ...
##  $ GML2_ler        : int  9 12 8 NA 8 9 15 NA NA NA ...
##  $ GML2_rer        : int  0 0 3 NA 1 0 1 NA NA NA ...
##  $ GML2_per        : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ GML2_cmv        : int  28 28 28 NA 28 28 28 NA NA NA ...
##  $ GML2_rth        : int  9 12 10 NA 9 9 16 NA NA NA ...
##  $ GML2_Completion : logi  NA NA NA NA NA NA ...
##  $ GML2_Integrity  : logi  NA NA NA NA NA NA ...
##  $ GML3_GMLidx     : int  5 17 11 NA 18 11 11 NA NA NA ...
##  $ GML3_mps        : num  0.693 0.724 0.581 NA 0.785 ...
##  $ GML3_dur        : int  40408 38680 48161 NA 35649 34233 59837 NA NA NA ...
##  $ GML3_ter        : int  10 15 11 NA 6 9 19 NA NA NA ...
##  $ GML3_ler        : int  10 13 7 NA 6 5 15 NA NA NA ...
##  $ GML3_rer        : int  0 2 4 NA 0 4 4 NA NA NA ...
##  $ GML3_per        : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ GML3_cmv        : int  28 28 28 NA 28 28 28 NA NA NA ...
##  $ GML3_rth        : int  11 14 8 NA 6 6 18 NA NA NA ...
##  $ GML3_Completion : logi  NA NA NA NA NA NA ...
##  $ GML3_Integrity  : logi  NA NA NA NA NA NA ...
##  $ GML4_GMLidx     : int  5 17 11 NA 18 11 11 NA NA NA ...
##  $ GML4_mps        : num  0.716 0.847 0.617 NA 0.822 ...
##  $ GML4_dur        : int  39097 33049 45368 NA 34054 28666 52822 NA NA NA ...
##  $ GML4_ter        : int  6 11 4 NA 5 5 9 NA NA NA ...
##  $ GML4_ler        : int  6 11 4 NA 5 5 9 NA NA NA ...
##  $ GML4_rer        : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ GML4_per        : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ GML4_cmv        : int  28 28 28 NA 28 28 28 NA NA NA ...
##  $ GML4_rth        : int  6 11 4 NA 5 5 9 NA NA NA ...
##  $ GML4_Completion : logi  NA NA NA NA NA NA ...
##  $ GML4_Integrity  : logi  NA NA NA NA NA NA ...
##  $ GML5_GMLidx     : int  5 17 11 NA 18 11 11 NA NA NA ...
##  $ GML5_mps        : num  0.713 1.134 0.456 NA 1.025 ...
##  $ GML5_dur        : int  39250 24688 61368 NA 27309 31745 44045 NA NA NA ...
##  $ GML5_ter        : int  7 7 9 NA 4 7 6 NA NA NA ...
##  $ GML5_ler        : int  6 7 6 NA 4 7 6 NA NA NA ...
##  $ GML5_rer        : int  1 0 3 NA 0 0 0 NA NA NA ...
##  $ GML5_per        : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ GML5_cmv        : int  28 28 28 NA 28 28 28 NA NA NA ...
##  $ GML5_rth        : int  6 7 9 NA 4 7 6 NA NA NA ...
##  $ GML5_Completion : logi  NA NA NA NA NA NA ...
##  $ GML5_Integrity  : logi  NA NA NA NA NA NA ...
##  $ GMR_GMLidx      : int  5 17 11 NA 18 11 11 NA NA NA ...
##  $ GMR_mps         : num  0.554 1.101 0.442 NA 0.851 ...
##  $ GMR_dur         : int  50497 25425 63305 NA 32916 25688 56223 NA NA NA ...
##  $ GMR_ter         : int  9 7 5 NA 7 2 10 NA NA NA ...
##  $ GMR_ler         : int  8 7 3 NA 7 2 9 NA NA NA ...
##  $ GMR_rer         : int  1 0 2 NA 0 0 1 NA NA NA ...
##  $ GMR_per         : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ GMR_cmv         : int  28 28 28 NA 28 28 28 NA NA NA ...
##  $ GMR_rth         : int  8 7 3 NA 7 2 9 NA NA NA ...
##  $ GMR_Completion  : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ GMR_Integrity   : logi  NA NA NA NA NA NA ...
##  $ IDN_lmn         : num  NA NA NA NA NA NA NA NA NA NA ...
##  $ IDN_lsd         : num  NA NA NA NA NA NA NA NA NA NA ...
##  $ IDN_acc         : num  NA NA NA NA NA NA NA NA NA NA ...
##  $ IDN_cor         : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ IDN_err         : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ IDN_presnt      : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ IDN_sti         : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ IDN_Completion  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ IDN_Integrity   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ OCL_lmn         : num  3.06 3.09 3.34 NA 3.04 ...
##  $ OCL_lsd         : num  0.127 0.146 0.268 NA 0.114 ...
##  $ OCL_acc         : num  0.975 0.984 0.909 NA 0.959 ...
##  $ OCL_cor         : int  61 61 56 NA 59 64 59 NA NA NA ...
##  $ OCL_err         : int  28 27 34 NA 29 32 31 NA NA NA ...
##  $ OCL_presnt      : int  89 88 90 NA 88 96 90 NA NA NA ...
##  $ OCL_sti         : int  88 88 88 NA 88 88 88 NA NA NA ...
##  $ OCL_Completion  : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ OCL_Integrity   : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ ONB_lmn         : num  2.89 2.96 3.09 NA 2.96 ...
##  $ ONB_lsd         : num  0.1555 0.0997 0.1382 NA 0.1353 ...
##  $ ONB_acc         : num  1.27 1.32 1.23 NA 1.27 ...
##  $ ONB_cor         : int  31 31 31 NA 31 31 31 NA NA NA ...
##  $ ONB_err         : int  3 2 4 NA 3 0 0 NA NA NA ...
##  $ ONB_presnt      : int  34 33 35 NA 34 31 31 NA NA NA ...
##  $ ONB_sti         : int  34 32 34 NA 34 31 31 NA NA NA ...
##  $ ONB_Completion  : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ ONB_Integrity   : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ TWOB_lmn        : num  2.98 3.07 3.06 NA 3.01 ...
##  $ TWOB_lsd        : num  0.145 0.143 0.197 NA 0.154 ...
##  $ TWOB_acc        : num  1.27 1.27 1.19 NA 1.23 ...
##  $ TWOB_cor        : int  32 32 32 NA 32 32 32 NA NA NA ...
##  $ TWOB_err        : int  3 3 5 NA 4 0 0 NA NA NA ...
##  $ TWOB_presnt     : int  35 35 37 NA 36 32 32 NA NA NA ...
##  $ TWOB_sti        : int  35 35 37 NA 36 32 32 NA NA NA ...
##  $ TWOB_Completion : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ TWOB_Integrity  : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ TestNotPerformed: int  NA NA NA 1 NA NA NA 1 1 1 ...
##  $ update_stamp    : chr  &quot;2025-01-24 08:16:58.0&quot; &quot;2025-01-24 08:16:58.0&quot; &quot;2025-01-24 08:16:58.0&quot; &quot;2024-09-13 08:52:04.0&quot; ...</code></pre>
</details>
<pre class="r"><code>## check table frequency for all variables

## convert logi columns to int
## these columns were confirmed to only have NA values in the raw data
logicols &lt;- colnames(CogStateData)[sapply(CogStateData, is.logical)]
CogStateData[logicols] &lt;- lapply(CogStateData[logicols], as.integer)

## drop update_stamp column
CogStateData &lt;- subset(CogStateData,select = -c(update_stamp)) ## 227 columns left</code></pre>
<pre class="r"><code>str(CogStateData, max.level = 99, list.len = 99999)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    2283 obs. of  227 variables:
##  $ wrapnum         : chr  &quot;wrap0002&quot; &quot;wrap0148&quot; &quot;wrap0305&quot; &quot;wrap0401&quot; ...
##  $ WRAPNo          : chr  &quot;0002&quot; &quot;0148&quot; &quot;0305&quot; &quot;0401&quot; ...
##  $ VisNo           : int  5 5 5 6 6 5 1 3 4 1 ...
##  $ Hand            : chr  &quot;Right&quot; &quot;Right&quot; &quot;Right&quot; &quot;&quot; ...
##  $ Sex             : chr  &quot;Female&quot; &quot;Female&quot; &quot;Female&quot; &quot;&quot; ...
##  $ CPAL_dur        : int  360688 370426 531773 NA 343473 232056 483750 NA NA NA ...
##  $ CPAL_lmn        : num  3.51 3.48 3.61 NA 3.43 ...
##  $ CPAL_lsd        : num  0.299 0.375 0.44 NA 0.301 ...
##  $ CPAL_acc        : num  0.572 0.647 0.772 NA 0.857 ...
##  $ CPAL_cor        : int  56 56 56 NA 56 56 56 NA NA NA ...
##  $ CPAL_err        : int  135 98 59 NA 42 2 74 NA NA NA ...
##  $ CPAL_presnt     : int  191 154 115 NA 98 58 130 NA NA NA ...
##  $ CPAL_sti        : int  56 56 56 NA 56 56 56 NA NA NA ...
##  $ CPAL_res        : int  191 154 115 NA 98 58 130 NA NA NA ...
##  $ CPAL_Completion : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ CPAL_Integrity  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CPAL1_dur       : int  34209 33274 45506 NA 30696 34671 35425 NA NA NA ...
##  $ CPAL1_lmn       : num  3.35 3.27 3.53 NA 3.27 ...
##  $ CPAL1_lsd       : num  0.1367 0.2606 0.2196 NA 0.0963 ...
##  $ CPAL1_acc       : num  1.57 1.02 1.23 NA 1.57 ...
##  $ CPAL1_cor       : int  8 8 8 NA 8 8 8 NA NA NA ...
##  $ CPAL1_err       : int  0 3 1 NA 0 0 0 NA NA NA ...
##  $ CPAL1_presnt    : int  8 11 9 NA 8 8 8 NA NA NA ...
##  $ CPAL1_sti       : int  8 8 8 NA 8 8 8 NA NA NA ...
##  $ CPAL1_res       : int  8 11 9 NA 8 8 8 NA NA NA ...
##  $ CPAL1_Completion: int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CPAL1_Integrity : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CPAL2_dur       : int  47155 72612 115482 NA 71387 32087 75817 NA NA NA ...
##  $ CPAL2_lmn       : num  3.52 3.67 3.91 NA 3.75 ...
##  $ CPAL2_lsd       : num  0.259 0.467 0.48 NA 0.335 ...
##  $ CPAL2_acc       : num  0.543 0.553 0.498 NA 0.615 ...
##  $ CPAL2_cor       : int  8 8 8 NA 8 8 8 NA NA NA ...
##  $ CPAL2_err       : int  22 21 27 NA 16 2 13 NA NA NA ...
##  $ CPAL2_presnt    : int  30 29 35 NA 24 10 21 NA NA NA ...
##  $ CPAL2_sti       : int  8 8 8 NA 8 8 8 NA NA NA ...
##  $ CPAL2_res       : int  30 29 35 NA 24 10 21 NA NA NA ...
##  $ CPAL2_Completion: int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CPAL2_Integrity : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CPAL3_dur       : int  39352 61915 86112 NA 52415 26305 76206 NA NA NA ...
##  $ CPAL3_lmn       : num  3.37 3.6 3.67 NA 3.52 ...
##  $ CPAL3_lsd       : num  0.317 0.438 0.51 NA 0.343 ...
##  $ CPAL3_acc       : num  0.588 0.553 0.665 NA 0.706 ...
##  $ CPAL3_cor       : int  8 8 8 NA 8 8 8 NA NA NA ...
##  $ CPAL3_err       : int  18 21 13 NA 11 0 18 NA NA NA ...
##  $ CPAL3_presnt    : int  26 29 21 NA 19 8 26 NA NA NA ...
##  $ CPAL3_sti       : int  8 8 8 NA 8 8 8 NA NA NA ...
##  $ CPAL3_res       : int  26 29 21 NA 19 8 26 NA NA NA ...
##  $ CPAL3_Completion: int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CPAL3_Integrity : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CPAL4_dur       : int  40612 38835 71408 NA 49569 26061 69068 NA NA NA ...
##  $ CPAL4_lmn       : num  3.45 3.34 3.64 NA 3.5 ...
##  $ CPAL4_lsd       : num  0.227 0.334 0.444 NA 0.356 ...
##  $ CPAL4_acc       : num  0.601 0.73 0.756 NA 0.785 ...
##  $ CPAL4_cor       : int  8 8 8 NA 8 8 8 NA NA NA ...
##  $ CPAL4_err       : int  17 10 9 NA 8 0 9 NA NA NA ...
##  $ CPAL4_presnt    : int  25 18 17 NA 16 8 17 NA NA NA ...
##  $ CPAL4_sti       : int  8 8 8 NA 8 8 8 NA NA NA ...
##  $ CPAL4_res       : int  25 18 17 NA 16 8 17 NA NA NA ...
##  $ CPAL4_Completion: int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CPAL4_Integrity : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CPAL5_dur       : int  54466 48133 51421 NA 34614 25035 81750 NA NA NA ...
##  $ CPAL5_lmn       : num  3.6 3.51 3.49 NA 3.31 ...
##  $ CPAL5_lsd       : num  0.315 0.333 0.362 NA 0.252 ...
##  $ CPAL5_acc       : num  0.484 0.631 1.021 NA 1.021 ...
##  $ CPAL5_cor       : int  8 8 8 NA 8 8 8 NA NA NA ...
##  $ CPAL5_err       : int  29 15 3 NA 3 0 18 NA NA NA ...
##  $ CPAL5_presnt    : int  37 23 11 NA 11 8 26 NA NA NA ...
##  $ CPAL5_sti       : int  8 8 8 NA 8 8 8 NA NA NA ...
##  $ CPAL5_res       : int  37 23 11 NA 11 8 26 NA NA NA ...
##  $ CPAL5_Completion: int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CPAL5_Integrity : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CPAL6_dur       : int  58755 54312 73568 NA 31100 24328 64003 NA NA NA ...
##  $ CPAL6_lmn       : num  3.64 3.58 3.55 NA 3.27 ...
##  $ CPAL6_lsd       : num  0.335 0.336 0.518 NA 0.137 ...
##  $ CPAL6_acc       : num  0.498 0.588 0.857 NA 1.571 ...
##  $ CPAL6_cor       : int  8 8 8 NA 8 8 8 NA NA NA ...
##  $ CPAL6_err       : int  27 18 6 NA 0 0 8 NA NA NA ...
##  $ CPAL6_presnt    : int  35 26 14 NA 8 8 16 NA NA NA ...
##  $ CPAL6_sti       : int  8 8 8 NA 8 8 8 NA NA NA ...
##  $ CPAL6_res       : int  35 26 14 NA 8 8 16 NA NA NA ...
##  $ CPAL6_Completion: int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CPAL6_Integrity : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CPAL7_dur       : int  63087 43512 60684 NA 38679 23943 55046 NA NA NA ...
##  $ CPAL7_lmn       : num  3.62 3.4 3.51 NA 3.4 ...
##  $ CPAL7_lsd       : num  0.396 0.372 0.486 NA 0.253 ...
##  $ CPAL7_acc       : num  0.543 0.73 1.571 NA 0.955 ...
##  $ CPAL7_cor       : int  8 8 8 NA 8 8 8 NA NA NA ...
##  $ CPAL7_err       : int  22 10 0 NA 4 0 8 NA NA NA ...
##  $ CPAL7_presnt    : int  30 18 8 NA 12 8 16 NA NA NA ...
##  $ CPAL7_sti       : int  8 8 8 NA 8 8 8 NA NA NA ...
##  $ CPAL7_res       : int  30 18 8 NA 12 8 16 NA NA NA ...
##  $ CPAL7_Completion: int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CPAL7_Integrity : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ DET_lmn         : num  NA NA NA NA NA NA NA NA NA NA ...
##  $ DET_lsd         : num  NA NA NA NA NA NA NA NA NA NA ...
##  $ DET_acc         : num  NA NA NA NA NA NA NA NA NA NA ...
##  $ DET_cor         : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ DET_err         : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ DET_presnt      : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ DET_sti         : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ DET_Completion  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ DET_Integrity   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ GMCT_GMLidx     : int  -1 -1 -1 NA -1 -1 -1 NA NA NA ...
##  $ GMCT_mps        : num  1.33 1.4 1.3 NA 1.1 ...
##  $ GMCT_dur        : int  30011 30005 30002 NA 30004 30013 30001 NA NA NA ...
##  $ GMCT_ter        : int  0 0 0 NA 0 0 3 NA NA NA ...
##  $ GMCT_ler        : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ GMCT_rer        : int  0 0 0 NA 0 0 3 NA NA NA ...
##  $ GMCT_per        : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ GMCT_cmv        : int  40 42 39 NA 33 35 31 NA NA NA ...
##  $ GMCT_rth        : int  0 0 0 NA 0 0 1 NA NA NA ...
##  $ GMCT_Completion : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ GMCT_Integrity  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ GML_GMLidx      : int  5 17 11 NA 18 11 11 NA NA NA ...
##  $ GML_mps         : num  0.498 0.71 0.508 NA 0.552 ...
##  $ GML_dur         : int  280889 197144 275833 NA 253652 206585 303546 NA NA NA ...
##  $ GML_ter         : int  50 62 54 NA 34 42 67 NA NA NA ...
##  $ GML_ler         : int  49 56 42 NA 33 38 61 NA NA NA ...
##  $ GML_rer         : int  1 6 12 NA 1 4 6 NA NA NA ...
##  $ GML_per         : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ GML_cmv         : int  140 140 140 NA 140 140 140 NA NA NA ...
##  $ GML_rth         : int  50 59 49 NA 34 39 65 NA NA NA ...
##  $ GML_Completion  : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ GML_Integrity   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ GML1_GMLidx     : int  5 17 11 NA 18 11 11 NA NA NA ...
##  $ GML1_mps        : num  0.475 0.627 0.514 NA 0.513 ...
##  $ GML1_dur        : int  58969 44641 54513 NA 54528 49329 57109 NA NA NA ...
##  $ GML1_ter        : int  18 17 19 NA 10 12 17 NA NA NA ...
##  $ GML1_ler        : int  18 13 17 NA 10 12 16 NA NA NA ...
##  $ GML1_rer        : int  0 4 2 NA 0 0 1 NA NA NA ...
##  $ GML1_per        : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ GML1_cmv        : int  28 28 28 NA 28 28 28 NA NA NA ...
##  $ GML1_rth        : int  18 15 18 NA 10 12 16 NA NA NA ...
##  $ GML1_Completion : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ GML1_Integrity  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ GML2_GMLidx     : int  5 17 11 NA 18 11 11 NA NA NA ...
##  $ GML2_mps        : num  0.634 0.853 0.619 NA 0.549 ...
##  $ GML2_dur        : int  44184 32825 45265 NA 51030 37921 56949 NA NA NA ...
##  $ GML2_ter        : int  9 12 11 NA 9 9 16 NA NA NA ...
##  $ GML2_ler        : int  9 12 8 NA 8 9 15 NA NA NA ...
##  $ GML2_rer        : int  0 0 3 NA 1 0 1 NA NA NA ...
##  $ GML2_per        : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ GML2_cmv        : int  28 28 28 NA 28 28 28 NA NA NA ...
##  $ GML2_rth        : int  9 12 10 NA 9 9 16 NA NA NA ...
##  $ GML2_Completion : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ GML2_Integrity  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ GML3_GMLidx     : int  5 17 11 NA 18 11 11 NA NA NA ...
##  $ GML3_mps        : num  0.693 0.724 0.581 NA 0.785 ...
##  $ GML3_dur        : int  40408 38680 48161 NA 35649 34233 59837 NA NA NA ...
##  $ GML3_ter        : int  10 15 11 NA 6 9 19 NA NA NA ...
##  $ GML3_ler        : int  10 13 7 NA 6 5 15 NA NA NA ...
##  $ GML3_rer        : int  0 2 4 NA 0 4 4 NA NA NA ...
##  $ GML3_per        : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ GML3_cmv        : int  28 28 28 NA 28 28 28 NA NA NA ...
##  $ GML3_rth        : int  11 14 8 NA 6 6 18 NA NA NA ...
##  $ GML3_Completion : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ GML3_Integrity  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ GML4_GMLidx     : int  5 17 11 NA 18 11 11 NA NA NA ...
##  $ GML4_mps        : num  0.716 0.847 0.617 NA 0.822 ...
##  $ GML4_dur        : int  39097 33049 45368 NA 34054 28666 52822 NA NA NA ...
##  $ GML4_ter        : int  6 11 4 NA 5 5 9 NA NA NA ...
##  $ GML4_ler        : int  6 11 4 NA 5 5 9 NA NA NA ...
##  $ GML4_rer        : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ GML4_per        : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ GML4_cmv        : int  28 28 28 NA 28 28 28 NA NA NA ...
##  $ GML4_rth        : int  6 11 4 NA 5 5 9 NA NA NA ...
##  $ GML4_Completion : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ GML4_Integrity  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ GML5_GMLidx     : int  5 17 11 NA 18 11 11 NA NA NA ...
##  $ GML5_mps        : num  0.713 1.134 0.456 NA 1.025 ...
##  $ GML5_dur        : int  39250 24688 61368 NA 27309 31745 44045 NA NA NA ...
##  $ GML5_ter        : int  7 7 9 NA 4 7 6 NA NA NA ...
##  $ GML5_ler        : int  6 7 6 NA 4 7 6 NA NA NA ...
##  $ GML5_rer        : int  1 0 3 NA 0 0 0 NA NA NA ...
##  $ GML5_per        : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ GML5_cmv        : int  28 28 28 NA 28 28 28 NA NA NA ...
##  $ GML5_rth        : int  6 7 9 NA 4 7 6 NA NA NA ...
##  $ GML5_Completion : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ GML5_Integrity  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ GMR_GMLidx      : int  5 17 11 NA 18 11 11 NA NA NA ...
##  $ GMR_mps         : num  0.554 1.101 0.442 NA 0.851 ...
##  $ GMR_dur         : int  50497 25425 63305 NA 32916 25688 56223 NA NA NA ...
##  $ GMR_ter         : int  9 7 5 NA 7 2 10 NA NA NA ...
##  $ GMR_ler         : int  8 7 3 NA 7 2 9 NA NA NA ...
##  $ GMR_rer         : int  1 0 2 NA 0 0 1 NA NA NA ...
##  $ GMR_per         : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ GMR_cmv         : int  28 28 28 NA 28 28 28 NA NA NA ...
##  $ GMR_rth         : int  8 7 3 NA 7 2 9 NA NA NA ...
##  $ GMR_Completion  : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ GMR_Integrity   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ IDN_lmn         : num  NA NA NA NA NA NA NA NA NA NA ...
##  $ IDN_lsd         : num  NA NA NA NA NA NA NA NA NA NA ...
##  $ IDN_acc         : num  NA NA NA NA NA NA NA NA NA NA ...
##  $ IDN_cor         : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ IDN_err         : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ IDN_presnt      : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ IDN_sti         : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ IDN_Completion  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ IDN_Integrity   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ OCL_lmn         : num  3.06 3.09 3.34 NA 3.04 ...
##  $ OCL_lsd         : num  0.127 0.146 0.268 NA 0.114 ...
##  $ OCL_acc         : num  0.975 0.984 0.909 NA 0.959 ...
##  $ OCL_cor         : int  61 61 56 NA 59 64 59 NA NA NA ...
##  $ OCL_err         : int  28 27 34 NA 29 32 31 NA NA NA ...
##  $ OCL_presnt      : int  89 88 90 NA 88 96 90 NA NA NA ...
##  $ OCL_sti         : int  88 88 88 NA 88 88 88 NA NA NA ...
##  $ OCL_Completion  : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ OCL_Integrity   : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ ONB_lmn         : num  2.89 2.96 3.09 NA 2.96 ...
##  $ ONB_lsd         : num  0.1555 0.0997 0.1382 NA 0.1353 ...
##  $ ONB_acc         : num  1.27 1.32 1.23 NA 1.27 ...
##  $ ONB_cor         : int  31 31 31 NA 31 31 31 NA NA NA ...
##  $ ONB_err         : int  3 2 4 NA 3 0 0 NA NA NA ...
##  $ ONB_presnt      : int  34 33 35 NA 34 31 31 NA NA NA ...
##  $ ONB_sti         : int  34 32 34 NA 34 31 31 NA NA NA ...
##  $ ONB_Completion  : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ ONB_Integrity   : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ TWOB_lmn        : num  2.98 3.07 3.06 NA 3.01 ...
##  $ TWOB_lsd        : num  0.145 0.143 0.197 NA 0.154 ...
##  $ TWOB_acc        : num  1.27 1.27 1.19 NA 1.23 ...
##  $ TWOB_cor        : int  32 32 32 NA 32 32 32 NA NA NA ...
##  $ TWOB_err        : int  3 3 5 NA 4 0 0 NA NA NA ...
##  $ TWOB_presnt     : int  35 35 37 NA 36 32 32 NA NA NA ...
##  $ TWOB_sti        : int  35 35 37 NA 36 32 32 NA NA NA ...
##  $ TWOB_Completion : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ TWOB_Integrity  : int  0 0 0 NA 0 0 0 NA NA NA ...
##  $ TestNotPerformed: int  NA NA NA 1 NA NA NA 1 1 1 ...</code></pre>
</details>
<p><br> <br></p>
</div>
<div id="compositescorecalcvalues" class="section level2">
<h2>CompositeScoreCalcValues</h2>
<p>Raw scores used in computation of CompositeScores values</p>
<pre class="r"><code>## load data
CompositeScoreCalcValues &lt;- read.csv(paste0(wrap_directory, &quot;CompositeScoreCalcValues_05Jun2025.csv&quot;)) 
info(CompositeScoreCalcValues,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:7996, cols:17, inds:1799</code></pre>
<pre class="r"><code>str(CompositeScoreCalcValues)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    7996 obs. of  17 variables:
##  $ wrapnum     : chr  &quot;wrap0002&quot; &quot;wrap0068&quot; &quot;wrap0146&quot; &quot;wrap0148&quot; ...
##  $ WRAPNo      : chr  &quot;0002&quot; &quot;0068&quot; &quot;0146&quot; &quot;0148&quot; ...
##  $ VisNo       : int  5 2 4 5 5 4 6 4 7 4 ...
##  $ ttotal      : int  47 44 40 37 57 57 41 59 37 45 ...
##  $ bvmttot     : num  26 24 23 19 24 27 21 28 NA 25 ...
##  $ wmsrtot     : num  22 NA 24 32 26 24 23 35 28 20 ...
##  $ drraw       : num  10 8 10 10 11 13 9 14 6 9 ...
##  $ bvmtdel     : num  11 11 10 6 12 11 8 9 NA 10 ...
##  $ wmsrtot2    : num  21 NA 21 33 26 26 13 27 26 16 ...
##  $ trlb        : num  41 50 115 27 73 52 59 44 NA 103 ...
##  $ stroocw     : num  105 NA 100 113 93 114 130 117 NA 87 ...
##  $ waisrtot    : num  68 72 40 66 51 64 48 63 NA 50 ...
##  $ animtotraw  : num  30 NA 13 31 26 26 16 25 21 28 ...
##  $ flucfl      : num  48 54 68 44 44 34 48 48 25 52 ...
##  $ mmsetot     : num  30 28 30 30 30 30 30 30 NA 30 ...
##  $ stroocwe    : num  1 NA 2 3 3 0 1 1 NA 2 ...
##  $ update_stamp: chr  &quot;2025-01-24 08:17:01.0&quot; &quot;2025-01-24 08:17:01.0&quot; &quot;2025-01-24 08:17:01.0&quot; &quot;2025-01-24 08:17:01.0&quot; ...</code></pre>
</details>
<pre class="r"><code>## check table frequency for all variables

## drop update_stamp column
CompositeScoreCalcValues &lt;- subset(CompositeScoreCalcValues,select = -c(update_stamp)) ## 16 columns left</code></pre>
<pre class="r"><code>str(CompositeScoreCalcValues)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    7996 obs. of  16 variables:
##  $ wrapnum   : chr  &quot;wrap0002&quot; &quot;wrap0068&quot; &quot;wrap0146&quot; &quot;wrap0148&quot; ...
##  $ WRAPNo    : chr  &quot;0002&quot; &quot;0068&quot; &quot;0146&quot; &quot;0148&quot; ...
##  $ VisNo     : int  5 2 4 5 5 4 6 4 7 4 ...
##  $ ttotal    : int  47 44 40 37 57 57 41 59 37 45 ...
##  $ bvmttot   : num  26 24 23 19 24 27 21 28 NA 25 ...
##  $ wmsrtot   : num  22 NA 24 32 26 24 23 35 28 20 ...
##  $ drraw     : num  10 8 10 10 11 13 9 14 6 9 ...
##  $ bvmtdel   : num  11 11 10 6 12 11 8 9 NA 10 ...
##  $ wmsrtot2  : num  21 NA 21 33 26 26 13 27 26 16 ...
##  $ trlb      : num  41 50 115 27 73 52 59 44 NA 103 ...
##  $ stroocw   : num  105 NA 100 113 93 114 130 117 NA 87 ...
##  $ waisrtot  : num  68 72 40 66 51 64 48 63 NA 50 ...
##  $ animtotraw: num  30 NA 13 31 26 26 16 25 21 28 ...
##  $ flucfl    : num  48 54 68 44 44 34 48 48 25 52 ...
##  $ mmsetot   : num  30 28 30 30 30 30 30 30 NA 30 ...
##  $ stroocwe  : num  1 NA 2 3 3 0 1 1 NA 2 ...</code></pre>
</details>
<p><br> <br></p>
</div>
<div id="compositescores" class="section level2">
<h2>CompositeScores</h2>
<p>Domain-specific and global cognitive composite scores (see Jonaitis
et al 2019)</p>
<pre class="r"><code>## load data
CompositeScores &lt;- read.csv(paste0(wrap_directory, &quot;CompositeScores_05Jun2025.csv&quot;))
info(CompositeScores,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:7996, cols:32, inds:1799</code></pre>
<pre class="r"><code>str(CompositeScores)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    7996 obs. of  32 variables:
##  $ wrapnum            : chr  &quot;wrapL0308&quot; &quot;wrap0401&quot; &quot;wrapL0005&quot; &quot;wrap0434&quot; ...
##  $ WRAPNo             : chr  &quot;L0308&quot; &quot;0401&quot; &quot;L0005&quot; &quot;0434&quot; ...
##  $ VisNo              : int  2 6 5 7 1 5 1 5 2 2 ...
##  $ z_imm_lrn          : num  0.0851 -1.1886 0.4592 NA 0.9849 ...
##  $ std_imm_lrn        : num  101.3 82.2 106.9 NA 114.8 ...
##  $ z_del_rec          : num  -1.005 -1.376 0.82 NA 0.906 ...
##  $ std_del_rec        : num  84.9 79.4 112.3 NA 113.6 ...
##  $ z_exec_fn          : num  -0.71 0.063 -0.194 NA 0.628 ...
##  $ std_exec_fn        : num  89.3 100.9 97.1 NA 109.4 ...
##  $ z_exec_fn_animtot  : num  0.145 -0.566 0.681 NA 1.495 ...
##  $ std_exec_fn_animtot: num  102.2 91.5 110.2 NA 122.4 ...
##  $ z_exec_fn_cfl      : num  0.162 -0.217 -0.3 NA 0.231 ...
##  $ std_exec_fn_cfl    : num  102.4 96.7 95.5 NA 103.5 ...
##  $ z_pacc3_ds         : num  -0.8235 -1.7449 -0.0785 NA 0.7462 ...
##  $ std_pacc3_ds       : num  87.6 73.8 98.8 NA 111.2 ...
##  $ z_pacc3_tmtb       : num  -0.352 -1.331 0.247 NA 0.987 ...
##  $ std_pacc3_tmtb     : num  94.7 80 103.7 NA 114.8 ...
##  $ z_pacc3_animtot    : num  -0.795 -1.975 0.464 -0.921 1.546 ...
##  $ std_pacc3_animtot  : num  88.1 70.4 107 86.2 123.2 ...
##  $ z_pacc3_cfl        : num  -0.469 -1.234 -0.284 -1.531 0.418 ...
##  $ std_pacc3_cfl      : num  93 81.5 95.7 77 106.3 ...
##  $ z_pacc4_ds_mmse    : num  -1.662 -1.206 -0.228 NA 0.458 ...
##  $ std_pacc4_ds_mmse  : num  75.1 81.9 96.6 NA 106.9 ...
##  $ z_pacc4_tmtb_mmse  : num  -1.2825 -0.8556 0.0384 NA 0.6515 ...
##  $ std_pacc4_tmtb_mmse: num  80.8 87.2 100.6 NA 109.8 ...
##  $ z_pacc4_ds_cfl     : num  -0.56 -1.313 -0.342 NA 0.43 ...
##  $ std_pacc4_ds_cfl   : num  91.6 80.3 94.9 NA 106.4 ...
##  $ z_pacc5_animtot    : num  -1.463 -1.376 0.142 NA 1.064 ...
##  $ std_pacc5_animtot  : num  78.1 79.4 102.1 NA 116 ...
##  $ z_pacc5_cfl        : num  -0.292 -1.054 -0.161 NA 0.581 ...
##  $ std_pacc5_cfl      : num  95.6 84.2 97.6 NA 108.7 ...
##  $ update_stamp       : chr  &quot;2025-01-24 08:17:07.0&quot; &quot;2025-01-24 08:17:07.0&quot; &quot;2025-01-24 08:17:07.0&quot; &quot;2025-01-24 08:17:07.0&quot; ...</code></pre>
</details>
<pre class="r"><code>## check table frequency for all variables

## drop update_stamp column
CompositeScores &lt;- subset(CompositeScores,select = -c(update_stamp)) ## 31 columns left</code></pre>
<pre class="r"><code>str(CompositeScores)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    7996 obs. of  31 variables:
##  $ wrapnum            : chr  &quot;wrapL0308&quot; &quot;wrap0401&quot; &quot;wrapL0005&quot; &quot;wrap0434&quot; ...
##  $ WRAPNo             : chr  &quot;L0308&quot; &quot;0401&quot; &quot;L0005&quot; &quot;0434&quot; ...
##  $ VisNo              : int  2 6 5 7 1 5 1 5 2 2 ...
##  $ z_imm_lrn          : num  0.0851 -1.1886 0.4592 NA 0.9849 ...
##  $ std_imm_lrn        : num  101.3 82.2 106.9 NA 114.8 ...
##  $ z_del_rec          : num  -1.005 -1.376 0.82 NA 0.906 ...
##  $ std_del_rec        : num  84.9 79.4 112.3 NA 113.6 ...
##  $ z_exec_fn          : num  -0.71 0.063 -0.194 NA 0.628 ...
##  $ std_exec_fn        : num  89.3 100.9 97.1 NA 109.4 ...
##  $ z_exec_fn_animtot  : num  0.145 -0.566 0.681 NA 1.495 ...
##  $ std_exec_fn_animtot: num  102.2 91.5 110.2 NA 122.4 ...
##  $ z_exec_fn_cfl      : num  0.162 -0.217 -0.3 NA 0.231 ...
##  $ std_exec_fn_cfl    : num  102.4 96.7 95.5 NA 103.5 ...
##  $ z_pacc3_ds         : num  -0.8235 -1.7449 -0.0785 NA 0.7462 ...
##  $ std_pacc3_ds       : num  87.6 73.8 98.8 NA 111.2 ...
##  $ z_pacc3_tmtb       : num  -0.352 -1.331 0.247 NA 0.987 ...
##  $ std_pacc3_tmtb     : num  94.7 80 103.7 NA 114.8 ...
##  $ z_pacc3_animtot    : num  -0.795 -1.975 0.464 -0.921 1.546 ...
##  $ std_pacc3_animtot  : num  88.1 70.4 107 86.2 123.2 ...
##  $ z_pacc3_cfl        : num  -0.469 -1.234 -0.284 -1.531 0.418 ...
##  $ std_pacc3_cfl      : num  93 81.5 95.7 77 106.3 ...
##  $ z_pacc4_ds_mmse    : num  -1.662 -1.206 -0.228 NA 0.458 ...
##  $ std_pacc4_ds_mmse  : num  75.1 81.9 96.6 NA 106.9 ...
##  $ z_pacc4_tmtb_mmse  : num  -1.2825 -0.8556 0.0384 NA 0.6515 ...
##  $ std_pacc4_tmtb_mmse: num  80.8 87.2 100.6 NA 109.8 ...
##  $ z_pacc4_ds_cfl     : num  -0.56 -1.313 -0.342 NA 0.43 ...
##  $ std_pacc4_ds_cfl   : num  91.6 80.3 94.9 NA 106.4 ...
##  $ z_pacc5_animtot    : num  -1.463 -1.376 0.142 NA 1.064 ...
##  $ std_pacc5_animtot  : num  78.1 79.4 102.1 NA 116 ...
##  $ z_pacc5_cfl        : num  -0.292 -1.054 -0.161 NA 0.581 ...
##  $ std_pacc5_cfl      : num  95.6 84.2 97.6 NA 108.7 ...</code></pre>
</details>
<p><br> <br></p>
</div>
<div id="consensusconference" class="section level2">
<h2>ConsensusConference</h2>
<p>Consensus Conference Results</p>
<pre class="r"><code>## load data
ConsensusConference &lt;- read.csv(paste0(wrap_directory, &quot;ConsensusConference_05Jun2025.csv&quot;))
info(ConsensusConference,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:8176, cols:96, inds:1808</code></pre>
<pre class="r"><code>str(ConsensusConference)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    8176 obs. of  96 variables:
##  $ wrapnum                : chr  &quot;wrap0002&quot; &quot;wrap0068&quot; &quot;wrap0146&quot; &quot;wrap0148&quot; ...
##  $ WRAPNo                 : chr  &quot;0002&quot; &quot;0068&quot; &quot;0146&quot; &quot;0148&quot; ...
##  $ VisNo                  : int  5 2 4 5 5 4 6 4 7 4 ...
##  $ CC1                    : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ CC2                    : int  1 1 0 1 1 1 0 1 NA 0 ...
##  $ CC2b_CDR               : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ CC2b_IADL              : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ CC2b_IQ                : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ CC3                    : int  NA NA 0 NA NA NA 0 NA NA 0 ...
##  $ CC4_Early              : int  0 0 -1 0 0 0 -1 0 -1 -1 ...
##  $ CC4_Clinical           : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ CC4a                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC4b                   : int  0 NA 0 0 0 0 1 0 0 1 ...
##  $ CC4b_Lang              : int  0 NA 0 0 0 0 1 0 0 0 ...
##  $ CC4b_Att               : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC4b_Exec              : int  0 NA 0 0 0 0 0 0 0 1 ...
##  $ CC4b_Vis               : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC4c                   : int  0 NA 1 0 0 0 0 0 0 0 ...
##  $ CC4c_Lang              : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC4c_Att               : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC4c_Exec              : int  0 NA 1 0 0 0 0 0 0 0 ...
##  $ CC4c_Vis               : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC4d                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC4d_Lang              : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC4d_Att               : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC4d_Exec              : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC4d_Vis               : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC4e                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC4e_longstanding      : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ Calculated_Consensus_dx: chr  &quot;Cog_Unimpaired_Stable&quot; &quot;Cog_Unimpaired_Stable&quot; &quot;Cog_Unimpaired_Declining&quot; &quot;Cog_Unimpaired_Stable&quot; ...
##  $ CC5                    : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC5a                   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC6                    : int  0 NA 0 0 0 0 0 0 0 1 ...
##  $ CC6a                   : int  NA NA NA NA NA NA NA NA NA 1 ...
##  $ CC7                    : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC7a                   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC8                    : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC8a                   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC9                    : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC9a                   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC10                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC10a                  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC11                   : int  0 NA 0 0 0 0 1 0 0 0 ...
##  $ CC11a                  : int  NA NA NA NA NA NA 1 NA NA NA ...
##  $ CC12                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC12a                  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC13                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC13a                  : logi  NA NA NA NA NA NA ...
##  $ CC13_1                 : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC13_2                 : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC13_3                 : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC13_4                 : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC14                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC14a                  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC15                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC15a                  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC16                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC16a                  : logi  NA NA NA NA NA NA ...
##  $ CC17                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC17a                  : logi  NA NA NA NA NA NA ...
##  $ CC18                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC18a                  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC19                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC19a                  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC20                   : int  0 NA 0 0 0 0 1 0 0 0 ...
##  $ CC20a                  : int  NA NA NA NA NA NA 2 NA NA NA ...
##  $ CC21                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC21a                  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC22                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC22a                  : logi  NA NA NA NA NA NA ...
##  $ CC23                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC23a                  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC24                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC24a                  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC25                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC25a                  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC26                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC26a                  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC27                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC27a                  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC28                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC28a                  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC29                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC29a                  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC30                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC30a                  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ RcvPhone               : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ CogDec                 : int  0 0 0 0 0 0 0 0 -1 0 ...
##  $ NoAction               : int  0 0 0 0 0 0 -1 0 0 -1 ...
##  $ ActOth                 : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ cons_memory            : int  0 0 0 0 0 0 1 0 0 1 ...
##  $ cons_language          : int  0 0 0 0 0 0 1 0 0 0 ...
##  $ cons_attention         : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ cons_execfunc          : int  0 0 1 0 0 0 0 0 0 1 ...
##  $ cons_visuospatial      : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ update_stamp           : chr  &quot;2024-09-13 08:52:18.0&quot; &quot;2024-09-13 08:52:18.0&quot; &quot;2024-09-13 08:52:18.0&quot; &quot;2024-09-13 08:52:18.0&quot; ...</code></pre>
</details>
<pre class="r"><code>## check table frequency for all variables

## convert logi columns to int
## these columns were confirmed to only have NA values in the raw data
logicols &lt;- colnames(ConsensusConference)[sapply(ConsensusConference, is.logical)]
ConsensusConference[logicols] &lt;- lapply(ConsensusConference[logicols], as.integer)

## drop update_stamp column
ConsensusConference &lt;- subset(ConsensusConference,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(ConsensusConference)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    8176 obs. of  95 variables:
##  $ wrapnum                : chr  &quot;wrap0002&quot; &quot;wrap0068&quot; &quot;wrap0146&quot; &quot;wrap0148&quot; ...
##  $ WRAPNo                 : chr  &quot;0002&quot; &quot;0068&quot; &quot;0146&quot; &quot;0148&quot; ...
##  $ VisNo                  : int  5 2 4 5 5 4 6 4 7 4 ...
##  $ CC1                    : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ CC2                    : int  1 1 0 1 1 1 0 1 NA 0 ...
##  $ CC2b_CDR               : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ CC2b_IADL              : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ CC2b_IQ                : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ CC3                    : int  NA NA 0 NA NA NA 0 NA NA 0 ...
##  $ CC4_Early              : int  0 0 -1 0 0 0 -1 0 -1 -1 ...
##  $ CC4_Clinical           : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ CC4a                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC4b                   : int  0 NA 0 0 0 0 1 0 0 1 ...
##  $ CC4b_Lang              : int  0 NA 0 0 0 0 1 0 0 0 ...
##  $ CC4b_Att               : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC4b_Exec              : int  0 NA 0 0 0 0 0 0 0 1 ...
##  $ CC4b_Vis               : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC4c                   : int  0 NA 1 0 0 0 0 0 0 0 ...
##  $ CC4c_Lang              : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC4c_Att               : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC4c_Exec              : int  0 NA 1 0 0 0 0 0 0 0 ...
##  $ CC4c_Vis               : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC4d                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC4d_Lang              : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC4d_Att               : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC4d_Exec              : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC4d_Vis               : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC4e                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC4e_longstanding      : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ Calculated_Consensus_dx: chr  &quot;Cog_Unimpaired_Stable&quot; &quot;Cog_Unimpaired_Stable&quot; &quot;Cog_Unimpaired_Declining&quot; &quot;Cog_Unimpaired_Stable&quot; ...
##  $ CC5                    : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC5a                   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC6                    : int  0 NA 0 0 0 0 0 0 0 1 ...
##  $ CC6a                   : int  NA NA NA NA NA NA NA NA NA 1 ...
##  $ CC7                    : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC7a                   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC8                    : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC8a                   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC9                    : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC9a                   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC10                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC10a                  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC11                   : int  0 NA 0 0 0 0 1 0 0 0 ...
##  $ CC11a                  : int  NA NA NA NA NA NA 1 NA NA NA ...
##  $ CC12                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC12a                  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC13                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC13a                  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC13_1                 : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC13_2                 : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC13_3                 : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC13_4                 : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC14                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC14a                  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC15                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC15a                  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC16                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC16a                  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC17                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC17a                  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC18                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC18a                  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC19                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC19a                  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC20                   : int  0 NA 0 0 0 0 1 0 0 0 ...
##  $ CC20a                  : int  NA NA NA NA NA NA 2 NA NA NA ...
##  $ CC21                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC21a                  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC22                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC22a                  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC23                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC23a                  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC24                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC24a                  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC25                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC25a                  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC26                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC26a                  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC27                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC27a                  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC28                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC28a                  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC29                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC29a                  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ CC30                   : int  0 NA 0 0 0 0 0 0 0 0 ...
##  $ CC30a                  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ RcvPhone               : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ CogDec                 : int  0 0 0 0 0 0 0 0 -1 0 ...
##  $ NoAction               : int  0 0 0 0 0 0 -1 0 0 -1 ...
##  $ ActOth                 : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ cons_memory            : int  0 0 0 0 0 0 1 0 0 1 ...
##  $ cons_language          : int  0 0 0 0 0 0 1 0 0 0 ...
##  $ cons_attention         : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ cons_execfunc          : int  0 0 1 0 0 0 0 0 0 1 ...
##  $ cons_visuospatial      : int  0 0 0 0 0 0 0 0 0 0 ...</code></pre>
</details>
<p><br> <br></p>
</div>
<div id="dailyactivity" class="section level2">
<h2>DailyActivity</h2>
<p>Legacy - Daily Activities (Prior to inclusion in General Health and
Daily Activities)</p>
<pre class="r"><code>## load data
DailyActivity &lt;- read.csv(paste0(wrap_directory, &quot;DailyActivity_05Jun2025.csv&quot;))
info(DailyActivity,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:1263, cols:85, inds:1263</code></pre>
<pre class="r"><code>str(DailyActivity)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    1263 obs. of  85 variables:
##  $ wrapnum      : chr  &quot;wrap0068&quot; &quot;wrap1006&quot; &quot;wrapA0081&quot; &quot;wrapL0240&quot; ...
##  $ WRAPNo       : chr  &quot;0068&quot; &quot;1006&quot; &quot;A0081&quot; &quot;L0240&quot; ...
##  $ VisNo        : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ RelPar       : int  -1 NA NA NA NA NA NA -1 0 0 ...
##  $ RelMIL       : int  0 NA NA NA NA NA NA 0 0 0 ...
##  $ RelFIL       : int  0 NA NA NA NA NA NA 0 0 0 ...
##  $ RelSib       : int  0 NA NA NA NA NA NA 0 -1 0 ...
##  $ RelOth       : int  0 NA NA NA NA NA NA 0 0 0 ...
##  $ RelDemOth    : chr  &quot;&quot; &quot;&quot; &quot;grandmother&quot; &quot;&quot; ...
##  $ SuppAct      : int  1 NA NA NA 1 NA NA 1 1 NA ...
##  $ RptSchool    : int  8 8 8 8 8 8 0 8 0 8 ...
##  $ RptSchoolAge : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ SentAway     : int  8 8 -1 8 8 8 8 8 8 8 ...
##  $ SentAwayAge  : chr  &quot;&quot; &quot;&quot; &quot;12&quot; &quot;&quot; ...
##  $ FMNoJob      : int  8 8 -1 8 8 8 8 8 8 8 ...
##  $ FMNoJobAge   : chr  &quot;&quot; &quot;&quot; &quot;10&quot; &quot;&quot; ...
##  $ FMDrank      : int  8 8 -1 8 -2 8 8 8 8 -1 ...
##  $ FMDrankAge   : chr  &quot;&quot; &quot;&quot; &quot;10&quot; &quot;&quot; ...
##  $ FMDrugs      : int  8 8 1 8 8 8 8 8 8 8 ...
##  $ FMDrugsAge   : chr  &quot;&quot; &quot;&quot; &quot;13-18&quot; &quot;&quot; ...
##  $ DropSchool   : int  8 8 8 8 8 8 8 8 8 8 ...
##  $ DropSchoolAge: num  NA NA NA NA NA NA NA NA NA NA ...
##  $ ExpSchool    : int  8 8 8 8 8 8 8 8 8 8 ...
##  $ ExpSchoolAge : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ Flunk        : int  8 8 8 8 8 8 8 8 8 8 ...
##  $ FlunkAge     : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ FlunkChk     : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ FiredJob     : int  0 8 8 8 8 8 2 8 8 1 ...
##  $ FiredJobAge  : chr  &quot;57&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ FiredChk     : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ NoJob        : int  8 8 -1 8 8 8 -1 8 8 8 ...
##  $ NoJobAge     : chr  &quot;&quot; &quot;&quot; &quot;53&quot; &quot;&quot; ...
##  $ NoJobChk     : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ FMDied       : int  1 -1 8 1 8 -2 8 NA NA 8 ...
##  $ FMDiedAge    : chr  &quot;57&quot; &quot;46&quot; &quot;&quot; &quot;40&quot; ...
##  $ FMDiedChk    : int  NA NA NA NA NA -1 NA NA -1 NA ...
##  $ FMDivorce    : int  8 8 -1 8 8 8 8 8 8 8 ...
##  $ FMDivorceAge : chr  &quot;&quot; &quot;&quot; &quot;4&quot; &quot;&quot; ...
##  $ FMDivorceChk : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ Infidelity   : int  8 8 8 8 8 8 8 8 NA 8 ...
##  $ InfidelityAge: chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ InfidelityChk: int  NA NA NA NA NA NA NA NA NA NA ...
##  $ DiffInLaws   : int  8 8 8 8 8 8 -1 8 8 8 ...
##  $ DiffInLawsAge: chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ DiffInLawsChk: int  NA NA NA NA NA NA NA NA NA NA ...
##  $ SibDied      : int  -1 8 8 8 -2 8 0 8 8 8 ...
##  $ SibDiedAge   : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ SibDiedChk   : int  NA NA NA NA -1 NA NA NA NA NA ...
##  $ ChildDied    : int  8 8 8 8 8 8 8 8 8 8 ...
##  $ ChildDiedAge : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ ChildDiedChk : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ ChildAccid   : int  -1 8 8 8 8 8 0 8 8 8 ...
##  $ ChildAccidAge: chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ ChildAccidChk: int  NA NA NA NA NA NA NA NA NA NA ...
##  $ LostHome     : int  8 8 8 8 8 8 8 8 8 8 ...
##  $ LostHomeAge  : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ LostHomeChk  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ PhysAtt      : int  8 8 8 8 8 8 8 8 8 8 ...
##  $ PhysAttAge   : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ PhysAttChk   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ SexAtt       : int  8 8 8 8 8 8 8 8 8 8 ...
##  $ SexAttAge    : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ SexAttChk    : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ LegalDiff    : int  8 -1 -1 8 8 8 8 8 8 8 ...
##  $ LegalDiffAge : chr  &quot;&quot; &quot;40&quot; &quot;22&quot; &quot;&quot; ...
##  $ LegalDiffChk : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ DetJail      : int  -1 8 8 8 8 8 0 8 8 8 ...
##  $ DetJailAge   : chr  &quot;37&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ DetJailChk   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ Bankrupt     : int  8 8 8 8 8 8 8 8 8 8 ...
##  $ BankruptAge  : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ BankruptChk  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ FinLoss      : int  -1 8 -1 8 -2 8 2 8 8 8 ...
##  $ FinLossAge   : chr  &quot;36&quot; &quot;&quot; &quot;53&quot; &quot;&quot; ...
##  $ FinLossChk   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ Welfare      : int  8 8 8 8 -2 0 8 8 NA 8 ...
##  $ WelfareAge   : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ WelfareChk   : logi  NA NA NA NA NA NA ...
##  $ ArmedFrc     : int  8 8 8 8 8 8 8 8 NA 1 ...
##  $ ArmedFrcAge  : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ ArmedFrcChk  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ Combat       : int  8 8 8 8 8 8 8 8 8 8 ...
##  $ CombatAge    : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ CombatChk    : logi  NA NA NA NA NA NA ...
##  $ update_stamp : chr  &quot;2024-09-13 08:52:20.0&quot; &quot;2024-09-13 08:52:20.0&quot; &quot;2024-09-13 08:52:20.0&quot; &quot;2024-09-13 08:52:20.0&quot; ...</code></pre>
</details>
<pre class="r"><code>## check table frequency for all variables

## convert logi columns to int
## ## these columns were confirmed to only have NA values in the raw data
logicols &lt;- colnames(DailyActivity)[sapply(DailyActivity, is.logical)]
DailyActivity[logicols] &lt;- lapply(DailyActivity[logicols], as.integer)

## extract all chr columns and recode &quot;&quot; to NA
chrcols &lt;- colnames(DailyActivity)[sapply(DailyActivity, is.character)]
DailyActivity[chrcols] &lt;- lapply(DailyActivity[chrcols], function(x) ifelse(x == &quot;&quot;, NA, x))

## drop update_stamp column
DailyActivity &lt;- subset(DailyActivity,select = -c(update_stamp)) ## 84 columns left</code></pre>
<pre class="r"><code>str(DailyActivity)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    1263 obs. of  84 variables:
##  $ wrapnum      : chr  &quot;wrap0068&quot; &quot;wrap1006&quot; &quot;wrapA0081&quot; &quot;wrapL0240&quot; ...
##  $ WRAPNo       : chr  &quot;0068&quot; &quot;1006&quot; &quot;A0081&quot; &quot;L0240&quot; ...
##  $ VisNo        : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ RelPar       : int  -1 NA NA NA NA NA NA -1 0 0 ...
##  $ RelMIL       : int  0 NA NA NA NA NA NA 0 0 0 ...
##  $ RelFIL       : int  0 NA NA NA NA NA NA 0 0 0 ...
##  $ RelSib       : int  0 NA NA NA NA NA NA 0 -1 0 ...
##  $ RelOth       : int  0 NA NA NA NA NA NA 0 0 0 ...
##  $ RelDemOth    : chr  NA NA &quot;grandmother&quot; NA ...
##  $ SuppAct      : int  1 NA NA NA 1 NA NA 1 1 NA ...
##  $ RptSchool    : int  8 8 8 8 8 8 0 8 0 8 ...
##  $ RptSchoolAge : chr  NA NA NA NA ...
##  $ SentAway     : int  8 8 -1 8 8 8 8 8 8 8 ...
##  $ SentAwayAge  : chr  NA NA &quot;12&quot; NA ...
##  $ FMNoJob      : int  8 8 -1 8 8 8 8 8 8 8 ...
##  $ FMNoJobAge   : chr  NA NA &quot;10&quot; NA ...
##  $ FMDrank      : int  8 8 -1 8 -2 8 8 8 8 -1 ...
##  $ FMDrankAge   : chr  NA NA &quot;10&quot; NA ...
##  $ FMDrugs      : int  8 8 1 8 8 8 8 8 8 8 ...
##  $ FMDrugsAge   : chr  NA NA &quot;13-18&quot; NA ...
##  $ DropSchool   : int  8 8 8 8 8 8 8 8 8 8 ...
##  $ DropSchoolAge: num  NA NA NA NA NA NA NA NA NA NA ...
##  $ ExpSchool    : int  8 8 8 8 8 8 8 8 8 8 ...
##  $ ExpSchoolAge : chr  NA NA NA NA ...
##  $ Flunk        : int  8 8 8 8 8 8 8 8 8 8 ...
##  $ FlunkAge     : chr  NA NA NA NA ...
##  $ FlunkChk     : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ FiredJob     : int  0 8 8 8 8 8 2 8 8 1 ...
##  $ FiredJobAge  : chr  &quot;57&quot; NA NA NA ...
##  $ FiredChk     : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ NoJob        : int  8 8 -1 8 8 8 -1 8 8 8 ...
##  $ NoJobAge     : chr  NA NA &quot;53&quot; NA ...
##  $ NoJobChk     : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ FMDied       : int  1 -1 8 1 8 -2 8 NA NA 8 ...
##  $ FMDiedAge    : chr  &quot;57&quot; &quot;46&quot; NA &quot;40&quot; ...
##  $ FMDiedChk    : int  NA NA NA NA NA -1 NA NA -1 NA ...
##  $ FMDivorce    : int  8 8 -1 8 8 8 8 8 8 8 ...
##  $ FMDivorceAge : chr  NA NA &quot;4&quot; NA ...
##  $ FMDivorceChk : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ Infidelity   : int  8 8 8 8 8 8 8 8 NA 8 ...
##  $ InfidelityAge: chr  NA NA NA NA ...
##  $ InfidelityChk: int  NA NA NA NA NA NA NA NA NA NA ...
##  $ DiffInLaws   : int  8 8 8 8 8 8 -1 8 8 8 ...
##  $ DiffInLawsAge: chr  NA NA NA NA ...
##  $ DiffInLawsChk: int  NA NA NA NA NA NA NA NA NA NA ...
##  $ SibDied      : int  -1 8 8 8 -2 8 0 8 8 8 ...
##  $ SibDiedAge   : chr  NA NA NA NA ...
##  $ SibDiedChk   : int  NA NA NA NA -1 NA NA NA NA NA ...
##  $ ChildDied    : int  8 8 8 8 8 8 8 8 8 8 ...
##  $ ChildDiedAge : chr  NA NA NA NA ...
##  $ ChildDiedChk : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ ChildAccid   : int  -1 8 8 8 8 8 0 8 8 8 ...
##  $ ChildAccidAge: chr  NA NA NA NA ...
##  $ ChildAccidChk: int  NA NA NA NA NA NA NA NA NA NA ...
##  $ LostHome     : int  8 8 8 8 8 8 8 8 8 8 ...
##  $ LostHomeAge  : chr  NA NA NA NA ...
##  $ LostHomeChk  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ PhysAtt      : int  8 8 8 8 8 8 8 8 8 8 ...
##  $ PhysAttAge   : chr  NA NA NA NA ...
##  $ PhysAttChk   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ SexAtt       : int  8 8 8 8 8 8 8 8 8 8 ...
##  $ SexAttAge    : chr  NA NA NA NA ...
##  $ SexAttChk    : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ LegalDiff    : int  8 -1 -1 8 8 8 8 8 8 8 ...
##  $ LegalDiffAge : chr  NA &quot;40&quot; &quot;22&quot; NA ...
##  $ LegalDiffChk : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ DetJail      : int  -1 8 8 8 8 8 0 8 8 8 ...
##  $ DetJailAge   : chr  &quot;37&quot; NA NA NA ...
##  $ DetJailChk   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ Bankrupt     : int  8 8 8 8 8 8 8 8 8 8 ...
##  $ BankruptAge  : chr  NA NA NA NA ...
##  $ BankruptChk  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ FinLoss      : int  -1 8 -1 8 -2 8 2 8 8 8 ...
##  $ FinLossAge   : chr  &quot;36&quot; NA &quot;53&quot; NA ...
##  $ FinLossChk   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ Welfare      : int  8 8 8 8 -2 0 8 8 NA 8 ...
##  $ WelfareAge   : chr  NA NA NA NA ...
##  $ WelfareChk   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ ArmedFrc     : int  8 8 8 8 8 8 8 8 NA 1 ...
##  $ ArmedFrcAge  : chr  NA NA NA NA ...
##  $ ArmedFrcChk  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ Combat       : int  8 8 8 8 8 8 8 8 8 8 ...
##  $ CombatAge    : chr  NA NA NA NA ...
##  $ CombatChk    : int  NA NA NA NA NA NA NA NA NA NA ...</code></pre>
</details>
<p><br> <br></p>
</div>
<div id="demographics" class="section level2">
<h2>Demographics</h2>
<p>Participant Demographic Information</p>
<pre class="r"><code>## load data
Demographics &lt;- read.csv(paste0(wrap_directory, &quot;Demographics_05Jun2025.csv&quot;))
info(Demographics,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:1811, cols:66, inds:1811</code></pre>
<pre class="r"><code>str(Demographics)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    1811 obs. of  66 variables:
##  $ wrapnum               : chr  &quot;wrap0543&quot; &quot;wrap0579&quot; &quot;wrap0596&quot; &quot;wrap0671&quot; ...
##  $ WRAPNo                : chr  &quot;0543&quot; &quot;0579&quot; &quot;0596&quot; &quot;0671&quot; ...
##  $ gender                : int  2 2 2 2 2 1 2 1 1 2 ...
##  $ race1                 : int  1 1 1 1 1 1 1 1 2 1 ...
##  $ race2                 : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ race3                 : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ raceoth               : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ race_native_tribe     : logi  NA NA NA NA NA NA ...
##  $ hispanic_or_latino    : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ other_ethnicity       : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ BioKids               : int  3 1 4 2 0 1 1 4 3 3 ...
##  $ hand                  : int  1 1 1 1 2 1 1 1 1 1 ...
##  $ frstlang              : int  NA NA NA NA 1 1 1 1 1 1 ...
##  $ speclang              : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ moloss                : int  2 1 2 2 1 2 2 3 2 2 ...
##  $ moepisz               : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ mostroke              : int  2 2 2 2 2 2 1 2 1 2 ...
##  $ moms                  : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ mopark                : int  2 2 2 2 2 2 2 3 2 2 ...
##  $ momen                 : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ mohdache              : int  1 2 1 2 2 2 2 2 2 2 ...
##  $ moneuro               : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ motneuro              : int  2 2 NA NA 2 2 2 2 2 2 ...
##  $ specmoth              : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ faloss                : int  2 2 2 1 2 2 1 3 2 2 ...
##  $ faepisz               : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ fastroke              : int  2 2 2 2 1 2 1 2 2 2 ...
##  $ fams                  : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ fapark                : int  2 2 2 2 2 2 1 2 2 2 ...
##  $ famen                 : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ fahdache              : int  2 2 2 1 2 2 2 2 2 2 ...
##  $ faneuro               : int  2 2 2 2 1 2 2 2 2 2 ...
##  $ foneuro               : int  2 2 NA 2 1 2 2 2 NA 2 ...
##  $ specfath              : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ famdown               : int  2 2 2 2 2 2 2 2 2 1 ...
##  $ numsis                : int  NA 1 1 3 1 1 1 1 1 2 ...
##  $ numbro                : int  1 3 2 0 2 2 2 2 3 3 ...
##  $ ldschool              : int  2 2 1 2 2 2 2 1 3 2 ...
##  $ add                   : int  NA NA 2 NA NA 2 2 2 2 2 ...
##  $ adhd                  : int  NA NA 2 NA NA 2 2 2 2 2 ...
##  $ ld                    : int  NA NA 2 NA NA 2 2 1 2 2 ...
##  $ otherld               : int  NA NA 1 NA NA 2 2 2 2 2 ...
##  $ specld                : chr  &quot;&quot; &quot;&quot; &quot;hard time remembering things&quot; &quot;&quot; ...
##  $ LearnProb             : int  2 2 3 2 2 2 2 NA NA 2 ...
##  $ RelLearnProb          : int  NA NA NA 2 1 2 1 1 2 2 ...
##  $ Farm                  : int  1 2 1 2 2 2 2 1 1 2 ...
##  $ FarmYrs               : int  1 NA 2 NA NA NA NA 1 3 NA ...
##  $ Color                 : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Grade                 : int  6 5 5 8 11 10 7 7 4 6 ...
##  $ EducYrs               : int  13 12 13 16 26 22 15 15 11 12 ...
##  $ OC_Imputed_grade      : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ OC_Reporting_EducYears: int  13 12 12 16 20 18 14 14 11 13 ...
##  $ DxLearnDis            : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ DxADHD                : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ DxAttDD               : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ DxOth                 : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ DxOthTxt              : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ interv                : int  6 7 6 6 6 8 9 31 39 9 ...
##  $ nochild               : int  3 1 4 2 0 1 1 NA NA 3 ...
##  $ FH6Bro                : int  1 3 1 0 2 2 2 NA NA 3 ...
##  $ FH6Sis                : int  0 1 1 3 1 1 1 NA NA 2 ...
##  $ interest              : int  1 3 1 1 1 2 1 NA NA 2 ...
##  $ parntres              : num  1 2 1 1 1 1 1 NA NA 2 ...
##  $ Lang                  : int  1 1 1 1 1 1 1 NA NA 1 ...
##  $ LangOth               : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ update_stamp          : chr  &quot;2025-01-24 08:17:16.0&quot; &quot;2025-01-24 08:17:16.0&quot; &quot;2025-01-24 08:17:16.0&quot; &quot;2025-01-24 08:17:16.0&quot; ...</code></pre>
</details>
<pre class="r"><code>## check table frequency for all variables

## DD has 1=yes,2=no coding but data has 0/1 coding.
## Confirmed with other 0/1 variables that 0 is no and 1 stays yes, so will update the dataset to match the DD
Demographics$OC_Imputed_grade[Demographics$OC_Imputed_grade] &lt;- 2
Demographics$OC_Imputed_grade &lt;- as.integer(Demographics$OC_Imputed_grade)

## convert logi columns to int
logicols &lt;- colnames(Demographics)[sapply(Demographics, is.logical)]
Demographics[logicols] &lt;- lapply(Demographics[logicols], as.integer)

## convert parntres from num to int
Demographics$parntres &lt;- as.integer(Demographics$parntres)

## convert &quot;&quot; to NA for chr columns
chrcols &lt;- colnames(Demographics)[sapply(Demographics, is.character)]
Demographics[chrcols] &lt;- lapply(Demographics[chrcols], function(x) ifelse(x == &quot;&quot;, NA, x))

## drop update_stamp column
Demographics &lt;- subset(Demographics,select = -c(update_stamp)) ## 84 columns left</code></pre>
<pre class="r"><code>str(Demographics)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    1811 obs. of  65 variables:
##  $ wrapnum               : chr  &quot;wrap0543&quot; &quot;wrap0579&quot; &quot;wrap0596&quot; &quot;wrap0671&quot; ...
##  $ WRAPNo                : chr  &quot;0543&quot; &quot;0579&quot; &quot;0596&quot; &quot;0671&quot; ...
##  $ gender                : int  2 2 2 2 2 1 2 1 1 2 ...
##  $ race1                 : int  1 1 1 1 1 1 1 1 2 1 ...
##  $ race2                 : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ race3                 : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ raceoth               : chr  NA NA NA NA ...
##  $ race_native_tribe     : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ hispanic_or_latino    : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ other_ethnicity       : chr  NA NA NA NA ...
##  $ BioKids               : int  3 1 4 2 0 1 1 4 3 3 ...
##  $ hand                  : int  1 1 1 1 2 1 1 1 1 1 ...
##  $ frstlang              : int  NA NA NA NA 1 1 1 1 1 1 ...
##  $ speclang              : chr  NA NA NA NA ...
##  $ moloss                : int  2 1 2 2 1 2 2 3 2 2 ...
##  $ moepisz               : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ mostroke              : int  2 2 2 2 2 2 1 2 1 2 ...
##  $ moms                  : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ mopark                : int  2 2 2 2 2 2 2 3 2 2 ...
##  $ momen                 : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ mohdache              : int  1 2 1 2 2 2 2 2 2 2 ...
##  $ moneuro               : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ motneuro              : int  2 2 NA NA 2 2 2 2 2 2 ...
##  $ specmoth              : chr  NA NA NA NA ...
##  $ faloss                : int  2 2 2 1 2 2 1 3 2 2 ...
##  $ faepisz               : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ fastroke              : int  2 2 2 2 1 2 1 2 2 2 ...
##  $ fams                  : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ fapark                : int  2 2 2 2 2 2 1 2 2 2 ...
##  $ famen                 : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ fahdache              : int  2 2 2 1 2 2 2 2 2 2 ...
##  $ faneuro               : int  2 2 2 2 1 2 2 2 2 2 ...
##  $ foneuro               : int  2 2 NA 2 1 2 2 2 NA 2 ...
##  $ specfath              : chr  NA NA NA NA ...
##  $ famdown               : int  2 2 2 2 2 2 2 2 2 1 ...
##  $ numsis                : int  NA 1 1 3 1 1 1 1 1 2 ...
##  $ numbro                : int  1 3 2 0 2 2 2 2 3 3 ...
##  $ ldschool              : int  2 2 1 2 2 2 2 1 3 2 ...
##  $ add                   : int  NA NA 2 NA NA 2 2 2 2 2 ...
##  $ adhd                  : int  NA NA 2 NA NA 2 2 2 2 2 ...
##  $ ld                    : int  NA NA 2 NA NA 2 2 1 2 2 ...
##  $ otherld               : int  NA NA 1 NA NA 2 2 2 2 2 ...
##  $ specld                : chr  NA NA &quot;hard time remembering things&quot; NA ...
##  $ LearnProb             : int  2 2 3 2 2 2 2 NA NA 2 ...
##  $ RelLearnProb          : int  NA NA NA 2 1 2 1 1 2 2 ...
##  $ Farm                  : int  1 2 1 2 2 2 2 1 1 2 ...
##  $ FarmYrs               : int  1 NA 2 NA NA NA NA 1 3 NA ...
##  $ Color                 : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Grade                 : int  6 5 5 8 11 10 7 7 4 6 ...
##  $ EducYrs               : int  13 12 13 16 26 22 15 15 11 12 ...
##  $ OC_Imputed_grade      : int  2 0 0 0 0 0 0 0 0 0 ...
##  $ OC_Reporting_EducYears: int  13 12 12 16 20 18 14 14 11 13 ...
##  $ DxLearnDis            : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ DxADHD                : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ DxAttDD               : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ DxOth                 : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ DxOthTxt              : chr  NA NA NA NA ...
##  $ interv                : int  6 7 6 6 6 8 9 31 39 9 ...
##  $ nochild               : int  3 1 4 2 0 1 1 NA NA 3 ...
##  $ FH6Bro                : int  1 3 1 0 2 2 2 NA NA 3 ...
##  $ FH6Sis                : int  0 1 1 3 1 1 1 NA NA 2 ...
##  $ interest              : int  1 3 1 1 1 2 1 NA NA 2 ...
##  $ parntres              : int  1 2 1 1 1 1 1 NA NA 2 ...
##  $ Lang                  : int  1 1 1 1 1 1 1 NA NA 1 ...
##  $ LangOth               : chr  NA NA NA NA ...</code></pre>
</details>
<p><br> <br></p>
</div>
<div id="factorscores" class="section level2">
<h2>FactorScores</h2>
<p>Factor scores based on analysis of Wave 1 and Wave 2
neuropsychological test data (see Dowling et al. 2010).</p>
<pre class="r"><code>## load data
FactorScores &lt;- read.csv(paste0(wrap_directory, &quot;FactorScores_05Jun2025.csv&quot;))
info(FactorScores,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:7996, cols:14, inds:1799</code></pre>
<pre class="r"><code>str(FactorScores)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    7996 obs. of  14 variables:
##  $ wrapnum        : chr  &quot;wrap0002&quot; &quot;wrap0068&quot; &quot;wrap0146&quot; &quot;wrap0148&quot; ...
##  $ WRAPNo         : chr  &quot;0002&quot; &quot;0068&quot; &quot;0146&quot; &quot;0148&quot; ...
##  $ VisNo          : int  5 2 4 5 5 4 6 4 7 4 ...
##  $ z_verb_abil    : num  NA 1.37 NA NA NA ...
##  $ z_vis_sptl     : num  NA 1.2 NA NA NA ...
##  $ z_spd_flex     : num  0.931 NA -1.257 2.416 -0.76 ...
##  $ z_wrkg_mem     : num  -1.006 1.323 -0.966 -1.537 1.398 ...
##  $ z_vbl_lng      : num  0.0892 -1.1008 -0.4225 -1.2874 0.6987 ...
##  $ z_imm_mem      : num  -1.362 -0.187 -2.192 -1.285 0.41 ...
##  $ z_vis_lng      : num  0.592 0.4286 -0.0637 -1.4973 0.8993 ...
##  $ z_stry_rcl     : num  -0.954 NA -0.81 0.766 -0.278 ...
##  $ z_verb_abil_adj: num  NA 1.37 NA NA NA ...
##  $ z_imm_mem_adj  : num  -1.362 -0.187 -2.192 -1.285 0.41 ...
##  $ update_stamp   : chr  &quot;2025-01-24 08:17:20.0&quot; &quot;2025-01-24 08:17:20.0&quot; &quot;2025-01-24 08:17:20.0&quot; &quot;2025-01-24 08:17:20.0&quot; ...</code></pre>
</details>
<pre class="r"><code>## check table frequency for all variables

## drop update_stamp column
FactorScores &lt;- subset(FactorScores,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(FactorScores)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    7996 obs. of  13 variables:
##  $ wrapnum        : chr  &quot;wrap0002&quot; &quot;wrap0068&quot; &quot;wrap0146&quot; &quot;wrap0148&quot; ...
##  $ WRAPNo         : chr  &quot;0002&quot; &quot;0068&quot; &quot;0146&quot; &quot;0148&quot; ...
##  $ VisNo          : int  5 2 4 5 5 4 6 4 7 4 ...
##  $ z_verb_abil    : num  NA 1.37 NA NA NA ...
##  $ z_vis_sptl     : num  NA 1.2 NA NA NA ...
##  $ z_spd_flex     : num  0.931 NA -1.257 2.416 -0.76 ...
##  $ z_wrkg_mem     : num  -1.006 1.323 -0.966 -1.537 1.398 ...
##  $ z_vbl_lng      : num  0.0892 -1.1008 -0.4225 -1.2874 0.6987 ...
##  $ z_imm_mem      : num  -1.362 -0.187 -2.192 -1.285 0.41 ...
##  $ z_vis_lng      : num  0.592 0.4286 -0.0637 -1.4973 0.8993 ...
##  $ z_stry_rcl     : num  -0.954 NA -0.81 0.766 -0.278 ...
##  $ z_verb_abil_adj: num  NA 1.37 NA NA NA ...
##  $ z_imm_mem_adj  : num  -1.362 -0.187 -2.192 -1.285 0.41 ...</code></pre>
</details>
<p><br> <br></p>
</div>
<div id="fqrystatisticaldata" class="section level2">
<h2>fqryStatisticalData</h2>
<p>Participant/Visit-level study data</p>
<pre class="r"><code>## load data
fqryStatisticalData &lt;- read.csv(paste0(wrap_directory, &quot;fqryStatisticalData_05Jun2025.csv&quot;))
info(fqryStatisticalData,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:8212, cols:10, inds:1814</code></pre>
<pre class="r"><code>str(fqryStatisticalData)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    8212 obs. of  10 variables:
##  $ WRAPNo             : chr  &quot;0002&quot; &quot;0068&quot; &quot;0146&quot; &quot;0148&quot; ...
##  $ wrapnum            : chr  &quot;wrap0002&quot; &quot;wrap0068&quot; &quot;wrap0146&quot; &quot;wrap0148&quot; ...
##  $ VisNo              : int  5 2 4 5 5 4 6 4 7 4 ...
##  $ Visit_Type         : chr  &quot;in_person&quot; &quot;in_person&quot; &quot;in_person&quot; &quot;in_person&quot; ...
##  $ Visit_Status       : chr  &quot;attended&quot; &quot;attended&quot; &quot;attended&quot; &quot;attended&quot; ...
##  $ Age_At_Visit_Over89: chr  &quot;No&quot; &quot;No&quot; &quot;No&quot; &quot;No&quot; ...
##  $ TesterNo           : int  22 11 22 25 22 14 27 14 39 14 ...
##  $ Age_At_Baseline_Int: int  57 58 62 55 56 42 50 46 43 55 ...
##  $ Days_Since_Baseline: int  5123 2254 3555 4756 4523 3275 5199 3856 6119 3355 ...
##  $ update_stamp       : chr  &quot;2024-09-13 08:52:29.0&quot; &quot;2024-09-13 08:52:29.0&quot; &quot;2024-09-13 08:52:29.0&quot; &quot;2024-09-13 08:52:29.0&quot; ...</code></pre>
</details>
<pre class="r"><code>## check table frequency for all variables

## convert &quot;&quot; to NA for chr columns
chrcols &lt;- colnames(fqryStatisticalData)[sapply(fqryStatisticalData, is.character)]
fqryStatisticalData[chrcols] &lt;- lapply(fqryStatisticalData[chrcols], function(x) ifelse(x == &quot;&quot;, NA, x))

## drop update_stamp column
fqryStatisticalData &lt;- subset(fqryStatisticalData,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(fqryStatisticalData)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    8212 obs. of  9 variables:
##  $ WRAPNo             : chr  &quot;0002&quot; &quot;0068&quot; &quot;0146&quot; &quot;0148&quot; ...
##  $ wrapnum            : chr  &quot;wrap0002&quot; &quot;wrap0068&quot; &quot;wrap0146&quot; &quot;wrap0148&quot; ...
##  $ VisNo              : int  5 2 4 5 5 4 6 4 7 4 ...
##  $ Visit_Type         : chr  &quot;in_person&quot; &quot;in_person&quot; &quot;in_person&quot; &quot;in_person&quot; ...
##  $ Visit_Status       : chr  &quot;attended&quot; &quot;attended&quot; &quot;attended&quot; &quot;attended&quot; ...
##  $ Age_At_Visit_Over89: chr  &quot;No&quot; &quot;No&quot; &quot;No&quot; &quot;No&quot; ...
##  $ TesterNo           : int  22 11 22 25 22 14 27 14 39 14 ...
##  $ Age_At_Baseline_Int: int  57 58 62 55 56 42 50 46 43 55 ...
##  $ Days_Since_Baseline: int  5123 2254 3555 4756 4523 3275 5199 3856 6119 3355 ...</code></pre>
</details>
<p><br> <br></p>
</div>
<div id="gene_datacollection" class="section level2">
<h2>Gene_DataCollection</h2>
<p>Gene Data Collection</p>
<pre class="r"><code>## load data
Gene_DataCollection &lt;- read.csv(paste0(wrap_directory, &quot;Gene_DataCollection_05Jun2025.csv&quot;))
info(Gene_DataCollection,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:1018, cols:15, inds:1018</code></pre>
<pre class="r"><code>str(Gene_DataCollection)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    1018 obs. of  15 variables:
##  $ wrapnum     : chr  &quot;wrap0001&quot; &quot;wrap0002&quot; &quot;wrap0006&quot; &quot;wrap0008&quot; ...
##  $ WRAPNo      : chr  &quot;0001&quot; &quot;0002&quot; &quot;0006&quot; &quot;0008&quot; ...
##  $ C677T       : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ A1298C      : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ CBS         : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ T833C       : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ A2756G      : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ MTRR        : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ T471C       : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ APOE_C609T  : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ APOE112_158 : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ A209G       : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ G1615A      : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ G248A       : chr  &quot;C&quot; &quot;H&quot; &quot;C&quot; &quot;C&quot; ...
##  $ update_stamp: chr  &quot;2024-09-13 08:52:30.0&quot; &quot;2024-09-13 08:52:30.0&quot; &quot;2024-09-13 08:52:30.0&quot; &quot;2024-09-13 08:52:30.0&quot; ...</code></pre>
</details>
<pre class="r"><code>## check table frequency for all variables

## convert &quot;&quot; to NA for chr columns
chrcols &lt;- colnames(Gene_DataCollection)[sapply(Gene_DataCollection, is.character)]
Gene_DataCollection[chrcols] &lt;- lapply(Gene_DataCollection[chrcols], function(x) ifelse(x == &quot;&quot;, NA, x))

## drop update_stamp column
Gene_DataCollection &lt;- subset(Gene_DataCollection,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(Gene_DataCollection)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    1018 obs. of  14 variables:
##  $ wrapnum    : chr  &quot;wrap0001&quot; &quot;wrap0002&quot; &quot;wrap0006&quot; &quot;wrap0008&quot; ...
##  $ WRAPNo     : chr  &quot;0001&quot; &quot;0002&quot; &quot;0006&quot; &quot;0008&quot; ...
##  $ C677T      : chr  NA NA NA NA ...
##  $ A1298C     : chr  NA NA NA NA ...
##  $ CBS        : chr  NA NA NA NA ...
##  $ T833C      : chr  NA NA NA NA ...
##  $ A2756G     : chr  NA NA NA NA ...
##  $ MTRR       : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ T471C      : chr  NA NA NA NA ...
##  $ APOE_C609T : chr  NA NA NA NA ...
##  $ APOE112_158: int  NA NA NA NA NA NA NA NA NA NA ...
##  $ A209G      : chr  NA NA NA NA ...
##  $ G1615A     : chr  NA NA NA NA ...
##  $ G248A      : chr  &quot;C&quot; &quot;H&quot; &quot;C&quot; &quot;C&quot; ...</code></pre>
</details>
<p><br> <br></p>
</div>
<div id="genhealthdailyact" class="section level2">
<h2>GenHealthDailyAct</h2>
<p>General Health and Daily Activities</p>
<pre class="r"><code>## load data
GenHealthDailyAct &lt;- read.csv(paste0(wrap_directory, &quot;GenHealthDailyAct_05Jun2025.csv&quot;))
info(GenHealthDailyAct,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:8168, cols:220, inds:1796</code></pre>
<pre class="r"><code>str(GenHealthDailyAct,max.level = 99, list.len = 99999)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    8168 obs. of  220 variables:
##  $ wrapnum          : chr  &quot;wrap0002&quot; &quot;wrap0068&quot; &quot;wrap0146&quot; &quot;wrap0148&quot; ...
##  $ WRAPNo           : chr  &quot;0002&quot; &quot;0068&quot; &quot;0146&quot; &quot;0148&quot; ...
##  $ VisNo            : int  5 2 4 5 5 4 6 4 7 4 ...
##  $ CurHealth        : int  3 5 5 3 4 4 4 3 4 5 ...
##  $ Htft             : int  5 6 5 5 5 5 5 5 5 5 ...
##  $ Htin             : int  6 1 4 9 4 5 10 7 9 5 ...
##  $ Tall18ft         : int  NA 6 NA NA NA NA NA NA NA NA ...
##  $ Tall18in         : int  NA 0 NA NA NA NA NA NA NA NA ...
##  $ Wt18             : int  NA 175 NA NA NA NA NA NA NA NA ...
##  $ Wt35             : int  NA 185 NA NA NA NA NA NA NA NA ...
##  $ Wt50             : int  NA 195 NA NA NA NA NA NA NA NA ...
##  $ Wt50NA           : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ WtMax            : int  NA 202 NA NA NA NA NA NA NA NA ...
##  $ Wtlbs            : int  190 195 200 210 134 143 168 223 188 139 ...
##  $ Smoke            : int  NA 2 NA NA NA NA NA NA NA NA ...
##  $ SmokeYrs         : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ SmokeNo          : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ SmokePast        : int  2 NA 2 2 2 2 2 2 2 2 ...
##  $ SmokePastNo      : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ TobProd          : int  2 NA 2 2 2 2 2 2 2 1 ...
##  $ LiveSmoke        : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ LiveYrs          : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ LivCur           : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ SmokeJob         : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ JobYrs           : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ JobCur           : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ AlcPast          : int  1 1 1 1 2 1 1 1 1 1 ...
##  $ AlcDays          : int  1 5 1 1 NA 2 4 4 1 5 ...
##  $ AlcNo            : int  1 2 1 2 NA 2 2 2 3 2 ...
##  $ AlcProb          : int  2 NA 2 2 2 2 2 2 2 2 ...
##  $ DrugProb         : int  NA 2 NA NA NA NA NA NA NA NA ...
##  $ AlcDrugProb      : int  NA 2 NA NA NA NA NA NA NA NA ...
##  $ Prob6            : int  NA 2 NA NA NA NA NA NA NA NA ...
##  $ Caffeine         : int  2 2 1 2 1 3 2 4 1 3 ...
##  $ MemProb          : int  1 2 2 2 2 2 3 3 2 2 ...
##  $ ToldMem          : int  NA 2 NA NA NA NA NA NA NA NA ...
##  $ MemRate          : int  3 6 6 6 6 6 6 4 5 5 ...
##  $ memocclm         : int  2 7 6 6 6 5 5 5 5 6 ...
##  $ memoccyr         : int  2 7 6 6 5 5 5 5 5 6 ...
##  $ memoccfy         : int  2 7 6 6 5 4 4 5 5 6 ...
##  $ memoccty         : int  2 7 6 5 4 4 4 4 5 6 ...
##  $ freqtna          : int  3 5 4 6 7 3 4 5 2 5 ...
##  $ freqfac          : int  6 7 7 7 7 5 6 6 6 6 ...
##  $ freqapp          : int  5 7 7 7 5 6 6 7 7 6 ...
##  $ freqitem         : int  5 6 7 6 6 5 6 5 6 4 ...
##  $ freqcho          : int  6 7 7 7 7 7 6 7 6 7 ...
##  $ freqdir          : int  5 7 7 6 5 4 NA 7 6 7 ...
##  $ freqphno         : int  2 6 7 7 6 5 8 6 6 7 ...
##  $ freqfph          : int  5 7 7 7 6 6 8 6 7 6 ...
##  $ freqtel          : int  6 6 7 6 6 5 5 6 6 5 ...
##  $ freqcor          : int  6 6 4 7 5 5 6 7 6 6 ...
##  $ freqpdat         : int  5 7 7 7 5 5 7 7 2 7 ...
##  $ freqwor          : int  2 5 5 6 5 5 6 5 5 6 ...
##  $ freqbuy          : int  5 7 7 6 7 6 8 6 6 7 ...
##  $ freqtes          : int  8 7 8 5 8 8 8 8 4 8 ...
##  $ freqdo           : int  4 7 7 7 7 6 6 6 6 5 ...
##  $ freqconv         : int  4 7 7 6 7 6 6 4 6 5 ...
##  $ freqpubs         : int  8 7 7 7 6 8 7 8 8 6 ...
##  $ freqrep          : int  3 6 7 6 6 5 5 5 6 5 ...
##  $ forgtna          : int  6 6 6 7 6 6 6 6 4 6 ...
##  $ forgfac          : int  6 8 7 7 7 6 6 6 6 6 ...
##  $ forgapp          : int  6 8 7 5 6 5 3 4 6 6 ...
##  $ forgitem         : int  6 8 7 6 7 6 5 4 6 6 ...
##  $ forgcho          : int  6 8 7 5 6 6 7 6 6 6 ...
##  $ forgdir          : int  6 8 7 6 6 6 6 5 6 6 ...
##  $ forgphno         : int  6 7 7 6 7 6 7 6 6 6 ...
##  $ forgfph          : int  6 8 7 6 6 6 7 4 6 6 ...
##  $ forgtel          : int  6 7 6 5 6 6 6 3 6 6 ...
##  $ forgcor          : int  6 7 6 6 6 6 6 3 6 6 ...
##  $ forgpdat         : int  6 8 7 5 6 6 7 6 4 6 ...
##  $ forgwor          : int  6 6 6 5 6 6 6 3 6 6 ...
##  $ forgbuy          : int  6 8 7 6 6 6 7 5 6 7 ...
##  $ forgtes          : int  8 8 8 5 8 8 8 8 6 6 ...
##  $ forgdo           : int  6 8 7 4 6 6 6 4 6 6 ...
##  $ forgconv         : int  6 8 7 6 6 6 6 4 6 6 ...
##  $ forgpubs         : int  8 8 7 4 6 8 8 8 8 5 ...
##  $ forgrep          : int  6 7 7 6 7 6 6 5 6 6 ...
##  $ retmem1          : int  4 4 5 4 4 4 3 4 4 3 ...
##  $ retmem5          : int  4 4 4 3 3 4 3 4 4 5 ...
##  $ retmem10         : int  2 4 3 3 3 5 2 3 4 6 ...
##  $ retmem20         : int  2 4 3 3 3 3 2 3 4 6 ...
##  $ retmem18         : int  1 4 3 3 3 3 1 2 4 7 ...
##  $ mnemappt         : int  2 1 6 1 2 1 1 1 7 2 ...
##  $ mnemnote         : int  2 2 6 1 1 1 2 4 2 2 ...
##  $ mnemlist         : int  2 1 6 1 3 1 2 7 2 2 ...
##  $ mnemgrol         : int  2 1 4 1 3 1 1 2 1 2 ...
##  $ mnemsche         : int  2 1 5 1 4 3 2 1 6 2 ...
##  $ mnemrep          : int  4 3 5 3 5 4 3 4 4 2 ...
##  $ mnemasoc         : int  2 4 5 3 4 4 3 4 3 7 ...
##  $ mnemprom         : int  1 2 6 4 4 2 2 1 3 3 ...
##  $ PhysAct          : int  4 4 4 3 4 4 4 4 4 4 ...
##  $ WalkOut          : int  1 5 4 3 4 4 6 4 5 4 ...
##  $ WalkMin          : int  NA 2 2 2 1 2 4 2 3 3 ...
##  $ WalkSp           : int  1 2 1 2 3 3 3 2 2 3 ...
##  $ HardEx           : int  1 4 5 1 2 5 4 1 2 2 ...
##  $ HardExTime       : int  NA 3 2 NA 1 2 2 NA 2 1 ...
##  $ ModEx            : int  4 3 1 1 2 3 1 1 1 6 ...
##  $ ModExTime        : int  3 1 NA NA 1 2 NA NA 1 3 ...
##  $ MildEx           : int  1 3 1 2 2 2 1 4 2 4 ...
##  $ MildExTime       : int  NA 3 NA 2 1 2 NA 1 3 3 ...
##  $ Chores           : int  1 1 3 2 1 1 1 2 1 2 ...
##  $ Yard             : int  4 5 2 3 2 3 6 5 5 4 ...
##  $ YardHrs          : int  4 4 2 2 2 2 6 2 1 5 ...
##  $ Sitting          : int  3 4 3 4 4 5 2 5 5 5 ...
##  $ ExTypical        : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ TV               : int  5 3 5 5 2 4 5 4 5 4 ...
##  $ TVHrs            : num  4 NA 3 3 NA NA 1 NA 2.5 NA ...
##  $ Radio            : int  1 5 5 5 5 5 5 5 4 5 ...
##  $ RadioHrs         : num  NA 2 1 4 1 2 2 6 NA 3 ...
##  $ Paper            : int  4 4 5 5 4 3 4 3 3 5 ...
##  $ Mag              : int  4 5 5 5 4 4 3 3 2 4 ...
##  $ Book             : int  5 5 3 5 3 2 4 2 2 5 ...
##  $ Games            : int  5 4 5 5 2 5 3 3 2 2 ...
##  $ Museum           : int  2 2 2 1 2 1 1 1 1 2 ...
##  $ Plays            : int  2 1 2 1 3 2 3 2 1 2 ...
##  $ Educ             : int  3 2 2 1 3 1 3 1 1 1 ...
##  $ Comp             : int  5 5 5 5 5 5 5 4 3 5 ...
##  $ CompHrs          : num  3 2 1 3 6 9 1 NA NA 3 ...
##  $ SmartPhonTbltUse : int  4 NA NA 5 5 NA 4 NA 5 NA ...
##  $ SmartPhonTbltHrs : int  NA NA NA 4 1 NA NA NA 1 NA ...
##  $ KidClose         : int  NA 2 4 1 3 2 3 4 NA 1 ...
##  $ RelFr            : int  2 6 5 4 6 4 8 15 10 4 ...
##  $ LiveWith         : int  1 2 2 2 2 2 2 2 2 2 ...
##  $ LiveWithOth      : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ Pet              : int  2 1 1 2 1 2 1 1 1 1 ...
##  $ Dog              : int  0 0 0 0 0 0 0 -1 -1 -1 ...
##  $ Cat              : int  0 -1 -1 0 -1 0 0 0 0 -1 ...
##  $ Bird             : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ Fish             : int  0 0 0 0 0 0 -1 0 0 0 ...
##  $ OthPet           : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ PetOth           : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ Help             : int  2 2 2 1 2 2 2 2 2 2 ...
##  $ HelpTime         : int  NA NA NA 2 NA NA NA NA NA NA ...
##  $ HelpAD           : int  NA 1 NA 1 NA NA NA NA NA NA ...
##  $ HelpRel          : int  NA 1 NA 2 NA NA NA NA NA NA ...
##  $ CareLiveWith     : int  NA 2 NA 2 NA NA NA NA NA NA ...
##  $ CareLiveClose    : int  NA 1 NA 1 NA NA NA NA NA NA ...
##  $ CareFac          : int  NA 2 NA 1 NA NA NA NA NA NA ...
##  $ CareLvl          : int  NA 3 NA 3 NA NA NA NA NA NA ...
##  $ CarePrim         : int  NA 2 NA 2 NA NA NA NA NA NA ...
##  $ CarePhone        : int  NA 4 NA 4 NA NA NA NA NA NA ...
##  $ CareSee          : int  NA 4 NA 4 NA NA NA NA NA NA ...
##  $ CareShare        : int  NA 1 NA 1 NA NA NA NA NA NA ...
##  $ CareShareWho     : int  NA 1 NA 1 NA NA NA NA NA NA ...
##  $ CarePleasant     : int  NA NA NA 4 NA NA NA NA NA NA ...
##  $ Listen           : int  4 5 5 5 5 4 5 5 5 4 ...
##  $ Advice           : int  4 5 4 5 5 3 5 5 5 4 ...
##  $ Doctor           : int  2 5 5 4 5 4 5 5 5 5 ...
##  $ GoodTime         : int  4 5 3 5 5 4 4 5 5 4 ...
##  $ Understand       : int  4 5 4 5 5 4 5 5 5 5 ...
##  $ DailyCh          : int  2 5 4 5 5 3 5 4 5 4 ...
##  $ Share            : int  2 5 3 5 5 4 4 5 5 5 ...
##  $ Fun              : int  3 5 5 5 5 4 4 4 5 4 ...
##  $ Love             : int  2 5 4 5 5 4 5 5 5 5 ...
##  $ SpDie            : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ SpIll            : int  1 1 1 1 1 1 1 2 1 1 ...
##  $ FrDie            : int  1 2 3 1 1 1 1 3 2 1 ...
##  $ Money            : int  1 1 1 1 1 2 1 1 1 1 ...
##  $ SpDivorce        : int  1 3 1 1 1 1 1 1 1 1 ...
##  $ FrDivorce        : int  1 2 1 1 1 1 1 2 1 1 ...
##  $ Conflict         : int  1 1 1 3 1 1 1 1 1 1 ...
##  $ Accid            : int  2 1 1 1 1 1 1 1 1 1 ...
##  $ LoseJob          : int  1 1 1 1 1 1 1 2 2 1 ...
##  $ PetDie           : int  1 1 1 1 1 1 1 1 1 2 ...
##  $ MCIDEM           : int  2 NA 2 2 2 2 2 2 2 2 ...
##  $ MCIDEMTYPE       : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ SelfIllness      : int  1 NA NA 1 1 NA 1 NA 1 NA ...
##  $ SleepHrs         : int  NA 8 NA NA NA NA NA NA NA NA ...
##  $ Church           : int  1 3 5 2 4 4 NA 2 NA 1 ...
##  $ Comfort          : int  3 3 3 2 3 3 NA 2 NA 1 ...
##  $ Meetings         : int  2 1 3 2 1 1 NA 1 NA 1 ...
##  $ PhAbused         : int  NA 1 1 NA NA 1 NA 1 NA 1 ...
##  $ VAbused          : int  NA 1 1 NA NA 1 NA 1 NA 1 ...
##  $ FTFSPPART        : int  6 NA 2 4 3 3 NA 5 NA 3 ...
##  $ FTFSPPART_ET     : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ FTFSPPARTPL      : int  NA NA 5 5 5 5 NA 5 NA 4 ...
##  $ FTFFAM           : int  6 NA 6 2 3 3 NA 4 NA 1 ...
##  $ FTFFAM_ET        : num  NA NA NA NA NA NA NA NA NA NA ...
##  $ FTFFAMPL         : int  NA NA NA 5 5 5 NA 5 NA 5 ...
##  $ FTFFRIEND        : int  5 NA 4 3 2 5 NA 2 NA 5 ...
##  $ FTFFRIEND_ET     : int  NA NA NA NA NA 180 NA NA NA NA ...
##  $ FTFFRIENDPL      : int  5 NA 5 5 4 5 NA 5 NA 5 ...
##  $ FTFCHURCH        : int  6 NA 2 1 2 1 NA 1 NA 6 ...
##  $ FTFCHURCH_ET     : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ FTFCHURCHPL      : int  NA NA 5 4 5 5 NA 5 NA NA ...
##  $ FTFMEETINGS      : int  4 NA 2 4 6 2 NA 6 NA 6 ...
##  $ FTFMEETINGS_ET   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ FTFMEETINSSPL    : int  3 NA 5 5 NA 5 NA NA NA NA ...
##  $ DAYSWORK         : int  6 NA 6 5 5 4 NA 4 NA 6 ...
##  $ FTFWORK          : int  NA NA NA 5 2 5 NA 5 NA NA ...
##  $ FTFWORK_ET       : int  NA NA NA 300 NA 240 NA 180 NA NA ...
##  $ FTFWORKPL        : int  NA NA NA 5 3 4 NA 5 NA NA ...
##  $ FTFSTRANGER      : int  2 NA 2 2 5 1 NA 1 NA 1 ...
##  $ FTFSTRANGER_ET   : int  NA NA NA NA 1800 NA NA NA NA NA ...
##  $ FTFSTRANGERPL    : int  4 NA 4 5 4 5 NA 5 NA 5 ...
##  $ FTFOTHER         : int  2 NA 1 2 2 2 NA 1 NA 2 ...
##  $ FTFOTHER_TXT     : chr  &quot;&quot; &quot;&quot; &quot;volunteering, choir, bells, quilting&quot; &quot;&quot; ...
##  $ FTFOTHERTIME     : int  NA NA 300 NA NA NA NA 1500 NA NA ...
##  $ FTFOTHERPL       : int  NA NA 5 NA NA NA NA 5 NA NA ...
##  $ VERBTEL          : int  2 NA 1 5 1 2 NA 2 NA 1 ...
##  $ VERBTEL_TXT      : int  NA NA NA 300 NA NA NA NA NA NA ...
##  $ VERBTELPL        : int  5 NA 5 4 4 5 NA 5 NA 4 ...
##  $ WRITCONVIM       : int  6 NA 1 2 1 2 NA 6 NA 1 ...
##  $ WRITCONVIM_TXT   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ WRITCONVIMPL     : int  NA NA 5 4 4 4 NA NA NA 5 ...
##  $ WRITCONVEMAIL    : int  3 NA 1 5 1 4 NA 1 NA 1 ...
##  $ WRITCONVEMAIL_TXT: int  NA NA NA 240 NA NA NA NA NA NA ...
##  $ WRITCONVEMAILPL  : int  3 NA 5 4 5 4 NA 4 NA 5 ...
##  $ WRITLETTERS      : int  1 NA 1 6 1 2 NA 1 NA 6 ...
##  $ WRITLETTERS_TXT  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ HELPRELPL        : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ xswkexer         : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ rateexer         : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ HardHrWk         : num  0 3 3 0 0.25 3 2.25 0 0.75 0.25 ...
##  $ ModHrWk          : num  3 0.5 0 0 0.25 1.5 0 0 0 5 ...
##  $ MildHrWk         : num  0 2 0 0.75 0.25 0.75 0 0.75 1 3 ...
##  $ WalkMet          : int  0 3 0 3 4 4 4 3 3 4 ...
##  $ WalkHrWk         : num  0 2.5 1.25 0.5 0.425 ...
##  $ MetHrWk          : num  12 36.5 21 3.75 5.2 ...
##  $ update_stamp     : chr  &quot;2025-01-24 08:17:47.0&quot; &quot;2025-01-24 08:17:47.0&quot; &quot;2025-01-24 08:17:47.0&quot; &quot;2025-01-24 08:17:47.0&quot; ...</code></pre>
</details>
<pre class="r"><code>## check table frequency for all variables

## convert 6 to NA for columns: YardHrs,FTFFAMPL,FTFMEETINSSPL,FTFSTRANGERPL
tmp &lt;- c(&quot;YardHrs&quot;,&quot;FTFFAMPL&quot;,&quot;FTFMEETINSSPL&quot;,&quot;FTFSTRANGERPL&quot;)
GenHealthDailyAct[tmp] &lt;- lapply(GenHealthDailyAct[tmp], function(x) {
  x[x == 6] &lt;- NA
  return(x)
})

## convert 5 to NA for column: FTFOTHER
GenHealthDailyAct$FTFOTHER[GenHealthDailyAct$FTFOTHER==5] &lt;- NA

## convert 8 to NA for column: FTFSPPARTPL
GenHealthDailyAct$FTFSPPARTPL[GenHealthDailyAct$FTFSPPARTPL==8] &lt;- NA

## convert &quot;&quot; to NA for chr columns
chrcols &lt;- colnames(GenHealthDailyAct)[sapply(GenHealthDailyAct, is.character)]
GenHealthDailyAct[chrcols] &lt;- lapply(GenHealthDailyAct[chrcols], function(x) ifelse(x == &quot;&quot;, NA, x))

## drop update_stamp column
GenHealthDailyAct &lt;- subset(GenHealthDailyAct,select = -c(update_stamp)) ## 219 columns left</code></pre>
<pre class="r"><code>str(GenHealthDailyAct,max.level = 99, list.len = 99999)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    8168 obs. of  219 variables:
##  $ wrapnum          : chr  &quot;wrap0002&quot; &quot;wrap0068&quot; &quot;wrap0146&quot; &quot;wrap0148&quot; ...
##  $ WRAPNo           : chr  &quot;0002&quot; &quot;0068&quot; &quot;0146&quot; &quot;0148&quot; ...
##  $ VisNo            : int  5 2 4 5 5 4 6 4 7 4 ...
##  $ CurHealth        : int  3 5 5 3 4 4 4 3 4 5 ...
##  $ Htft             : int  5 6 5 5 5 5 5 5 5 5 ...
##  $ Htin             : int  6 1 4 9 4 5 10 7 9 5 ...
##  $ Tall18ft         : int  NA 6 NA NA NA NA NA NA NA NA ...
##  $ Tall18in         : int  NA 0 NA NA NA NA NA NA NA NA ...
##  $ Wt18             : int  NA 175 NA NA NA NA NA NA NA NA ...
##  $ Wt35             : int  NA 185 NA NA NA NA NA NA NA NA ...
##  $ Wt50             : int  NA 195 NA NA NA NA NA NA NA NA ...
##  $ Wt50NA           : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ WtMax            : int  NA 202 NA NA NA NA NA NA NA NA ...
##  $ Wtlbs            : int  190 195 200 210 134 143 168 223 188 139 ...
##  $ Smoke            : int  NA 2 NA NA NA NA NA NA NA NA ...
##  $ SmokeYrs         : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ SmokeNo          : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ SmokePast        : int  2 NA 2 2 2 2 2 2 2 2 ...
##  $ SmokePastNo      : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ TobProd          : int  2 NA 2 2 2 2 2 2 2 1 ...
##  $ LiveSmoke        : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ LiveYrs          : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ LivCur           : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ SmokeJob         : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ JobYrs           : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ JobCur           : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ AlcPast          : int  1 1 1 1 2 1 1 1 1 1 ...
##  $ AlcDays          : int  1 5 1 1 NA 2 4 4 1 5 ...
##  $ AlcNo            : int  1 2 1 2 NA 2 2 2 3 2 ...
##  $ AlcProb          : int  2 NA 2 2 2 2 2 2 2 2 ...
##  $ DrugProb         : int  NA 2 NA NA NA NA NA NA NA NA ...
##  $ AlcDrugProb      : int  NA 2 NA NA NA NA NA NA NA NA ...
##  $ Prob6            : int  NA 2 NA NA NA NA NA NA NA NA ...
##  $ Caffeine         : int  2 2 1 2 1 3 2 4 1 3 ...
##  $ MemProb          : int  1 2 2 2 2 2 3 3 2 2 ...
##  $ ToldMem          : int  NA 2 NA NA NA NA NA NA NA NA ...
##  $ MemRate          : int  3 6 6 6 6 6 6 4 5 5 ...
##  $ memocclm         : int  2 7 6 6 6 5 5 5 5 6 ...
##  $ memoccyr         : int  2 7 6 6 5 5 5 5 5 6 ...
##  $ memoccfy         : int  2 7 6 6 5 4 4 5 5 6 ...
##  $ memoccty         : int  2 7 6 5 4 4 4 4 5 6 ...
##  $ freqtna          : int  3 5 4 6 7 3 4 5 2 5 ...
##  $ freqfac          : int  6 7 7 7 7 5 6 6 6 6 ...
##  $ freqapp          : int  5 7 7 7 5 6 6 7 7 6 ...
##  $ freqitem         : int  5 6 7 6 6 5 6 5 6 4 ...
##  $ freqcho          : int  6 7 7 7 7 7 6 7 6 7 ...
##  $ freqdir          : int  5 7 7 6 5 4 NA 7 6 7 ...
##  $ freqphno         : int  2 6 7 7 6 5 8 6 6 7 ...
##  $ freqfph          : int  5 7 7 7 6 6 8 6 7 6 ...
##  $ freqtel          : int  6 6 7 6 6 5 5 6 6 5 ...
##  $ freqcor          : int  6 6 4 7 5 5 6 7 6 6 ...
##  $ freqpdat         : int  5 7 7 7 5 5 7 7 2 7 ...
##  $ freqwor          : int  2 5 5 6 5 5 6 5 5 6 ...
##  $ freqbuy          : int  5 7 7 6 7 6 8 6 6 7 ...
##  $ freqtes          : int  8 7 8 5 8 8 8 8 4 8 ...
##  $ freqdo           : int  4 7 7 7 7 6 6 6 6 5 ...
##  $ freqconv         : int  4 7 7 6 7 6 6 4 6 5 ...
##  $ freqpubs         : int  8 7 7 7 6 8 7 8 8 6 ...
##  $ freqrep          : int  3 6 7 6 6 5 5 5 6 5 ...
##  $ forgtna          : int  6 6 6 7 6 6 6 6 4 6 ...
##  $ forgfac          : int  6 8 7 7 7 6 6 6 6 6 ...
##  $ forgapp          : int  6 8 7 5 6 5 3 4 6 6 ...
##  $ forgitem         : int  6 8 7 6 7 6 5 4 6 6 ...
##  $ forgcho          : int  6 8 7 5 6 6 7 6 6 6 ...
##  $ forgdir          : int  6 8 7 6 6 6 6 5 6 6 ...
##  $ forgphno         : int  6 7 7 6 7 6 7 6 6 6 ...
##  $ forgfph          : int  6 8 7 6 6 6 7 4 6 6 ...
##  $ forgtel          : int  6 7 6 5 6 6 6 3 6 6 ...
##  $ forgcor          : int  6 7 6 6 6 6 6 3 6 6 ...
##  $ forgpdat         : int  6 8 7 5 6 6 7 6 4 6 ...
##  $ forgwor          : int  6 6 6 5 6 6 6 3 6 6 ...
##  $ forgbuy          : int  6 8 7 6 6 6 7 5 6 7 ...
##  $ forgtes          : int  8 8 8 5 8 8 8 8 6 6 ...
##  $ forgdo           : int  6 8 7 4 6 6 6 4 6 6 ...
##  $ forgconv         : int  6 8 7 6 6 6 6 4 6 6 ...
##  $ forgpubs         : int  8 8 7 4 6 8 8 8 8 5 ...
##  $ forgrep          : int  6 7 7 6 7 6 6 5 6 6 ...
##  $ retmem1          : int  4 4 5 4 4 4 3 4 4 3 ...
##  $ retmem5          : int  4 4 4 3 3 4 3 4 4 5 ...
##  $ retmem10         : int  2 4 3 3 3 5 2 3 4 6 ...
##  $ retmem20         : int  2 4 3 3 3 3 2 3 4 6 ...
##  $ retmem18         : int  1 4 3 3 3 3 1 2 4 7 ...
##  $ mnemappt         : int  2 1 6 1 2 1 1 1 7 2 ...
##  $ mnemnote         : int  2 2 6 1 1 1 2 4 2 2 ...
##  $ mnemlist         : int  2 1 6 1 3 1 2 7 2 2 ...
##  $ mnemgrol         : int  2 1 4 1 3 1 1 2 1 2 ...
##  $ mnemsche         : int  2 1 5 1 4 3 2 1 6 2 ...
##  $ mnemrep          : int  4 3 5 3 5 4 3 4 4 2 ...
##  $ mnemasoc         : int  2 4 5 3 4 4 3 4 3 7 ...
##  $ mnemprom         : int  1 2 6 4 4 2 2 1 3 3 ...
##  $ PhysAct          : int  4 4 4 3 4 4 4 4 4 4 ...
##  $ WalkOut          : int  1 5 4 3 4 4 6 4 5 4 ...
##  $ WalkMin          : int  NA 2 2 2 1 2 4 2 3 3 ...
##  $ WalkSp           : int  1 2 1 2 3 3 3 2 2 3 ...
##  $ HardEx           : int  1 4 5 1 2 5 4 1 2 2 ...
##  $ HardExTime       : int  NA 3 2 NA 1 2 2 NA 2 1 ...
##  $ ModEx            : int  4 3 1 1 2 3 1 1 1 6 ...
##  $ ModExTime        : int  3 1 NA NA 1 2 NA NA 1 3 ...
##  $ MildEx           : int  1 3 1 2 2 2 1 4 2 4 ...
##  $ MildExTime       : int  NA 3 NA 2 1 2 NA 1 3 3 ...
##  $ Chores           : int  1 1 3 2 1 1 1 2 1 2 ...
##  $ Yard             : int  4 5 2 3 2 3 6 5 5 4 ...
##  $ YardHrs          : int  4 4 2 2 2 2 NA 2 1 5 ...
##  $ Sitting          : int  3 4 3 4 4 5 2 5 5 5 ...
##  $ ExTypical        : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ TV               : int  5 3 5 5 2 4 5 4 5 4 ...
##  $ TVHrs            : num  4 NA 3 3 NA NA 1 NA 2.5 NA ...
##  $ Radio            : int  1 5 5 5 5 5 5 5 4 5 ...
##  $ RadioHrs         : num  NA 2 1 4 1 2 2 6 NA 3 ...
##  $ Paper            : int  4 4 5 5 4 3 4 3 3 5 ...
##  $ Mag              : int  4 5 5 5 4 4 3 3 2 4 ...
##  $ Book             : int  5 5 3 5 3 2 4 2 2 5 ...
##  $ Games            : int  5 4 5 5 2 5 3 3 2 2 ...
##  $ Museum           : int  2 2 2 1 2 1 1 1 1 2 ...
##  $ Plays            : int  2 1 2 1 3 2 3 2 1 2 ...
##  $ Educ             : int  3 2 2 1 3 1 3 1 1 1 ...
##  $ Comp             : int  5 5 5 5 5 5 5 4 3 5 ...
##  $ CompHrs          : num  3 2 1 3 6 9 1 NA NA 3 ...
##  $ SmartPhonTbltUse : int  4 NA NA 5 5 NA 4 NA 5 NA ...
##  $ SmartPhonTbltHrs : int  NA NA NA 4 1 NA NA NA 1 NA ...
##  $ KidClose         : int  NA 2 4 1 3 2 3 4 NA 1 ...
##  $ RelFr            : int  2 6 5 4 6 4 8 15 10 4 ...
##  $ LiveWith         : int  1 2 2 2 2 2 2 2 2 2 ...
##  $ LiveWithOth      : chr  NA NA NA NA ...
##  $ Pet              : int  2 1 1 2 1 2 1 1 1 1 ...
##  $ Dog              : int  0 0 0 0 0 0 0 -1 -1 -1 ...
##  $ Cat              : int  0 -1 -1 0 -1 0 0 0 0 -1 ...
##  $ Bird             : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ Fish             : int  0 0 0 0 0 0 -1 0 0 0 ...
##  $ OthPet           : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ PetOth           : chr  NA NA NA NA ...
##  $ Help             : int  2 2 2 1 2 2 2 2 2 2 ...
##  $ HelpTime         : int  NA NA NA 2 NA NA NA NA NA NA ...
##  $ HelpAD           : int  NA 1 NA 1 NA NA NA NA NA NA ...
##  $ HelpRel          : int  NA 1 NA 2 NA NA NA NA NA NA ...
##  $ CareLiveWith     : int  NA 2 NA 2 NA NA NA NA NA NA ...
##  $ CareLiveClose    : int  NA 1 NA 1 NA NA NA NA NA NA ...
##  $ CareFac          : int  NA 2 NA 1 NA NA NA NA NA NA ...
##  $ CareLvl          : int  NA 3 NA 3 NA NA NA NA NA NA ...
##  $ CarePrim         : int  NA 2 NA 2 NA NA NA NA NA NA ...
##  $ CarePhone        : int  NA 4 NA 4 NA NA NA NA NA NA ...
##  $ CareSee          : int  NA 4 NA 4 NA NA NA NA NA NA ...
##  $ CareShare        : int  NA 1 NA 1 NA NA NA NA NA NA ...
##  $ CareShareWho     : int  NA 1 NA 1 NA NA NA NA NA NA ...
##  $ CarePleasant     : int  NA NA NA 4 NA NA NA NA NA NA ...
##  $ Listen           : int  4 5 5 5 5 4 5 5 5 4 ...
##  $ Advice           : int  4 5 4 5 5 3 5 5 5 4 ...
##  $ Doctor           : int  2 5 5 4 5 4 5 5 5 5 ...
##  $ GoodTime         : int  4 5 3 5 5 4 4 5 5 4 ...
##  $ Understand       : int  4 5 4 5 5 4 5 5 5 5 ...
##  $ DailyCh          : int  2 5 4 5 5 3 5 4 5 4 ...
##  $ Share            : int  2 5 3 5 5 4 4 5 5 5 ...
##  $ Fun              : int  3 5 5 5 5 4 4 4 5 4 ...
##  $ Love             : int  2 5 4 5 5 4 5 5 5 5 ...
##  $ SpDie            : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ SpIll            : int  1 1 1 1 1 1 1 2 1 1 ...
##  $ FrDie            : int  1 2 3 1 1 1 1 3 2 1 ...
##  $ Money            : int  1 1 1 1 1 2 1 1 1 1 ...
##  $ SpDivorce        : int  1 3 1 1 1 1 1 1 1 1 ...
##  $ FrDivorce        : int  1 2 1 1 1 1 1 2 1 1 ...
##  $ Conflict         : int  1 1 1 3 1 1 1 1 1 1 ...
##  $ Accid            : int  2 1 1 1 1 1 1 1 1 1 ...
##  $ LoseJob          : int  1 1 1 1 1 1 1 2 2 1 ...
##  $ PetDie           : int  1 1 1 1 1 1 1 1 1 2 ...
##  $ MCIDEM           : int  2 NA 2 2 2 2 2 2 2 2 ...
##  $ MCIDEMTYPE       : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ SelfIllness      : int  1 NA NA 1 1 NA 1 NA 1 NA ...
##  $ SleepHrs         : int  NA 8 NA NA NA NA NA NA NA NA ...
##  $ Church           : int  1 3 5 2 4 4 NA 2 NA 1 ...
##  $ Comfort          : int  3 3 3 2 3 3 NA 2 NA 1 ...
##  $ Meetings         : int  2 1 3 2 1 1 NA 1 NA 1 ...
##  $ PhAbused         : int  NA 1 1 NA NA 1 NA 1 NA 1 ...
##  $ VAbused          : int  NA 1 1 NA NA 1 NA 1 NA 1 ...
##  $ FTFSPPART        : int  6 NA 2 4 3 3 NA 5 NA 3 ...
##  $ FTFSPPART_ET     : chr  NA NA NA NA ...
##  $ FTFSPPARTPL      : int  NA NA 5 5 5 5 NA 5 NA 4 ...
##  $ FTFFAM           : int  6 NA 6 2 3 3 NA 4 NA 1 ...
##  $ FTFFAM_ET        : num  NA NA NA NA NA NA NA NA NA NA ...
##  $ FTFFAMPL         : int  NA NA NA 5 5 5 NA 5 NA 5 ...
##  $ FTFFRIEND        : int  5 NA 4 3 2 5 NA 2 NA 5 ...
##  $ FTFFRIEND_ET     : int  NA NA NA NA NA 180 NA NA NA NA ...
##  $ FTFFRIENDPL      : int  5 NA 5 5 4 5 NA 5 NA 5 ...
##  $ FTFCHURCH        : int  6 NA 2 1 2 1 NA 1 NA 6 ...
##  $ FTFCHURCH_ET     : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ FTFCHURCHPL      : int  NA NA 5 4 5 5 NA 5 NA NA ...
##  $ FTFMEETINGS      : int  4 NA 2 4 6 2 NA 6 NA 6 ...
##  $ FTFMEETINGS_ET   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ FTFMEETINSSPL    : int  3 NA 5 5 NA 5 NA NA NA NA ...
##  $ DAYSWORK         : int  6 NA 6 5 5 4 NA 4 NA 6 ...
##  $ FTFWORK          : int  NA NA NA 5 2 5 NA 5 NA NA ...
##  $ FTFWORK_ET       : int  NA NA NA 300 NA 240 NA 180 NA NA ...
##  $ FTFWORKPL        : int  NA NA NA 5 3 4 NA 5 NA NA ...
##  $ FTFSTRANGER      : int  2 NA 2 2 5 1 NA 1 NA 1 ...
##  $ FTFSTRANGER_ET   : int  NA NA NA NA 1800 NA NA NA NA NA ...
##  $ FTFSTRANGERPL    : int  4 NA 4 5 4 5 NA 5 NA 5 ...
##  $ FTFOTHER         : int  2 NA 1 2 2 2 NA 1 NA 2 ...
##  $ FTFOTHER_TXT     : chr  NA NA &quot;volunteering, choir, bells, quilting&quot; NA ...
##  $ FTFOTHERTIME     : int  NA NA 300 NA NA NA NA 1500 NA NA ...
##  $ FTFOTHERPL       : int  NA NA 5 NA NA NA NA 5 NA NA ...
##  $ VERBTEL          : int  2 NA 1 5 1 2 NA 2 NA 1 ...
##  $ VERBTEL_TXT      : int  NA NA NA 300 NA NA NA NA NA NA ...
##  $ VERBTELPL        : int  5 NA 5 4 4 5 NA 5 NA 4 ...
##  $ WRITCONVIM       : int  6 NA 1 2 1 2 NA 6 NA 1 ...
##  $ WRITCONVIM_TXT   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ WRITCONVIMPL     : int  NA NA 5 4 4 4 NA NA NA 5 ...
##  $ WRITCONVEMAIL    : int  3 NA 1 5 1 4 NA 1 NA 1 ...
##  $ WRITCONVEMAIL_TXT: int  NA NA NA 240 NA NA NA NA NA NA ...
##  $ WRITCONVEMAILPL  : int  3 NA 5 4 5 4 NA 4 NA 5 ...
##  $ WRITLETTERS      : int  1 NA 1 6 1 2 NA 1 NA 6 ...
##  $ WRITLETTERS_TXT  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ HELPRELPL        : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ xswkexer         : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ rateexer         : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ HardHrWk         : num  0 3 3 0 0.25 3 2.25 0 0.75 0.25 ...
##  $ ModHrWk          : num  3 0.5 0 0 0.25 1.5 0 0 0 5 ...
##  $ MildHrWk         : num  0 2 0 0.75 0.25 0.75 0 0.75 1 3 ...
##  $ WalkMet          : int  0 3 0 3 4 4 4 3 3 4 ...
##  $ WalkHrWk         : num  0 2.5 1.25 0.5 0.425 ...
##  $ MetHrWk          : num  12 36.5 21 3.75 5.2 ...</code></pre>
</details>
<p><br> <br></p>
</div>
<div id="headinjuries" class="section level2">
<h2>HeadInjuries</h2>
<p>Head Injuries</p>
<pre class="r"><code>## load data
HeadInjuries &lt;- read.csv(paste0(wrap_directory, &quot;HeadInjuries_05Jun2025.csv&quot;))
info(HeadInjuries,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:551, cols:5, inds:468</code></pre>
<pre class="r"><code>str(HeadInjuries)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    551 obs. of  5 variables:
##  $ wrapnum     : chr  &quot;wrap0324&quot; &quot;wrap0398&quot; &quot;wrapA0016&quot; &quot;wrapL0110&quot; ...
##  $ WRAPNo      : chr  &quot;0324&quot; &quot;0398&quot; &quot;A0016&quot; &quot;L0110&quot; ...
##  $ VisNo       : int  6 6 1 1 3 1 5 1 1 7 ...
##  $ numhi       : int  1 1 1 2 1 1 1 3 1 1 ...
##  $ update_stamp: chr  &quot;2024-09-13 08:52:54.0&quot; &quot;2024-09-13 08:52:54.0&quot; &quot;2024-09-13 08:52:54.0&quot; &quot;2024-09-13 08:52:54.0&quot; ...</code></pre>
</details>
<pre class="r"><code>## check table frequency for all variables

## drop update_stamp column
HeadInjuries &lt;- subset(HeadInjuries,select = -c(update_stamp)) ## 4 columns left</code></pre>
<pre class="r"><code>str(HeadInjuries)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    551 obs. of  4 variables:
##  $ wrapnum: chr  &quot;wrap0324&quot; &quot;wrap0398&quot; &quot;wrapA0016&quot; &quot;wrapL0110&quot; ...
##  $ WRAPNo : chr  &quot;0324&quot; &quot;0398&quot; &quot;A0016&quot; &quot;L0110&quot; ...
##  $ VisNo  : int  6 6 1 1 3 1 5 1 1 7 ...
##  $ numhi  : int  1 1 1 2 1 1 1 3 1 1 ...</code></pre>
</details>
<p><br> <br></p>
</div>
<div id="headinjurysub" class="section level2">
<h2>HeadInjurySub</h2>
<p>Head Injuries - Additional Data</p>
<pre class="r"><code>## load data
HeadInjurySub &lt;- read.csv(paste0(wrap_directory, &quot;HeadInjurySub_05Jun2025.csv&quot;))
info(HeadInjurySub,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:759, cols:20, inds:489</code></pre>
<pre class="r"><code>str(HeadInjurySub)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    759 obs. of  20 variables:
##  $ wrapnum     : chr  &quot;wrapA0016&quot; &quot;wrap0292&quot; &quot;wrap0119&quot; &quot;wrapM0027&quot; ...
##  $ WRAPNo      : chr  &quot;A0016&quot; &quot;0292&quot; &quot;0119&quot; &quot;M0027&quot; ...
##  $ VisNo       : int  1 6 1 1 2 1 1 1 6 2 ...
##  $ RowNumber   : int  1 2 1 1 1 1 1 1 1 1 ...
##  $ agehi       : int  6 75 14 4 37 37 13 60 67 30 ...
##  $ hicause     : int  2 2 2 2 1 1 2 1 2 1 ...
##  $ memhi       : int  1 1 1 1 1 2 1 2 1 1 ...
##  $ loc         : int  2 2 2 2 1 1 1 3 2 1 ...
##  $ locdur      : int  NA NA NA NA 1 7 2 2 NA 1 ...
##  $ awake       : int  NA 1 1 NA 1 1 1 1 1 1 ...
##  $ mdhi        : int  1 2 1 1 1 1 2 2 2 2 ...
##  $ scanhi      : int  1 2 1 2 2 1 2 2 2 2 ...
##  $ hosphi      : int  1 2 1 2 2 1 2 2 2 2 ...
##  $ othinj      : int  3 0 0 0 0 1 0 0 0 4 ...
##  $ treata      : int  1 5 5 NA NA 1 NA NA NA NA ...
##  $ treatb      : int  NA NA NA NA NA 2 NA NA NA NA ...
##  $ treatc      : int  NA NA NA NA NA 3 NA NA NA NA ...
##  $ cognhi      : int  2 2 2 2 2 1 2 2 1 2 ...
##  $ cognow      : int  NA 2 2 2 2 1 2 NA 2 2 ...
##  $ update_stamp: chr  &quot;2025-01-24 08:17:49.0&quot; &quot;2025-01-24 08:17:49.0&quot; &quot;2025-01-24 08:17:49.0&quot; &quot;2025-01-24 08:17:49.0&quot; ...</code></pre>
</details>
<pre class="r"><code>## check table frequency for all variables

## drop update_stamp column
HeadInjurySub &lt;- subset(HeadInjurySub,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(HeadInjurySub)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    759 obs. of  19 variables:
##  $ wrapnum  : chr  &quot;wrapA0016&quot; &quot;wrap0292&quot; &quot;wrap0119&quot; &quot;wrapM0027&quot; ...
##  $ WRAPNo   : chr  &quot;A0016&quot; &quot;0292&quot; &quot;0119&quot; &quot;M0027&quot; ...
##  $ VisNo    : int  1 6 1 1 2 1 1 1 6 2 ...
##  $ RowNumber: int  1 2 1 1 1 1 1 1 1 1 ...
##  $ agehi    : int  6 75 14 4 37 37 13 60 67 30 ...
##  $ hicause  : int  2 2 2 2 1 1 2 1 2 1 ...
##  $ memhi    : int  1 1 1 1 1 2 1 2 1 1 ...
##  $ loc      : int  2 2 2 2 1 1 1 3 2 1 ...
##  $ locdur   : int  NA NA NA NA 1 7 2 2 NA 1 ...
##  $ awake    : int  NA 1 1 NA 1 1 1 1 1 1 ...
##  $ mdhi     : int  1 2 1 1 1 1 2 2 2 2 ...
##  $ scanhi   : int  1 2 1 2 2 1 2 2 2 2 ...
##  $ hosphi   : int  1 2 1 2 2 1 2 2 2 2 ...
##  $ othinj   : int  3 0 0 0 0 1 0 0 0 4 ...
##  $ treata   : int  1 5 5 NA NA 1 NA NA NA NA ...
##  $ treatb   : int  NA NA NA NA NA 2 NA NA NA NA ...
##  $ treatc   : int  NA NA NA NA NA 3 NA NA NA NA ...
##  $ cognhi   : int  2 2 2 2 2 1 2 2 1 2 ...
##  $ cognow   : int  NA 2 2 2 2 1 2 NA 2 2 ...</code></pre>
</details>
<p><br> <br></p>
</div>
<div id="iadl" class="section level2">
<h2>IADL</h2>
<p>Lawton Instrumental Activities of Daily Living (IADL)</p>
<pre class="r"><code>## load data
IADL &lt;- read.csv(paste0(wrap_directory, &quot;IADL_05Jun2025.csv&quot;))
info(IADL,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:5913, cols:17, inds:1496</code></pre>
<pre class="r"><code>str(IADL)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    5913 obs. of  17 variables:
##  $ wrapnum                                    : chr  &quot;wrap0068&quot; &quot;wrap0146&quot; &quot;wrap0148&quot; &quot;wrap0305&quot; ...
##  $ WRAPNo                                     : chr  &quot;0068&quot; &quot;0146&quot; &quot;0148&quot; &quot;0305&quot; ...
##  $ VisNo                                      : int  2 4 5 5 4 6 4 7 4 8 ...
##  $ UsePhone                                   : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Walking                                    : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Shopping                                   : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Meals                                      : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ HouseWk                                    : int  2 1 2 2 2 2 2 2 2 2 ...
##  $ Laundry                                    : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ TakeMeds                                   : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Money                                      : int  2 1 2 2 2 2 2 2 2 2 ...
##  $ Total                                      : int  16 14 16 16 16 16 16 16 16 16 ...
##  $ Returned_Days_From_Visit                   : int  NA 26 12 113 10 28 11 105 18 48 ...
##  $ estimated_questionnaire_days_after_baseline: int  2259 3581 4765 4633 3285 5131 3867 6204 3373 6409 ...
##  $ RelPPT                                     : int  1 2 1 2 1 6 1 NA 1 4 ...
##  $ RelLive                                    : int  NA 1 1 NA 1 NA 1 NA 1 NA ...
##  $ update_stamp                               : chr  &quot;2024-09-13 08:52:56.0&quot; &quot;2024-09-13 08:52:56.0&quot; &quot;2024-09-13 08:52:56.0&quot; &quot;2024-09-13 08:52:56.0&quot; ...</code></pre>
</details>
<pre class="r"><code>## check table frequency for all variables

## DD has 1=yes,2=no coding but data has 0/1 coding. 
## Confirmed with other 0/1 variables that 0 is no and 1 stays yes, so will update the dataset to match the DD
IADL$RelLive[IADL$RelLive==0] &lt;- 2
IADL$RelLive &lt;- as.integer(IADL$RelLive)

## drop update_stamp column
IADL &lt;- subset(IADL,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(IADL)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    5913 obs. of  16 variables:
##  $ wrapnum                                    : chr  &quot;wrap0068&quot; &quot;wrap0146&quot; &quot;wrap0148&quot; &quot;wrap0305&quot; ...
##  $ WRAPNo                                     : chr  &quot;0068&quot; &quot;0146&quot; &quot;0148&quot; &quot;0305&quot; ...
##  $ VisNo                                      : int  2 4 5 5 4 6 4 7 4 8 ...
##  $ UsePhone                                   : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Walking                                    : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Shopping                                   : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Meals                                      : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ HouseWk                                    : int  2 1 2 2 2 2 2 2 2 2 ...
##  $ Laundry                                    : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ TakeMeds                                   : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Money                                      : int  2 1 2 2 2 2 2 2 2 2 ...
##  $ Total                                      : int  16 14 16 16 16 16 16 16 16 16 ...
##  $ Returned_Days_From_Visit                   : int  NA 26 12 113 10 28 11 105 18 48 ...
##  $ estimated_questionnaire_days_after_baseline: int  2259 3581 4765 4633 3285 5131 3867 6204 3373 6409 ...
##  $ RelPPT                                     : int  1 2 1 2 1 6 1 NA 1 4 ...
##  $ RelLive                                    : int  NA 1 1 NA 1 NA 1 NA 1 NA ...</code></pre>
</details>
<p><br> <br></p>
</div>
<div id="iqcode" class="section level2">
<h2>IQCode</h2>
<pre class="r"><code>## load data
IQCode &lt;- read.csv(paste0(wrap_directory, &quot;IQCode_05Jun2025.csv&quot;))
info(IQCode,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:5896, cols:25, inds:1497</code></pre>
<pre class="r"><code>str(IQCode)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    5896 obs. of  25 variables:
##  $ wrapnum                                    : chr  &quot;wrap0068&quot; &quot;wrap0146&quot; &quot;wrap0148&quot; &quot;wrap0305&quot; ...
##  $ WRAPNo                                     : chr  &quot;0068&quot; &quot;0146&quot; &quot;0148&quot; &quot;0305&quot; ...
##  $ VisNo                                      : int  2 4 5 5 4 6 4 7 4 8 ...
##  $ FamDates                                   : int  3 3 3 3 3 3 3 3 3 3 ...
##  $ Recent                                     : int  3 3 3 3 3 3 3 3 4 3 ...
##  $ Convers                                    : int  3 4 3 3 3 3 3 3 4 3 ...
##  $ Address                                    : int  3 3 3 3 3 3 3 3 3 3 ...
##  $ DayMo                                      : int  3 3 3 3 3 3 3 3 3 3 ...
##  $ Kept                                       : int  3 3 3 3 3 3 3 3 4 3 ...
##  $ Find                                       : int  3 3 3 3 3 3 3 3 4 3 ...
##  $ Machines                                   : int  3 3 3 3 3 3 3 3 3 3 ...
##  $ NewMach                                    : int  3 4 3 3 3 3 3 3 3 3 ...
##  $ NewThings                                  : int  3 3 3 3 3 3 3 3 3 3 ...
##  $ Story                                      : int  3 3 3 3 3 3 3 3 3 3 ...
##  $ Decisions                                  : int  3 3 3 3 3 3 3 3 3 3 ...
##  $ MonShop                                    : int  3 3 3 3 3 3 3 3 3 3 ...
##  $ Finance                                    : int  3 3 3 3 3 3 3 3 3 3 ...
##  $ MathProb                                   : int  3 3 3 3 3 3 3 3 3 3 ...
##  $ Intell                                     : int  3 3 3 3 3 3 3 3 3 3 ...
##  $ Total                                      : int  48 50 48 48 48 48 48 48 52 48 ...
##  $ Returned_Days_From_Visit                   : int  NA 26 12 113 10 28 11 105 18 48 ...
##  $ estimated_questionnaire_days_after_baseline: int  2259 3581 4765 4633 3285 5131 3867 6204 3373 6409 ...
##  $ RelPPT                                     : int  1 2 1 2 1 6 1 NA 1 4 ...
##  $ RelLive                                    : int  NA 1 1 NA 1 NA 1 NA 1 NA ...
##  $ update_stamp                               : chr  &quot;2024-09-13 08:52:57.0&quot; &quot;2024-09-13 08:52:57.0&quot; &quot;2024-09-13 08:52:57.0&quot; &quot;2024-09-13 08:52:57.0&quot; ...</code></pre>
</details>
<pre class="r"><code>## check table frequency for all variables

## DD has 1=yes,2=no coding but data has 0/1 coding. 
## Confirmed with other 0/1 variables that 0 is no and 1 stays yes, so will update the dataset to match the DD
IQCode$RelLive[IQCode$RelLive==0] &lt;- 2
IQCode$RelLive &lt;- as.integer(IQCode$RelLive)

## drop update_stamp column
IQCode &lt;- subset(IQCode,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(IQCode)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    5896 obs. of  24 variables:
##  $ wrapnum                                    : chr  &quot;wrap0068&quot; &quot;wrap0146&quot; &quot;wrap0148&quot; &quot;wrap0305&quot; ...
##  $ WRAPNo                                     : chr  &quot;0068&quot; &quot;0146&quot; &quot;0148&quot; &quot;0305&quot; ...
##  $ VisNo                                      : int  2 4 5 5 4 6 4 7 4 8 ...
##  $ FamDates                                   : int  3 3 3 3 3 3 3 3 3 3 ...
##  $ Recent                                     : int  3 3 3 3 3 3 3 3 4 3 ...
##  $ Convers                                    : int  3 4 3 3 3 3 3 3 4 3 ...
##  $ Address                                    : int  3 3 3 3 3 3 3 3 3 3 ...
##  $ DayMo                                      : int  3 3 3 3 3 3 3 3 3 3 ...
##  $ Kept                                       : int  3 3 3 3 3 3 3 3 4 3 ...
##  $ Find                                       : int  3 3 3 3 3 3 3 3 4 3 ...
##  $ Machines                                   : int  3 3 3 3 3 3 3 3 3 3 ...
##  $ NewMach                                    : int  3 4 3 3 3 3 3 3 3 3 ...
##  $ NewThings                                  : int  3 3 3 3 3 3 3 3 3 3 ...
##  $ Story                                      : int  3 3 3 3 3 3 3 3 3 3 ...
##  $ Decisions                                  : int  3 3 3 3 3 3 3 3 3 3 ...
##  $ MonShop                                    : int  3 3 3 3 3 3 3 3 3 3 ...
##  $ Finance                                    : int  3 3 3 3 3 3 3 3 3 3 ...
##  $ MathProb                                   : int  3 3 3 3 3 3 3 3 3 3 ...
##  $ Intell                                     : int  3 3 3 3 3 3 3 3 3 3 ...
##  $ Total                                      : int  48 50 48 48 48 48 48 48 52 48 ...
##  $ Returned_Days_From_Visit                   : int  NA 26 12 113 10 28 11 105 18 48 ...
##  $ estimated_questionnaire_days_after_baseline: int  2259 3581 4765 4633 3285 5131 3867 6204 3373 6409 ...
##  $ RelPPT                                     : int  1 2 1 2 1 6 1 NA 1 4 ...
##  $ RelLive                                    : int  NA 1 1 NA 1 NA 1 NA 1 NA ...</code></pre>
</details>
<p><br> <br></p>
</div>
<div id="learningdifficulties" class="section level2">
<h2>LearningDifficulties</h2>
<p>Learning Difficulties interview</p>
<pre class="r"><code>## load data
LearningDifficulties &lt;- read.csv(paste0(wrap_directory, &quot;LearningDifficulties_05Jun2025.csv&quot;))
info(LearningDifficulties,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:213, cols:27, inds:204</code></pre>
<pre class="r"><code>str(LearningDifficulties)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    213 obs. of  27 variables:
##  $ wrapnum         : chr  &quot;wrap0733&quot; &quot;wrapL0005&quot; &quot;wrap0191&quot; &quot;wrap0505&quot; ...
##  $ WRAPNo          : chr  &quot;0733&quot; &quot;L0005&quot; &quot;0191&quot; &quot;0505&quot; ...
##  $ VisNo           : int  1 2 1 4 2 1 2 1 2 1 ...
##  $ catdiff1        : int  1 1 1 2 2 1 3 3 1 3 ...
##  $ catdiff2        : int  NA NA NA NA NA 3 NA NA NA NA ...
##  $ evalp           : int  2 1 1 1 1 1 2 1 1 2 ...
##  $ subdiff1        : int  1 1 1 4 4 1 4 6 1 4 ...
##  $ subdiff2        : int  NA 3 4 NA NA 3 5 NA NA NA ...
##  $ subdiff3        : int  NA NA NA NA NA 4 NA NA NA NA ...
##  $ subdiff4        : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ subdiff5        : logi  NA NA NA NA NA NA ...
##  $ lowgrad         : int  2 2 2 1 1 1 1 2 3 2 ...
##  $ latrpt          : int  2 1 2 2 1 2 2 2 1 2 ...
##  $ illinj          : int  2 2 2 1 1 3 3 2 1 2 ...
##  $ specillinj      : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;hit on head with baseball bat&quot; ...
##  $ wrkhrd          : int  1 1 1 2 1 1 1 1 1 NA ...
##  $ workhrd_than_who: chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ sphelpyn1       : int  0 0 4 0 1 1 2 4 3 0 ...
##  $ sphelpyn2       : int  NA NA NA NA NA 3 NA NA NA NA ...
##  $ sphelpyn3       : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ relld1          : int  3 2 2 6 1 0 2 2 0 NA ...
##  $ relld2          : int  NA NA 1 NA 5 NA 4 3 NA NA ...
##  $ relld3          : int  NA NA 3 NA 4 NA 3 NA NA NA ...
##  $ relld4          : int  NA NA 3 NA NA NA NA NA NA NA ...
##  $ relld5          : int  NA NA 0 NA NA NA NA NA NA NA ...
##  $ ovecomp         : int  1 1 1 1 1 1 1 1 1 3 ...
##  $ update_stamp    : chr  &quot;2024-09-13 08:52:58.0&quot; &quot;2024-09-13 08:52:58.0&quot; &quot;2024-09-13 08:52:58.0&quot; &quot;2024-09-13 08:52:58.0&quot; ...</code></pre>
</details>
<pre class="r"><code>## check table frequency for all variables

## convert logi columns to int
## these columns were confirmed to only have NA values in the raw data
logicols &lt;- colnames(LearningDifficulties)[sapply(LearningDifficulties, is.logical)]
LearningDifficulties[logicols] &lt;- lapply(LearningDifficulties[logicols], as.integer)

## convert &quot;&quot; to NA for chr columns
chrcols &lt;- colnames(LearningDifficulties)[sapply(LearningDifficulties, is.character)]
LearningDifficulties[chrcols] &lt;- lapply(LearningDifficulties[chrcols], function(x) ifelse(x == &quot;&quot;, NA, x))

## drop update_stamp column
LearningDifficulties &lt;- subset(LearningDifficulties,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(LearningDifficulties)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    213 obs. of  26 variables:
##  $ wrapnum         : chr  &quot;wrap0733&quot; &quot;wrapL0005&quot; &quot;wrap0191&quot; &quot;wrap0505&quot; ...
##  $ WRAPNo          : chr  &quot;0733&quot; &quot;L0005&quot; &quot;0191&quot; &quot;0505&quot; ...
##  $ VisNo           : int  1 2 1 4 2 1 2 1 2 1 ...
##  $ catdiff1        : int  1 1 1 2 2 1 3 3 1 3 ...
##  $ catdiff2        : int  NA NA NA NA NA 3 NA NA NA NA ...
##  $ evalp           : int  2 1 1 1 1 1 2 1 1 2 ...
##  $ subdiff1        : int  1 1 1 4 4 1 4 6 1 4 ...
##  $ subdiff2        : int  NA 3 4 NA NA 3 5 NA NA NA ...
##  $ subdiff3        : int  NA NA NA NA NA 4 NA NA NA NA ...
##  $ subdiff4        : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ subdiff5        : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ lowgrad         : int  2 2 2 1 1 1 1 2 3 2 ...
##  $ latrpt          : int  2 1 2 2 1 2 2 2 1 2 ...
##  $ illinj          : int  2 2 2 1 1 3 3 2 1 2 ...
##  $ specillinj      : chr  NA NA NA &quot;hit on head with baseball bat&quot; ...
##  $ wrkhrd          : int  1 1 1 2 1 1 1 1 1 NA ...
##  $ workhrd_than_who: chr  NA NA NA NA ...
##  $ sphelpyn1       : int  0 0 4 0 1 1 2 4 3 0 ...
##  $ sphelpyn2       : int  NA NA NA NA NA 3 NA NA NA NA ...
##  $ sphelpyn3       : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ relld1          : int  3 2 2 6 1 0 2 2 0 NA ...
##  $ relld2          : int  NA NA 1 NA 5 NA 4 3 NA NA ...
##  $ relld3          : int  NA NA 3 NA 4 NA 3 NA NA NA ...
##  $ relld4          : int  NA NA 3 NA NA NA NA NA NA NA ...
##  $ relld5          : int  NA NA 0 NA NA NA NA NA NA NA ...
##  $ ovecomp         : int  1 1 1 1 1 1 1 1 1 3 ...</code></pre>
</details>
<p><br> <br></p>
</div>
<div id="libra" class="section level2">
<h2>LIBRA</h2>
<p>Lifestyle for BRAin health</p>
<pre class="r"><code>## load data
LIBRA &lt;- read.csv(paste0(wrap_directory, &quot;LIBRA_05Jun2025.csv&quot;))
info(LIBRA,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:8173, cols:18, inds:1818</code></pre>
<pre class="r"><code>str(LIBRA)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    8173 obs. of  18 variables:
##  $ wrapnum                 : chr  &quot;wrap0001&quot; &quot;wrap0001&quot; &quot;wrap0001&quot; &quot;wrap0001&quot; ...
##  $ WRAPNo                  : chr  &quot;0001&quot; &quot;0001&quot; &quot;0001&quot; &quot;0001&quot; ...
##  $ VisNo                   : int  1 2 3 4 5 1 2 3 4 5 ...
##  $ Smoker_LIBRA            : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ Combined_Alc            : int  NA 0 0 0 0 0 0 0 1 0 ...
##  $ PhysicallyInactive_LIBRA: int  1 1 1 1 1 0 1 1 1 0 ...
##  $ Cog_LIBRA               : int  NA 0 0 1 0 NA 0 1 0 0 ...
##  $ Combined_Depression     : int  1 0 0 0 0 0 0 0 0 0 ...
##  $ Combined_CAD            : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ Combined_Renal          : int  0 0 1 0 0 0 0 0 0 0 ...
##  $ Combined_Diabetes       : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ Combined_Hypertension   : int  1 1 1 1 0 0 1 0 0 0 ...
##  $ Combined_HighCholes     : int  1 0 0 0 0 1 0 0 0 0 ...
##  $ Obesity_LIBRA           : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ LIBRA_Score             : num  7.8 0.1 1.2 3.3 -1.5 2 0.1 1.7 -1.5 -2.6 ...
##  $ LIBRA_tertile           : chr  &quot;High&quot; &quot;Moderate&quot; &quot;Moderate&quot; &quot;High&quot; ...
##  $ N_Missing               : int  2 0 0 0 0 1 0 0 0 0 ...
##  $ update_stamp            : chr  &quot;2025-01-24 08:17:55.0&quot; &quot;2025-01-24 08:17:55.0&quot; &quot;2025-01-24 08:17:55.0&quot; &quot;2025-01-24 08:17:55.0&quot; ...</code></pre>
</details>
<pre class="r"><code>## check table frequency for all variables

## drop update_stamp column
LIBRA &lt;- subset(LIBRA,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(LIBRA)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    8173 obs. of  17 variables:
##  $ wrapnum                 : chr  &quot;wrap0001&quot; &quot;wrap0001&quot; &quot;wrap0001&quot; &quot;wrap0001&quot; ...
##  $ WRAPNo                  : chr  &quot;0001&quot; &quot;0001&quot; &quot;0001&quot; &quot;0001&quot; ...
##  $ VisNo                   : int  1 2 3 4 5 1 2 3 4 5 ...
##  $ Smoker_LIBRA            : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ Combined_Alc            : int  NA 0 0 0 0 0 0 0 1 0 ...
##  $ PhysicallyInactive_LIBRA: int  1 1 1 1 1 0 1 1 1 0 ...
##  $ Cog_LIBRA               : int  NA 0 0 1 0 NA 0 1 0 0 ...
##  $ Combined_Depression     : int  1 0 0 0 0 0 0 0 0 0 ...
##  $ Combined_CAD            : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ Combined_Renal          : int  0 0 1 0 0 0 0 0 0 0 ...
##  $ Combined_Diabetes       : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ Combined_Hypertension   : int  1 1 1 1 0 0 1 0 0 0 ...
##  $ Combined_HighCholes     : int  1 0 0 0 0 1 0 0 0 0 ...
##  $ Obesity_LIBRA           : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ LIBRA_Score             : num  7.8 0.1 1.2 3.3 -1.5 2 0.1 1.7 -1.5 -2.6 ...
##  $ LIBRA_tertile           : chr  &quot;High&quot; &quot;Moderate&quot; &quot;Moderate&quot; &quot;High&quot; ...
##  $ N_Missing               : int  2 0 0 0 0 1 0 0 0 0 ...</code></pre>
</details>
<p><br> <br></p>
</div>
<div id="med_lab" class="section level2">
<h2>Med_Lab</h2>
<p>Med/Lab Scores</p>
<pre class="r"><code>## load data
Med_Lab &lt;- read.csv(paste0(wrap_directory, &quot;Med_Lab_05Jun2025.csv&quot;))
info(Med_Lab,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:8094, cols:31, inds:1797</code></pre>
<pre class="r"><code>str(Med_Lab)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    8094 obs. of  31 variables:
##  $ wrapnum               : chr  &quot;wrapL0308&quot; &quot;wrap0401&quot; &quot;wrapL0005&quot; &quot;wrap0434&quot; ...
##  $ WRAPNo                : chr  &quot;L0308&quot; &quot;0401&quot; &quot;L0005&quot; &quot;0434&quot; ...
##  $ VisNo                 : int  2 6 5 7 1 5 1 5 2 2 ...
##  $ nocrntpd              : int  0 0 0 0 0 0 NA 0 0 -1 ...
##  $ nonpd                 : int  -1 0 0 -1 0 0 NA 0 -1 -1 ...
##  $ ma                    : int  1 2 2 3 2 1 NA 1 2 1 ...
##  $ specma                : chr  &quot;sulfa&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ antipsychotic_reaction: int  NA NA NA NA NA NA NA NA NA NA ...
##  $ creat1                : num  0.87 NA NA NA NA ...
##  $ eGFR_Creatinine       : int  80 NA NA NA NA NA 72 NA 101 98 ...
##  $ choles1               : num  264 157 248 232 254 168 228 NA 129 231 ...
##  $ triglycerides         : num  NA 86 62 199 96 254 121 NA NA NA ...
##  $ LDL_Calc              : num  NA 71 151 114 163 71 110 NA NA NA ...
##  $ Non_HDL_Chol          : num  NA 88 163 154 182 122 134 NA NA NA ...
##  $ insulin               : num  28.7 6 18.4 12 12 10 9 NA 3.1 6 ...
##  $ HDL                   : int  58 69 85 78 72 46 94 NA 68 67 ...
##  $ CReProt               : chr  &quot;0.9&quot; &quot;0.4&quot; &quot;1.4&quot; &quot;1.6&quot; ...
##  $ B12                   : int  398 399 409 664 707 634 258 NA 401 373 ...
##  $ glucose               : int  103 92 82 102 95 95 90 NA 89 98 ...
##  $ VitaD2                : chr  &quot;&quot; &quot;&lt;5&quot; &quot;&lt;4.0&quot; &quot;&quot; ...
##  $ VitaD3                : chr  &quot;&quot; &quot;66&quot; &quot;52&quot; &quot;&quot; ...
##  $ TotVitamin            : int  NA 66 52 39 34 23 8 NA NA NA ...
##  $ autopsy               : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ AutopsyRet            : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ BrainBank             : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ folica1               : chr  &quot;13&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ homocy1               : num  8 NA NA NA NA NA NA NA 9 12 ...
##  $ IL6                   : chr  &quot;0.73&quot; &quot;1.342&quot; &quot;1.07&quot; &quot;&quot; ...
##  $ AllBott               : int  NA -1 NA NA -1 NA NA NA NA NA ...
##  $ SomeBott              : int  -1 NA -1 NA NA -1 NA NA NA NA ...
##  $ update_stamp          : chr  &quot;2025-01-24 08:18:05.0&quot; &quot;2025-01-24 08:18:05.0&quot; &quot;2025-01-24 08:18:05.0&quot; &quot;2025-01-24 08:18:05.0&quot; ...</code></pre>
</details>
<pre class="r"><code>## check table frequency for all variables

## DD has 1=Yes;0=No;3=Unknown coding but data has 0/1/2 coding. 
## I will convert 2 to 3 in the dataset
Med_Lab$antipsychotic_reaction[Med_Lab$antipsychotic_reaction==2] &lt;- 3
Med_Lab$antipsychotic_reaction &lt;- as.integer(Med_Lab$antipsychotic_reaction)

## convert &quot;&quot; to NA for chr columns
chrcols &lt;- colnames(Med_Lab)[sapply(Med_Lab, is.character)]
Med_Lab[chrcols] &lt;- lapply(Med_Lab[chrcols], function(x) ifelse(x == &quot;&quot;, NA, x))

## drop update_stamp column
Med_Lab &lt;- subset(Med_Lab,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(Med_Lab)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    8094 obs. of  30 variables:
##  $ wrapnum               : chr  &quot;wrapL0308&quot; &quot;wrap0401&quot; &quot;wrapL0005&quot; &quot;wrap0434&quot; ...
##  $ WRAPNo                : chr  &quot;L0308&quot; &quot;0401&quot; &quot;L0005&quot; &quot;0434&quot; ...
##  $ VisNo                 : int  2 6 5 7 1 5 1 5 2 2 ...
##  $ nocrntpd              : int  0 0 0 0 0 0 NA 0 0 -1 ...
##  $ nonpd                 : int  -1 0 0 -1 0 0 NA 0 -1 -1 ...
##  $ ma                    : int  1 2 2 3 2 1 NA 1 2 1 ...
##  $ specma                : chr  &quot;sulfa&quot; NA NA NA ...
##  $ antipsychotic_reaction: int  NA NA NA NA NA NA NA NA NA NA ...
##  $ creat1                : num  0.87 NA NA NA NA ...
##  $ eGFR_Creatinine       : int  80 NA NA NA NA NA 72 NA 101 98 ...
##  $ choles1               : num  264 157 248 232 254 168 228 NA 129 231 ...
##  $ triglycerides         : num  NA 86 62 199 96 254 121 NA NA NA ...
##  $ LDL_Calc              : num  NA 71 151 114 163 71 110 NA NA NA ...
##  $ Non_HDL_Chol          : num  NA 88 163 154 182 122 134 NA NA NA ...
##  $ insulin               : num  28.7 6 18.4 12 12 10 9 NA 3.1 6 ...
##  $ HDL                   : int  58 69 85 78 72 46 94 NA 68 67 ...
##  $ CReProt               : chr  &quot;0.9&quot; &quot;0.4&quot; &quot;1.4&quot; &quot;1.6&quot; ...
##  $ B12                   : int  398 399 409 664 707 634 258 NA 401 373 ...
##  $ glucose               : int  103 92 82 102 95 95 90 NA 89 98 ...
##  $ VitaD2                : chr  NA &quot;&lt;5&quot; &quot;&lt;4.0&quot; NA ...
##  $ VitaD3                : chr  NA &quot;66&quot; &quot;52&quot; NA ...
##  $ TotVitamin            : int  NA 66 52 39 34 23 8 NA NA NA ...
##  $ autopsy               : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ AutopsyRet            : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ BrainBank             : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ folica1               : chr  &quot;13&quot; NA NA NA ...
##  $ homocy1               : num  8 NA NA NA NA NA NA NA 9 12 ...
##  $ IL6                   : chr  &quot;0.73&quot; &quot;1.342&quot; &quot;1.07&quot; NA ...
##  $ AllBott               : int  NA -1 NA NA -1 NA NA NA NA NA ...
##  $ SomeBott              : int  -1 NA -1 NA NA -1 NA NA NA NA ...</code></pre>
</details>
<p><br> <br></p>
</div>
<div id="med_lab_npdc" class="section level2">
<h2>Med_Lab_NPDC</h2>
<p>Med/Lab Non-Prescription Drug Therapy</p>
<pre class="r"><code>## load data
Med_Lab_NPDC &lt;- read.csv(paste0(wrap_directory, &quot;Med_Lab_NPDC_05Jun2025.csv&quot;))
info(Med_Lab_NPDC,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:30019, cols:10, inds:1631</code></pre>
<pre class="r"><code>str(Med_Lab_NPDC)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    30019 obs. of  10 variables:
##  $ wrapnum        : chr  &quot;wrap0002&quot; &quot;wrap0002&quot; &quot;wrap0002&quot; &quot;wrap0002&quot; ...
##  $ WRAPNo         : chr  &quot;0002&quot; &quot;0002&quot; &quot;0002&quot; &quot;0002&quot; ...
##  $ VisNo          : int  5 5 5 5 5 2 2 2 2 2 ...
##  $ npdc_repeat_key: int  5 4 3 2 1 7 6 5 4 3 ...
##  $ npdcd          : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ mg             : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ npdcf          : int  2 1 2 2 2 7 7 7 7 7 ...
##  $ npdcname       : chr  &quot;Zuo Gui Wan&quot; &quot;Leaf Herbals Slim Waist&quot; &quot;Leaf Herbals Knee Pain Relief&quot; &quot;Leaf Herbals Circulation Enhancer&quot; ...
##  $ npdcclass      : chr  &quot;Herbal&quot; &quot;Herbal&quot; &quot;Herbal&quot; &quot;Herbal&quot; ...
##  $ update_stamp   : chr  &quot;2024-09-13 08:53:25.0&quot; &quot;2024-09-13 08:53:25.0&quot; &quot;2024-09-13 08:53:25.0&quot; &quot;2024-09-13 08:53:25.0&quot; ...</code></pre>
</details>
<pre class="r"><code>## check table frequency for all variables

## convert &quot;&quot; to NA for chr columns
chrcols &lt;- colnames(Med_Lab_NPDC)[sapply(Med_Lab_NPDC, is.character)]
Med_Lab_NPDC[chrcols] &lt;- lapply(Med_Lab_NPDC[chrcols], function(x) ifelse(x == &quot;&quot;, NA, x))

## drop update_stamp column
Med_Lab_NPDC &lt;- subset(Med_Lab_NPDC,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(Med_Lab_NPDC)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    30019 obs. of  9 variables:
##  $ wrapnum        : chr  &quot;wrap0002&quot; &quot;wrap0002&quot; &quot;wrap0002&quot; &quot;wrap0002&quot; ...
##  $ WRAPNo         : chr  &quot;0002&quot; &quot;0002&quot; &quot;0002&quot; &quot;0002&quot; ...
##  $ VisNo          : int  5 5 5 5 5 2 2 2 2 2 ...
##  $ npdc_repeat_key: int  5 4 3 2 1 7 6 5 4 3 ...
##  $ npdcd          : chr  NA NA NA NA ...
##  $ mg             : chr  NA NA NA NA ...
##  $ npdcf          : int  2 1 2 2 2 7 7 7 7 7 ...
##  $ npdcname       : chr  &quot;Zuo Gui Wan&quot; &quot;Leaf Herbals Slim Waist&quot; &quot;Leaf Herbals Knee Pain Relief&quot; &quot;Leaf Herbals Circulation Enhancer&quot; ...
##  $ npdcclass      : chr  &quot;Herbal&quot; &quot;Herbal&quot; &quot;Herbal&quot; &quot;Herbal&quot; ...</code></pre>
</details>
<p><br> <br></p>
</div>
<div id="med_lab_pdc" class="section level2">
<h2>Med_Lab_PDC</h2>
<p>Med/Lab Prescription Drug Therapy</p>
<pre class="r"><code>## load data
Med_Lab_PDC &lt;- read.csv(paste0(wrap_directory, &quot;Med_Lab_PDC_05Jun2025.csv&quot;))
info(Med_Lab_PDC,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:24379, cols:10, inds:1577</code></pre>
<pre class="r"><code>str(Med_Lab_PDC)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    24379 obs. of  10 variables:
##  $ wrapnum       : chr  &quot;wrap0002&quot; &quot;wrap0148&quot; &quot;wrap0148&quot; &quot;wrap0148&quot; ...
##  $ WRAPNo        : chr  &quot;0002&quot; &quot;0148&quot; &quot;0148&quot; &quot;0148&quot; ...
##  $ VisNo         : int  5 5 5 5 5 5 5 6 6 6 ...
##  $ pdc_repeat_key: int  1 3 2 1 3 2 1 5 4 3 ...
##  $ pdcd          : chr  &quot;20&quot; &quot;180&quot; &quot;10&quot; &quot;5&quot; ...
##  $ mg            : chr  &quot;mg&quot; &quot;mg&quot; &quot;mg&quot; &quot;mg&quot; ...
##  $ pdcf          : int  1 1 1 7 7 7 7 6 1 1 ...
##  $ pdcname       : chr  &quot;Simvastatin&quot; &quot;Cartia&quot; &quot;Atorvastatin&quot; &quot;Warfarin&quot; ...
##  $ pdcclass      : chr  &quot;Cholesterol Lowering - Statin (BBB)&quot; &quot;CCB (BBB)&quot; &quot;Cholesterol Lowering - Statin (No BBB)&quot; &quot;Anticoag/Thrombolytics&quot; ...
##  $ update_stamp  : chr  &quot;2024-09-13 08:53:28.0&quot; &quot;2024-09-13 08:53:28.0&quot; &quot;2024-09-13 08:53:28.0&quot; &quot;2024-09-13 08:53:28.0&quot; ...</code></pre>
</details>
<pre class="r"><code>## check table frequency for all variables

## convert &quot;&quot; to NA for chr columns
chrcols &lt;- colnames(Med_Lab_PDC)[sapply(Med_Lab_PDC, is.character)]
Med_Lab_PDC[chrcols] &lt;- lapply(Med_Lab_PDC[chrcols], function(x) ifelse(x == &quot;&quot;, NA, x))

## drop update_stamp column
Med_Lab_PDC &lt;- subset(Med_Lab_PDC,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(Med_Lab_PDC)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    24379 obs. of  9 variables:
##  $ wrapnum       : chr  &quot;wrap0002&quot; &quot;wrap0148&quot; &quot;wrap0148&quot; &quot;wrap0148&quot; ...
##  $ WRAPNo        : chr  &quot;0002&quot; &quot;0148&quot; &quot;0148&quot; &quot;0148&quot; ...
##  $ VisNo         : int  5 5 5 5 5 5 5 6 6 6 ...
##  $ pdc_repeat_key: int  1 3 2 1 3 2 1 5 4 3 ...
##  $ pdcd          : chr  &quot;20&quot; &quot;180&quot; &quot;10&quot; &quot;5&quot; ...
##  $ mg            : chr  &quot;mg&quot; &quot;mg&quot; &quot;mg&quot; &quot;mg&quot; ...
##  $ pdcf          : int  1 1 1 7 7 7 7 6 1 1 ...
##  $ pdcname       : chr  &quot;Simvastatin&quot; &quot;Cartia&quot; &quot;Atorvastatin&quot; &quot;Warfarin&quot; ...
##  $ pdcclass      : chr  &quot;Cholesterol Lowering - Statin (BBB)&quot; &quot;CCB (BBB)&quot; &quot;Cholesterol Lowering - Statin (No BBB)&quot; &quot;Anticoag/Thrombolytics&quot; ...</code></pre>
</details>
<p><br> <br></p>
</div>
<div id="medhistory" class="section level2">
<h2>MedHistory</h2>
<p>Medical History</p>
<pre class="r"><code>## load data
MedHistory &lt;- read.csv(paste0(wrap_directory, &quot;MedHistory_05Jun2025.csv&quot;))
info(MedHistory,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:8205, cols:71, inds:1810</code></pre>
<pre class="r"><code>str(MedHistory)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    8205 obs. of  71 variables:
##  $ wrapnum             : chr  &quot;wrapL0308&quot; &quot;wrap0401&quot; &quot;wrapL0005&quot; &quot;wrap0434&quot; ...
##  $ WRAPNo              : chr  &quot;L0308&quot; &quot;0401&quot; &quot;L0005&quot; &quot;0434&quot; ...
##  $ VisNo               : int  2 6 5 7 1 5 1 5 2 2 ...
##  $ MedCare             : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ LastVis             : int  2 1 1 1 3 2 1 2 2 3 ...
##  $ AdmitS              : int  2 2 2 2 1 2 1 2 2 2 ...
##  $ SurgBF              : int  2 NA NA NA 1 NA 1 NA 1 2 ...
##  $ SurgS               : int  1 1 2 1 NA 2 NA 2 2 2 ...
##  $ HrtDis_E            : int  2 2 2 3 2 2 2 2 2 2 ...
##  $ HrtAtt_E            : int  2 2 2 2 2 2 2 2 1 2 ...
##  $ HrtFail_E           : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Chest_E             : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ IrrHB_E             : int  2 2 2 1 1 1 1 2 2 2 ...
##  $ CorBypass_E         : int  2 2 2 2 2 2 NA 2 2 2 ...
##  $ OthHrt_E            : int  2 2 2 1 2 2 NA 2 2 2 ...
##  $ Hyperten_E          : int  1 2 2 1 2 1 1 1 2 2 ...
##  $ Orthostatic_Hyperten: int  NA NA NA NA NA NA 2 NA NA NA ...
##  $ Syncope             : int  NA NA NA NA NA NA 2 NA NA NA ...
##  $ Presyncope          : int  NA NA NA NA NA NA 2 NA NA NA ...
##  $ Diab_E              : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ HighChol_E          : int  2 1 1 2 2 1 2 1 1 2 ...
##  $ Dep_E               : int  1 1 2 2 2 2 2 2 2 1 ...
##  $ Anx_E               : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Schiz_E             : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Bip_E               : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ OthMHC_E            : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ HeadInj_E           : int  2 2 2 2 2 2 2 1 2 2 ...
##  $ HeadInj_Age         : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ Unc_E               : int  2 2 2 2 2 2 2 1 2 2 ...
##  $ Unc_Age             : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ Stroke_E            : int  2 2 2 2 2 2 NA 2 2 2 ...
##  $ TIA_E               : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Park_E              : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ ParkTh_E            : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ Parkinsonism        : int  NA NA NA NA NA NA 2 NA NA NA ...
##  $ MS_E                : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Epi_E               : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Menin_E             : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Migr_E              : int  2 2 2 2 2 2 2 1 2 2 ...
##  $ OthND_E             : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Neuro_E             : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Neuro_Age           : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ Sleep_E             : int  1 1 2 2 2 2 2 2 1 2 ...
##  $ Apnea_E             : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Lung_E              : int  2 2 2 2 2 2 2 1 2 2 ...
##  $ ChronBron_E         : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Emph_E              : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Asthma_E            : int  2 2 2 2 1 2 1 3 2 2 ...
##  $ OthLung_E           : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Arth_E              : int  1 1 1 1 1 2 1 1 2 2 ...
##  $ ThyDis_E            : int  2 2 1 2 2 2 1 2 2 2 ...
##  $ Cancer_E            : int  1 1 2 1 2 2 2 2 2 2 ...
##  $ LivDis_E            : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ AIDS_E              : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ KidDis_E            : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Vision_E            : int  2 1 2 2 2 2 1 2 2 2 ...
##  $ HearLoss_E          : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ OthMajor_E          : int  1 1 2 2 2 2 2 1 2 2 ...
##  $ OthCogTest          : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ PsychDis_E          : int  NA 2 2 2 2 2 2 2 NA NA ...
##  $ StrokeCons_E        : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ StrokeSpells_E      : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ StrokeParal_E       : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ StrokeVision_E      : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ StrokeLang_E        : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ StrokeWeak_E        : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ StrokeSense_E       : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ TumorOrCancer_E     : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ hosp5yr             : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ seehear             : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ update_stamp        : chr  &quot;2024-09-13 08:53:10.0&quot; &quot;2024-09-13 08:53:10.0&quot; &quot;2024-09-13 08:53:10.0&quot; &quot;2024-09-13 08:53:10.0&quot; ...</code></pre>
</details>
<pre class="r"><code>## check table frequency for all variables

## values in columns [AdmitS,SurgS - HeadInj_E, Unc_E, Stroke_E - Neuro_E, Sleep_E - TumorOrCancer_E, seehear] should be in {1,2,3}
cols = c(&quot;AdmitS&quot;,
         names(MedHistory)[which(names(MedHistory) == &quot;SurgS&quot;) : which(names(MedHistory) == &quot;HeadInj_E&quot;)],
         &quot;Unc_E&quot;,
         names(MedHistory)[which(names(MedHistory) == &quot;Stroke_E&quot;) : which(names(MedHistory) == &quot;Neuro_E&quot;)],
         names(MedHistory)[which(names(MedHistory) == &quot;Sleep_E&quot;) : which(names(MedHistory) == &quot;TumorOrCancer_E&quot;)],
         &quot;seehear&quot;
        )

for (col in cols) {
  vals &lt;- unique(MedHistory[[col]])
  invalid_vals &lt;- vals[!vals %in% c(1,2,3, NA)]
  
  if (length(invalid_vals) &gt; 0) {
    cat(&quot;Column:&quot;, col, &quot;- Invalid values:&quot;, paste(invalid_vals, collapse = &quot;, &quot;), &quot;\n&quot;)
  }
}

## drop update_stamp column
MedHistory &lt;- subset(MedHistory,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(MedHistory)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    8205 obs. of  70 variables:
##  $ wrapnum             : chr  &quot;wrapL0308&quot; &quot;wrap0401&quot; &quot;wrapL0005&quot; &quot;wrap0434&quot; ...
##  $ WRAPNo              : chr  &quot;L0308&quot; &quot;0401&quot; &quot;L0005&quot; &quot;0434&quot; ...
##  $ VisNo               : int  2 6 5 7 1 5 1 5 2 2 ...
##  $ MedCare             : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ LastVis             : int  2 1 1 1 3 2 1 2 2 3 ...
##  $ AdmitS              : int  2 2 2 2 1 2 1 2 2 2 ...
##  $ SurgBF              : int  2 NA NA NA 1 NA 1 NA 1 2 ...
##  $ SurgS               : int  1 1 2 1 NA 2 NA 2 2 2 ...
##  $ HrtDis_E            : int  2 2 2 3 2 2 2 2 2 2 ...
##  $ HrtAtt_E            : int  2 2 2 2 2 2 2 2 1 2 ...
##  $ HrtFail_E           : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Chest_E             : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ IrrHB_E             : int  2 2 2 1 1 1 1 2 2 2 ...
##  $ CorBypass_E         : int  2 2 2 2 2 2 NA 2 2 2 ...
##  $ OthHrt_E            : int  2 2 2 1 2 2 NA 2 2 2 ...
##  $ Hyperten_E          : int  1 2 2 1 2 1 1 1 2 2 ...
##  $ Orthostatic_Hyperten: int  NA NA NA NA NA NA 2 NA NA NA ...
##  $ Syncope             : int  NA NA NA NA NA NA 2 NA NA NA ...
##  $ Presyncope          : int  NA NA NA NA NA NA 2 NA NA NA ...
##  $ Diab_E              : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ HighChol_E          : int  2 1 1 2 2 1 2 1 1 2 ...
##  $ Dep_E               : int  1 1 2 2 2 2 2 2 2 1 ...
##  $ Anx_E               : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Schiz_E             : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Bip_E               : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ OthMHC_E            : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ HeadInj_E           : int  2 2 2 2 2 2 2 1 2 2 ...
##  $ HeadInj_Age         : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ Unc_E               : int  2 2 2 2 2 2 2 1 2 2 ...
##  $ Unc_Age             : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ Stroke_E            : int  2 2 2 2 2 2 NA 2 2 2 ...
##  $ TIA_E               : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Park_E              : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ ParkTh_E            : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ Parkinsonism        : int  NA NA NA NA NA NA 2 NA NA NA ...
##  $ MS_E                : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Epi_E               : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Menin_E             : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Migr_E              : int  2 2 2 2 2 2 2 1 2 2 ...
##  $ OthND_E             : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Neuro_E             : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Neuro_Age           : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ Sleep_E             : int  1 1 2 2 2 2 2 2 1 2 ...
##  $ Apnea_E             : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Lung_E              : int  2 2 2 2 2 2 2 1 2 2 ...
##  $ ChronBron_E         : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Emph_E              : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Asthma_E            : int  2 2 2 2 1 2 1 3 2 2 ...
##  $ OthLung_E           : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Arth_E              : int  1 1 1 1 1 2 1 1 2 2 ...
##  $ ThyDis_E            : int  2 2 1 2 2 2 1 2 2 2 ...
##  $ Cancer_E            : int  1 1 2 1 2 2 2 2 2 2 ...
##  $ LivDis_E            : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ AIDS_E              : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ KidDis_E            : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Vision_E            : int  2 1 2 2 2 2 1 2 2 2 ...
##  $ HearLoss_E          : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ OthMajor_E          : int  1 1 2 2 2 2 2 1 2 2 ...
##  $ OthCogTest          : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ PsychDis_E          : int  NA 2 2 2 2 2 2 2 NA NA ...
##  $ StrokeCons_E        : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ StrokeSpells_E      : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ StrokeParal_E       : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ StrokeVision_E      : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ StrokeLang_E        : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ StrokeWeak_E        : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ StrokeSense_E       : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ TumorOrCancer_E     : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ hosp5yr             : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ seehear             : int  NA NA NA NA NA NA NA NA NA NA ...</code></pre>
</details>
<p><br> <br></p>
</div>
<div id="morbidity_mortality" class="section level2">
<h2>Morbidity_Mortality</h2>
<p>Participant Morbidity and Mortality Details</p>
<pre class="r"><code>## load data
Morbidity_Mortality &lt;- read.csv(paste0(wrap_directory, &quot;Morbidity_Mortality_05Jun2025.csv&quot;))
info(Morbidity_Mortality,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:94, cols:18, inds:94</code></pre>
<pre class="r"><code>str(Morbidity_Mortality)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    94 obs. of  18 variables:
##  $ wrapnum                  : chr  &quot;wrap0065&quot; &quot;wrap0126&quot; &quot;wrap0127&quot; &quot;wrap0128&quot; ...
##  $ WRAPNo                   : chr  &quot;0065&quot; &quot;0126&quot; &quot;0127&quot; &quot;0128&quot; ...
##  $ PTRel                    : int  NA NA NA NA 2 NA NA NA NA NA ...
##  $ ageAtDeath               : num  61.2 68.2 74.3 58.5 69 ...
##  $ death_days_after_baseline: int  3253 5793 5300 4906 5484 5500 4559 6512 4752 4189 ...
##  $ ageAtDeath_Over89        : chr  &quot;No&quot; &quot;No&quot; &quot;No&quot; &quot;No&quot; ...
##  $ CauseDeath               : chr  &quot;Stomach cancer&quot; &quot;&quot; &quot;Pulmonary fibrosis&quot; &quot;pancreatic cancer&quot; ...
##  $ BrainAutopsy             : int  0 NA NA 0 0 0 0 0 NA 1 ...
##  $ yearAutopsy              : int  NA NA NA NA NA NA NA NA NA 2015 ...
##  $ yearADDiagnosis          : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ WhereAutopsy             : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ AutopsyRes               : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ MemProb                  : int  0 NA 0 0 0 0 0 NA NA 1 ...
##  $ Confuse                  : int  0 NA 0 0 0 NA 0 NA NA 1 ...
##  $ Dementia                 : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ MCI                      : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ WBDPTissueAvail          : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ update_stamp             : chr  &quot;2024-09-13 08:53:33.0&quot; &quot;2024-09-13 08:53:33.0&quot; &quot;2024-09-13 08:53:33.0&quot; &quot;2024-09-13 08:53:33.0&quot; ...</code></pre>
</details>
<pre class="r"><code>## check table frequency for all variables

## convert &quot;&quot; to NA for chr columns
chrcols &lt;- colnames(Morbidity_Mortality)[sapply(Morbidity_Mortality, is.character)]
Morbidity_Mortality[chrcols] &lt;- lapply(Morbidity_Mortality[chrcols], function(x) ifelse(x == &quot;&quot;, NA, x))

## drop update_stamp column
Morbidity_Mortality &lt;- subset(Morbidity_Mortality,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(Morbidity_Mortality)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    94 obs. of  17 variables:
##  $ wrapnum                  : chr  &quot;wrap0065&quot; &quot;wrap0126&quot; &quot;wrap0127&quot; &quot;wrap0128&quot; ...
##  $ WRAPNo                   : chr  &quot;0065&quot; &quot;0126&quot; &quot;0127&quot; &quot;0128&quot; ...
##  $ PTRel                    : int  NA NA NA NA 2 NA NA NA NA NA ...
##  $ ageAtDeath               : num  61.2 68.2 74.3 58.5 69 ...
##  $ death_days_after_baseline: int  3253 5793 5300 4906 5484 5500 4559 6512 4752 4189 ...
##  $ ageAtDeath_Over89        : chr  &quot;No&quot; &quot;No&quot; &quot;No&quot; &quot;No&quot; ...
##  $ CauseDeath               : chr  &quot;Stomach cancer&quot; NA &quot;Pulmonary fibrosis&quot; &quot;pancreatic cancer&quot; ...
##  $ BrainAutopsy             : int  0 NA NA 0 0 0 0 0 NA 1 ...
##  $ yearAutopsy              : int  NA NA NA NA NA NA NA NA NA 2015 ...
##  $ yearADDiagnosis          : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ WhereAutopsy             : chr  NA NA NA NA ...
##  $ AutopsyRes               : chr  NA NA NA NA ...
##  $ MemProb                  : int  0 NA 0 0 0 0 0 NA NA 1 ...
##  $ Confuse                  : int  0 NA 0 0 0 NA 0 NA NA 1 ...
##  $ Dementia                 : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ MCI                      : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ WBDPTissueAvail          : int  NA NA NA NA NA NA NA NA NA NA ...</code></pre>
</details>
<p><br> <br></p>
</div>
<div id="medhistory_admit" class="section level2">
<h2>MedHistory_Admit</h2>
<p>Medical History (Admitted to Hospital)</p>
<pre class="r"><code>## load data
MedHistory_Admit &lt;- read.csv(paste0(wrap_directory, &quot;MedHistory_Admit_05Jun2025.csv&quot;))
info(MedHistory_Admit,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:1052, cols:6, inds:603</code></pre>
<pre class="r"><code>str(MedHistory_Admit)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    1052 obs. of  6 variables:
##  $ wrapnum          : chr  &quot;wrap0146&quot; &quot;wrap0146&quot; &quot;wrap1081&quot; &quot;wrap1081&quot; ...
##  $ WRAPNo           : chr  &quot;0146&quot; &quot;0146&quot; &quot;1081&quot; &quot;1081&quot; ...
##  $ VisNo            : int  4 4 1 1 4 4 2 7 6 6 ...
##  $ AdmitSRsn        : chr  &quot;right knee replacement&quot; &quot;trimaleolar fx of left lower leg&quot; &quot;knee replacement&quot; &quot;knee replacement&quot; ...
##  $ AdmitS_Repeat_Key: int  2 1 2 1 2 1 1 1 2 1 ...
##  $ update_stamp     : chr  &quot;2024-09-13 08:53:12.0&quot; &quot;2024-09-13 08:53:12.0&quot; &quot;2024-09-13 08:53:12.0&quot; &quot;2024-09-13 08:53:12.0&quot; ...</code></pre>
</details>
<pre class="r"><code>## check table frequency for all variables

## convert &quot;&quot; to NA for chr columns
chrcols &lt;- colnames(MedHistory_Admit)[sapply(MedHistory_Admit, is.character)]
MedHistory_Admit[chrcols] &lt;- lapply(MedHistory_Admit[chrcols], function(x) ifelse(x == &quot;&quot;, NA, x))

## drop update_stamp column
MedHistory_Admit &lt;- subset(MedHistory_Admit,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(MedHistory_Admit)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    1052 obs. of  5 variables:
##  $ wrapnum          : chr  &quot;wrap0146&quot; &quot;wrap0146&quot; &quot;wrap1081&quot; &quot;wrap1081&quot; ...
##  $ WRAPNo           : chr  &quot;0146&quot; &quot;0146&quot; &quot;1081&quot; &quot;1081&quot; ...
##  $ VisNo            : int  4 4 1 1 4 4 2 7 6 6 ...
##  $ AdmitSRsn        : chr  &quot;right knee replacement&quot; &quot;trimaleolar fx of left lower leg&quot; &quot;knee replacement&quot; &quot;knee replacement&quot; ...
##  $ AdmitS_Repeat_Key: int  2 1 2 1 2 1 1 1 2 1 ...</code></pre>
</details>
<p><br> <br></p>
</div>
<div id="medhistory_surgbf" class="section level2">
<h2>MedHistory_SurgBF</h2>
<p>Medical History (Surgery since last visit)</p>
<pre class="r"><code>## load data
MedHistory_SurgBF &lt;- read.csv(paste0(wrap_directory, &quot;MedHistory_SurgBF_05Jun2025.csv&quot;))
info(MedHistory_SurgBF,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:1327, cols:6, inds:1115</code></pre>
<pre class="r"><code>str(MedHistory_SurgBF)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    1327 obs. of  6 variables:
##  $ wrapnum          : chr  &quot;wrap1006&quot; &quot;wrap1081&quot; &quot;wrap1081&quot; &quot;wrap1081&quot; ...
##  $ WRAPNo           : chr  &quot;1006&quot; &quot;1081&quot; &quot;1081&quot; &quot;1081&quot; ...
##  $ VisNo            : int  2 1 1 1 2 2 2 2 2 1 ...
##  $ SurgBfRsn        : chr  &quot;nasal polypectomy&quot; &quot;L &amp; R thumbs&quot; &quot;knee replacements&quot; &quot;knee replacements&quot; ...
##  $ SurgBF_Repeat_Key: int  1 3 2 1 1 1 1 1 1 1 ...
##  $ update_stamp     : chr  &quot;2024-09-13 08:53:12.0&quot; &quot;2024-09-13 08:53:12.0&quot; &quot;2024-09-13 08:53:12.0&quot; &quot;2024-09-13 08:53:12.0&quot; ...</code></pre>
</details>
<pre class="r"><code>## check table frequency for all variables

## convert &quot;&quot; to NA for chr columns
chrcols &lt;- colnames(MedHistory_SurgBF)[sapply(MedHistory_SurgBF, is.character)]
MedHistory_SurgBF[chrcols] &lt;- lapply(MedHistory_SurgBF[chrcols], function(x) ifelse(x == &quot;&quot;, NA, x))

## drop update_stamp column
MedHistory_SurgBF &lt;- subset(MedHistory_SurgBF,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(MedHistory_SurgBF)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    1327 obs. of  5 variables:
##  $ wrapnum          : chr  &quot;wrap1006&quot; &quot;wrap1081&quot; &quot;wrap1081&quot; &quot;wrap1081&quot; ...
##  $ WRAPNo           : chr  &quot;1006&quot; &quot;1081&quot; &quot;1081&quot; &quot;1081&quot; ...
##  $ VisNo            : int  2 1 1 1 2 2 2 2 2 1 ...
##  $ SurgBfRsn        : chr  &quot;nasal polypectomy&quot; &quot;L &amp; R thumbs&quot; &quot;knee replacements&quot; &quot;knee replacements&quot; ...
##  $ SurgBF_Repeat_Key: int  1 3 2 1 1 1 1 1 1 1 ...</code></pre>
<p><br> <br></p>
</div>
<div id="medhistory_surgs" class="section level2">
<h2>MedHistory_SurgS</h2>
<p>Medical History (Surgery since last visit)</p>
<pre class="r"><code>## load data
MedHistory_SurgS &lt;- read.csv(paste0(wrap_directory, &quot;MedHistory_SurgS_05Jun2025.csv&quot;))
info(MedHistory_SurgS,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:3438, cols:6, inds:1116</code></pre>
<pre class="r"><code>str(MedHistory_SurgS)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    3438 obs. of  6 variables:
##  $ wrapnum         : chr  &quot;wrap0002&quot; &quot;wrap0146&quot; &quot;wrap0146&quot; &quot;wrap0401&quot; ...
##  $ WRAPNo          : chr  &quot;0002&quot; &quot;0146&quot; &quot;0146&quot; &quot;0401&quot; ...
##  $ VisNo           : int  5 4 4 6 6 6 7 7 6 3 ...
##  $ SurgSRsn        : chr  &quot;colonoscopy&quot; &quot;spinal for knee replacement&quot; &quot;general anesthesia for trimaleolar fx&quot; &quot;removal of 2 small skin spots&quot; ...
##  $ SurgS_Repeat_Key: int  1 2 1 3 2 1 2 1 1 1 ...
##  $ update_stamp    : chr  &quot;2024-09-13 08:53:13.0&quot; &quot;2024-09-13 08:53:13.0&quot; &quot;2024-09-13 08:53:13.0&quot; &quot;2024-09-13 08:53:13.0&quot; ...</code></pre>
<pre class="r"><code>## check table frequency for all variables

## convert &quot;&quot; to NA for chr columns
chrcols &lt;- colnames(MedHistory_SurgS)[sapply(MedHistory_SurgS, is.character)]
MedHistory_SurgS[chrcols] &lt;- lapply(MedHistory_SurgS[chrcols], function(x) ifelse(x == &quot;&quot;, NA, x))

## drop update_stamp column
MedHistory_SurgS &lt;- subset(MedHistory_SurgS,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(MedHistory_SurgS)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    3438 obs. of  5 variables:
##  $ wrapnum         : chr  &quot;wrap0002&quot; &quot;wrap0146&quot; &quot;wrap0146&quot; &quot;wrap0401&quot; ...
##  $ WRAPNo          : chr  &quot;0002&quot; &quot;0146&quot; &quot;0146&quot; &quot;0401&quot; ...
##  $ VisNo           : int  5 4 4 6 6 6 7 7 6 3 ...
##  $ SurgSRsn        : chr  &quot;colonoscopy&quot; &quot;spinal for knee replacement&quot; &quot;general anesthesia for trimaleolar fx&quot; &quot;removal of 2 small skin spots&quot; ...
##  $ SurgS_Repeat_Key: int  1 2 1 3 2 1 2 1 1 1 ...</code></pre>
<p><br> <br></p>
</div>
<div id="mind_diet" class="section level2">
<h2>MIND_Diet</h2>
<p>MIND Diet</p>
<pre class="r"><code>## load data
MIND_Diet &lt;- read.csv(paste0(wrap_directory, &quot;MIND_Diet_05Jun2025.csv&quot;))
info(MIND_Diet,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:3621, cols:19, inds:1416</code></pre>
<pre class="r"><code>str(MIND_Diet)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    3621 obs. of  19 variables:
##  $ wrapnum            : chr  &quot;wrap0002&quot; &quot;wrap0148&quot; &quot;wrap0305&quot; &quot;wrap0401&quot; ...
##  $ WRAPNo             : chr  &quot;0002&quot; &quot;0148&quot; &quot;0305&quot; &quot;0401&quot; ...
##  $ VisNo              : int  5 5 5 6 7 8 6 5 3 1 ...
##  $ MIND_OlivOil_Dy    : num  0 0 1 2 0 0.1 1 2 0 0.5 ...
##  $ MIND_GrnLfyVg_SrvDy: num  0 1 1 1 0.5 0.3 1 1.5 1 2 ...
##  $ MIND_OthrVg_SrvDy  : num  1 1 3 1 0 0.3 2 1.5 1 2 ...
##  $ MIND_Berries_SrvWk : num  0 1 2 7 1 0.1 1 6 0 1 ...
##  $ MIND_RedMeat_SrvWk : num  5 4 4 2 3.5 5 0 3 6 4 ...
##  $ MIND_Fish_SrvWk    : num  0.5 1 1 4 1.5 0.2 0.5 0 1 1 ...
##  $ MIND_Chickn_SrvWk  : num  3 1 2 1 1.5 0.2 2 3 0 1 ...
##  $ MIND_Cheese_SrvWk  : num  7 2 1 3 0 0.3 4 3 9 3 ...
##  $ MIND_Butter_SrvDy  : num  9 1 0.25 1 2 0.2 1 0.5 1 0.5 ...
##  $ MIND_Bean_SrvWk    : num  0.5 2 1 1 1 0.1 2 3 2 1 ...
##  $ MIND_WhlGrn_Dy     : num  0 1 1 1 1 2 2 4 2 1 ...
##  $ MIND_Sweet_Wk      : num  10 4 2 3 5 10 1 4 5 3 ...
##  $ MIND_Nut_SrvWk     : num  1 10 7 6 1 10 2 6 1 3 ...
##  $ MIND_FastFood_Wk   : num  1 0 0 0 0 4 0 0.25 0 0 ...
##  $ MIND_Alchl_SrvDy   : num  0 0 0 1.5 0 2 1.5 1.5 0 2 ...
##  $ update_stamp       : chr  &quot;2025-01-24 08:18:14.0&quot; &quot;2025-01-24 08:18:14.0&quot; &quot;2025-01-24 08:18:14.0&quot; &quot;2025-01-24 08:18:14.0&quot; ...</code></pre>
<pre class="r"><code>## check table frequency for all variables

## drop update_stamp column
MIND_Diet &lt;- subset(MIND_Diet,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(MIND_Diet)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    3621 obs. of  18 variables:
##  $ wrapnum            : chr  &quot;wrap0002&quot; &quot;wrap0148&quot; &quot;wrap0305&quot; &quot;wrap0401&quot; ...
##  $ WRAPNo             : chr  &quot;0002&quot; &quot;0148&quot; &quot;0305&quot; &quot;0401&quot; ...
##  $ VisNo              : int  5 5 5 6 7 8 6 5 3 1 ...
##  $ MIND_OlivOil_Dy    : num  0 0 1 2 0 0.1 1 2 0 0.5 ...
##  $ MIND_GrnLfyVg_SrvDy: num  0 1 1 1 0.5 0.3 1 1.5 1 2 ...
##  $ MIND_OthrVg_SrvDy  : num  1 1 3 1 0 0.3 2 1.5 1 2 ...
##  $ MIND_Berries_SrvWk : num  0 1 2 7 1 0.1 1 6 0 1 ...
##  $ MIND_RedMeat_SrvWk : num  5 4 4 2 3.5 5 0 3 6 4 ...
##  $ MIND_Fish_SrvWk    : num  0.5 1 1 4 1.5 0.2 0.5 0 1 1 ...
##  $ MIND_Chickn_SrvWk  : num  3 1 2 1 1.5 0.2 2 3 0 1 ...
##  $ MIND_Cheese_SrvWk  : num  7 2 1 3 0 0.3 4 3 9 3 ...
##  $ MIND_Butter_SrvDy  : num  9 1 0.25 1 2 0.2 1 0.5 1 0.5 ...
##  $ MIND_Bean_SrvWk    : num  0.5 2 1 1 1 0.1 2 3 2 1 ...
##  $ MIND_WhlGrn_Dy     : num  0 1 1 1 1 2 2 4 2 1 ...
##  $ MIND_Sweet_Wk      : num  10 4 2 3 5 10 1 4 5 3 ...
##  $ MIND_Nut_SrvWk     : num  1 10 7 6 1 10 2 6 1 3 ...
##  $ MIND_FastFood_Wk   : num  1 0 0 0 0 4 0 0.25 0 0 ...
##  $ MIND_Alchl_SrvDy   : num  0 0 0 1.5 0 2 1.5 1.5 0 2 ...</code></pre>
<p><br> <br></p>
</div>
<div id="mind_diet_scoredvalues" class="section level2">
<h2>MIND_Diet_ScoredValues</h2>
<p>MIND Diet Scored Values</p>
<p>This dataset is exactly same as the NIND_Diet dataset. So I won’t be
including this file when do the final merge</p>
<pre class="r"><code>## load data
MIND_Diet_ScoredValues &lt;- read.csv(paste0(wrap_directory, &quot;MIND_Diet_ScoredValues_05Jun2025.csv&quot;))
info(MIND_Diet_ScoredValues,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:3621, cols:22, inds:1416</code></pre>
<pre class="r"><code>str(MIND_Diet_ScoredValues)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    3621 obs. of  22 variables:
##  $ wrapnum                 : chr  &quot;wrap0002&quot; &quot;wrap0148&quot; &quot;wrap0305&quot; &quot;wrap0401&quot; ...
##  $ WRAPNo                  : chr  &quot;0002&quot; &quot;0148&quot; &quot;0305&quot; &quot;0401&quot; ...
##  $ VisNo                   : int  5 5 5 6 7 8 6 5 3 1 ...
##  $ MINDScoredValue_GrnLfyVg: int  0 1 1 1 1 1 1 1 1 1 ...
##  $ MINDScoredValue_OthrVg  : int  1 1 1 1 0 0 1 1 1 1 ...
##  $ MINDScoredValue_Berries : int  0 1 1 1 1 0 1 1 0 1 ...
##  $ MINDScoredValue_RedMeat : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ MINDScoredValue_Fish    : int  1 1 1 1 1 0 1 0 1 1 ...
##  $ MINDScoredValue_Chicken : int  1 1 1 1 1 0 1 1 0 1 ...
##  $ MINDScoredValue_Cheese  : int  0 1 1 1 1 1 1 1 0 1 ...
##  $ MINDScoredValue_Butter  : int  0 1 1 1 1 1 1 1 1 1 ...
##  $ MINDScoredValue_Beans   : int  0 1 1 1 1 0 1 1 1 1 ...
##  $ MINDScoredValue_WhlGrn  : int  0 1 1 1 1 1 1 1 1 1 ...
##  $ MINDScoredValue_Sweets  : int  0 1 1 1 1 0 1 1 1 1 ...
##  $ MINDScoredValue_Nuts    : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ MINDScoredValue_FastFood: int  1 1 1 1 1 0 1 1 1 1 ...
##  $ MINDScoredValue_Alcohol : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ MINDScoredValue_OliveOil: int  0 0 1 1 0 0 0 1 0 0 ...
##  $ MIND_Straight_Sum       : num  4 9.5 12 11.5 8 4.5 9.5 12 7 9.5 ...
##  $ MIND_Extrapolated_Sum   : num  4 9.5 12 11.5 8 4.5 9.5 12 7 9.5 ...
##  $ MIND_NumberItemsMissing : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ update_stamp            : chr  &quot;2025-01-24 08:18:15.0&quot; &quot;2024-09-13 08:53:31.0&quot; &quot;2025-01-24 08:18:15.0&quot; &quot;2024-09-13 08:53:31.0&quot; ...</code></pre>
<pre class="r"><code>## check table frequency for all variables

## drop update_stamp column
MIND_Diet &lt;- subset(MIND_Diet_ScoredValues,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(MIND_Diet_ScoredValues)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    3621 obs. of  22 variables:
##  $ wrapnum                 : chr  &quot;wrap0002&quot; &quot;wrap0148&quot; &quot;wrap0305&quot; &quot;wrap0401&quot; ...
##  $ WRAPNo                  : chr  &quot;0002&quot; &quot;0148&quot; &quot;0305&quot; &quot;0401&quot; ...
##  $ VisNo                   : int  5 5 5 6 7 8 6 5 3 1 ...
##  $ MINDScoredValue_GrnLfyVg: int  0 1 1 1 1 1 1 1 1 1 ...
##  $ MINDScoredValue_OthrVg  : int  1 1 1 1 0 0 1 1 1 1 ...
##  $ MINDScoredValue_Berries : int  0 1 1 1 1 0 1 1 0 1 ...
##  $ MINDScoredValue_RedMeat : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ MINDScoredValue_Fish    : int  1 1 1 1 1 0 1 0 1 1 ...
##  $ MINDScoredValue_Chicken : int  1 1 1 1 1 0 1 1 0 1 ...
##  $ MINDScoredValue_Cheese  : int  0 1 1 1 1 1 1 1 0 1 ...
##  $ MINDScoredValue_Butter  : int  0 1 1 1 1 1 1 1 1 1 ...
##  $ MINDScoredValue_Beans   : int  0 1 1 1 1 0 1 1 1 1 ...
##  $ MINDScoredValue_WhlGrn  : int  0 1 1 1 1 1 1 1 1 1 ...
##  $ MINDScoredValue_Sweets  : int  0 1 1 1 1 0 1 1 1 1 ...
##  $ MINDScoredValue_Nuts    : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ MINDScoredValue_FastFood: int  1 1 1 1 1 0 1 1 1 1 ...
##  $ MINDScoredValue_Alcohol : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ MINDScoredValue_OliveOil: int  0 0 1 1 0 0 0 1 0 0 ...
##  $ MIND_Straight_Sum       : num  4 9.5 12 11.5 8 4.5 9.5 12 7 9.5 ...
##  $ MIND_Extrapolated_Sum   : num  4 9.5 12 11.5 8 4.5 9.5 12 7 9.5 ...
##  $ MIND_NumberItemsMissing : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ update_stamp            : chr  &quot;2025-01-24 08:18:15.0&quot; &quot;2024-09-13 08:53:31.0&quot; &quot;2025-01-24 08:18:15.0&quot; &quot;2024-09-13 08:53:31.0&quot; ...</code></pre>
<p><br> <br></p>
</div>
<div id="moca" class="section level2">
<h2>MOCA</h2>
<p>Montreal Cognitive Assessment-Blind. Administered as part of remote
assessment (T-Cog)</p>
<pre class="r"><code>## load data
MOCA &lt;- read.csv(paste0(wrap_directory, &quot;MOCA_05Jun2025.csv&quot;))
info(MOCA,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:316, cols:66, inds:309</code></pre>
<pre class="r"><code>str(MOCA)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    316 obs. of  66 variables:
##  $ wrapnum                    : chr  &quot;wrap0434&quot; &quot;wrap1113&quot; &quot;wrap0212&quot; &quot;wrap0219&quot; ...
##  $ WRAPNo                     : chr  &quot;0434&quot; &quot;1113&quot; &quot;0212&quot; &quot;0219&quot; ...
##  $ VisNo                      : int  7 1 7 7 1 5 7 1 7 7 ...
##  $ MOCA_EdYrsLE12             : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ MOCA_Mem_TestPerf          : int  -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 ...
##  $ MOCA_Mem_Trl1_Face         : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ MOCA_Mem_Trl1_Velvet       : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ MOCA_Mem_Trl1_Church       : int  1 1 1 1 1 1 0 1 1 1 ...
##  $ MOCA_Mem_Trl1_Daisy        : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ MOCA_Mem_Trl1_Red          : int  1 1 1 1 1 1 1 1 0 1 ...
##  $ MOCA_Mem_Trl2_Face         : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ MOCA_Mem_Trl2_Velvet       : int  0 1 1 1 1 1 1 1 1 1 ...
##  $ MOCA_Mem_Trl2_Church       : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ MOCA_Mem_Trl2_Daisy        : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ MOCA_Mem_Trl2_Red          : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ MOCA_Attn_Digits_Forw      : int  1 1 1 1 1 1 0 1 1 1 ...
##  $ MOCA_Attn_Digits_Backw     : int  1 1 1 1 1 1 1 0 1 1 ...
##  $ MOCA_Attn_Digits_Total     : int  2 2 2 2 2 2 1 1 2 2 ...
##  $ MOCA_Attn_Tap_A_Errors     : int  0 1 0 0 0 3 1 0 0 0 ...
##  $ MOCA_Attn_Tap_A_Total      : int  1 1 1 1 1 0 1 1 1 1 ...
##  $ MOCA_Attn_Ser7_93          : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ MOCA_Attn_Ser7_86          : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ MOCA_Attn_Ser7_79          : int  1 1 1 1 1 1 1 0 0 1 ...
##  $ MOCA_Attn_Ser7_72          : int  1 1 1 1 1 1 1 0 1 1 ...
##  $ MOCA_Attn_Ser7_65          : int  1 1 1 1 1 1 1 0 1 1 ...
##  $ MOCA_Attn_Ser7_Total       : int  3 3 3 3 3 3 3 2 3 3 ...
##  $ MOCA_Attn_Domain_Total     : int  6 6 6 6 6 5 5 4 6 6 ...
##  $ MOCA_Lang_JohnHelp         : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ MOCA_Lang_CatHid           : int  0 0 1 1 1 1 1 1 1 0 ...
##  $ MOCA_Lang_Total            : int  1 1 2 2 2 2 2 2 2 1 ...
##  $ MOCA_Lang_WordsF           : int  15 12 9 16 14 12 11 17 11 21 ...
##  $ MOCA_Lang_WordsF_Score     : int  1 1 0 1 1 1 1 1 1 1 ...
##  $ MOCA_Lang_Domain_Total     : int  2 2 2 3 3 3 3 3 3 2 ...
##  $ MOCA_Abstr_Sim_TrainBicycle: int  0 1 1 1 1 1 1 1 1 0 ...
##  $ MOCA_Abstr_Sim_WatchRuler  : int  1 1 0 1 1 1 1 1 1 1 ...
##  $ MOCA_Abstr_Sim_Total       : int  1 2 1 2 2 2 2 2 2 1 ...
##  $ MOCA_DelRec_NoCue_Face     : int  1 0 1 1 1 1 1 0 0 1 ...
##  $ MOCA_DelRec_NoCue_Velvet   : int  1 0 1 1 1 1 0 1 1 1 ...
##  $ MOCA_DelRec_NoCue_Church   : int  0 1 1 0 1 0 1 1 1 1 ...
##  $ MOCA_DelRec_NoCue_Daisy    : int  1 1 1 0 1 0 0 1 1 1 ...
##  $ MOCA_DelRec_NoCue_Red      : int  0 1 1 1 1 0 0 0 1 1 ...
##  $ MOCA_DelRec_NoCue_Total    : int  3 3 5 3 5 2 2 3 4 5 ...
##  $ MOCA_DelRec_CatCue_Face    : int  NA 0 0 NA NA NA NA 1 1 NA ...
##  $ MOCA_DelRec_CatCue_Velvet  : int  NA 0 0 NA NA NA 0 NA NA NA ...
##  $ MOCA_DelRec_CatCue_Church  : int  1 NA 0 0 NA 1 NA NA NA NA ...
##  $ MOCA_DelRec_CatCue_Daisy   : int  NA NA 0 0 NA 1 1 NA NA NA ...
##  $ MOCA_DelRec_CatCue_Red     : int  1 NA 0 NA NA 1 1 1 NA NA ...
##  $ MOCA_DelRec_CatCue_Total   : int  2 0 0 0 0 3 2 2 1 0 ...
##  $ MOCA_DelRec_MultCue_Face   : int  NA 0 0 NA NA NA NA NA NA NA ...
##  $ MOCA_DelRec_MultCue_Velvet : int  NA 1 0 NA NA NA 1 NA NA NA ...
##  $ MOCA_DelRec_MultCue_Church : int  NA NA 0 1 NA NA NA NA NA NA ...
##  $ MOCA_DelRec_MultCue_Daisy  : int  NA NA 0 0 NA NA NA NA NA NA ...
##  $ MOCA_DelRec_MultCue_Red    : int  NA NA 0 NA NA NA NA NA NA NA ...
##  $ MOCA_DelRec_MultCue_Total  : int  0 1 0 1 0 0 1 0 0 0 ...
##  $ MOCA_DelRec_AllCue_Total   : int  13 10 15 10 15 12 11 13 14 15 ...
##  $ MOCA_Orient_Date           : int  1 1 1 1 1 0 1 1 1 1 ...
##  $ MOCA_Orient_Month          : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ MOCA_Orient_Year           : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ MOCA_Orient_Day            : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ MOCA_Orient_Place          : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ MOCA_Orient_City           : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ MOCA_Orient_Total          : int  6 6 6 6 6 5 6 6 6 6 ...
##  $ MOCA_NumScoresMissing      : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ MOCA_TotalScore            : int  18 19 20 20 22 17 18 18 21 20 ...
##  $ MOCA_TotalScoreStatus      : chr  &quot;Normal - With Memory Scores&quot; &quot;Normal - With Memory Scores&quot; &quot;Normal - With Memory Scores&quot; &quot;Normal - With Memory Scores&quot; ...
##  $ update_stamp               : chr  &quot;2024-09-13 08:53:33.0&quot; &quot;2024-09-13 08:53:33.0&quot; &quot;2024-09-13 08:53:33.0&quot; &quot;2024-09-13 08:53:33.0&quot; ...</code></pre>
<pre class="r"><code>## check table frequency for all variables

## drop update_stamp column
MOCA &lt;- subset(MOCA,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(MOCA)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    316 obs. of  65 variables:
##  $ wrapnum                    : chr  &quot;wrap0434&quot; &quot;wrap1113&quot; &quot;wrap0212&quot; &quot;wrap0219&quot; ...
##  $ WRAPNo                     : chr  &quot;0434&quot; &quot;1113&quot; &quot;0212&quot; &quot;0219&quot; ...
##  $ VisNo                      : int  7 1 7 7 1 5 7 1 7 7 ...
##  $ MOCA_EdYrsLE12             : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ MOCA_Mem_TestPerf          : int  -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 ...
##  $ MOCA_Mem_Trl1_Face         : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ MOCA_Mem_Trl1_Velvet       : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ MOCA_Mem_Trl1_Church       : int  1 1 1 1 1 1 0 1 1 1 ...
##  $ MOCA_Mem_Trl1_Daisy        : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ MOCA_Mem_Trl1_Red          : int  1 1 1 1 1 1 1 1 0 1 ...
##  $ MOCA_Mem_Trl2_Face         : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ MOCA_Mem_Trl2_Velvet       : int  0 1 1 1 1 1 1 1 1 1 ...
##  $ MOCA_Mem_Trl2_Church       : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ MOCA_Mem_Trl2_Daisy        : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ MOCA_Mem_Trl2_Red          : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ MOCA_Attn_Digits_Forw      : int  1 1 1 1 1 1 0 1 1 1 ...
##  $ MOCA_Attn_Digits_Backw     : int  1 1 1 1 1 1 1 0 1 1 ...
##  $ MOCA_Attn_Digits_Total     : int  2 2 2 2 2 2 1 1 2 2 ...
##  $ MOCA_Attn_Tap_A_Errors     : int  0 1 0 0 0 3 1 0 0 0 ...
##  $ MOCA_Attn_Tap_A_Total      : int  1 1 1 1 1 0 1 1 1 1 ...
##  $ MOCA_Attn_Ser7_93          : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ MOCA_Attn_Ser7_86          : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ MOCA_Attn_Ser7_79          : int  1 1 1 1 1 1 1 0 0 1 ...
##  $ MOCA_Attn_Ser7_72          : int  1 1 1 1 1 1 1 0 1 1 ...
##  $ MOCA_Attn_Ser7_65          : int  1 1 1 1 1 1 1 0 1 1 ...
##  $ MOCA_Attn_Ser7_Total       : int  3 3 3 3 3 3 3 2 3 3 ...
##  $ MOCA_Attn_Domain_Total     : int  6 6 6 6 6 5 5 4 6 6 ...
##  $ MOCA_Lang_JohnHelp         : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ MOCA_Lang_CatHid           : int  0 0 1 1 1 1 1 1 1 0 ...
##  $ MOCA_Lang_Total            : int  1 1 2 2 2 2 2 2 2 1 ...
##  $ MOCA_Lang_WordsF           : int  15 12 9 16 14 12 11 17 11 21 ...
##  $ MOCA_Lang_WordsF_Score     : int  1 1 0 1 1 1 1 1 1 1 ...
##  $ MOCA_Lang_Domain_Total     : int  2 2 2 3 3 3 3 3 3 2 ...
##  $ MOCA_Abstr_Sim_TrainBicycle: int  0 1 1 1 1 1 1 1 1 0 ...
##  $ MOCA_Abstr_Sim_WatchRuler  : int  1 1 0 1 1 1 1 1 1 1 ...
##  $ MOCA_Abstr_Sim_Total       : int  1 2 1 2 2 2 2 2 2 1 ...
##  $ MOCA_DelRec_NoCue_Face     : int  1 0 1 1 1 1 1 0 0 1 ...
##  $ MOCA_DelRec_NoCue_Velvet   : int  1 0 1 1 1 1 0 1 1 1 ...
##  $ MOCA_DelRec_NoCue_Church   : int  0 1 1 0 1 0 1 1 1 1 ...
##  $ MOCA_DelRec_NoCue_Daisy    : int  1 1 1 0 1 0 0 1 1 1 ...
##  $ MOCA_DelRec_NoCue_Red      : int  0 1 1 1 1 0 0 0 1 1 ...
##  $ MOCA_DelRec_NoCue_Total    : int  3 3 5 3 5 2 2 3 4 5 ...
##  $ MOCA_DelRec_CatCue_Face    : int  NA 0 0 NA NA NA NA 1 1 NA ...
##  $ MOCA_DelRec_CatCue_Velvet  : int  NA 0 0 NA NA NA 0 NA NA NA ...
##  $ MOCA_DelRec_CatCue_Church  : int  1 NA 0 0 NA 1 NA NA NA NA ...
##  $ MOCA_DelRec_CatCue_Daisy   : int  NA NA 0 0 NA 1 1 NA NA NA ...
##  $ MOCA_DelRec_CatCue_Red     : int  1 NA 0 NA NA 1 1 1 NA NA ...
##  $ MOCA_DelRec_CatCue_Total   : int  2 0 0 0 0 3 2 2 1 0 ...
##  $ MOCA_DelRec_MultCue_Face   : int  NA 0 0 NA NA NA NA NA NA NA ...
##  $ MOCA_DelRec_MultCue_Velvet : int  NA 1 0 NA NA NA 1 NA NA NA ...
##  $ MOCA_DelRec_MultCue_Church : int  NA NA 0 1 NA NA NA NA NA NA ...
##  $ MOCA_DelRec_MultCue_Daisy  : int  NA NA 0 0 NA NA NA NA NA NA ...
##  $ MOCA_DelRec_MultCue_Red    : int  NA NA 0 NA NA NA NA NA NA NA ...
##  $ MOCA_DelRec_MultCue_Total  : int  0 1 0 1 0 0 1 0 0 0 ...
##  $ MOCA_DelRec_AllCue_Total   : int  13 10 15 10 15 12 11 13 14 15 ...
##  $ MOCA_Orient_Date           : int  1 1 1 1 1 0 1 1 1 1 ...
##  $ MOCA_Orient_Month          : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ MOCA_Orient_Year           : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ MOCA_Orient_Day            : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ MOCA_Orient_Place          : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ MOCA_Orient_City           : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ MOCA_Orient_Total          : int  6 6 6 6 6 5 6 6 6 6 ...
##  $ MOCA_NumScoresMissing      : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ MOCA_TotalScore            : int  18 19 20 20 22 17 18 18 21 20 ...
##  $ MOCA_TotalScoreStatus      : chr  &quot;Normal - With Memory Scores&quot; &quot;Normal - With Memory Scores&quot; &quot;Normal - With Memory Scores&quot; &quot;Normal - With Memory Scores&quot; ...</code></pre>
<p><br> <br></p>
</div>
<div id="neuropsychscores" class="section level2">
<h2>NeuropsychScores</h2>
<p>Neuropsych Testing Scores</p>
<pre class="r"><code>## load data
NeuropsychScores &lt;- read.csv(paste0(wrap_directory, &quot;NeuropsychScores_05Jun2025.csv&quot;))
info(NeuropsychScores,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:7996, cols:157, inds:1799</code></pre>
<pre class="r"><code>str(NeuropsychScores, max.level = 99, list.len = 99999)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    7996 obs. of  157 variables:
##  $ wrapnum     : chr  &quot;wrap0943&quot; &quot;wrap0068&quot; &quot;wrap0759&quot; &quot;wrapA0078&quot; ...
##  $ WRAPNo      : chr  &quot;0943&quot; &quot;0068&quot; &quot;0759&quot; &quot;A0078&quot; ...
##  $ VisNo       : int  3 2 3 4 5 4 2 1 3 6 ...
##  $ curage      : int  60 64 64 65 65 56 64 52 68 70 ...
##  $ iqvocr      : int  NA 68 NA NA NA NA 50 62 NA NA ...
##  $ iqvocs      : int  NA 61 NA NA NA NA 44 53 NA NA ...
##  $ iqbdr       : int  NA 62 NA NA NA NA 28 NA NA NA ...
##  $ iqbds       : int  NA 69 NA NA NA NA 47 NA NA NA ...
##  $ iqsimr      : int  NA 45 NA NA NA NA 35 NA NA NA ...
##  $ iqsims      : int  NA 67 NA NA NA NA 53 NA NA NA ...
##  $ iqmrr       : int  NA 25 NA NA NA NA 22 23 NA NA ...
##  $ iqmrs       : int  NA 59 NA NA NA NA 55 53 NA NA ...
##  $ iq4verbr    : int  NA 128 NA NA NA NA 97 NA NA NA ...
##  $ iq4verbs    : int  NA 123 NA NA NA NA 98 NA NA NA ...
##  $ iq4verbp    : int  NA 94 NA NA NA NA 45 NA NA NA ...
##  $ iq4perfr    : int  NA 128 NA NA NA NA 102 NA NA NA ...
##  $ iq4perfs    : int  NA 125 NA NA NA NA 101 NA NA NA ...
##  $ iq4perfp    : int  NA 95 NA NA NA NA 53 NA NA NA ...
##  $ iq4fullr    : int  NA 256 NA NA NA NA 199 NA NA NA ...
##  $ iq4fulls    : int  NA 127 NA NA NA NA 100 NA NA NA ...
##  $ iq4fullp    : int  NA 96 NA NA NA NA 50 NA NA NA ...
##  $ iq2fullr    : int  NA 120 NA NA NA NA 99 106 NA NA ...
##  $ iq2fulls    : int  NA 118 NA NA NA NA 99 105 NA NA ...
##  $ iq2fullp    : int  NA 88 NA NA NA NA 47 NA NA NA ...
##  $ iqdspf      : int  11 14 11 8 9 8 7 8 7 11 ...
##  $ iqdspb      : int  5 10 9 6 7 7 5 5 7 9 ...
##  $ iqdspr      : int  16 24 20 14 16 15 12 13 14 20 ...
##  $ iqdsps      : int  10 15 13 9 10 9 7 7 9 17 ...
##  $ iqlnsr      : int  10 11 9 8 6 11 9 NA 9 5 ...
##  $ iqlnss      : int  11 12 10 9 6 12 10 NA 10 6 ...
##  $ readraw     : int  NA 57 NA NA NA NA 49 50 NA NA ...
##  $ readstn     : int  NA 119 NA NA NA NA 103 103 NA NA ...
##  $ flucfl      : int  48 54 72 32 41 48 48 45 38 60 ...
##  $ flutot      : int  51 57 78 38 47 52 54 48 41 63 ...
##  $ bntspon     : int  58 58 56 56 54 57 55 NA 56 59 ...
##  $ bntstim     : int  2 2 4 4 5 3 5 NA 4 1 ...
##  $ bntstimc    : int  0 0 3 0 1 3 0 NA 3 0 ...
##  $ bntphon     : int  2 2 1 4 4 0 5 NA 1 1 ...
##  $ bntphoc     : int  1 2 1 1 2 0 2 NA 0 0 ...
##  $ bntrecu     : int  1 0 0 3 2 0 3 NA 1 1 ...
##  $ bntreco     : int  1 0 0 3 1 0 3 NA 1 1 ...
##  $ bnttot      : int  58 58 59 56 55 60 55 NA 59 59 ...
##  $ bnteng      : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ t1raw       : int  5 6 8 6 5 6 8 3 7 9 ...
##  $ t2raw       : int  8 9 10 7 7 12 8 5 11 13 ...
##  $ t3raw       : int  8 9 11 10 9 12 14 8 12 13 ...
##  $ t4raw       : int  10 10 13 12 11 15 13 10 15 13 ...
##  $ t5raw       : int  11 10 11 13 12 14 14 11 13 15 ...
##  $ tTotal      : int  42 44 53 48 44 59 57 37 58 63 ...
##  $ lbraw       : int  7 8 6 6 3 6 3 7 6 5 ...
##  $ t6raw       : int  13 8 13 10 10 14 9 7 13 12 ...
##  $ drraw       : int  10 8 14 8 9 14 8 6 12 13 ...
##  $ recraw      : int  15 12 15 15 13 15 12 13 15 14 ...
##  $ errraw      : int  1 0 0 0 1 0 0 0 1 1 ...
##  $ avltPresRaw : int  1 NA 9 4 2 14 2 5 7 5 ...
##  $ avltIntrRaw : int  0 NA 13 0 6 0 0 1 4 3 ...
##  $ strooc      : int  232 267 306 172 252 244 189 NA 212 290 ...
##  $ strooce     : int  0 0 0 1 0 0 2 NA 0 0 ...
##  $ stroocw     : int  114 NA 119 60 96 117 49 NA 104 NA ...
##  $ stroocwe    : int  0 NA 1 4 1 1 5 NA 1 NA ...
##  $ jolobent    : int  NA 30 NA NA NA NA 17 NA NA NA ...
##  $ joloplus    : int  NA 60 NA NA NA NA 46 NA NA NA ...
##  $ trla        : int  NA 20 23 35 17 23 34 25 30 20 ...
##  $ trlaerr     : int  NA 0 0 0 0 0 0 1 0 1 ...
##  $ trlass      : int  NA 12 11 8 14 11 8 10 9 12 ...
##  $ trlats      : int  NA 60 63 51 75 61 51 53 53 64 ...
##  $ trlb        : int  40 50 47 71 67 44 78 113 85 84 ...
##  $ trlberr     : int  0 0 0 0 1 0 1 1 1 1 ...
##  $ trlbss      : int  14 12 12 9 9 13 9 6 8 8 ...
##  $ trlbts      : int  72 60 67 54 54 69 54 37 46 45 ...
##  $ wmfacir     : int  NA 41 NA NA NA NA 31 NA NA NA ...
##  $ wmfacis     : int  NA 14 NA NA NA NA 7 NA NA NA ...
##  $ wmfacdr     : int  NA 42 NA NA NA NA 32 NA NA NA ...
##  $ wmfacds     : int  NA 16 NA NA NA NA 8 NA NA NA ...
##  $ wcscat      : int  NA NA NA NA NA NA 1 NA NA NA ...
##  $ wcspr       : int  NA NA NA NA NA NA 16 NA NA NA ...
##  $ wcsterr     : int  NA NA NA NA NA NA 32 NA NA NA ...
##  $ wcsbrk      : int  NA NA NA NA NA NA 1 NA NA NA ...
##  $ cdtts       : int  10 10 10 10 10 10 9 10 10 9 ...
##  $ cestyp      : int  2 1 1 1 1 1 1 1 1 1 ...
##  $ cesdts      : int  16 2 2 10 28 3 36 17 0 0 ...
##  $ bvmt1r      : int  7 4 6 3 5 7 7 6 9 8 ...
##  $ bvmt2r      : int  9 9 7 5 8 11 6 8 9 11 ...
##  $ bvmt3r      : int  10 11 8 6 8 10 9 9 11 12 ...
##  $ bvmtTot     : int  26 24 21 14 21 28 22 23 29 31 ...
##  $ bvmtDel     : int  10 11 8 6 8 9 9 8 11 12 ...
##  $ bvmtRec     : int  6 6 4 6 6 6 5 6 6 6 ...
##  $ bvmtFA      : int  0 0 0 1 0 0 0 0 0 0 ...
##  $ mmseTot     : int  30 28 29 29 30 30 30 30 30 30 ...
##  $ wmsrAr      : int  13 NA 16 11 14 17 17 11 14 16 ...
##  $ wmsrBr      : int  16 NA 17 7 8 18 13 13 11 9 ...
##  $ wmsrTot     : int  29 NA 33 18 22 35 30 24 25 25 ...
##  $ wmsrAr2     : int  10 NA 11 10 13 14 11 10 11 16 ...
##  $ wmsrBr2     : int  14 NA 12 6 9 13 10 14 11 9 ...
##  $ wmsrTot2    : int  24 NA 23 16 22 27 21 24 22 25 ...
##  $ waisrTot    : int  60 72 60 40 50 63 41 57 51 63 ...
##  $ waisrSca    : int  13 16 13 11 13 14 9 12 13 17 ...
##  $ animTotRaw  : int  28 NA 18 20 23 25 21 20 31 29 ...
##  $ animTotRep  : int  0 NA 0 0 1 0 0 0 0 0 ...
##  $ animTotErr  : int  0 NA 1 1 1 0 0 0 0 1 ...
##  $ cflTotRep   : int  2 0 1 0 0 1 0 1 0 2 ...
##  $ cflTotErr   : int  0 0 1 2 1 1 1 1 2 0 ...
##  $ LangSampComp: int  1 NA 1 1 1 1 1 1 1 1 ...
##  $ minttot     : int  NA NA NA NA NA NA NA 29 NA NA ...
##  $ mintspon    : int  NA NA NA NA NA NA NA 28 NA NA ...
##  $ mintstim    : int  NA NA NA NA NA NA NA 3 NA NA ...
##  $ mintstimc   : int  NA NA NA NA NA NA NA 1 NA NA ...
##  $ mintphon    : int  NA NA NA NA NA NA NA 3 NA NA ...
##  $ mintphonc   : int  NA NA NA NA NA NA NA 0 NA NA ...
##  $ interv      : int  22 11 14 28 24 14 3 28 14 24 ...
##  $ NoTesting   : num  NA NA NA NA NA NA NA NA NA NA ...
##  $ TestNotComp : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ iqarr       : int  19 20 19 NA NA 12 15 NA 17 NA ...
##  $ iqars       : int  15 16 15 NA NA 9 11 NA 13 NA ...
##  $ iqwmemr     : int  36 43 38 NA NA 30 28 NA 32 NA ...
##  $ iqwmems     : int  111 126 115 NA NA 99 95 NA 104 NA ...
##  $ iqwmemp     : int  77 96 84 NA NA 47 37 NA 61 NA ...
##  $ fluper      : int  88 96 96 NA 82 89 96 NA 63 NA ...
##  $ flueng      : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ lotraw      : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ lass        : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ lotss       : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ lbss        : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ t6ss        : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ drss        : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ recss       : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ stprraw     : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ ltprraw     : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ stprss      : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ ltprss      : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ avlteff     : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ avltdr      : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ avltpr      : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ wmfacpr     : logi  NA NA NA NA NA NA ...
##  $ wmfacps     : logi  NA NA NA NA NA NA ...
##  $ cesd_call   : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ bvmt1p      : chr  &quot;79&quot; &quot;27&quot; &quot;66&quot; &quot;&quot; ...
##  $ bvmt2p      : chr  &quot;69&quot; &quot;73&quot; &quot;34&quot; &quot;&quot; ...
##  $ bvmt3p      : chr  &quot;69&quot; &quot;84&quot; &quot;34&quot; &quot;&quot; ...
##  $ bvmtTotp    : chr  &quot;73&quot; &quot;66&quot; &quot;42&quot; &quot;&quot; ...
##  $ bvmtDelp    : chr  &quot;76&quot; &quot;88&quot; &quot;42&quot; &quot;&quot; ...
##  $ bvmtRecp    : chr  &quot;&gt;16&quot; &quot;&gt;16&quot; &quot;6-10&quot; &quot;&quot; ...
##  $ bvmtFAp     : chr  &quot;&gt;16&quot; &quot;&gt;16&quot; &quot;&gt;16&quot; &quot;&quot; ...
##  $ wmsrp       : int  83 NA 92 NA 54 96 86 NA 70 NA ...
##  $ wmsrp2      : int  79 NA 76 NA 75 87 68 NA 75 NA ...
##  $ animSemMean : num  2.22 NA 1 NA 1 1.78 1.63 NA 1.38 NA ...
##  $ animSemSwi  : int  8 NA 8 NA 11 8 7 NA 12 NA ...
##  $ animPhoMean : logi  NA NA NA NA NA NA ...
##  $ animPhoSwi  : logi  NA NA NA NA NA NA ...
##  $ cflTotRaw   : logi  NA NA NA NA NA NA ...
##  $ cflSemMean  : logi  NA NA NA NA NA NA ...
##  $ cflSemSwi   : logi  NA NA NA NA NA NA ...
##  $ cflPhoMean  : num  0.45 0.79 0.85 NA 0.32 0.45 0.42 NA 0.81 NA ...
##  $ cflPhoSwi   : int  32 28 38 NA 30 32 32 NA 20 NA ...
##  $ StroopC_Cor : int  232 267 306 171 252 244 187 NA 212 290 ...
##  $ StroopCW_Cor: int  114 NA 118 56 95 116 44 NA 103 NA ...
##  $ update_stamp: chr  &quot;2024-09-13 08:53:45.0&quot; &quot;2024-09-13 08:53:45.0&quot; &quot;2025-01-24 08:18:29.0&quot; &quot;2024-09-13 08:53:45.0&quot; ...</code></pre>
<pre class="r"><code>## convert &quot;&quot; to NA for chr columns
chrcols &lt;- colnames(NeuropsychScores)[sapply(NeuropsychScores, is.character)]
NeuropsychScores[chrcols] &lt;- lapply(NeuropsychScores[chrcols], function(x) ifelse(x == &quot;&quot;, NA, x))

## convert from logi to int for columns: wmfacpr wmfacps animPhoMean animPhoSwi cflTotRaw cflSemMean cflSemSwi
## these columns were confirmed to only have NA values in the raw data
logicols &lt;- c(&quot;wmfacpr&quot;,&quot;wmfacps&quot;,&quot;animPhoMean&quot;,&quot;animPhoSwi&quot;,&quot;cflTotRaw&quot;,&quot;cflSemMean&quot;,&quot;cflSemSwi&quot;)
NeuropsychScores[logicols] &lt;- lapply(NeuropsychScores[logicols], as.integer)

## drop update_stamp column
NeuropsychScores &lt;- subset(NeuropsychScores,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(NeuropsychScores, max.level = 99, list.len = 99999)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    7996 obs. of  156 variables:
##  $ wrapnum     : chr  &quot;wrap0943&quot; &quot;wrap0068&quot; &quot;wrap0759&quot; &quot;wrapA0078&quot; ...
##  $ WRAPNo      : chr  &quot;0943&quot; &quot;0068&quot; &quot;0759&quot; &quot;A0078&quot; ...
##  $ VisNo       : int  3 2 3 4 5 4 2 1 3 6 ...
##  $ curage      : int  60 64 64 65 65 56 64 52 68 70 ...
##  $ iqvocr      : int  NA 68 NA NA NA NA 50 62 NA NA ...
##  $ iqvocs      : int  NA 61 NA NA NA NA 44 53 NA NA ...
##  $ iqbdr       : int  NA 62 NA NA NA NA 28 NA NA NA ...
##  $ iqbds       : int  NA 69 NA NA NA NA 47 NA NA NA ...
##  $ iqsimr      : int  NA 45 NA NA NA NA 35 NA NA NA ...
##  $ iqsims      : int  NA 67 NA NA NA NA 53 NA NA NA ...
##  $ iqmrr       : int  NA 25 NA NA NA NA 22 23 NA NA ...
##  $ iqmrs       : int  NA 59 NA NA NA NA 55 53 NA NA ...
##  $ iq4verbr    : int  NA 128 NA NA NA NA 97 NA NA NA ...
##  $ iq4verbs    : int  NA 123 NA NA NA NA 98 NA NA NA ...
##  $ iq4verbp    : int  NA 94 NA NA NA NA 45 NA NA NA ...
##  $ iq4perfr    : int  NA 128 NA NA NA NA 102 NA NA NA ...
##  $ iq4perfs    : int  NA 125 NA NA NA NA 101 NA NA NA ...
##  $ iq4perfp    : int  NA 95 NA NA NA NA 53 NA NA NA ...
##  $ iq4fullr    : int  NA 256 NA NA NA NA 199 NA NA NA ...
##  $ iq4fulls    : int  NA 127 NA NA NA NA 100 NA NA NA ...
##  $ iq4fullp    : int  NA 96 NA NA NA NA 50 NA NA NA ...
##  $ iq2fullr    : int  NA 120 NA NA NA NA 99 106 NA NA ...
##  $ iq2fulls    : int  NA 118 NA NA NA NA 99 105 NA NA ...
##  $ iq2fullp    : int  NA 88 NA NA NA NA 47 NA NA NA ...
##  $ iqdspf      : int  11 14 11 8 9 8 7 8 7 11 ...
##  $ iqdspb      : int  5 10 9 6 7 7 5 5 7 9 ...
##  $ iqdspr      : int  16 24 20 14 16 15 12 13 14 20 ...
##  $ iqdsps      : int  10 15 13 9 10 9 7 7 9 17 ...
##  $ iqlnsr      : int  10 11 9 8 6 11 9 NA 9 5 ...
##  $ iqlnss      : int  11 12 10 9 6 12 10 NA 10 6 ...
##  $ readraw     : int  NA 57 NA NA NA NA 49 50 NA NA ...
##  $ readstn     : int  NA 119 NA NA NA NA 103 103 NA NA ...
##  $ flucfl      : int  48 54 72 32 41 48 48 45 38 60 ...
##  $ flutot      : int  51 57 78 38 47 52 54 48 41 63 ...
##  $ bntspon     : int  58 58 56 56 54 57 55 NA 56 59 ...
##  $ bntstim     : int  2 2 4 4 5 3 5 NA 4 1 ...
##  $ bntstimc    : int  0 0 3 0 1 3 0 NA 3 0 ...
##  $ bntphon     : int  2 2 1 4 4 0 5 NA 1 1 ...
##  $ bntphoc     : int  1 2 1 1 2 0 2 NA 0 0 ...
##  $ bntrecu     : int  1 0 0 3 2 0 3 NA 1 1 ...
##  $ bntreco     : int  1 0 0 3 1 0 3 NA 1 1 ...
##  $ bnttot      : int  58 58 59 56 55 60 55 NA 59 59 ...
##  $ bnteng      : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ t1raw       : int  5 6 8 6 5 6 8 3 7 9 ...
##  $ t2raw       : int  8 9 10 7 7 12 8 5 11 13 ...
##  $ t3raw       : int  8 9 11 10 9 12 14 8 12 13 ...
##  $ t4raw       : int  10 10 13 12 11 15 13 10 15 13 ...
##  $ t5raw       : int  11 10 11 13 12 14 14 11 13 15 ...
##  $ tTotal      : int  42 44 53 48 44 59 57 37 58 63 ...
##  $ lbraw       : int  7 8 6 6 3 6 3 7 6 5 ...
##  $ t6raw       : int  13 8 13 10 10 14 9 7 13 12 ...
##  $ drraw       : int  10 8 14 8 9 14 8 6 12 13 ...
##  $ recraw      : int  15 12 15 15 13 15 12 13 15 14 ...
##  $ errraw      : int  1 0 0 0 1 0 0 0 1 1 ...
##  $ avltPresRaw : int  1 NA 9 4 2 14 2 5 7 5 ...
##  $ avltIntrRaw : int  0 NA 13 0 6 0 0 1 4 3 ...
##  $ strooc      : int  232 267 306 172 252 244 189 NA 212 290 ...
##  $ strooce     : int  0 0 0 1 0 0 2 NA 0 0 ...
##  $ stroocw     : int  114 NA 119 60 96 117 49 NA 104 NA ...
##  $ stroocwe    : int  0 NA 1 4 1 1 5 NA 1 NA ...
##  $ jolobent    : int  NA 30 NA NA NA NA 17 NA NA NA ...
##  $ joloplus    : int  NA 60 NA NA NA NA 46 NA NA NA ...
##  $ trla        : int  NA 20 23 35 17 23 34 25 30 20 ...
##  $ trlaerr     : int  NA 0 0 0 0 0 0 1 0 1 ...
##  $ trlass      : int  NA 12 11 8 14 11 8 10 9 12 ...
##  $ trlats      : int  NA 60 63 51 75 61 51 53 53 64 ...
##  $ trlb        : int  40 50 47 71 67 44 78 113 85 84 ...
##  $ trlberr     : int  0 0 0 0 1 0 1 1 1 1 ...
##  $ trlbss      : int  14 12 12 9 9 13 9 6 8 8 ...
##  $ trlbts      : int  72 60 67 54 54 69 54 37 46 45 ...
##  $ wmfacir     : int  NA 41 NA NA NA NA 31 NA NA NA ...
##  $ wmfacis     : int  NA 14 NA NA NA NA 7 NA NA NA ...
##  $ wmfacdr     : int  NA 42 NA NA NA NA 32 NA NA NA ...
##  $ wmfacds     : int  NA 16 NA NA NA NA 8 NA NA NA ...
##  $ wcscat      : int  NA NA NA NA NA NA 1 NA NA NA ...
##  $ wcspr       : int  NA NA NA NA NA NA 16 NA NA NA ...
##  $ wcsterr     : int  NA NA NA NA NA NA 32 NA NA NA ...
##  $ wcsbrk      : int  NA NA NA NA NA NA 1 NA NA NA ...
##  $ cdtts       : int  10 10 10 10 10 10 9 10 10 9 ...
##  $ cestyp      : int  2 1 1 1 1 1 1 1 1 1 ...
##  $ cesdts      : int  16 2 2 10 28 3 36 17 0 0 ...
##  $ bvmt1r      : int  7 4 6 3 5 7 7 6 9 8 ...
##  $ bvmt2r      : int  9 9 7 5 8 11 6 8 9 11 ...
##  $ bvmt3r      : int  10 11 8 6 8 10 9 9 11 12 ...
##  $ bvmtTot     : int  26 24 21 14 21 28 22 23 29 31 ...
##  $ bvmtDel     : int  10 11 8 6 8 9 9 8 11 12 ...
##  $ bvmtRec     : int  6 6 4 6 6 6 5 6 6 6 ...
##  $ bvmtFA      : int  0 0 0 1 0 0 0 0 0 0 ...
##  $ mmseTot     : int  30 28 29 29 30 30 30 30 30 30 ...
##  $ wmsrAr      : int  13 NA 16 11 14 17 17 11 14 16 ...
##  $ wmsrBr      : int  16 NA 17 7 8 18 13 13 11 9 ...
##  $ wmsrTot     : int  29 NA 33 18 22 35 30 24 25 25 ...
##  $ wmsrAr2     : int  10 NA 11 10 13 14 11 10 11 16 ...
##  $ wmsrBr2     : int  14 NA 12 6 9 13 10 14 11 9 ...
##  $ wmsrTot2    : int  24 NA 23 16 22 27 21 24 22 25 ...
##  $ waisrTot    : int  60 72 60 40 50 63 41 57 51 63 ...
##  $ waisrSca    : int  13 16 13 11 13 14 9 12 13 17 ...
##  $ animTotRaw  : int  28 NA 18 20 23 25 21 20 31 29 ...
##  $ animTotRep  : int  0 NA 0 0 1 0 0 0 0 0 ...
##  $ animTotErr  : int  0 NA 1 1 1 0 0 0 0 1 ...
##  $ cflTotRep   : int  2 0 1 0 0 1 0 1 0 2 ...
##  $ cflTotErr   : int  0 0 1 2 1 1 1 1 2 0 ...
##  $ LangSampComp: int  1 NA 1 1 1 1 1 1 1 1 ...
##  $ minttot     : int  NA NA NA NA NA NA NA 29 NA NA ...
##  $ mintspon    : int  NA NA NA NA NA NA NA 28 NA NA ...
##  $ mintstim    : int  NA NA NA NA NA NA NA 3 NA NA ...
##  $ mintstimc   : int  NA NA NA NA NA NA NA 1 NA NA ...
##  $ mintphon    : int  NA NA NA NA NA NA NA 3 NA NA ...
##  $ mintphonc   : int  NA NA NA NA NA NA NA 0 NA NA ...
##  $ interv      : int  22 11 14 28 24 14 3 28 14 24 ...
##  $ NoTesting   : num  NA NA NA NA NA NA NA NA NA NA ...
##  $ TestNotComp : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ iqarr       : int  19 20 19 NA NA 12 15 NA 17 NA ...
##  $ iqars       : int  15 16 15 NA NA 9 11 NA 13 NA ...
##  $ iqwmemr     : int  36 43 38 NA NA 30 28 NA 32 NA ...
##  $ iqwmems     : int  111 126 115 NA NA 99 95 NA 104 NA ...
##  $ iqwmemp     : int  77 96 84 NA NA 47 37 NA 61 NA ...
##  $ fluper      : int  88 96 96 NA 82 89 96 NA 63 NA ...
##  $ flueng      : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ lotraw      : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ lass        : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ lotss       : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ lbss        : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ t6ss        : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ drss        : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ recss       : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ stprraw     : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ ltprraw     : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ stprss      : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ ltprss      : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ avlteff     : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ avltdr      : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ avltpr      : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ wmfacpr     : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ wmfacps     : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ cesd_call   : chr  NA NA NA NA ...
##  $ bvmt1p      : chr  &quot;79&quot; &quot;27&quot; &quot;66&quot; NA ...
##  $ bvmt2p      : chr  &quot;69&quot; &quot;73&quot; &quot;34&quot; NA ...
##  $ bvmt3p      : chr  &quot;69&quot; &quot;84&quot; &quot;34&quot; NA ...
##  $ bvmtTotp    : chr  &quot;73&quot; &quot;66&quot; &quot;42&quot; NA ...
##  $ bvmtDelp    : chr  &quot;76&quot; &quot;88&quot; &quot;42&quot; NA ...
##  $ bvmtRecp    : chr  &quot;&gt;16&quot; &quot;&gt;16&quot; &quot;6-10&quot; NA ...
##  $ bvmtFAp     : chr  &quot;&gt;16&quot; &quot;&gt;16&quot; &quot;&gt;16&quot; NA ...
##  $ wmsrp       : int  83 NA 92 NA 54 96 86 NA 70 NA ...
##  $ wmsrp2      : int  79 NA 76 NA 75 87 68 NA 75 NA ...
##  $ animSemMean : num  2.22 NA 1 NA 1 1.78 1.63 NA 1.38 NA ...
##  $ animSemSwi  : int  8 NA 8 NA 11 8 7 NA 12 NA ...
##  $ animPhoMean : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ animPhoSwi  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ cflTotRaw   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ cflSemMean  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ cflSemSwi   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ cflPhoMean  : num  0.45 0.79 0.85 NA 0.32 0.45 0.42 NA 0.81 NA ...
##  $ cflPhoSwi   : int  32 28 38 NA 30 32 32 NA 20 NA ...
##  $ StroopC_Cor : int  232 267 306 171 252 244 187 NA 212 290 ...
##  $ StroopCW_Cor: int  114 NA 118 56 95 116 44 NA 103 NA ...</code></pre>
<p><br> <br></p>
</div>
<div id="persfammedback" class="section level2">
<h2>PersFamMedBack</h2>
<p>Personal Family Medical Background</p>
<pre class="r"><code>## load data
PersFamMedBack &lt;- read.csv(paste0(wrap_directory, &quot;PersFamMedBack_05Jun2025.csv&quot;))
info(PersFamMedBack,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:8176, cols:50, inds:1802</code></pre>
<pre class="r"><code>str(PersFamMedBack)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    8176 obs. of  50 variables:
##  $ wrapnum               : chr  &quot;wrap0002&quot; &quot;wrap0068&quot; &quot;wrap0146&quot; &quot;wrap0148&quot; ...
##  $ WRAPNo                : chr  &quot;0002&quot; &quot;0068&quot; &quot;0146&quot; &quot;0148&quot; ...
##  $ VisNo                 : int  5 2 4 5 5 4 6 4 7 4 ...
##  $ CurJob                : int  2 4 1 4 4 4 5 5 4 2 ...
##  $ RetireAge             : int  68 NA NA NA NA NA NA NA NA 58 ...
##  $ RetireAge_Over89      : chr  &quot;No&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ ChangeJob             : int  NA NA NA 2 2 2 2 2 2 NA ...
##  $ Volunt                : int  1 NA 1 NA NA NA NA NA NA 2 ...
##  $ VolHrs                : int  65 NA 15 NA NA NA NA NA NA NA ...
##  $ MarStat               : int  4 1 1 1 1 1 6 1 1 1 ...
##  $ AddSchool             : int  2 NA 2 2 2 2 2 2 2 2 ...
##  $ Income                : int  4 5 6 NA 7 6 7 NA 7 7 ...
##  $ FH1                   : int  2 NA 2 2 2 2 2 2 2 2 ...
##  $ FH1DK                 : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ FH1MD                 : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ FH1AD_Mo              : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ FH2                   : int  3 1 3 3 3 3 1 3 3 3 ...
##  $ FH1AD_Fa              : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ FH1_MO_LewyBodyDem_dx : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ FH1_FA_LewyBodyDem_dx : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ FH1_MO_Parkinsons_dx  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ FH1_FA_Parkinsons_dx  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ FH1_MO_Parkinsonism_dx: int  NA NA NA NA NA NA NA NA NA NA ...
##  $ FH1_FA_Parkinsonism_dx: int  NA NA NA NA NA NA NA NA NA NA ...
##  $ FHMoLiving            : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ FH3                   : int  3 1 3 3 3 3 3 3 1 3 ...
##  $ FHFaLiving            : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ FH4MP                 : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ FH4MS                 : int  2 2 2 2 2 2 2 1 2 2 ...
##  $ FH4MB                 : int  2 2 2 2 2 1 2 2 2 2 ...
##  $ FH5FP                 : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ FH5FS                 : int  2 2 2 2 2 1 2 2 2 2 ...
##  $ FH5FB                 : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ HR60M                 : int  2 NA 2 2 2 2 2 2 2 2 ...
##  $ HR60F                 : int  2 NA 2 2 2 2 2 2 2 2 ...
##  $ FH7                   : int  2 2 2 2 2 2 1 2 2 2 ...
##  $ FH8                   : int  NA 2 NA NA NA NA NA NA NA NA ...
##  $ modement_W1           : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ fadement_W1           : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ gender                : int  2 1 2 2 2 2 1 2 1 2 ...
##  $ gender_other          : logi  NA NA NA NA NA NA ...
##  $ gender_identity       : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ gender_identity_other : logi  NA NA NA NA NA NA ...
##  $ prefix                : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ employ                : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ ses                   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ jobyrs1               : chr  &quot;&quot; &quot;38&quot; &quot;&quot; &quot;&quot; ...
##  $ jobyrs2               : num  NA 3 NA NA NA NA NA NA NA NA ...
##  $ jobyrs3               : num  NA 3 NA NA NA NA NA NA NA NA ...
##  $ update_stamp          : chr  &quot;2024-09-13 08:53:50.0&quot; &quot;2025-01-24 08:18:33.0&quot; &quot;2024-09-13 08:53:50.0&quot; &quot;2024-09-13 08:53:50.0&quot; ...</code></pre>
<pre class="r"><code>## check table frequency for all variables

## for column PersFamMedBack, in DD it got values:1=Yes;0=No
## but in the dataset, it has value:
# table(PersFamMedBack$Volunt,useNA = &quot;ifany&quot;)
# 
#    0    1    2 &lt;NA&gt; 
#  114 1935 1530 4597
## by comparing with how other value being coded, I decide to convert 2 to 0
PersFamMedBack$Volunt[PersFamMedBack$Volunt==2] &lt;- 0

## convert &quot;&quot; to NA for chr columns
chrcols &lt;- colnames(PersFamMedBack)[sapply(PersFamMedBack, is.character)]
PersFamMedBack[chrcols] &lt;- lapply(PersFamMedBack[chrcols], function(x) ifelse(x == &quot;&quot;, NA, x))

## convert logi to chr for column: gender_other,gender_identity_other
logicols &lt;- c(&quot;gender_other&quot;,&quot;gender_identity_other&quot;)
PersFamMedBack[logicols] &lt;- lapply(PersFamMedBack[logicols], as.character)

## drop update_stamp column
PersFamMedBack &lt;- subset(PersFamMedBack,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(PersFamMedBack)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    8176 obs. of  49 variables:
##  $ wrapnum               : chr  &quot;wrap0002&quot; &quot;wrap0068&quot; &quot;wrap0146&quot; &quot;wrap0148&quot; ...
##  $ WRAPNo                : chr  &quot;0002&quot; &quot;0068&quot; &quot;0146&quot; &quot;0148&quot; ...
##  $ VisNo                 : int  5 2 4 5 5 4 6 4 7 4 ...
##  $ CurJob                : int  2 4 1 4 4 4 5 5 4 2 ...
##  $ RetireAge             : int  68 NA NA NA NA NA NA NA NA 58 ...
##  $ RetireAge_Over89      : chr  &quot;No&quot; NA NA NA ...
##  $ ChangeJob             : int  NA NA NA 2 2 2 2 2 2 NA ...
##  $ Volunt                : num  1 NA 1 NA NA NA NA NA NA 0 ...
##  $ VolHrs                : int  65 NA 15 NA NA NA NA NA NA NA ...
##  $ MarStat               : int  4 1 1 1 1 1 6 1 1 1 ...
##  $ AddSchool             : int  2 NA 2 2 2 2 2 2 2 2 ...
##  $ Income                : int  4 5 6 NA 7 6 7 NA 7 7 ...
##  $ FH1                   : int  2 NA 2 2 2 2 2 2 2 2 ...
##  $ FH1DK                 : chr  NA NA NA NA ...
##  $ FH1MD                 : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ FH1AD_Mo              : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ FH2                   : int  3 1 3 3 3 3 1 3 3 3 ...
##  $ FH1AD_Fa              : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ FH1_MO_LewyBodyDem_dx : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ FH1_FA_LewyBodyDem_dx : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ FH1_MO_Parkinsons_dx  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ FH1_FA_Parkinsons_dx  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ FH1_MO_Parkinsonism_dx: int  NA NA NA NA NA NA NA NA NA NA ...
##  $ FH1_FA_Parkinsonism_dx: int  NA NA NA NA NA NA NA NA NA NA ...
##  $ FHMoLiving            : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ FH3                   : int  3 1 3 3 3 3 3 3 1 3 ...
##  $ FHFaLiving            : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ FH4MP                 : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ FH4MS                 : int  2 2 2 2 2 2 2 1 2 2 ...
##  $ FH4MB                 : int  2 2 2 2 2 1 2 2 2 2 ...
##  $ FH5FP                 : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ FH5FS                 : int  2 2 2 2 2 1 2 2 2 2 ...
##  $ FH5FB                 : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ HR60M                 : int  2 NA 2 2 2 2 2 2 2 2 ...
##  $ HR60F                 : int  2 NA 2 2 2 2 2 2 2 2 ...
##  $ FH7                   : int  2 2 2 2 2 2 1 2 2 2 ...
##  $ FH8                   : int  NA 2 NA NA NA NA NA NA NA NA ...
##  $ modement_W1           : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ fadement_W1           : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ gender                : int  2 1 2 2 2 2 1 2 1 2 ...
##  $ gender_other          : chr  NA NA NA NA ...
##  $ gender_identity       : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ gender_identity_other : chr  NA NA NA NA ...
##  $ prefix                : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ employ                : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ ses                   : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ jobyrs1               : chr  NA &quot;38&quot; NA NA ...
##  $ jobyrs2               : num  NA 3 NA NA NA NA NA NA NA NA ...
##  $ jobyrs3               : num  NA 3 NA NA NA NA NA NA NA NA ...</code></pre>
<p><br> <br></p>
</div>
<div id="pfmb_sibwad_or_othermemissue" class="section level2">
<h2>PFMB_SibWAD_or_OtherMemIssue</h2>
<p>Personal Family Medical Background - Siblings with AD or Other Memory
Issue</p>
<pre class="r"><code>## load data
PFMB_SibWAD_or_OtherMemIssue &lt;- read.csv(paste0(wrap_directory, &quot;PFMB_SibWAD_or_OtherMemIssue_05Jun2025.csv&quot;))
info(PFMB_SibWAD_or_OtherMemIssue,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:297, cols:6, inds:175</code></pre>
<pre class="r"><code>str(PFMB_SibWAD_or_OtherMemIssue)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    297 obs. of  6 variables:
##  $ wrapnum                                  : chr  &quot;wrap0401&quot; &quot;wrap1058&quot; &quot;wrap1081&quot; &quot;wrap0157&quot; ...
##  $ WRAPNo                                   : chr  &quot;0401&quot; &quot;1058&quot; &quot;1081&quot; &quot;0157&quot; ...
##  $ VisNo                                    : int  6 3 1 2 7 6 3 1 7 4 ...
##  $ FH7SiblingWithAD_OtherMemIssue_Repeat_Key: int  1 1 1 1 1 1 1 1 1 1 ...
##  $ FH7BS1                                   : chr  &quot;brother&quot; &quot;brother&quot; &quot;brother&quot; &quot;brother&quot; ...
##  $ update_stamp                             : chr  &quot;2024-09-13 08:53:51.0&quot; &quot;2024-09-13 08:53:51.0&quot; &quot;2024-09-13 08:53:51.0&quot; &quot;2024-09-13 08:53:51.0&quot; ...</code></pre>
<pre class="r"><code>## check table frequency for all variables

## convert &quot;&quot; to NA for chr columns
chrcols &lt;- colnames(PFMB_SibWAD_or_OtherMemIssue)[sapply(PFMB_SibWAD_or_OtherMemIssue, is.character)]
PFMB_SibWAD_or_OtherMemIssue[chrcols] &lt;- lapply(PFMB_SibWAD_or_OtherMemIssue[chrcols], function(x) ifelse(x == &quot;&quot;, NA, x))

## drop update_stamp column
PFMB_SibWAD_or_OtherMemIssue &lt;- subset(PFMB_SibWAD_or_OtherMemIssue,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(PFMB_SibWAD_or_OtherMemIssue)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    297 obs. of  5 variables:
##  $ wrapnum                                  : chr  &quot;wrap0401&quot; &quot;wrap1058&quot; &quot;wrap1081&quot; &quot;wrap0157&quot; ...
##  $ WRAPNo                                   : chr  &quot;0401&quot; &quot;1058&quot; &quot;1081&quot; &quot;0157&quot; ...
##  $ VisNo                                    : int  6 3 1 2 7 6 3 1 7 4 ...
##  $ FH7SiblingWithAD_OtherMemIssue_Repeat_Key: int  1 1 1 1 1 1 1 1 1 1 ...
##  $ FH7BS1                                   : chr  &quot;brother&quot; &quot;brother&quot; &quot;brother&quot; &quot;brother&quot; ...</code></pre>
<p><br> <br></p>
</div>
<div id="quickdementia" class="section level2">
<h2>QuickDementia</h2>
<p>Quick Dementia Rating Scale</p>
<pre class="r"><code>## load data
QuickDementia &lt;- read.csv(paste0(wrap_directory, &quot;QuickDementia_05Jun2025.csv&quot;))
info(QuickDementia,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:3006, cols:23, inds:1303</code></pre>
<pre class="r"><code>str(QuickDementia)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    3006 obs. of  23 variables:
##  $ wrapnum                                    : chr  &quot;wrap0148&quot; &quot;wrap0305&quot; &quot;wrap0401&quot; &quot;wrap0434&quot; ...
##  $ WRAPNo                                     : chr  &quot;0148&quot; &quot;0305&quot; &quot;0401&quot; &quot;0434&quot; ...
##  $ VisNo                                      : int  5 5 6 7 8 6 5 6 3 1 ...
##  $ QD_MemRecall                               : num  0 0 0 0 0 0 0 0 0.5 0 ...
##  $ QD_Orientation                             : num  0 0 0 0 0 0 0 0 0.5 0 ...
##  $ QD_DMPSA                                   : num  0 0 0 0 0 0 0 0 0 0 ...
##  $ QD_ActOutHome                              : num  0 0 0 0 0 0 0 0 0 0 ...
##  $ QD_FuncHome                                : num  0 0 0 0 0 0 0 0 0 0 ...
##  $ QD_Hygeine                                 : num  0 0 0 0 0 0 0 0 0 0 ...
##  $ QD_Behavior                                : num  0 0 0 0 0 0 0 0.5 0 0 ...
##  $ QD_Language                                : num  0 0 NA 0 0 0 0 0 0 0 ...
##  $ QD_Mood                                    : num  0 0 NA 0 0 0 0.5 0 0 0 ...
##  $ QD_AttConc                                 : num  0 0 NA 0 0 0 0 0.5 0 0 ...
##  $ QD_CogSubTotal                             : num  0 0 NA 0 0 0 0 0 1 0 ...
##  $ QD_BehSubTotal                             : num  0 0 NA 0 0 0 0.5 1 0 0 ...
##  $ QD_TotalScore                              : num  0 0 NA 0 0 0 0.5 1 1 0 ...
##  $ QD_SumOfBoxes                              : num  0 0 0 0 0 0 0 0 1 0 ...
##  $ Returned_Days_From_Visit                   : int  12 113 28 105 48 13 28 24 56 25 ...
##  $ estimated_questionnaire_days_after_baseline: int  4765 4633 5131 6204 6409 4742 3765 4887 1887 11 ...
##  $ RelPPT                                     : int  1 2 6 NA 4 1 6 4 7 1 ...
##  $ RelLive                                    : int  1 NA NA NA NA NA NA NA NA NA ...
##  $ QDRS_Rating                                : num  0 0 0 0 0 0 0 0 0.5 0 ...
##  $ update_stamp                               : chr  &quot;2025-01-24 08:18:35.0&quot; &quot;2025-01-24 08:18:35.0&quot; &quot;2025-01-24 08:18:35.0&quot; &quot;2025-01-24 08:18:35.0&quot; ...</code></pre>
<pre class="r"><code>## check table frequency for all variables

## DD has 1=yes,2=no coding but data has 0/1 coding. 
## Confirmed with other 0/1 variables that 0 is no and 1 stays yes, so will update the dataset to match the DD
QuickDementia$RelLive[QuickDementia$RelLive==0] &lt;- 2
QuickDementia$RelLive &lt;- as.integer(QuickDementia$RelLive)

## drop update_stamp column
QuickDementia &lt;- subset(QuickDementia,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(QuickDementia)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    3006 obs. of  22 variables:
##  $ wrapnum                                    : chr  &quot;wrap0148&quot; &quot;wrap0305&quot; &quot;wrap0401&quot; &quot;wrap0434&quot; ...
##  $ WRAPNo                                     : chr  &quot;0148&quot; &quot;0305&quot; &quot;0401&quot; &quot;0434&quot; ...
##  $ VisNo                                      : int  5 5 6 7 8 6 5 6 3 1 ...
##  $ QD_MemRecall                               : num  0 0 0 0 0 0 0 0 0.5 0 ...
##  $ QD_Orientation                             : num  0 0 0 0 0 0 0 0 0.5 0 ...
##  $ QD_DMPSA                                   : num  0 0 0 0 0 0 0 0 0 0 ...
##  $ QD_ActOutHome                              : num  0 0 0 0 0 0 0 0 0 0 ...
##  $ QD_FuncHome                                : num  0 0 0 0 0 0 0 0 0 0 ...
##  $ QD_Hygeine                                 : num  0 0 0 0 0 0 0 0 0 0 ...
##  $ QD_Behavior                                : num  0 0 0 0 0 0 0 0.5 0 0 ...
##  $ QD_Language                                : num  0 0 NA 0 0 0 0 0 0 0 ...
##  $ QD_Mood                                    : num  0 0 NA 0 0 0 0.5 0 0 0 ...
##  $ QD_AttConc                                 : num  0 0 NA 0 0 0 0 0.5 0 0 ...
##  $ QD_CogSubTotal                             : num  0 0 NA 0 0 0 0 0 1 0 ...
##  $ QD_BehSubTotal                             : num  0 0 NA 0 0 0 0.5 1 0 0 ...
##  $ QD_TotalScore                              : num  0 0 NA 0 0 0 0.5 1 1 0 ...
##  $ QD_SumOfBoxes                              : num  0 0 0 0 0 0 0 0 1 0 ...
##  $ Returned_Days_From_Visit                   : int  12 113 28 105 48 13 28 24 56 25 ...
##  $ estimated_questionnaire_days_after_baseline: int  4765 4633 5131 6204 6409 4742 3765 4887 1887 11 ...
##  $ RelPPT                                     : int  1 2 6 NA 4 1 6 4 7 1 ...
##  $ RelLive                                    : int  1 NA NA NA NA NA NA NA NA NA ...
##  $ QDRS_Rating                                : num  0 0 0 0 0 0 0 0 0.5 0 ...</code></pre>
<p><br> <br></p>
</div>
<div id="robustnormzscores" class="section level2">
<h2>RobustNormZScores</h2>
<p>Demographically-adjusted robust norm z scores (see Koscik et
al. 2014, Clark et al. 2016)</p>
<pre class="r"><code>## load data
RobustNormZScores &lt;- read.csv(paste0(wrap_directory, &quot;RobustNormZScores_05Jun2025.csv&quot;))
info(RobustNormZScores,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:7996, cols:22, inds:1799</code></pre>
<pre class="r"><code>str(RobustNormZScores)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    7996 obs. of  22 variables:
##  $ wrapnum                                                        : chr  &quot;wrapL0308&quot; &quot;wrap0401&quot; &quot;wrap0943&quot; &quot;wrap0002&quot; ...
##  $ WRAPNo                                                         : chr  &quot;L0308&quot; &quot;0401&quot; &quot;0943&quot; &quot;0002&quot; ...
##  $ VisNo                                                          : int  2 6 3 5 5 1 4 3 2 3 ...
##  $ RNZ_WAISIIIDigitSpanForward_iqdspf                             : num  2.1219 -0.9091 -0.436 0.0109 0.0794 ...
##  $ RNZ_WAISIIIDigitSpanBackward_iqdspb                            : num  1.429 -0.694 -1.776 -0.741 1.778 ...
##  $ RNZ_WAISIIILetterNumberSequenceRawScore_iqlnsr                 : num  0.408 -0.686 -0.872 -1.447 1.324 ...
##  $ RNZ_AVLTRawTrialTotal_tTotal                                   : num  0.266 -1.042 -1.489 -0.537 0.571 ...
##  $ RNZ_AVLTDelayedRecallRaw_drraw                                 : num  -1.2808 0.0291 -0.1409 0.073 0.0756 ...
##  $ RNZ_LetterSemanticFluencyCFLRaw_flucfl                         : num  0.75068 0.57745 0.00136 0.72564 -0.12097 ...
##  $ RNZ_STROOPNumberNamedForColorWord_stroocw                      : num  -1.504 1.835 0.184 0.594 -0.605 ...
##  $ RNZ_TrailMakingTestTrailATime_trla                             : num  1.406 1.148 NA 1.51 -0.201 ...
##  $ RNZ_TrailMakingTestTrailBTime_trlb                             : num  0.876 0.747 1.231 2.158 -0.273 ...
##  $ RNZ_ClockDrawCDTTotalScore_cdtts                               : num  0.932 0.657 0.364 0.936 0.703 ...
##  $ RNZ_WAISRDigitSymbolTotalRaw_waisrTot                          : num  -1.067 -0.251 0.549 1.847 -0.48 ...
##  $ RNZ_AnimalTestingTotalRaw_animTotRaw                           : num  -0.313 -1.337 0.659 1.688 0.672 ...
##  $ RNZ_BVMTRTotalRaw_bvmtTot                                      : num  0.1781 -0.0415 0.4426 0.8698 0.1319 ...
##  $ RNZ_BVMTRDelayedRaw_bvmtDel                                    : num  0.042 -0.482 0.24 1.172 1.458 ...
##  $ RNZ_MMSETotalScore_mmseTot                                     : num  -2.349 1.065 0.584 1.209 0.863 ...
##  $ RNZ_WMSRLogicalMemoryITotalScore_wmsrTot                       : num  -0.699 -0.938 -0.382 -1.011 -0.685 ...
##  $ RNZ_WMSRLogicalMemoryIITotalScore_wmsrTot2                     : num  -1.9402 -1.8603 -0.654 -0.4535 -0.0814 ...
##  $ RNZ_BostonNamingTestTotal_bnttot_MultiLingualNamingTest_minttot: num  -0.865 -2.106 -0.845 0.98 0.414 ...
##  $ update_stamp                                                   : chr  &quot;2025-01-24 08:18:38.0&quot; &quot;2025-01-24 08:18:38.0&quot; &quot;2025-01-24 08:18:38.0&quot; &quot;2025-01-24 08:18:38.0&quot; ...</code></pre>
<pre class="r"><code>## check table frequency for all variables

## drop update_stamp column
RobustNormZScores &lt;- subset(RobustNormZScores,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(RobustNormZScores)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    7996 obs. of  21 variables:
##  $ wrapnum                                                        : chr  &quot;wrapL0308&quot; &quot;wrap0401&quot; &quot;wrap0943&quot; &quot;wrap0002&quot; ...
##  $ WRAPNo                                                         : chr  &quot;L0308&quot; &quot;0401&quot; &quot;0943&quot; &quot;0002&quot; ...
##  $ VisNo                                                          : int  2 6 3 5 5 1 4 3 2 3 ...
##  $ RNZ_WAISIIIDigitSpanForward_iqdspf                             : num  2.1219 -0.9091 -0.436 0.0109 0.0794 ...
##  $ RNZ_WAISIIIDigitSpanBackward_iqdspb                            : num  1.429 -0.694 -1.776 -0.741 1.778 ...
##  $ RNZ_WAISIIILetterNumberSequenceRawScore_iqlnsr                 : num  0.408 -0.686 -0.872 -1.447 1.324 ...
##  $ RNZ_AVLTRawTrialTotal_tTotal                                   : num  0.266 -1.042 -1.489 -0.537 0.571 ...
##  $ RNZ_AVLTDelayedRecallRaw_drraw                                 : num  -1.2808 0.0291 -0.1409 0.073 0.0756 ...
##  $ RNZ_LetterSemanticFluencyCFLRaw_flucfl                         : num  0.75068 0.57745 0.00136 0.72564 -0.12097 ...
##  $ RNZ_STROOPNumberNamedForColorWord_stroocw                      : num  -1.504 1.835 0.184 0.594 -0.605 ...
##  $ RNZ_TrailMakingTestTrailATime_trla                             : num  1.406 1.148 NA 1.51 -0.201 ...
##  $ RNZ_TrailMakingTestTrailBTime_trlb                             : num  0.876 0.747 1.231 2.158 -0.273 ...
##  $ RNZ_ClockDrawCDTTotalScore_cdtts                               : num  0.932 0.657 0.364 0.936 0.703 ...
##  $ RNZ_WAISRDigitSymbolTotalRaw_waisrTot                          : num  -1.067 -0.251 0.549 1.847 -0.48 ...
##  $ RNZ_AnimalTestingTotalRaw_animTotRaw                           : num  -0.313 -1.337 0.659 1.688 0.672 ...
##  $ RNZ_BVMTRTotalRaw_bvmtTot                                      : num  0.1781 -0.0415 0.4426 0.8698 0.1319 ...
##  $ RNZ_BVMTRDelayedRaw_bvmtDel                                    : num  0.042 -0.482 0.24 1.172 1.458 ...
##  $ RNZ_MMSETotalScore_mmseTot                                     : num  -2.349 1.065 0.584 1.209 0.863 ...
##  $ RNZ_WMSRLogicalMemoryITotalScore_wmsrTot                       : num  -0.699 -0.938 -0.382 -1.011 -0.685 ...
##  $ RNZ_WMSRLogicalMemoryIITotalScore_wmsrTot2                     : num  -1.9402 -1.8603 -0.654 -0.4535 -0.0814 ...
##  $ RNZ_BostonNamingTestTotal_bnttot_MultiLingualNamingTest_minttot: num  -0.865 -2.106 -0.845 0.98 0.414 ...</code></pre>
<p><br> <br></p>
</div>
<div id="sleep_mentalact" class="section level2">
<h2>Sleep_MentalAct</h2>
<p>Sleep/Mental Activities</p>
<pre class="r"><code>## load data
Sleep_MentalAct &lt;- read.csv(paste0(wrap_directory, &quot;Sleep_MentalAct_05Jun2025.csv&quot;))
info(Sleep_MentalAct,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:4208, cols:64, inds:1462</code></pre>
<pre class="r"><code>str(Sleep_MentalAct)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    4208 obs. of  64 variables:
##  $ wrapnum           : chr  &quot;wrap0002&quot; &quot;wrap0146&quot; &quot;wrap0148&quot; &quot;wrap0305&quot; ...
##  $ WRAPNo            : chr  &quot;0002&quot; &quot;0146&quot; &quot;0148&quot; &quot;0305&quot; ...
##  $ VisNo             : int  5 4 5 5 4 6 4 7 4 8 ...
##  $ FCAS_Chess        : int  4 1 -1 1 1 1 1 -1 -1 1 ...
##  $ FCAS_Checkers     : int  -1 -1 -1 1 1 1 1 -1 -1 -1 ...
##  $ FCAS_Puzzles      : int  4 0 4 -1 4 -1 0 0 -1 3 ...
##  $ FCAS_TVRadio      : int  4 4 4 4 4 4 4 4 4 4 ...
##  $ FCAS_Music        : int  1 4 4 4 4 4 4 3 4 4 ...
##  $ FCAS_Garden       : int  3 3 4 1 1 -1 3 0 3 -1 ...
##  $ FCAS_News         : int  3 4 4 3 2 3 2 3 2 2 ...
##  $ FCAS_Books        : int  4 2 4 3 1 3 1 1 4 3 ...
##  $ FCAS_Letters      : int  -1 2 1 1 1 1 -1 0 3 1 ...
##  $ FCAS_Phone        : int  3 4 3 3 3 3 3 2 3 4 ...
##  $ FCAS_Draw         : int  2 -1 0 -1 -1 -1 -1 0 2 -1 ...
##  $ FCAS_Knit         : int  3 0 3 -1 -1 0 0 0 3 1 ...
##  $ FCAS_CompRep      : int  -1 0 0 -1 -1 -1 -1 1 2 0 ...
##  $ FCAS_SimpRep      : int  -1 0 0 1 -1 1 1 2 2 -1 ...
##  $ FCAS_NewFood      : int  1 4 1 2 2 1 3 0 3 1 ...
##  $ FCAS_CookFam      : int  2 3 4 3 3 3 4 0 4 2 ...
##  $ FCAS_LeadDisc     : int  0 1 1 1 2 3 2 0 1 0 ...
##  $ FCAS_Course       : int  1 2 0 -1 -1 3 1 0 -1 3 ...
##  $ FCAS_ManageFin    : int  4 0 0 0 0 1 0 2 0 1 ...
##  $ FCAS_RoutFin      : int  4 3 4 3 3 2 4 4 3 2 ...
##  $ FCAS_TravelMap    : int  1 2 1 3 1 1 3 1 3 1 ...
##  $ FCAS_TravelFam    : int  3 4 4 4 4 4 4 1 3 4 ...
##  $ FCAS_SocialClub   : int  0 2 2 -1 2 0 0 0 -1 1 ...
##  $ FCAS_Church       : int  -1 3 1 3 2 2 1 2 0 1 ...
##  $ FCAS_Shop         : int  3 4 3 2 2 3 3 -1 3 3 ...
##  $ FCAS_VideoGame    : int  4 NA 3 0 NA 0 NA 0 NA 0 ...
##  $ FCAS_BrainGame    : int  4 NA 4 0 NA 0 NA 0 NA 4 ...
##  $ Rand_FallAsleep   : int  1 1 1 1 1 1 1 1 1 3 ...
##  $ Rand_HrsSleep     : int  7 8 7 6 5 7 7 8 10 5 ...
##  $ Rand_NotQuiet     : int  6 4 5 6 3 4 3 4 5 6 ...
##  $ Rand_FeelRested   : int  4 2 2 2 2 3 4 2 1 6 ...
##  $ Rand_ShortBreath  : int  6 6 6 6 6 6 6 6 6 5 ...
##  $ Rand_Drowsy       : int  4 5 6 6 5 5 4 5 4 4 ...
##  $ Rand_TroubleFall  : int  6 6 4 6 5 6 5 6 5 3 ...
##  $ Rand_FallAgain    : int  6 6 4 6 5 4 3 4 6 2 ...
##  $ Rand_StayAwake    : int  5 6 6 6 6 5 5 6 5 5 ...
##  $ Rand_Snore        : int  NA 4 4 6 5 6 3 1 4 1 ...
##  $ Rand_Naps         : int  5 4 6 6 6 6 5 6 4 5 ...
##  $ Rand_AmtNeed      : int  4 2 2 3 4 2 2 2 1 6 ...
##  $ Epw_SitRead       : int  1 2 1 1 1 0 2 0 1 1 ...
##  $ Epw_TV            : int  1 1 1 2 1 1 2 0 1 1 ...
##  $ Epw_SitInact      : int  2 0 0 1 1 1 1 0 0 0 ...
##  $ Epw_Car           : int  1 1 0 2 1 1 0 0 1 0 ...
##  $ Epw_LayDown       : int  1 1 1 2 1 2 3 2 1 2 ...
##  $ Epw_Talking       : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ Epw_Lunch         : int  0 0 0 0 0 0 0 0 0 1 ...
##  $ Epw_Traffic       : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ ISI_Satisfied     : int  2 NA 2 3 NA 3 NA 2 NA 4 ...
##  $ ISI_DiffFallAsleep: int  1 NA 1 1 NA 1 NA 1 NA 2 ...
##  $ ISI_DiffStayAwake : int  1 NA 1 1 NA 2 NA 2 NA 3 ...
##  $ ISI_ProbWakeEarly : int  1 NA 1 1 NA 3 NA 1 NA 4 ...
##  $ ISI_Noticeable    : int  NA NA NA NA NA 1 NA 1 NA 1 ...
##  $ ISI_Worried       : int  NA NA NA NA NA 2 NA 1 NA 2 ...
##  $ ISI_Interfere     : int  NA NA NA NA NA 2 NA 1 NA 2 ...
##  $ SA_DiagInsomnia   : int  2 NA 2 2 NA 2 NA 2 NA 2 ...
##  $ SA_DiagRestLegSynd: int  2 NA 2 2 NA 2 NA 2 NA 2 ...
##  $ SA_DiagSleepApnea : int  2 NA 2 2 NA 2 NA 2 NA 1 ...
##  $ SA_CPAPUse        : int  NA NA NA NA NA 2 NA NA NA 1 ...
##  $ SA_CPAPFreq       : int  NA NA NA NA NA 2 NA NA NA 3 ...
##  $ SA_UseOthDev      : int  NA NA NA NA NA 1 NA NA NA 2 ...
##  $ update_stamp      : chr  &quot;2024-09-13 08:53:59.0&quot; &quot;2024-09-13 08:53:59.0&quot; &quot;2024-09-13 08:53:59.0&quot; &quot;2024-09-13 08:53:59.0&quot; ...</code></pre>
<pre class="r"><code>## check table frequency for all variables

## drop update_stamp column
Sleep_MentalAct &lt;- subset(Sleep_MentalAct,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(Sleep_MentalAct)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    4208 obs. of  63 variables:
##  $ wrapnum           : chr  &quot;wrap0002&quot; &quot;wrap0146&quot; &quot;wrap0148&quot; &quot;wrap0305&quot; ...
##  $ WRAPNo            : chr  &quot;0002&quot; &quot;0146&quot; &quot;0148&quot; &quot;0305&quot; ...
##  $ VisNo             : int  5 4 5 5 4 6 4 7 4 8 ...
##  $ FCAS_Chess        : int  4 1 -1 1 1 1 1 -1 -1 1 ...
##  $ FCAS_Checkers     : int  -1 -1 -1 1 1 1 1 -1 -1 -1 ...
##  $ FCAS_Puzzles      : int  4 0 4 -1 4 -1 0 0 -1 3 ...
##  $ FCAS_TVRadio      : int  4 4 4 4 4 4 4 4 4 4 ...
##  $ FCAS_Music        : int  1 4 4 4 4 4 4 3 4 4 ...
##  $ FCAS_Garden       : int  3 3 4 1 1 -1 3 0 3 -1 ...
##  $ FCAS_News         : int  3 4 4 3 2 3 2 3 2 2 ...
##  $ FCAS_Books        : int  4 2 4 3 1 3 1 1 4 3 ...
##  $ FCAS_Letters      : int  -1 2 1 1 1 1 -1 0 3 1 ...
##  $ FCAS_Phone        : int  3 4 3 3 3 3 3 2 3 4 ...
##  $ FCAS_Draw         : int  2 -1 0 -1 -1 -1 -1 0 2 -1 ...
##  $ FCAS_Knit         : int  3 0 3 -1 -1 0 0 0 3 1 ...
##  $ FCAS_CompRep      : int  -1 0 0 -1 -1 -1 -1 1 2 0 ...
##  $ FCAS_SimpRep      : int  -1 0 0 1 -1 1 1 2 2 -1 ...
##  $ FCAS_NewFood      : int  1 4 1 2 2 1 3 0 3 1 ...
##  $ FCAS_CookFam      : int  2 3 4 3 3 3 4 0 4 2 ...
##  $ FCAS_LeadDisc     : int  0 1 1 1 2 3 2 0 1 0 ...
##  $ FCAS_Course       : int  1 2 0 -1 -1 3 1 0 -1 3 ...
##  $ FCAS_ManageFin    : int  4 0 0 0 0 1 0 2 0 1 ...
##  $ FCAS_RoutFin      : int  4 3 4 3 3 2 4 4 3 2 ...
##  $ FCAS_TravelMap    : int  1 2 1 3 1 1 3 1 3 1 ...
##  $ FCAS_TravelFam    : int  3 4 4 4 4 4 4 1 3 4 ...
##  $ FCAS_SocialClub   : int  0 2 2 -1 2 0 0 0 -1 1 ...
##  $ FCAS_Church       : int  -1 3 1 3 2 2 1 2 0 1 ...
##  $ FCAS_Shop         : int  3 4 3 2 2 3 3 -1 3 3 ...
##  $ FCAS_VideoGame    : int  4 NA 3 0 NA 0 NA 0 NA 0 ...
##  $ FCAS_BrainGame    : int  4 NA 4 0 NA 0 NA 0 NA 4 ...
##  $ Rand_FallAsleep   : int  1 1 1 1 1 1 1 1 1 3 ...
##  $ Rand_HrsSleep     : int  7 8 7 6 5 7 7 8 10 5 ...
##  $ Rand_NotQuiet     : int  6 4 5 6 3 4 3 4 5 6 ...
##  $ Rand_FeelRested   : int  4 2 2 2 2 3 4 2 1 6 ...
##  $ Rand_ShortBreath  : int  6 6 6 6 6 6 6 6 6 5 ...
##  $ Rand_Drowsy       : int  4 5 6 6 5 5 4 5 4 4 ...
##  $ Rand_TroubleFall  : int  6 6 4 6 5 6 5 6 5 3 ...
##  $ Rand_FallAgain    : int  6 6 4 6 5 4 3 4 6 2 ...
##  $ Rand_StayAwake    : int  5 6 6 6 6 5 5 6 5 5 ...
##  $ Rand_Snore        : int  NA 4 4 6 5 6 3 1 4 1 ...
##  $ Rand_Naps         : int  5 4 6 6 6 6 5 6 4 5 ...
##  $ Rand_AmtNeed      : int  4 2 2 3 4 2 2 2 1 6 ...
##  $ Epw_SitRead       : int  1 2 1 1 1 0 2 0 1 1 ...
##  $ Epw_TV            : int  1 1 1 2 1 1 2 0 1 1 ...
##  $ Epw_SitInact      : int  2 0 0 1 1 1 1 0 0 0 ...
##  $ Epw_Car           : int  1 1 0 2 1 1 0 0 1 0 ...
##  $ Epw_LayDown       : int  1 1 1 2 1 2 3 2 1 2 ...
##  $ Epw_Talking       : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ Epw_Lunch         : int  0 0 0 0 0 0 0 0 0 1 ...
##  $ Epw_Traffic       : int  0 0 0 0 0 0 0 0 0 0 ...
##  $ ISI_Satisfied     : int  2 NA 2 3 NA 3 NA 2 NA 4 ...
##  $ ISI_DiffFallAsleep: int  1 NA 1 1 NA 1 NA 1 NA 2 ...
##  $ ISI_DiffStayAwake : int  1 NA 1 1 NA 2 NA 2 NA 3 ...
##  $ ISI_ProbWakeEarly : int  1 NA 1 1 NA 3 NA 1 NA 4 ...
##  $ ISI_Noticeable    : int  NA NA NA NA NA 1 NA 1 NA 1 ...
##  $ ISI_Worried       : int  NA NA NA NA NA 2 NA 1 NA 2 ...
##  $ ISI_Interfere     : int  NA NA NA NA NA 2 NA 1 NA 2 ...
##  $ SA_DiagInsomnia   : int  2 NA 2 2 NA 2 NA 2 NA 2 ...
##  $ SA_DiagRestLegSynd: int  2 NA 2 2 NA 2 NA 2 NA 2 ...
##  $ SA_DiagSleepApnea : int  2 NA 2 2 NA 2 NA 2 NA 1 ...
##  $ SA_CPAPUse        : int  NA NA NA NA NA 2 NA NA NA 1 ...
##  $ SA_CPAPFreq       : int  NA NA NA NA NA 2 NA NA NA 3 ...
##  $ SA_UseOthDev      : int  NA NA NA NA NA 1 NA NA NA 2 ...</code></pre>
<p><br> <br></p>
</div>
<div id="sleep_mentalact_othdevice" class="section level2">
<h2>Sleep_MentalAct_OthDevice</h2>
<p>Sleep/Mental Activities - Sleep Apnea Other Devices</p>
<pre class="r"><code>## load data
Sleep_MentalAct_OthDevice &lt;- read.csv(paste0(wrap_directory, &quot;Sleep_MentalAct_OthDevice_05Jun2025.csv&quot;))
info(Sleep_MentalAct_OthDevice,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:49, cols:6, inds:37</code></pre>
<pre class="r"><code>str(Sleep_MentalAct_OthDevice)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    49 obs. of  6 variables:
##  $ wrapnum              : chr  &quot;wrap0401&quot; &quot;wrapA0114&quot; &quot;wrap0812&quot; &quot;wrap0248&quot; ...
##  $ WRAPNo               : chr  &quot;0401&quot; &quot;A0114&quot; &quot;0812&quot; &quot;0248&quot; ...
##  $ VisNo                : int  6 4 4 7 4 4 5 5 3 5 ...
##  $ oth_device_repeat_key: int  1 1 1 1 1 1 1 1 1 1 ...
##  $ SA_OthDevFrq         : int  3 1 1 3 3 2 3 3 3 3 ...
##  $ update_stamp         : chr  &quot;2024-09-13 08:54:00.0&quot; &quot;2024-09-13 08:54:00.0&quot; &quot;2024-09-13 08:54:00.0&quot; &quot;2024-09-13 08:54:00.0&quot; ...</code></pre>
<pre class="r"><code>## check table frequency for all variables

## drop update_stamp column
Sleep_MentalAct_OthDevice &lt;- subset(Sleep_MentalAct_OthDevice,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(Sleep_MentalAct_OthDevice)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    49 obs. of  5 variables:
##  $ wrapnum              : chr  &quot;wrap0401&quot; &quot;wrapA0114&quot; &quot;wrap0812&quot; &quot;wrap0248&quot; ...
##  $ WRAPNo               : chr  &quot;0401&quot; &quot;A0114&quot; &quot;0812&quot; &quot;0248&quot; ...
##  $ VisNo                : int  6 4 4 7 4 4 5 5 3 5 ...
##  $ oth_device_repeat_key: int  1 1 1 1 1 1 1 1 1 1 ...
##  $ SA_OthDevFrq         : int  3 1 1 3 3 2 3 3 3 3 ...</code></pre>
<p><br> <br></p>
</div>
<div id="subjectivehearing" class="section level2">
<h2>SubjectiveHearing</h2>
<p>Subjective Hearing Complaints</p>
<pre class="r"><code>## load data
SubjectiveHearing &lt;- read.csv(paste0(wrap_directory, &quot;SubjectiveHearing_05Jun2025.csv&quot;))
info(SubjectiveHearing,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:2732, cols:9, inds:1312</code></pre>
<pre class="r"><code>str(SubjectiveHearing)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    2732 obs. of  9 variables:
##  $ wrapnum               : chr  &quot;wrap0401&quot; &quot;wrap0434&quot; &quot;wrap0609&quot; &quot;wrap0624&quot; ...
##  $ WRAPNo                : chr  &quot;0401&quot; &quot;0434&quot; &quot;0609&quot; &quot;0624&quot; ...
##  $ VisNo                 : int  6 7 8 6 3 1 1 3 4 1 ...
##  $ SubjHear_HearProb     : int  1 2 1 4 1 2 1 1 8 1 ...
##  $ SubjHear_ConversPitch : int  8 8 8 7 10 8 9 1 8 10 ...
##  $ SubjHear_Concentrate  : int  7 1 4 4 2 3 1 5 3 2 ...
##  $ SubjHear_EverydaySound: int  10 10 10 7 10 9 10 9 3 10 ...
##  $ SubjHear_BusTruckFar  : int  9 10 10 8 10 8 8 9 7 10 ...
##  $ update_stamp          : chr  &quot;2024-09-13 08:54:01.0&quot; &quot;2024-09-13 08:54:01.0&quot; &quot;2024-09-13 08:54:01.0&quot; &quot;2024-09-13 08:54:01.0&quot; ...</code></pre>
<pre class="r"><code>## check table frequency for all variables

## drop update_stamp column
SubjectiveHearing &lt;- subset(SubjectiveHearing,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(SubjectiveHearing)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    2732 obs. of  8 variables:
##  $ wrapnum               : chr  &quot;wrap0401&quot; &quot;wrap0434&quot; &quot;wrap0609&quot; &quot;wrap0624&quot; ...
##  $ WRAPNo                : chr  &quot;0401&quot; &quot;0434&quot; &quot;0609&quot; &quot;0624&quot; ...
##  $ VisNo                 : int  6 7 8 6 3 1 1 3 4 1 ...
##  $ SubjHear_HearProb     : int  1 2 1 4 1 2 1 1 8 1 ...
##  $ SubjHear_ConversPitch : int  8 8 8 7 10 8 9 1 8 10 ...
##  $ SubjHear_Concentrate  : int  7 1 4 4 2 3 1 5 3 2 ...
##  $ SubjHear_EverydaySound: int  10 10 10 7 10 9 10 9 3 10 ...
##  $ SubjHear_BusTruckFar  : int  9 10 10 8 10 8 8 9 7 10 ...</code></pre>
<p><br> <br></p>
</div>
<div id="ticsm" class="section level2">
<h2>TICSM</h2>
<p>Modified Telephone Interview of Cognitive Status</p>
<pre class="r"><code>## load data
TICSM &lt;- read.csv(paste0(wrap_directory, &quot;TICSM_05Jun2025.csv&quot;))
info(TICSM,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:65, cols:45, inds:64</code></pre>
<pre class="r"><code>str(TICSM)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    65 obs. of  45 variables:
##  $ wrapnum                    : chr  &quot;wrap0792&quot; &quot;wrap0357&quot; &quot;wrap0163&quot; &quot;wrap0024&quot; ...
##  $ WRAPNo                     : chr  &quot;0792&quot; &quot;0357&quot; &quot;0163&quot; &quot;0024&quot; ...
##  $ VisNo                      : int  6 6 6 4 6 5 2 5 3 3 ...
##  $ TICSM_TesterID             : int  30 30 22 30 30 24 30 30 30 30 ...
##  $ TICSM_Full_Name            : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ TICSM_Todays_Date          : int  5 5 5 5 5 4 5 5 5 5 ...
##  $ TICSM_Age_Phone            : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ TICSM_Count_Back           : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ TICSM_ImmRecall_1_Cabin    : int  4 1 1 8 1 1 2 1 6 7 ...
##  $ TICSM_ImmRecall_2_Pipe     : int  NA 2 NA NA NA 3 NA NA 7 5 ...
##  $ TICSM_ImmRecall_3_Elephant : int  NA 3 2 4 3 2 NA 2 0 4 ...
##  $ TICSM_ImmRecall_4_Chest    : int  NA NA NA 7 0 NA NA NA 3 NA ...
##  $ TICSM_ImmRecall_5_Silk     : int  NA 6 4 5 4 NA NA NA 0 3 ...
##  $ TICSM_ImmRecall_6_Theater  : int  NA 7 NA 6 NA NA NA 3 4 NA ...
##  $ TICSM_ImmRecall_7_Watch    : int  3 5 NA NA 5 NA NA 4 0 NA ...
##  $ TICSM_ImmRecall_8_Whip     : int  NA 4 3 NA 6 NA NA NA 2 NA ...
##  $ TICSM_ImmRecall_9_Pillow   : int  2 NA NA 2 NA NA NA 5 5 2 ...
##  $ TICSM_ImmRecall_10_Giant   : int  1 NA NA 1 NA 4 1 6 1 1 ...
##  $ TICSM_ImmRecall_Repetitions: int  0 0 0 0 0 0 0 0 0 1 ...
##  $ TICSM_ImmRecall_Intrusions : int  0 0 1 1 1 0 2 0 0 0 ...
##  $ TICSM_Subtract7            : int  5 5 1 5 5 5 5 5 5 NA ...
##  $ TICSM_Vocabulary           : int  4 4 4 4 4 4 4 4 4 4 ...
##  $ TICSM_Repeat_Phrases       : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ TICSM_Pres_VP              : int  3 3 3 4 3 4 4 4 4 3 ...
##  $ TICSM_Finger_Tap           : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ TICSM_Opposites            : int  2 2 2 2 2 1 2 2 2 2 ...
##  $ TICSM_DelRecall_1_Cabin    : int  1 1 1 1 5 1 1 1 1 1 ...
##  $ TICSM_DelRecall_2_Pipe     : int  NA 3 NA NA NA NA NA NA 2 NA ...
##  $ TICSM_DelRecall_3_Elephant : int  NA 2 3 2 NA NA NA 2 0 2 ...
##  $ TICSM_DelRecall_4_Chest    : int  NA NA NA NA NA NA NA NA 4 NA ...
##  $ TICSM_DelRecall_5_Silk     : int  NA 5 2 4 NA NA NA NA 0 3 ...
##  $ TICSM_DelRecall_6_Theater  : int  NA 4 NA 3 NA NA NA 3 5 NA ...
##  $ TICSM_DelRecall_7_Watch    : int  NA 6 NA NA 1 NA NA 4 0 NA ...
##  $ TICSM_DelRecall_8_Whip     : int  NA NA 4 NA 2 NA NA NA 0 NA ...
##  $ TICSM_DelRecall_9_Pillow   : int  NA NA NA NA NA NA NA 5 3 NA ...
##  $ TICSM_DelRecall_10_Giant   : int  NA NA NA NA NA NA 2 6 0 NA ...
##  $ TICSM_DelRecall_Repetitions: int  0 0 0 0 0 0 0 0 0 0 ...
##  $ TICSM_DelRecall_Intrusions : int  0 0 1 0 2 0 2 0 0 0 ...
##  $ TICSM_ImmRecall_Score      : int  4 7 4 7 5 4 2 6 7 6 ...
##  $ TICSM_DelRecall_Score      : int  1 6 4 4 3 1 2 6 5 3 ...
##  $ TICSM_TotalRecall_Score    : int  5 13 8 11 8 5 4 12 12 9 ...
##  $ TICSM_Orientation          : int  12 12 12 13 12 12 13 13 13 12 ...
##  $ TICSM_Lang_Attn            : int  17 17 13 17 17 16 17 17 17 NA ...
##  $ TICSM_TotalScore           : int  34 42 33 41 37 33 34 42 42 NA ...
##  $ update_stamp               : chr  &quot;2024-09-13 08:54:01.0&quot; &quot;2024-09-13 08:54:01.0&quot; &quot;2024-09-13 08:54:01.0&quot; &quot;2024-09-13 08:54:01.0&quot; ...</code></pre>
<pre class="r"><code>## check table frequency for all variables

## drop update_stamp column
TICSM &lt;- subset(TICSM,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(TICSM)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    65 obs. of  44 variables:
##  $ wrapnum                    : chr  &quot;wrap0792&quot; &quot;wrap0357&quot; &quot;wrap0163&quot; &quot;wrap0024&quot; ...
##  $ WRAPNo                     : chr  &quot;0792&quot; &quot;0357&quot; &quot;0163&quot; &quot;0024&quot; ...
##  $ VisNo                      : int  6 6 6 4 6 5 2 5 3 3 ...
##  $ TICSM_TesterID             : int  30 30 22 30 30 24 30 30 30 30 ...
##  $ TICSM_Full_Name            : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ TICSM_Todays_Date          : int  5 5 5 5 5 4 5 5 5 5 ...
##  $ TICSM_Age_Phone            : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ TICSM_Count_Back           : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ TICSM_ImmRecall_1_Cabin    : int  4 1 1 8 1 1 2 1 6 7 ...
##  $ TICSM_ImmRecall_2_Pipe     : int  NA 2 NA NA NA 3 NA NA 7 5 ...
##  $ TICSM_ImmRecall_3_Elephant : int  NA 3 2 4 3 2 NA 2 0 4 ...
##  $ TICSM_ImmRecall_4_Chest    : int  NA NA NA 7 0 NA NA NA 3 NA ...
##  $ TICSM_ImmRecall_5_Silk     : int  NA 6 4 5 4 NA NA NA 0 3 ...
##  $ TICSM_ImmRecall_6_Theater  : int  NA 7 NA 6 NA NA NA 3 4 NA ...
##  $ TICSM_ImmRecall_7_Watch    : int  3 5 NA NA 5 NA NA 4 0 NA ...
##  $ TICSM_ImmRecall_8_Whip     : int  NA 4 3 NA 6 NA NA NA 2 NA ...
##  $ TICSM_ImmRecall_9_Pillow   : int  2 NA NA 2 NA NA NA 5 5 2 ...
##  $ TICSM_ImmRecall_10_Giant   : int  1 NA NA 1 NA 4 1 6 1 1 ...
##  $ TICSM_ImmRecall_Repetitions: int  0 0 0 0 0 0 0 0 0 1 ...
##  $ TICSM_ImmRecall_Intrusions : int  0 0 1 1 1 0 2 0 0 0 ...
##  $ TICSM_Subtract7            : int  5 5 1 5 5 5 5 5 5 NA ...
##  $ TICSM_Vocabulary           : int  4 4 4 4 4 4 4 4 4 4 ...
##  $ TICSM_Repeat_Phrases       : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ TICSM_Pres_VP              : int  3 3 3 4 3 4 4 4 4 3 ...
##  $ TICSM_Finger_Tap           : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ TICSM_Opposites            : int  2 2 2 2 2 1 2 2 2 2 ...
##  $ TICSM_DelRecall_1_Cabin    : int  1 1 1 1 5 1 1 1 1 1 ...
##  $ TICSM_DelRecall_2_Pipe     : int  NA 3 NA NA NA NA NA NA 2 NA ...
##  $ TICSM_DelRecall_3_Elephant : int  NA 2 3 2 NA NA NA 2 0 2 ...
##  $ TICSM_DelRecall_4_Chest    : int  NA NA NA NA NA NA NA NA 4 NA ...
##  $ TICSM_DelRecall_5_Silk     : int  NA 5 2 4 NA NA NA NA 0 3 ...
##  $ TICSM_DelRecall_6_Theater  : int  NA 4 NA 3 NA NA NA 3 5 NA ...
##  $ TICSM_DelRecall_7_Watch    : int  NA 6 NA NA 1 NA NA 4 0 NA ...
##  $ TICSM_DelRecall_8_Whip     : int  NA NA 4 NA 2 NA NA NA 0 NA ...
##  $ TICSM_DelRecall_9_Pillow   : int  NA NA NA NA NA NA NA 5 3 NA ...
##  $ TICSM_DelRecall_10_Giant   : int  NA NA NA NA NA NA 2 6 0 NA ...
##  $ TICSM_DelRecall_Repetitions: int  0 0 0 0 0 0 0 0 0 0 ...
##  $ TICSM_DelRecall_Intrusions : int  0 0 1 0 2 0 2 0 0 0 ...
##  $ TICSM_ImmRecall_Score      : int  4 7 4 7 5 4 2 6 7 6 ...
##  $ TICSM_DelRecall_Score      : int  1 6 4 4 3 1 2 6 5 3 ...
##  $ TICSM_TotalRecall_Score    : int  5 13 8 11 8 5 4 12 12 9 ...
##  $ TICSM_Orientation          : int  12 12 12 13 12 12 13 13 13 12 ...
##  $ TICSM_Lang_Attn            : int  17 17 13 17 17 16 17 17 17 NA ...
##  $ TICSM_TotalScore           : int  34 42 33 41 37 33 34 42 42 NA ...</code></pre>
<p><br> <br></p>
</div>
<div id="visitvitals" class="section level2">
<h2>VisitVitals</h2>
<p>WRAP Visit Vitals</p>
<pre class="r"><code>## load data
VisitVitals &lt;- read.csv(paste0(wrap_directory, &quot;VisitVitals_05Jun2025.csv&quot;))
info(VisitVitals,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:8205, cols:34, inds:1810</code></pre>
<pre class="r"><code>str(VisitVitals)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    8205 obs. of  34 variables:
##  $ wrapnum              : chr  &quot;wrap0002&quot; &quot;wrap0068&quot; &quot;wrap0146&quot; &quot;wrap0148&quot; ...
##  $ WRAPNo               : chr  &quot;0002&quot; &quot;0068&quot; &quot;0146&quot; &quot;0148&quot; ...
##  $ VisNo                : int  5 2 4 5 5 4 6 4 7 4 ...
##  $ FastTime             : int  14 15 16 14 13 14 14 13 13 15 ...
##  $ Fast_HM              : chr  &quot;Hrs&quot; &quot;Hrs&quot; &quot;Hrs&quot; &quot;Hrs&quot; ...
##  $ OTCTime              : int  37 48 15 15 27 24 2 48 23 48 ...
##  $ OTC_HM               : chr  &quot;Hrs&quot; &quot;Hrs&quot; &quot;Hrs&quot; &quot;Hrs&quot; ...
##  $ OTC_Meds_Taken       : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ PCDTime              : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ PCD_HM               : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ PCD_Meds_Taken       : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ caffeine             : int  14 1 22 17 48 22 19 24 19 19 ...
##  $ Caf_HM               : chr  &quot;Hrs&quot; &quot;Hrs&quot; &quot;Hrs&quot; &quot;Hrs&quot; ...
##  $ height               : int  166 183 164 171 161 166 177 170 174 164 ...
##  $ height_inches        : int  65 72 65 67 63 65 70 67 69 65 ...
##  $ weight               : int  88 87 93 95 62 65 77 101 86 64 ...
##  $ weight_lbs           : int  194 192 205 209 137 143 170 223 190 141 ...
##  $ temperature          : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ waist_hip_not_ordered: chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ waist                : int  102 92 94 93 76 79 86 122 93 79 ...
##  $ waist_inches         : num  40.2 36.2 37.1 36.4 29.8 ...
##  $ hip                  : int  118 101 125 122 99 100 101 125 101 93 ...
##  $ hip_inches           : int  46 40 49 48 39 39 40 49 40 36 ...
##  $ BP_not_ordered       : logi  NA NA NA NA NA NA ...
##  $ BP_single_measurement: logi  NA NA NA NA NA NA ...
##  $ s_bp                 : int  124 107 161 135 108 107 119 118 135 156 ...
##  $ d_bp                 : int  78 79 83 80 68 73 75 80 80 76 ...
##  $ Inflamed             : int  2 NA NA 2 2 NA 2 1 2 2 ...
##  $ Pulse                : int  64 72 NA 76 60 NA 55 56 71 67 ...
##  $ Resp                 : int  16 10 NA 14 14 NA 16 12 18 14 ...
##  $ CarotidUltrasoundComp: int  NA NA NA NA NA NA NA NA NA NA ...
##  $ s_bp_adj             : int  124 107 161 135 108 107 119 118 135 156 ...
##  $ d_bp_adj             : int  78 79 83 80 68 73 75 80 80 76 ...
##  $ update_stamp         : chr  &quot;2024-09-13 08:54:04.0&quot; &quot;2024-09-13 08:54:04.0&quot; &quot;2024-09-13 08:54:04.0&quot; &quot;2024-09-13 08:54:04.0&quot; ...</code></pre>
<pre class="r"><code>## convert &quot;&quot; to NA for chr columns
chrcols &lt;- colnames(VisitVitals)[sapply(VisitVitals, is.character)]
VisitVitals[chrcols] &lt;- lapply(VisitVitals[chrcols], function(x) ifelse(x == &quot;&quot;, NA, x))

## convert logi to chr for columns: BP_not_ordered BP_single_measurement
logicols &lt;- c(&quot;BP_not_ordered&quot;,&quot;BP_single_measurement&quot;)
VisitVitals[logicols] &lt;- lapply(VisitVitals[logicols], as.character)

## drop update_stamp column
VisitVitals &lt;- subset(VisitVitals,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(VisitVitals)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    8205 obs. of  33 variables:
##  $ wrapnum              : chr  &quot;wrap0002&quot; &quot;wrap0068&quot; &quot;wrap0146&quot; &quot;wrap0148&quot; ...
##  $ WRAPNo               : chr  &quot;0002&quot; &quot;0068&quot; &quot;0146&quot; &quot;0148&quot; ...
##  $ VisNo                : int  5 2 4 5 5 4 6 4 7 4 ...
##  $ FastTime             : int  14 15 16 14 13 14 14 13 13 15 ...
##  $ Fast_HM              : chr  &quot;Hrs&quot; &quot;Hrs&quot; &quot;Hrs&quot; &quot;Hrs&quot; ...
##  $ OTCTime              : int  37 48 15 15 27 24 2 48 23 48 ...
##  $ OTC_HM               : chr  &quot;Hrs&quot; &quot;Hrs&quot; &quot;Hrs&quot; &quot;Hrs&quot; ...
##  $ OTC_Meds_Taken       : chr  NA NA NA NA ...
##  $ PCDTime              : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ PCD_HM               : chr  NA NA NA NA ...
##  $ PCD_Meds_Taken       : chr  NA NA NA NA ...
##  $ caffeine             : int  14 1 22 17 48 22 19 24 19 19 ...
##  $ Caf_HM               : chr  &quot;Hrs&quot; &quot;Hrs&quot; &quot;Hrs&quot; &quot;Hrs&quot; ...
##  $ height               : int  166 183 164 171 161 166 177 170 174 164 ...
##  $ height_inches        : int  65 72 65 67 63 65 70 67 69 65 ...
##  $ weight               : int  88 87 93 95 62 65 77 101 86 64 ...
##  $ weight_lbs           : int  194 192 205 209 137 143 170 223 190 141 ...
##  $ temperature          : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ waist_hip_not_ordered: chr  NA NA NA NA ...
##  $ waist                : int  102 92 94 93 76 79 86 122 93 79 ...
##  $ waist_inches         : num  40.2 36.2 37.1 36.4 29.8 ...
##  $ hip                  : int  118 101 125 122 99 100 101 125 101 93 ...
##  $ hip_inches           : int  46 40 49 48 39 39 40 49 40 36 ...
##  $ BP_not_ordered       : chr  NA NA NA NA ...
##  $ BP_single_measurement: chr  NA NA NA NA ...
##  $ s_bp                 : int  124 107 161 135 108 107 119 118 135 156 ...
##  $ d_bp                 : int  78 79 83 80 68 73 75 80 80 76 ...
##  $ Inflamed             : int  2 NA NA 2 2 NA 2 1 2 2 ...
##  $ Pulse                : int  64 72 NA 76 60 NA 55 56 71 67 ...
##  $ Resp                 : int  16 10 NA 14 14 NA 16 12 18 14 ...
##  $ CarotidUltrasoundComp: int  NA NA NA NA NA NA NA NA NA NA ...
##  $ s_bp_adj             : int  124 107 161 135 108 107 119 118 135 156 ...
##  $ d_bp_adj             : int  78 79 83 80 68 73 75 80 80 76 ...</code></pre>
<p><br> <br></p>
</div>
<div id="womenquestions" class="section level2">
<h2>WomenQuestions</h2>
<p>Women Questions</p>
<pre class="r"><code>## load data
WomenQuestions &lt;- read.csv(paste0(wrap_directory, &quot;WomenQuestions_05Jun2025.csv&quot;))
info(WomenQuestions,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:5544, cols:68, inds:1249</code></pre>
<pre class="r"><code>str(WomenQuestions)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    5544 obs. of  68 variables:
##  $ wrapnum             : chr  &quot;wrap0002&quot; &quot;wrap0146&quot; &quot;wrap0148&quot; &quot;wrap0305&quot; ...
##  $ WRAPNo              : chr  &quot;0002&quot; &quot;0146&quot; &quot;0148&quot; &quot;0305&quot; ...
##  $ VisNo               : int  5 4 5 5 4 4 4 8 5 3 ...
##  $ Menopause           : int  3 4 3 3 3 3 4 3 3 3 ...
##  $ MenoChange          : int  2 2 2 2 3 2 2 2 2 2 ...
##  $ FirstPeriod         : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ Preg                : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Preg_Ever_Combined  : int  2 1 1 1 1 1 1 2 2 1 ...
##  $ PregNo              : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ Preg1mo             : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ Preg3mo             : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ Preg6mo             : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ PregFull            : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ BF6mo               : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ BF12mo              : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ BCP                 : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ OvaOut_InitRespSet  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ OvaOut_SubsqRespSet : int  5 5 5 5 5 5 5 5 5 5 ...
##  $ OvaOut_Text_Response: chr  &quot;Since the last Visit, I&#39;ve not had an ovary taken out.&quot; &quot;Since the last Visit, I&#39;ve not had an ovary taken out.&quot; &quot;Since the last Visit, I&#39;ve not had an ovary taken out.&quot; &quot;Since the last Visit, I&#39;ve not had an ovary taken out.&quot; ...
##  $ LastOvOp            : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ LastOvOp_ImputedAge : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ hyst                : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ HystType            : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ HystAge             : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ HystAge_ImputedAge  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ PerStop             : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ HTMed               : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ LastPerAge          : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ HotFlash            : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ Sweats              : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ ProbSleep           : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ MoodSw              : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ Depress             : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ SexDys              : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ NoSympt             : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ OthSympt            : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ SymptSev            : int  NA NA NA NA NA NA NA 2 NA NA ...
##  $ CurHTMed            : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ HTStart             : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ Hormones            : int  NA 2 NA NA 2 2 2 NA NA 2 ...
##  $ SERM                : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ BCPChange           : int  NA 2 NA NA 2 2 2 NA NA 2 ...
##  $ BCPStop             : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ BCPDoseA            : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ BCPDoseB            : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ BCPDoseC            : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ HTChange            : int  NA 2 NA NA 2 2 1 NA NA 2 ...
##  $ HTStop              : int  NA NA NA NA NA NA 1 NA NA NA ...
##  $ CurDose1            : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ CurDose2            : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ CurDose3            : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ HormStop            : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ HormDose1           : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ HormDose2           : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ HormDose3           : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ SERMStop            : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ SERMChange          : int  NA 2 NA NA 2 2 2 NA NA 2 ...
##  $ SERMDose1           : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ SERMDose2           : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ postmeno            : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ usedhrt             : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ nowhrt              : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ longhrt             : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ namehrt             : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ agehrt              : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ sidehrt             : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ yesside             : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ update_stamp        : chr  &quot;2024-09-13 08:54:08.0&quot; &quot;2024-09-13 08:54:08.0&quot; &quot;2024-09-13 08:54:08.0&quot; &quot;2024-09-13 08:54:08.0&quot; ...</code></pre>
<pre class="r"><code>## check table frequency for all variables

## in the DD, column nowhrt have three values: 1=Yes;2=No;3=Don&#39;t Know
## in the data, it got addition value 0:
# table(WomenQuestions$nowhrt,useNA=&quot;ifany&quot;)
# 
#    0    1    2    3 &lt;NA&gt; 
#    1  150  292    3 5098 

## I am going to convert 0 to NA:
WomenQuestions$nowhrt[WomenQuestions$nowhrt==0] &lt;- NA

## convert &quot;&quot; to NA for chr columns
chrcols &lt;- colnames(WomenQuestions)[sapply(WomenQuestions, is.character)]
WomenQuestions[chrcols] &lt;- lapply(WomenQuestions[chrcols], function(x) ifelse(x == &quot;&quot;, NA, x))

## drop update_stamp column
WomenQuestions &lt;- subset(WomenQuestions,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(WomenQuestions)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    5544 obs. of  67 variables:
##  $ wrapnum             : chr  &quot;wrap0002&quot; &quot;wrap0146&quot; &quot;wrap0148&quot; &quot;wrap0305&quot; ...
##  $ WRAPNo              : chr  &quot;0002&quot; &quot;0146&quot; &quot;0148&quot; &quot;0305&quot; ...
##  $ VisNo               : int  5 4 5 5 4 4 4 8 5 3 ...
##  $ Menopause           : int  3 4 3 3 3 3 4 3 3 3 ...
##  $ MenoChange          : int  2 2 2 2 3 2 2 2 2 2 ...
##  $ FirstPeriod         : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ Preg                : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Preg_Ever_Combined  : int  2 1 1 1 1 1 1 2 2 1 ...
##  $ PregNo              : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ Preg1mo             : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ Preg3mo             : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ Preg6mo             : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ PregFull            : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ BF6mo               : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ BF12mo              : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ BCP                 : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ OvaOut_InitRespSet  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ OvaOut_SubsqRespSet : int  5 5 5 5 5 5 5 5 5 5 ...
##  $ OvaOut_Text_Response: chr  &quot;Since the last Visit, I&#39;ve not had an ovary taken out.&quot; &quot;Since the last Visit, I&#39;ve not had an ovary taken out.&quot; &quot;Since the last Visit, I&#39;ve not had an ovary taken out.&quot; &quot;Since the last Visit, I&#39;ve not had an ovary taken out.&quot; ...
##  $ LastOvOp            : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ LastOvOp_ImputedAge : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ hyst                : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ HystType            : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ HystAge             : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ HystAge_ImputedAge  : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ PerStop             : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ HTMed               : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ LastPerAge          : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ HotFlash            : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ Sweats              : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ ProbSleep           : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ MoodSw              : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ Depress             : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ SexDys              : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ NoSympt             : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ OthSympt            : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ SymptSev            : int  NA NA NA NA NA NA NA 2 NA NA ...
##  $ CurHTMed            : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ HTStart             : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ Hormones            : int  NA 2 NA NA 2 2 2 NA NA 2 ...
##  $ SERM                : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ BCPChange           : int  NA 2 NA NA 2 2 2 NA NA 2 ...
##  $ BCPStop             : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ BCPDoseA            : chr  NA NA NA NA ...
##  $ BCPDoseB            : chr  NA NA NA NA ...
##  $ BCPDoseC            : chr  NA NA NA NA ...
##  $ HTChange            : int  NA 2 NA NA 2 2 1 NA NA 2 ...
##  $ HTStop              : int  NA NA NA NA NA NA 1 NA NA NA ...
##  $ CurDose1            : chr  NA NA NA NA ...
##  $ CurDose2            : chr  NA NA NA NA ...
##  $ CurDose3            : chr  NA NA NA NA ...
##  $ HormStop            : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ HormDose1           : chr  NA NA NA NA ...
##  $ HormDose2           : chr  NA NA NA NA ...
##  $ HormDose3           : chr  NA NA NA NA ...
##  $ SERMStop            : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ SERMChange          : int  NA 2 NA NA 2 2 2 NA NA 2 ...
##  $ SERMDose1           : chr  NA NA NA NA ...
##  $ SERMDose2           : chr  NA NA NA NA ...
##  $ postmeno            : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ usedhrt             : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ nowhrt              : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ longhrt             : chr  NA NA NA NA ...
##  $ namehrt             : chr  NA NA NA NA ...
##  $ agehrt              : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ sidehrt             : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ yesside             : chr  NA NA NA NA ...</code></pre>
<p><br> <br></p>
</div>
<div id="womenquestions_bcp" class="section level2">
<h2>WomenQuestions_BCP</h2>
<p>Women Questions</p>
<pre class="r"><code>## load data
WomenQuestions_BCP &lt;- read.csv(paste0(wrap_directory, &quot;WomenQuestions_BCP_05Jun2025.csv&quot;))
info(WomenQuestions_BCP,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:1112, cols:11, inds:866</code></pre>
<pre class="r"><code>str(WomenQuestions_BCP)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    1112 obs. of  11 variables:
##  $ wrapnum       : chr  &quot;wrap1006&quot; &quot;wrap1081&quot; &quot;wrapL0240&quot; &quot;wrapL0283&quot; ...
##  $ WRAPNo        : chr  &quot;1006&quot; &quot;1081&quot; &quot;L0240&quot; &quot;L0283&quot; ...
##  $ VisNo         : int  2 1 2 2 2 2 2 2 2 2 ...
##  $ BCP_Repeat_Key: int  1 1 1 1 1 1 1 2 1 1 ...
##  $ BCPName       : chr  &quot;The Pill&quot; &quot;&quot; &quot;&quot; &quot;birth control pills&quot; ...
##  $ BCPCur        : chr  &quot;no&quot; &quot;no&quot; &quot;no&quot; &quot;no&quot; ...
##  $ BCPType       : chr  &quot;pill&quot; &quot;pill&quot; &quot;pill&quot; &quot;pill&quot; ...
##  $ BCPAge        : int  19 20 20 20 20 22 27 18 46 23 ...
##  $ BCPLong       : chr  &quot;1 year&quot; &quot;1 year&quot; &quot;4 years&quot; &quot;7 years&quot; ...
##  $ BCPSide       : int  1 1 1 1 1 1 1 NA NA 1 ...
##  $ update_stamp  : chr  &quot;2024-09-13 08:54:09.0&quot; &quot;2024-09-13 08:54:09.0&quot; &quot;2024-09-13 08:54:09.0&quot; &quot;2024-09-13 08:54:09.0&quot; ...</code></pre>
<pre class="r"><code>## check table frequency for all variables

## convert &quot;&quot; to NA for chr columns
chrcols &lt;- colnames(WomenQuestions_BCP)[sapply(WomenQuestions_BCP, is.character)]
WomenQuestions_BCP[chrcols] &lt;- lapply(WomenQuestions_BCP[chrcols], function(x) ifelse(x == &quot;&quot;, NA, x))

## drop update_stamp column
WomenQuestions_BCP &lt;- subset(WomenQuestions_BCP,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(WomenQuestions_BCP)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    1112 obs. of  10 variables:
##  $ wrapnum       : chr  &quot;wrap1006&quot; &quot;wrap1081&quot; &quot;wrapL0240&quot; &quot;wrapL0283&quot; ...
##  $ WRAPNo        : chr  &quot;1006&quot; &quot;1081&quot; &quot;L0240&quot; &quot;L0283&quot; ...
##  $ VisNo         : int  2 1 2 2 2 2 2 2 2 2 ...
##  $ BCP_Repeat_Key: int  1 1 1 1 1 1 1 2 1 1 ...
##  $ BCPName       : chr  &quot;The Pill&quot; NA NA &quot;birth control pills&quot; ...
##  $ BCPCur        : chr  &quot;no&quot; &quot;no&quot; &quot;no&quot; &quot;no&quot; ...
##  $ BCPType       : chr  &quot;pill&quot; &quot;pill&quot; &quot;pill&quot; &quot;pill&quot; ...
##  $ BCPAge        : int  19 20 20 20 20 22 27 18 46 23 ...
##  $ BCPLong       : chr  &quot;1 year&quot; &quot;1 year&quot; &quot;4 years&quot; &quot;7 years&quot; ...
##  $ BCPSide       : int  1 1 1 1 1 1 1 NA NA 1 ...</code></pre>
<p><br> <br></p>
</div>
<div id="womenquestions_current_hormonetherapy" class="section level2">
<h2>WomenQuestions_Current_HormoneTherapy</h2>
<p>Women Questions - Current Hormone Therapy</p>
<pre class="r"><code>## load data
WomenQuestions_Current_HormoneTherapy &lt;- read.csv(paste0(wrap_directory, &quot;WomenQuestions_Current_HormoneTherapy_05Jun2025.csv&quot;))
info(WomenQuestions_Current_HormoneTherapy,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:466, cols:11, inds:210</code></pre>
<pre class="r"><code>str(WomenQuestions_Current_HormoneTherapy)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    466 obs. of  11 variables:
##  $ wrapnum          : chr  &quot;wrap0123&quot; &quot;wrap0228&quot; &quot;wrap0381&quot; &quot;wrap0381&quot; ...
##  $ WRAPNo           : chr  &quot;0123&quot; &quot;0228&quot; &quot;0381&quot; &quot;0381&quot; ...
##  $ VisNo            : int  2 4 2 2 2 2 3 2 4 4 ...
##  $ Cur_HT_Repeat_Key: int  1 1 3 2 1 1 1 1 1 3 ...
##  $ CurMed           : chr  &quot;&quot; &quot;&quot; &quot;Testosterone cream&quot; &quot;Vagifem&quot; ...
##  $ CurCur           : int  NA NA 1 1 1 1 NA 1 NA NA ...
##  $ CurType          : chr  &quot;&quot; &quot;&quot; &quot;cream&quot; &quot;tab&quot; ...
##  $ CurAge           : int  NA NA 51 52 51 46 NA 55 NA NA ...
##  $ CurLong          : chr  &quot;&quot; &quot;&quot; &quot;2.5 years&quot; &quot;1.5 years&quot; ...
##  $ CurSide          : int  NA NA 1 1 1 1 NA 1 NA NA ...
##  $ update_stamp     : chr  &quot;2024-09-13 08:54:10.0&quot; &quot;2024-09-13 08:54:10.0&quot; &quot;2024-09-13 08:54:10.0&quot; &quot;2024-09-13 08:54:10.0&quot; ...</code></pre>
<pre class="r"><code>## check table frequency for all variables

## DD has 1=yes,2=no coding but data has 0/1 coding. 
## Confirmed with other 0/1 variables that 0 is no and 1 stays yes, so will update the dataset to match the DD
WomenQuestions_Current_HormoneTherapy$CurCur[WomenQuestions_Current_HormoneTherapy$CurCur==0] &lt;- 2
WomenQuestions_Current_HormoneTherapy$CurCur &lt;- as.integer(WomenQuestions_Current_HormoneTherapy$CurCur)

## convert &quot;&quot; to NA for chr columns
chrcols &lt;- colnames(WomenQuestions_Current_HormoneTherapy)[sapply(WomenQuestions_Current_HormoneTherapy, is.character)]
WomenQuestions_Current_HormoneTherapy[chrcols] &lt;- lapply(WomenQuestions_Current_HormoneTherapy[chrcols], function(x) ifelse(x == &quot;&quot;, NA, x))

## drop update_stamp column
WomenQuestions_Current_HormoneTherapy &lt;- subset(WomenQuestions_Current_HormoneTherapy,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(WomenQuestions_Current_HormoneTherapy)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    466 obs. of  10 variables:
##  $ wrapnum          : chr  &quot;wrap0123&quot; &quot;wrap0228&quot; &quot;wrap0381&quot; &quot;wrap0381&quot; ...
##  $ WRAPNo           : chr  &quot;0123&quot; &quot;0228&quot; &quot;0381&quot; &quot;0381&quot; ...
##  $ VisNo            : int  2 4 2 2 2 2 3 2 4 4 ...
##  $ Cur_HT_Repeat_Key: int  1 1 3 2 1 1 1 1 1 3 ...
##  $ CurMed           : chr  NA NA &quot;Testosterone cream&quot; &quot;Vagifem&quot; ...
##  $ CurCur           : int  NA NA 1 1 1 1 NA 1 NA NA ...
##  $ CurType          : chr  NA NA &quot;cream&quot; &quot;tab&quot; ...
##  $ CurAge           : int  NA NA 51 52 51 46 NA 55 NA NA ...
##  $ CurLong          : chr  NA NA &quot;2.5 years&quot; &quot;1.5 years&quot; ...
##  $ CurSide          : int  NA NA 1 1 1 1 NA 1 NA NA ...</code></pre>
<p><br> <br></p>
</div>
<div id="womenquestions_nonpresc_hormonetherapy" class="section level2">
<h2>WomenQuestions_NonPresc_HormoneTherapy</h2>
<p>Women Questions - Non-Prescription Hormone Therapy</p>
<pre class="r"><code>## load data
WomenQuestions_NonPresc_HormoneTherapy &lt;- read.csv(paste0(wrap_directory, &quot;WomenQuestions_NonPresc_HormoneTherapy_05Jun2025.csv&quot;))
info(WomenQuestions_NonPresc_HormoneTherapy,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:300, cols:11, inds:198</code></pre>
<pre class="r"><code>str(WomenQuestions_NonPresc_HormoneTherapy)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    300 obs. of  11 variables:
##  $ wrapnum               : chr  &quot;wrap1006&quot; &quot;wrapA0042&quot; &quot;wrap0228&quot; &quot;wrap0381&quot; ...
##  $ WRAPNo                : chr  &quot;1006&quot; &quot;A0042&quot; &quot;0228&quot; &quot;0381&quot; ...
##  $ VisNo                 : int  2 2 4 2 3 1 1 1 1 2 ...
##  $ NonPresc_HT_Repeat_Key: int  1 1 1 1 1 2 1 2 1 2 ...
##  $ Horm                  : chr  &quot;Soy&quot; &quot;Black Cohosh&quot; &quot;I-Cool&quot; &quot;DHEA&quot; ...
##  $ HormCur               : int  2 1 2 1 1 0 0 0 0 2 ...
##  $ HormType              : chr  &quot;&quot; &quot;&quot; &quot;pill&quot; &quot;&quot; ...
##  $ HormAge               : int  48 49 NA 51 NA 48 48 58 55 53 ...
##  $ HormLong              : chr  &quot;1 week&quot; &quot;6 months&quot; &quot;1 month&quot; &quot;2 years&quot; ...
##  $ HormSide              : int  1 1 1 1 1 1 1 1 1 1 ...
##  $ update_stamp          : chr  &quot;2024-09-13 08:54:11.0&quot; &quot;2024-09-13 08:54:11.0&quot; &quot;2024-09-13 08:54:11.0&quot; &quot;2024-09-13 08:54:11.0&quot; ...</code></pre>
<pre class="r"><code>## check table frequency for all variables

## for column HormCur, it has additional value 0 in the data. I did not modify it for now.
table(WomenQuestions_NonPresc_HormoneTherapy$HormCur,useNA = &quot;ifany&quot;)</code></pre>
<pre><code>## 
##    0    1    2 &lt;NA&gt; 
##   11  107  173    9</code></pre>
<pre class="r"><code>## convert &quot;&quot; to NA for chr columns
chrcols &lt;- colnames(WomenQuestions_NonPresc_HormoneTherapy)[sapply(WomenQuestions_NonPresc_HormoneTherapy, is.character)]
WomenQuestions_NonPresc_HormoneTherapy[chrcols] &lt;- lapply(WomenQuestions_NonPresc_HormoneTherapy[chrcols], function(x) ifelse(x == &quot;&quot;, NA, x))

## drop update_stamp column
WomenQuestions_NonPresc_HormoneTherapy &lt;- subset(WomenQuestions_NonPresc_HormoneTherapy,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(WomenQuestions_NonPresc_HormoneTherapy)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    300 obs. of  10 variables:
##  $ wrapnum               : chr  &quot;wrap1006&quot; &quot;wrapA0042&quot; &quot;wrap0228&quot; &quot;wrap0381&quot; ...
##  $ WRAPNo                : chr  &quot;1006&quot; &quot;A0042&quot; &quot;0228&quot; &quot;0381&quot; ...
##  $ VisNo                 : int  2 2 4 2 3 1 1 1 1 2 ...
##  $ NonPresc_HT_Repeat_Key: int  1 1 1 1 1 2 1 2 1 2 ...
##  $ Horm                  : chr  &quot;Soy&quot; &quot;Black Cohosh&quot; &quot;I-Cool&quot; &quot;DHEA&quot; ...
##  $ HormCur               : int  2 1 2 1 1 0 0 0 0 2 ...
##  $ HormType              : chr  NA NA &quot;pill&quot; NA ...
##  $ HormAge               : int  48 49 NA 51 NA 48 48 58 55 53 ...
##  $ HormLong              : chr  &quot;1 week&quot; &quot;6 months&quot; &quot;1 month&quot; &quot;2 years&quot; ...
##  $ HormSide              : int  1 1 1 1 1 1 1 1 1 1 ...</code></pre>
<p><br> <br></p>
</div>
<div id="womenquestions_past_hormonetherapy" class="section level2">
<h2>WomenQuestions_Past_HormoneTherapy</h2>
<p>Women Questions - Past Hormone Therapy</p>
<pre class="r"><code>## load data
WomenQuestions_Past_HormoneTherapy &lt;- read.csv(paste0(wrap_directory, &quot;WomenQuestions_Past_HormoneTherapy_05Jun2025.csv&quot;))
info(WomenQuestions_Past_HormoneTherapy,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:492, cols:12, inds:379</code></pre>
<pre class="r"><code>str(WomenQuestions_Past_HormoneTherapy)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    492 obs. of  12 variables:
##  $ wrapnum           : chr  &quot;wrap0123&quot; &quot;wrap0825&quot; &quot;wrap0381&quot; &quot;wrap0540&quot; ...
##  $ WRAPNo            : chr  &quot;0123&quot; &quot;0825&quot; &quot;0381&quot; &quot;0540&quot; ...
##  $ VisNo             : int  2 2 2 2 2 1 2 2 2 2 ...
##  $ Past_HT_Repeat_Key: int  1 1 1 2 1 1 1 1 1 2 ...
##  $ PastMed           : chr  &quot;&quot; &quot;&quot; &quot;Vag estrogen cream&quot; &quot;Vagifem&quot; ...
##  $ PastCur           : int  NA 2 2 NA NA 0 2 2 2 NA ...
##  $ PastType          : chr  &quot;&quot; &quot;pill&quot; &quot;cream&quot; &quot;suppository cream&quot; ...
##  $ PastAge           : int  NA 53 51 58 54 45 55 47 47 56 ...
##  $ PastLong          : chr  &quot;&quot; &quot;10 years&quot; &quot;1 year&quot; &quot;4 yr&quot; ...
##  $ PastSide          : int  NA 1 1 1 1 1 1 1 1 1 ...
##  $ PastStop          : chr  &quot;&quot; &quot;&quot; &quot;too goopy&quot; &quot;&quot; ...
##  $ update_stamp      : chr  &quot;2024-09-13 08:54:12.0&quot; &quot;2024-09-13 08:54:12.0&quot; &quot;2024-09-13 08:54:12.0&quot; &quot;2024-09-13 08:54:12.0&quot; ...</code></pre>
<pre class="r"><code>## check table frequency for all variables

## for column PastCur, it has additional value 0 in the data. I did not modify it for now.
## in the dictionay, this variable should have two values: 1=Yes;2=No
## should I modify the DD? 
table(WomenQuestions_Past_HormoneTherapy$PastCur,useNA = &quot;ifany&quot;)</code></pre>
<pre><code>## 
##    0    2 &lt;NA&gt; 
##   26  266  200</code></pre>
<pre class="r"><code>## convert &quot;&quot; to NA for chr columns
chrcols &lt;- colnames(WomenQuestions_Past_HormoneTherapy)[sapply(WomenQuestions_Past_HormoneTherapy, is.character)]
WomenQuestions_Past_HormoneTherapy[chrcols] &lt;- lapply(WomenQuestions_Past_HormoneTherapy[chrcols], function(x) ifelse(x == &quot;&quot;, NA, x))

## drop update_stamp column
WomenQuestions_Past_HormoneTherapy &lt;- subset(WomenQuestions_Past_HormoneTherapy,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(WomenQuestions_Past_HormoneTherapy)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    492 obs. of  11 variables:
##  $ wrapnum           : chr  &quot;wrap0123&quot; &quot;wrap0825&quot; &quot;wrap0381&quot; &quot;wrap0540&quot; ...
##  $ WRAPNo            : chr  &quot;0123&quot; &quot;0825&quot; &quot;0381&quot; &quot;0540&quot; ...
##  $ VisNo             : int  2 2 2 2 2 1 2 2 2 2 ...
##  $ Past_HT_Repeat_Key: int  1 1 1 2 1 1 1 1 1 2 ...
##  $ PastMed           : chr  NA NA &quot;Vag estrogen cream&quot; &quot;Vagifem&quot; ...
##  $ PastCur           : int  NA 2 2 NA NA 0 2 2 2 NA ...
##  $ PastType          : chr  NA &quot;pill&quot; &quot;cream&quot; &quot;suppository cream&quot; ...
##  $ PastAge           : int  NA 53 51 58 54 45 55 47 47 56 ...
##  $ PastLong          : chr  NA &quot;10 years&quot; &quot;1 year&quot; &quot;4 yr&quot; ...
##  $ PastSide          : int  NA 1 1 1 1 1 1 1 1 1 ...
##  $ PastStop          : chr  NA NA &quot;too goopy&quot; NA ...</code></pre>
<p><br> <br></p>
</div>
<div id="womenquestions_serm" class="section level2">
<h2>WomenQuestions_SERM</h2>
<p>Women Questions - SERM (Antiestrogen medication)</p>
<pre class="r"><code>## load data
WomenQuestions_SERM &lt;- read.csv(paste0(wrap_directory, &quot;WomenQuestions_SERM_05Jun2025.csv&quot;))
info(WomenQuestions_SERM,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:76, cols:11, inds:58</code></pre>
<pre class="r"><code>str(WomenQuestions_SERM)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    76 obs. of  11 variables:
##  $ wrapnum        : chr  &quot;wrap0838&quot; &quot;wrap0027&quot; &quot;wrap0123&quot; &quot;wrap0743&quot; ...
##  $ WRAPNo         : chr  &quot;0838&quot; &quot;0027&quot; &quot;0123&quot; &quot;0743&quot; ...
##  $ VisNo          : int  3 3 2 2 2 2 2 2 2 4 ...
##  $ SERM_Repeat_Key: int  1 1 1 1 1 1 1 1 1 1 ...
##  $ SERM           : chr  &quot;&quot; &quot;&quot; &quot;Tamoxifen&quot; &quot;Tamoxifen (UW Star Study)&quot; ...
##  $ SERMCur        : int  NA NA 2 2 2 2 2 1 1 NA ...
##  $ SERMType       : logi  NA NA NA NA NA NA ...
##  $ SERMAge        : int  NA NA 56 54 53 41 35 59 61 NA ...
##  $ SERMLong       : chr  &quot;&quot; &quot;&quot; &quot;5 yrs&quot; &quot;2 yrs&quot; ...
##  $ SERMSide       : int  NA NA 1 1 1 1 1 1 1 NA ...
##  $ update_stamp   : chr  &quot;2025-01-24 08:18:53.0&quot; &quot;2024-09-13 08:54:12.0&quot; &quot;2024-09-13 08:54:12.0&quot; &quot;2024-09-13 08:54:12.0&quot; ...</code></pre>
<pre class="r"><code>## check table frequency for all variables

## for column SERMCur, it has additional value 0 in the data. I did not modify it for now.
table(WomenQuestions_SERM$SERMCur,useNA = &quot;ifany&quot;)</code></pre>
<pre><code>## 
##    0    1    2 &lt;NA&gt; 
##    3   21   25   27</code></pre>
<pre class="r"><code>## convert &quot;&quot; to NA for chr columns
chrcols &lt;- colnames(WomenQuestions_SERM)[sapply(WomenQuestions_SERM, is.character)]
WomenQuestions_SERM[chrcols] &lt;- lapply(WomenQuestions_SERM[chrcols], function(x) ifelse(x == &quot;&quot;, NA, x))

## convert column SERMType from logi to chr
WomenQuestions_SERM[[&quot;SERMType&quot;]] &lt;- as.character(WomenQuestions_SERM[[&quot;SERMType&quot;]])

## drop update_stamp column
WomenQuestions_SERM &lt;- subset(WomenQuestions_SERM,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(WomenQuestions_SERM)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    76 obs. of  10 variables:
##  $ wrapnum        : chr  &quot;wrap0838&quot; &quot;wrap0027&quot; &quot;wrap0123&quot; &quot;wrap0743&quot; ...
##  $ WRAPNo         : chr  &quot;0838&quot; &quot;0027&quot; &quot;0123&quot; &quot;0743&quot; ...
##  $ VisNo          : int  3 3 2 2 2 2 2 2 2 4 ...
##  $ SERM_Repeat_Key: int  1 1 1 1 1 1 1 1 1 1 ...
##  $ SERM           : chr  NA NA &quot;Tamoxifen&quot; &quot;Tamoxifen (UW Star Study)&quot; ...
##  $ SERMCur        : int  NA NA 2 2 2 2 2 1 1 NA ...
##  $ SERMType       : chr  NA NA NA NA ...
##  $ SERMAge        : int  NA NA 56 54 53 41 35 59 61 NA ...
##  $ SERMLong       : chr  NA NA &quot;5 yrs&quot; &quot;2 yrs&quot; ...
##  $ SERMSide       : int  NA NA 1 1 1 1 1 1 1 NA ...</code></pre>
<p><br> <br></p>
</div>
<div id="workhistory" class="section level2">
<h2>WorkHistory</h2>
<p>Participant Work History</p>
<pre class="r"><code>## load data
WorkHistory &lt;- read.csv(paste0(wrap_directory, &quot;WorkHistory_05Jun2025.csv&quot;))
info(WorkHistory,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:3752, cols:11, inds:1685</code></pre>
<pre class="r"><code>str(WorkHistory)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    3752 obs. of  11 variables:
##  $ wrapnum                : chr  &quot;wrap0001&quot; &quot;wrap0001&quot; &quot;wrap0001&quot; &quot;wrap0002&quot; ...
##  $ WRAPNo                 : chr  &quot;0001&quot; &quot;0001&quot; &quot;0001&quot; &quot;0002&quot; ...
##  $ work_history_repeat_key: int  1 2 3 2 3 1 2 1 3 2 ...
##  $ JobYrs                 : chr  &quot;3&quot; &quot;6&quot; &quot;5.5&quot; &quot;35&quot; ...
##  $ OccSoc                 : chr  &quot;43-3021.02&quot; &quot;43-6013.00&quot; &quot;43-3071.00&quot; &quot;11-1021.00&quot; ...
##  $ OccDOT                 : chr  &quot;214.362-014&quot; &quot;201.362-014&quot; &quot;211.362-018&quot; &quot;185.117-010&quot; ...
##  $ ComData                : int  3 3 3 1 1 3 1 3 3 0 ...
##  $ ComPeop                : int  6 6 6 1 2 8 3 6 5 3 ...
##  $ ComThg                 : int  2 2 2 7 1 2 7 2 7 7 ...
##  $ OccZone                : int  3 2 2 4 4 4 2 2 2 4 ...
##  $ update_stamp           : chr  &quot;2024-09-13 08:54:14.0&quot; &quot;2024-09-13 08:54:14.0&quot; &quot;2024-09-13 08:54:14.0&quot; &quot;2024-09-13 08:54:14.0&quot; ...</code></pre>
<pre class="r"><code>## check table frequency for all variables

## convert &quot;&quot; to NA for chr columns
chrcols &lt;- colnames(WorkHistory)[sapply(WorkHistory, is.character)]
WorkHistory[chrcols] &lt;- lapply(WorkHistory[chrcols], function(x) ifelse(x == &quot;&quot;, NA, x))

## drop update_stamp column
WorkHistory &lt;- subset(WorkHistory,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(WorkHistory)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    3752 obs. of  10 variables:
##  $ wrapnum                : chr  &quot;wrap0001&quot; &quot;wrap0001&quot; &quot;wrap0001&quot; &quot;wrap0002&quot; ...
##  $ WRAPNo                 : chr  &quot;0001&quot; &quot;0001&quot; &quot;0001&quot; &quot;0002&quot; ...
##  $ work_history_repeat_key: int  1 2 3 2 3 1 2 1 3 2 ...
##  $ JobYrs                 : chr  &quot;3&quot; &quot;6&quot; &quot;5.5&quot; &quot;35&quot; ...
##  $ OccSoc                 : chr  &quot;43-3021.02&quot; &quot;43-6013.00&quot; &quot;43-3071.00&quot; &quot;11-1021.00&quot; ...
##  $ OccDOT                 : chr  &quot;214.362-014&quot; &quot;201.362-014&quot; &quot;211.362-018&quot; &quot;185.117-010&quot; ...
##  $ ComData                : int  3 3 3 1 1 3 1 3 3 0 ...
##  $ ComPeop                : int  6 6 6 1 2 8 3 6 5 3 ...
##  $ ComThg                 : int  2 2 2 7 1 2 7 2 7 7 ...
##  $ OccZone                : int  3 2 2 4 4 4 2 2 2 4 ...</code></pre>
<p><br> <br></p>
</div>
<div id="workhistory_family" class="section level2">
<h2>WorkHistory_Family</h2>
<p>Participant Family Work History</p>
<pre class="r"><code>## load data
WorkHistory_Family &lt;- read.csv(paste0(wrap_directory, &quot;WorkHistory_Family_05Jun2025.csv&quot;))
info(WorkHistory_Family,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:1242, cols:17, inds:1242</code></pre>
<pre class="r"><code>str(WorkHistory_Family)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    1242 obs. of  17 variables:
##  $ wrapnum     : chr  &quot;wrap0003&quot; &quot;wrap0007&quot; &quot;wrap0014&quot; &quot;wrap0017&quot; ...
##  $ WRAPNo      : chr  &quot;0003&quot; &quot;0007&quot; &quot;0014&quot; &quot;0017&quot; ...
##  $ FaOccSoc    : chr  &quot;41-4012.00&quot; &quot;41-4012.00&quot; &quot;51-4111.00&quot; &quot;11-1021.00&quot; ...
##  $ FaDOT       : chr  &quot;279.357-014&quot; &quot;279.357-014&quot; &quot;601.260-010&quot; &quot;185.117-010&quot; ...
##  $ FaComData   : int  3 3 2 1 1 6 1 1 0 1 ...
##  $ FaComPeop   : int  5 5 6 1 3 6 6 1 6 6 ...
##  $ FaComThg    : int  7 7 0 7 0 3 2 7 1 7 ...
##  $ FaZone      : int  3 3 3 4 3 2 4 3 4 4 ...
##  $ FaSchool    : int  7 7 5 8 4 5 6 6 10 5 ...
##  $ MoOccSoc    : chr  &quot;11-1111.11&quot; &quot;11-1111.11&quot; &quot;43-3031.00&quot; &quot;11-1021.00&quot; ...
##  $ MoDOT       : chr  &quot;&quot; &quot;&quot; &quot;210.382-014&quot; &quot;185.117-010&quot; ...
##  $ MoComData   : int  NA NA 3 1 3 1 3 1 NA NA ...
##  $ MoComPeop   : int  NA NA 8 1 8 1 6 2 NA NA ...
##  $ MoComThg    : int  NA NA 2 7 2 7 2 4 NA NA ...
##  $ MoZone      : int  1 1 3 4 3 4 2 3 1 1 ...
##  $ MoSchool    : int  5 6 5 7 4 5 6 6 7 5 ...
##  $ update_stamp: chr  &quot;2024-09-13 08:54:15.0&quot; &quot;2024-09-13 08:54:15.0&quot; &quot;2024-09-13 08:54:15.0&quot; &quot;2024-09-13 08:54:15.0&quot; ...</code></pre>
<pre class="r"><code>## check table frequency for all variables

## convert &quot;&quot; to NA for chr columns
chrcols &lt;- colnames(WorkHistory_Family)[sapply(WorkHistory_Family, is.character)]
WorkHistory_Family[chrcols] &lt;- lapply(WorkHistory_Family[chrcols], function(x) ifelse(x == &quot;&quot;, NA, x))

## drop update_stamp column
WorkHistory_Family &lt;- subset(WorkHistory_Family,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(WorkHistory_Family)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    1242 obs. of  16 variables:
##  $ wrapnum  : chr  &quot;wrap0003&quot; &quot;wrap0007&quot; &quot;wrap0014&quot; &quot;wrap0017&quot; ...
##  $ WRAPNo   : chr  &quot;0003&quot; &quot;0007&quot; &quot;0014&quot; &quot;0017&quot; ...
##  $ FaOccSoc : chr  &quot;41-4012.00&quot; &quot;41-4012.00&quot; &quot;51-4111.00&quot; &quot;11-1021.00&quot; ...
##  $ FaDOT    : chr  &quot;279.357-014&quot; &quot;279.357-014&quot; &quot;601.260-010&quot; &quot;185.117-010&quot; ...
##  $ FaComData: int  3 3 2 1 1 6 1 1 0 1 ...
##  $ FaComPeop: int  5 5 6 1 3 6 6 1 6 6 ...
##  $ FaComThg : int  7 7 0 7 0 3 2 7 1 7 ...
##  $ FaZone   : int  3 3 3 4 3 2 4 3 4 4 ...
##  $ FaSchool : int  7 7 5 8 4 5 6 6 10 5 ...
##  $ MoOccSoc : chr  &quot;11-1111.11&quot; &quot;11-1111.11&quot; &quot;43-3031.00&quot; &quot;11-1021.00&quot; ...
##  $ MoDOT    : chr  NA NA &quot;210.382-014&quot; &quot;185.117-010&quot; ...
##  $ MoComData: int  NA NA 3 1 3 1 3 1 NA NA ...
##  $ MoComPeop: int  NA NA 8 1 8 1 6 2 NA NA ...
##  $ MoComThg : int  NA NA 2 7 2 7 2 4 NA NA ...
##  $ MoZone   : int  1 1 3 4 3 4 2 3 1 1 ...
##  $ MoSchool : int  5 6 5 7 4 5 6 6 7 5 ...</code></pre>
<p><br> <br></p>
</div>
<div id="wrap_to_dbid" class="section level2">
<h2>WRAP_to_DBID</h2>
<p>WRAP to DBID Identifier crosswalk table</p>
<pre class="r"><code>## load data
WRAP_to_DBID &lt;- read.csv(paste0(wrap_directory, &quot;WRAP_to_DBID_05Jun2025.csv&quot;))
info(WRAP_to_DBID,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:1717, cols:4, inds:1717</code></pre>
<pre class="r"><code>str(WRAP_to_DBID)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    1717 obs. of  4 variables:
##  $ wrapnum     : chr  &quot;wrap0001&quot; &quot;wrap0002&quot; &quot;wrap0003&quot; &quot;wrap0004&quot; ...
##  $ WRAPNo      : chr  &quot;0001&quot; &quot;0002&quot; &quot;0003&quot; &quot;0004&quot; ...
##  $ DBID        : int  1436 1430 1418 1443 1431 1442 1422 1441 1449 1462 ...
##  $ update_stamp: chr  &quot;2024-09-13 08:54:15.0&quot; &quot;2024-09-13 08:54:15.0&quot; &quot;2024-09-13 08:54:15.0&quot; &quot;2024-09-13 08:54:15.0&quot; ...</code></pre>
<pre class="r"><code>## drop update_stamp column
WRAP_to_DBID &lt;- subset(WRAP_to_DBID,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(WRAP_to_DBID)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    1717 obs. of  3 variables:
##  $ wrapnum: chr  &quot;wrap0001&quot; &quot;wrap0002&quot; &quot;wrap0003&quot; &quot;wrap0004&quot; ...
##  $ WRAPNo : chr  &quot;0001&quot; &quot;0002&quot; &quot;0003&quot; &quot;0004&quot; ...
##  $ DBID   : int  1436 1430 1418 1443 1431 1442 1422 1441 1449 1462 ...</code></pre>
<p><br> <br></p>
</div>
<div id="imaging_metadata" class="section level2">
<h2>Imaging_metadata</h2>
<p>WRAP Imaging Metadata</p>
<pre class="r"><code>## load data
Imaging_metadata &lt;- read.csv(paste0(wrap_directory, &quot;Imaging_metadata_05Jun2025.csv&quot;))
info(Imaging_metadata,&quot;wrapnum&quot;)</code></pre>
<pre><code>## #obs:14999, cols:45, inds:818</code></pre>
<pre class="r"><code>str(Imaging_metadata)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    14999 obs. of  45 variables:
##  $ wrapnum                    : chr  &quot;wrap0457&quot; &quot;wrap0908&quot; &quot;wrapL0238&quot; &quot;wrap0900&quot; ...
##  $ imaging_visit              : chr  &quot;Imaging Visit 10&quot; &quot;Imaging Visit 2&quot; &quot;Imaging Visit 5&quot; &quot;Imaging Visit 5&quot; ...
##  $ age                        : int  63 61 69 80 50 53 67 64 69 55 ...
##  $ Days_Since_Baseline        : int  6150 2775 3129 5337 1707 2222 2165 1175 1543 3076 ...
##  $ Modality                   : chr  &quot;MR&quot; &quot;MR&quot; &quot;MR&quot; &quot;MR&quot; ...
##  $ Manufacturer               : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ ManufacturersModelName     : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ Anaesthesia                : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ BodyPart                   : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ BodyWeight                 : num  NA NA NA NA NA NA NA NA NA NA ...
##  $ BodyWeightUnits            : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ TracerName                 : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ TracerIsotope              : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ InstitutionName            : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ DeviceSerialNumber         : num  NA NA NA NA NA NA NA NA NA NA ...
##  $ InjectedRadioactivity      : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ InjectedRadioactivityUnits : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ ScanStart                  : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ ScanStartUnits             : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ InjectionStart             : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ InjectionStartUnits        : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ ModeOfAdministration       : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ FrameTimesStart            : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ FrameTimesStartUnits       : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ FrameTimesEnd              : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ FrameTimesEndUnits         : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ FrameDuration              : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ FrameDurationUnits         : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ DecayCorrectionFactor      : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ ImageDecayCorrected        : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ ImageDecayCorrectionTime   : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ AcquisitionMode            : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ ReconMethodName            : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ ReconMethodParameterLabels : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ ReconMethodParameterUnits  : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ ReconMethodParameterValues : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ AttenuationCorrection      : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ MethodImplementationVersion: num  NA NA NA NA NA NA NA NA NA NA ...
##  $ ReconFilterType            : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ ReconFilterSize            : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ ReconFilterSizeUnits       : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ SoftwareVersion            : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ LONIImageID                : int  10896049 10920338 10914935 10914941 10920337 10920330 10914934 10914933 10914942 10914931 ...
##  $ CorrectedImage             : chr  &quot;&quot; &quot;&quot; &quot;&quot; &quot;&quot; ...
##  $ update_stamp               : chr  &quot;2024-09-12 14:44:30.0&quot; &quot;2024-09-12 14:44:30.0&quot; &quot;2024-09-12 14:44:30.0&quot; &quot;2024-09-12 14:44:30.0&quot; ...</code></pre>
<pre class="r"><code>## check table frequency for all variables

## convert &quot;&quot; to NA for chr columns
chrcols &lt;- colnames(Imaging_metadata)[sapply(Imaging_metadata, is.character)]
Imaging_metadata[chrcols] &lt;- lapply(Imaging_metadata[chrcols], function(x) ifelse(x == &quot;&quot;, NA, x))

## drop update_stamp column
Imaging_metadata &lt;- subset(Imaging_metadata,select = -c(update_stamp))</code></pre>
<pre class="r"><code>str(Imaging_metadata)</code></pre>
<details>
<summary>
Click for details
</summary>
<pre><code>## &#39;data.frame&#39;:    14999 obs. of  44 variables:
##  $ wrapnum                    : chr  &quot;wrap0457&quot; &quot;wrap0908&quot; &quot;wrapL0238&quot; &quot;wrap0900&quot; ...
##  $ imaging_visit              : chr  &quot;Imaging Visit 10&quot; &quot;Imaging Visit 2&quot; &quot;Imaging Visit 5&quot; &quot;Imaging Visit 5&quot; ...
##  $ age                        : int  63 61 69 80 50 53 67 64 69 55 ...
##  $ Days_Since_Baseline        : int  6150 2775 3129 5337 1707 2222 2165 1175 1543 3076 ...
##  $ Modality                   : chr  &quot;MR&quot; &quot;MR&quot; &quot;MR&quot; &quot;MR&quot; ...
##  $ Manufacturer               : chr  NA NA NA NA ...
##  $ ManufacturersModelName     : chr  NA NA NA NA ...
##  $ Anaesthesia                : chr  NA NA NA NA ...
##  $ BodyPart                   : chr  NA NA NA NA ...
##  $ BodyWeight                 : num  NA NA NA NA NA NA NA NA NA NA ...
##  $ BodyWeightUnits            : chr  NA NA NA NA ...
##  $ TracerName                 : chr  NA NA NA NA ...
##  $ TracerIsotope              : chr  NA NA NA NA ...
##  $ InstitutionName            : chr  NA NA NA NA ...
##  $ DeviceSerialNumber         : num  NA NA NA NA NA NA NA NA NA NA ...
##  $ InjectedRadioactivity      : int  NA NA NA NA NA NA NA NA NA NA ...
##  $ InjectedRadioactivityUnits : chr  NA NA NA NA ...
##  $ ScanStart                  : chr  NA NA NA NA ...
##  $ ScanStartUnits             : chr  NA NA NA NA ...
##  $ InjectionStart             : chr  NA NA NA NA ...
##  $ InjectionStartUnits        : chr  NA NA NA NA ...
##  $ ModeOfAdministration       : chr  NA NA NA NA ...
##  $ FrameTimesStart            : chr  NA NA NA NA ...
##  $ FrameTimesStartUnits       : chr  NA NA NA NA ...
##  $ FrameTimesEnd              : chr  NA NA NA NA ...
##  $ FrameTimesEndUnits         : chr  NA NA NA NA ...
##  $ FrameDuration              : chr  NA NA NA NA ...
##  $ FrameDurationUnits         : chr  NA NA NA NA ...
##  $ DecayCorrectionFactor      : chr  NA NA NA NA ...
##  $ ImageDecayCorrected        : chr  NA NA NA NA ...
##  $ ImageDecayCorrectionTime   : chr  NA NA NA NA ...
##  $ AcquisitionMode            : chr  NA NA NA NA ...
##  $ ReconMethodName            : chr  NA NA NA NA ...
##  $ ReconMethodParameterLabels : chr  NA NA NA NA ...
##  $ ReconMethodParameterUnits  : chr  NA NA NA NA ...
##  $ ReconMethodParameterValues : chr  NA NA NA NA ...
##  $ AttenuationCorrection      : chr  NA NA NA NA ...
##  $ MethodImplementationVersion: num  NA NA NA NA NA NA NA NA NA NA ...
##  $ ReconFilterType            : chr  NA NA NA NA ...
##  $ ReconFilterSize            : chr  NA NA NA NA ...
##  $ ReconFilterSizeUnits       : chr  NA NA NA NA ...
##  $ SoftwareVersion            : chr  NA NA NA NA ...
##  $ LONIImageID                : int  10896049 10920338 10914935 10914941 10920337 10920330 10914934 10914933 10914942 10914931 ...
##  $ CorrectedImage             : chr  NA NA NA NA ...</code></pre>
<p><br> <br></p>
</div>
</div>
<div id="get-sub-dataset-names" class="section level1">
<h1>Get Sub-Dataset Names</h1>
<pre class="r"><code>df_names &lt;- ls()[sapply(ls(), function(x) is.data.frame(get(x)))] ## 58 datasets</code></pre>
<p><br> <br></p>
</div>
<div id="check-missingness-in-wrapnum-wrapno-visno"
class="section level1">
<h1>Check Missingness in wrapnum | WRAPNo| VisNo</h1>
<pre class="r"><code>vars_to_check &lt;- c(&quot;wrapnum&quot;, &quot;WRAPNo&quot;, &quot;VisNo&quot;)
results_list &lt;- list()

for (df_name in df_names) {
  
  df &lt;- get(df_name)
  
  # For each variable, count NAs or mark Not present
  na_counts &lt;- sapply(vars_to_check, function(var) {
    if (var %in% names(df)) {
      sum(is.na(df[[var]]))
    } else {
      NA
    }
  })
  
  # Create a row with df_name and NA counts
  row &lt;- c(Dataframe = df_name, na_counts)
  results_list[[df_name]] &lt;- row
}
 
# Combine into one dataframe
results_df &lt;- bind_rows(lapply(results_list, as.data.frame.list))

DT::datatable(results_df)</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-0b89a59a8fa46b9346ad" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-0b89a59a8fa46b9346ad">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58"],["Abnormal_Factor_Flags_long","Abnormal_Factor_Flags_wide","AnimalNaming_ItemLevel","APG","AVLTPositionScores","BehavioralObservationChecklist","CA_Genome","CDR","CERAD","CES_D_Scale","CHAMPS","CogStateData","CompositeScoreCalcValues","CompositeScores","ConsensusConference","DailyActivity","Demographics","FactorScores","fqryStatisticalData","Gene_DataCollection","GenHealthDailyAct","HeadInjuries","HeadInjurySub","IADL","Imaging_metadata","IQCode","LearningDifficulties","LIBRA","Med_Lab","Med_Lab_NPDC","Med_Lab_PDC","MedHistory","MedHistory_Admit","MedHistory_SurgBF","MedHistory_SurgS","MIND_Diet","MIND_Diet_ScoredValues","MOCA","Morbidity_Mortality","NeuropsychScores","PersFamMedBack","PFMB_SibWAD_or_OtherMemIssue","QuickDementia","RobustNormZScores","Sleep_MentalAct","Sleep_MentalAct_OthDevice","SubjectiveHearing","TICSM","VisitVitals","WomenQuestions","WomenQuestions_BCP","WomenQuestions_Current_HormoneTherapy","WomenQuestions_NonPresc_HormoneTherapy","WomenQuestions_Past_HormoneTherapy","WomenQuestions_SERM","WorkHistory","WorkHistory_Family","WRAP_to_DBID"],["0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0"],["0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0",null,"0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0"],["0",null,"0",null,"0","0",null,"0",null,"0","0","0","0","0","0","0",null,"0","0",null,"0","0","0","0",null,"0","0","0","0","0","0","0","0","0","0","0","0","0",null,"0","0","0","0","0","0","0","0","0","0","0","0","0","0","0","0",null,null,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Dataframe<\/th>\n      <th>wrapnum<\/th>\n      <th>WRAPNo<\/th>\n      <th>VisNo<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Dataframe","targets":1},{"name":"wrapnum","targets":2},{"name":"WRAPNo","targets":3},{"name":"VisNo","targets":4}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><br> <br></p>
</div>
<div id="duplicates-handling" class="section level1">
<h1>Duplicates Handling</h1>
<div id="duplicates-detection" class="section level2">
<h2>Duplicates Detection</h2>
<pre class="r"><code># Get names of all data frames in the environment
longDfwithDuplicates &lt;- c()
otherwithDuplicates &lt;- c()
index = 0

## following function will do:
## 1. filter out the okay cross-sectional datasets
## 2. for the rest: return the longitudinal/cross-sectional dataset names if duplicates got detected
for (df_name in df_names) {
  df &lt;- get(df_name)
  
  ## filter out the okay cross-sectional datasets
  if (length(unique(df[[&quot;wrapnum&quot;]])) == nrow(df)) {
    index = index + 1
    cat(index,
        &quot;No duplicates found in cross-sectional dataset: &quot;,
        df_name,
        &quot;\n&quot;)
  } else{
    ## Check if columns ID and Visit exist (longitduinal data or not)
    if (all(c(&quot;wrapnum&quot;, &quot;VisNo&quot;) %in% colnames(df))) {
      # Find duplicates using dplyr
      dup_rows &lt;- df %&gt;%
        dplyr::group_by(wrapnum, VisNo) %&gt;%
        dplyr::filter(n() &gt; 1) %&gt;%
        dplyr::ungroup()
      
      # If any duplicates found, assign to new data frame with _Duplicates
      if (nrow(dup_rows) &gt; 0) {
        longDfwithDuplicates &lt;- c(longDfwithDuplicates, df_name)
      } else {
        index = index + 1
        cat(index,
            &quot;No duplicates found in longitudinal dataset: &quot;,
            df_name,
            &quot;\n&quot;)
      }
    } else {
      otherwithDuplicates &lt;- c(otherwithDuplicates, df_name)
    }
  }
}</code></pre>
<pre><code>## 1 No duplicates found in longitudinal dataset:  Abnormal_Factor_Flags_long 
## 2 No duplicates found in cross-sectional dataset:  Abnormal_Factor_Flags_wide 
## 3 No duplicates found in longitudinal dataset:  AnimalNaming_ItemLevel 
## 4 No duplicates found in cross-sectional dataset:  APG 
## 5 No duplicates found in longitudinal dataset:  AVLTPositionScores 
## 6 No duplicates found in longitudinal dataset:  BehavioralObservationChecklist 
## 7 No duplicates found in cross-sectional dataset:  CA_Genome 
## 8 No duplicates found in longitudinal dataset:  CDR 
## 9 No duplicates found in cross-sectional dataset:  CERAD 
## 10 No duplicates found in longitudinal dataset:  CES_D_Scale 
## 11 No duplicates found in longitudinal dataset:  CHAMPS 
## 12 No duplicates found in longitudinal dataset:  CogStateData 
## 13 No duplicates found in longitudinal dataset:  CompositeScoreCalcValues 
## 14 No duplicates found in longitudinal dataset:  CompositeScores 
## 15 No duplicates found in longitudinal dataset:  ConsensusConference 
## 16 No duplicates found in cross-sectional dataset:  DailyActivity 
## 17 No duplicates found in cross-sectional dataset:  Demographics 
## 18 No duplicates found in longitudinal dataset:  FactorScores 
## 19 No duplicates found in longitudinal dataset:  fqryStatisticalData 
## 20 No duplicates found in cross-sectional dataset:  Gene_DataCollection 
## 21 No duplicates found in longitudinal dataset:  GenHealthDailyAct 
## 22 No duplicates found in longitudinal dataset:  HeadInjuries 
## 23 No duplicates found in longitudinal dataset:  IADL 
## 24 No duplicates found in longitudinal dataset:  IQCode 
## 25 No duplicates found in longitudinal dataset:  LearningDifficulties 
## 26 No duplicates found in longitudinal dataset:  LIBRA 
## 27 No duplicates found in longitudinal dataset:  Med_Lab 
## 28 No duplicates found in longitudinal dataset:  MedHistory 
## 29 No duplicates found in longitudinal dataset:  MIND_Diet 
## 30 No duplicates found in longitudinal dataset:  MIND_Diet_ScoredValues 
## 31 No duplicates found in longitudinal dataset:  MOCA 
## 32 No duplicates found in cross-sectional dataset:  Morbidity_Mortality 
## 33 No duplicates found in longitudinal dataset:  NeuropsychScores 
## 34 No duplicates found in longitudinal dataset:  PersFamMedBack 
## 35 No duplicates found in longitudinal dataset:  QuickDementia 
## 36 No duplicates found in longitudinal dataset:  RobustNormZScores 
## 37 No duplicates found in longitudinal dataset:  Sleep_MentalAct 
## 38 No duplicates found in longitudinal dataset:  SubjectiveHearing 
## 39 No duplicates found in longitudinal dataset:  TICSM 
## 40 No duplicates found in longitudinal dataset:  VisitVitals 
## 41 No duplicates found in longitudinal dataset:  WomenQuestions 
## 42 No duplicates found in cross-sectional dataset:  WorkHistory_Family 
## 43 No duplicates found in cross-sectional dataset:  WRAP_to_DBID</code></pre>
<p><br> <br></p>
</div>
<div id="duplicates-handling-per-dataset" class="section level2">
<h2>Duplicates Handling Per Dataset</h2>
<pre class="r"><code>print(otherwithDuplicates)</code></pre>
<pre><code>## [1] &quot;Imaging_metadata&quot; &quot;WorkHistory&quot;</code></pre>
<div id="imaging_metadata-1" class="section level3">
<h3>Imaging_metadata</h3>
<pre class="r"><code>## the imaging_visit is coded inconsistenly across the observations, so I generated a new numeric column for visit index
Imaging_metadata &lt;- Imaging_metadata %&gt;%
  mutate(
    VisNo = str_extract(imaging_visit, &quot;\\d+&quot;) %&gt;% as.numeric()
  )

cat(&quot;Before duplicates handling - wrapnum*VisNo is: &quot;,dupFixCheck(Imaging_metadata,&quot;wrapnum&quot;,&quot;VisNo&quot;),&quot;\n&quot;)</code></pre>
<pre><code>## Before duplicates handling - wrapnum*VisNo is:  3443</code></pre>
<pre class="r"><code>## group by subject ID and visit index, we are going to retain only one obs per group. The selection criteria is as following: 
## 1. pick the obs with least missing values
## 2. if same, then randomly pick one

Imaging_metadata &lt;- Imaging_metadata %&gt;%
  mutate(missing_count = rowSums(is.na(.))) %&gt;%  # Count missing values per row
  group_by(wrapnum, VisNo) %&gt;%
  arrange(missing_count) %&gt;%
  slice_sample(n = 1, weight_by = (missing_count == min(missing_count))) %&gt;%
  ungroup() %&gt;%
  select(-missing_count) %&gt;% # Remove helper column
  mutate(WRAPNo = gsub(&quot;wrap&quot;, &quot;&quot;, wrapnum)) #make WRAPNo column to use for merging later on

cat(&quot;After duplicates handling - WRAPNo*VisNo is: &quot;,dupFixCheck(Imaging_metadata,&quot;WRAPNo&quot;,&quot;VisNo&quot;),&quot;\n&quot;)</code></pre>
<pre><code>## After duplicates handling - WRAPNo*VisNo is:  3443</code></pre>
<p><br></p>
</div>
<div id="workhistory-1" class="section level3">
<h3>WorkHistory</h3>
<pre class="r"><code>## check unique values for each columns, for columns with just one unique value,
## we leave it out from the pivot_wider function
result &lt;- lapply(names(WorkHistory), function(x) {
  cat(x, &quot; &quot;, length(unique(WorkHistory[[x]])), &quot;\n&quot;)
})</code></pre>
<pre><code>## wrapnum   1685 
## WRAPNo   1685 
## work_history_repeat_key   6 
## JobYrs   127 
## OccSoc   568 
## OccDOT   934 
## ComData   8 
## ComPeop   10 
## ComThg   9 
## OccZone   7</code></pre>
<pre class="r"><code>cat(&quot;Before duplicates handling: &quot;,dim(WorkHistory),&quot;\n&quot;)</code></pre>
<pre><code>## Before duplicates handling:  3752 10</code></pre>
<pre class="r"><code>tmp &lt;- WorkHistory %&gt;%
  pivot_wider(
    id_cols = c(wrapnum, WRAPNo),
    names_from = work_history_repeat_key,
    values_from = c(JobYrs:OccZone),
    names_sep = &quot;_&quot;
  )
## drop columns if all obs are missing
WorkHistory &lt;- tmp[, colSums(!is.na(tmp)) &gt; 0]

cat(&quot;After duplicates handling: &quot;,dim(WorkHistory),&quot;\n&quot;)</code></pre>
<pre><code>## After duplicates handling:  1685 44</code></pre>
<p><br></p>
</div>
<div id="headinjurysub-1" class="section level3">
<h3>HeadInjurySub</h3>
<pre class="r"><code>## check unique values for each columns, for columns with just one unique value,
## we leave it out from the pivot_wider function
result &lt;- lapply(names(HeadInjurySub), function(x) {
  cat(x, &quot; &quot;, length(unique(HeadInjurySub[[x]])), &quot;\n&quot;)
})</code></pre>
<pre><code>## wrapnum   489 
## WRAPNo   489 
## VisNo   9 
## RowNumber   10 
## agehi   80 
## hicause   5 
## memhi   4 
## loc   4 
## locdur   8 
## awake   4 
## mdhi   4 
## scanhi   4 
## hosphi   4 
## othinj   6 
## treata   7 
## treatb   7 
## treatc   7 
## cognhi   4 
## cognow   4</code></pre>
<pre class="r"><code>cat(&quot;Before duplicates handling - WRAPNo*VisNo is: &quot;,dupFixCheck(HeadInjurySub,&quot;WRAPNo&quot;,&quot;VisNo&quot;),&quot;\n&quot;)</code></pre>
<pre><code>## Before duplicates handling - WRAPNo*VisNo is:  578</code></pre>
<pre class="r"><code>tmp &lt;- HeadInjurySub %&gt;%
  pivot_wider(
    id_cols = c(wrapnum, WRAPNo, VisNo),
    names_from = RowNumber,
    values_from = c(agehi:cognow),
    names_sep = &quot;_&quot;
  )
## drop columns if all obs are missing
HeadInjurySub &lt;- tmp[, colSums(!is.na(tmp)) &gt; 0]

cat(&quot;After duplicates handling - WRAPNo*VisNo is: &quot;,dupFixCheck(HeadInjurySub,&quot;WRAPNo&quot;,&quot;VisNo&quot;),&quot;\n&quot;)</code></pre>
<pre><code>## After duplicates handling - WRAPNo*VisNo is:  578</code></pre>
<p><br></p>
</div>
<div id="med_lab_npdc-1" class="section level3">
<h3>Med_Lab_NPDC</h3>
<pre class="r"><code>result &lt;- lapply(names(Med_Lab_NPDC), function(x) {
  cat(x, &quot; &quot;, length(unique(Med_Lab_NPDC[[x]])), &quot;\n&quot;)
})</code></pre>
<pre><code>## wrapnum   1631 
## WRAPNo   1631 
## VisNo   9 
## npdc_repeat_key   33 
## npdcd   1005 
## mg   6 
## npdcf   8 
## npdcname   12437 
## npdcclass   377</code></pre>
<pre class="r"><code>cat(&quot;Before duplicates handling - WRAPNo*VisNo is: &quot;,dupFixCheck(Med_Lab_NPDC,&quot;WRAPNo&quot;,&quot;VisNo&quot;),&quot;\n&quot;)</code></pre>
<pre><code>## Before duplicates handling - WRAPNo*VisNo is:  6882</code></pre>
<pre class="r"><code>tmp &lt;- Med_Lab_NPDC %&gt;%
  pivot_wider(
    id_cols = c(wrapnum, WRAPNo, VisNo),
    names_from = npdc_repeat_key,
    values_from = c(npdcd:npdcclass),
    names_sep = &quot;_&quot;
  )
## drop columns if all obs are missing
Med_Lab_NPDC &lt;- tmp[, colSums(!is.na(tmp)) &gt; 0]

cat(&quot;After duplicates handling - WRAPNo*VisNo is: &quot;,dupFixCheck(Med_Lab_NPDC,&quot;WRAPNo&quot;,&quot;VisNo&quot;),&quot;\n&quot;)</code></pre>
<pre><code>## After duplicates handling - WRAPNo*VisNo is:  6882</code></pre>
<p><br></p>
</div>
<div id="med_lab_pdc-1" class="section level3">
<h3>Med_Lab_PDC</h3>
<pre class="r"><code>result &lt;- lapply(names(Med_Lab_PDC), function(x) {
  cat(x, &quot; &quot;, length(unique(Med_Lab_PDC[[x]])), &quot;\n&quot;)
})</code></pre>
<pre><code>## wrapnum   1577 
## WRAPNo   1577 
## VisNo   9 
## pdc_repeat_key   32 
## pdcd   754 
## mg   6 
## pdcf   8 
## pdcname   3723 
## pdcclass   440</code></pre>
<pre class="r"><code>cat(&quot;Before duplicates handling - WRAPNo*VisNo is: &quot;,dupFixCheck(Med_Lab_PDC,&quot;WRAPNo&quot;,&quot;VisNo&quot;),&quot;\n&quot;)</code></pre>
<pre><code>## Before duplicates handling - WRAPNo*VisNo is:  6298</code></pre>
<pre class="r"><code>tmp &lt;- Med_Lab_PDC %&gt;%
  pivot_wider(
    id_cols = c(wrapnum, WRAPNo, VisNo),
    names_from = pdc_repeat_key,
    values_from = c(pdcd:pdcclass),
    names_sep = &quot;_&quot;
  )
## drop columns if all obs are missing
Med_Lab_PDC &lt;- tmp[, colSums(!is.na(tmp)) &gt; 0]

cat(&quot;After duplicates handling - WRAPNo*VisNo is: &quot;,dupFixCheck(Med_Lab_PDC,&quot;WRAPNo&quot;,&quot;VisNo&quot;),&quot;\n&quot;)</code></pre>
<pre><code>## After duplicates handling - WRAPNo*VisNo is:  6298</code></pre>
<p><br></p>
</div>
<div id="medhistory_admit-1" class="section level3">
<h3>MedHistory_Admit</h3>
<pre class="r"><code>result &lt;- lapply(names(MedHistory_Admit), function(x) {
  cat(x, &quot; &quot;, length(unique(MedHistory_Admit[[x]])), &quot;\n&quot;)
})</code></pre>
<pre><code>## wrapnum   603 
## WRAPNo   603 
## VisNo   9 
## AdmitSRsn   763 
## AdmitS_Repeat_Key   8</code></pre>
<pre class="r"><code>cat(&quot;Before duplicates handling - WRAPNo*VisNo is: &quot;,dupFixCheck(MedHistory_Admit,&quot;WRAPNo&quot;,&quot;VisNo&quot;),&quot;\n&quot;)</code></pre>
<pre><code>## Before duplicates handling - WRAPNo*VisNo is:  782</code></pre>
<pre class="r"><code>tmp &lt;- MedHistory_Admit %&gt;%
  pivot_wider(
    id_cols = c(wrapnum, WRAPNo, VisNo),
    names_from = AdmitS_Repeat_Key,
    values_from = c(AdmitSRsn),
    names_prefix = &quot;AdmitSRsn_&quot;
  )
## drop columns if all obs are missing
MedHistory_Admit &lt;- tmp[, colSums(!is.na(tmp)) &gt; 0]

cat(&quot;After duplicates handling - WRAPNo*VisNo is: &quot;,dupFixCheck(MedHistory_Admit,&quot;WRAPNo&quot;,&quot;VisNo&quot;),&quot;\n&quot;)</code></pre>
<pre><code>## After duplicates handling - WRAPNo*VisNo is:  782</code></pre>
<p><br></p>
</div>
<div id="medhistory_surgbf-1" class="section level3">
<h3>MedHistory_SurgBF</h3>
<pre class="r"><code>result &lt;- lapply(names(MedHistory_SurgBF), function(x) {
  cat(x, &quot; &quot;, length(unique(MedHistory_SurgBF[[x]])), &quot;\n&quot;)
})</code></pre>
<pre><code>## wrapnum   1115 
## WRAPNo   1115 
## VisNo   3 
## SurgBfRsn   811 
## SurgBF_Repeat_Key   7</code></pre>
<pre class="r"><code>cat(&quot;Before duplicates handling - WRAPNo*VisNo is: &quot;,dupFixCheck(MedHistory_SurgBF,&quot;WRAPNo&quot;,&quot;VisNo&quot;),&quot;\n&quot;)</code></pre>
<pre><code>## Before duplicates handling - WRAPNo*VisNo is:  1115</code></pre>
<pre class="r"><code>tmp &lt;- MedHistory_SurgBF %&gt;%
  pivot_wider(
    id_cols = c(wrapnum, WRAPNo, VisNo),
    names_from = SurgBF_Repeat_Key,
    values_from = c(SurgBfRsn),
    names_prefix = &quot;SurgBfRsn_&quot;
  )
## drop columns if all obs are missing
MedHistory_SurgBF &lt;- tmp[, colSums(!is.na(tmp)) &gt; 0]

cat(&quot;After duplicates handling - WRAPNo*VisNo is: &quot;,dupFixCheck(MedHistory_SurgBF,&quot;WRAPNo&quot;,&quot;VisNo&quot;),&quot;\n&quot;)</code></pre>
<pre><code>## After duplicates handling - WRAPNo*VisNo is:  1115</code></pre>
<p><br></p>
</div>
<div id="medhistory_surgs-1" class="section level3">
<h3>MedHistory_SurgS</h3>
<pre class="r"><code>result &lt;- lapply(names(MedHistory_SurgS), function(x) {
  cat(x, &quot; &quot;, length(unique(MedHistory_SurgS[[x]])), &quot;\n&quot;)
})</code></pre>
<pre><code>## wrapnum   1116 
## WRAPNo   1116 
## VisNo   8 
## SurgSRsn   1959 
## SurgS_Repeat_Key   7</code></pre>
<pre class="r"><code>cat(&quot;Before duplicates handling - WRAPNo*VisNo is: &quot;,dupFixCheck(MedHistory_SurgS,&quot;WRAPNo&quot;,&quot;VisNo&quot;),&quot;\n&quot;)</code></pre>
<pre><code>## Before duplicates handling - WRAPNo*VisNo is:  2503</code></pre>
<pre class="r"><code>tmp &lt;- MedHistory_SurgS %&gt;%
  pivot_wider(
    id_cols = c(wrapnum, WRAPNo, VisNo),
    names_from = SurgS_Repeat_Key,
    values_from = c(SurgSRsn),
    names_prefix = &quot;SurgSRsn_&quot;
  )
## drop columns if all obs are missing
MedHistory_SurgS &lt;- tmp[, colSums(!is.na(tmp)) &gt; 0]

cat(&quot;After duplicates handling - WRAPNo*VisNo is: &quot;,dupFixCheck(MedHistory_SurgS,&quot;WRAPNo&quot;,&quot;VisNo&quot;),&quot;\n&quot;)</code></pre>
<pre><code>## After duplicates handling - WRAPNo*VisNo is:  2503</code></pre>
<p><br></p>
</div>
<div id="pfmb_sibwad_or_othermemissue-1" class="section level3">
<h3>PFMB_SibWAD_or_OtherMemIssue</h3>
<pre class="r"><code>result &lt;- lapply(names(PFMB_SibWAD_or_OtherMemIssue), function(x) {
  cat(x, &quot; &quot;, length(unique(PFMB_SibWAD_or_OtherMemIssue[[x]])), &quot;\n&quot;)
})</code></pre>
<pre><code>## wrapnum   175 
## WRAPNo   175 
## VisNo   8 
## FH7SiblingWithAD_OtherMemIssue_Repeat_Key   3 
## FH7BS1   5</code></pre>
<pre class="r"><code>cat(&quot;Before duplicates handling - WRAPNo*VisNo is: &quot;,dupFixCheck(PFMB_SibWAD_or_OtherMemIssue,&quot;WRAPNo&quot;,&quot;VisNo&quot;),&quot;\n&quot;)</code></pre>
<pre><code>## Before duplicates handling - WRAPNo*VisNo is:  265</code></pre>
<pre class="r"><code>tmp &lt;- PFMB_SibWAD_or_OtherMemIssue %&gt;%
  pivot_wider(
    id_cols = c(wrapnum, WRAPNo, VisNo),
    names_from = FH7SiblingWithAD_OtherMemIssue_Repeat_Key,
    values_from = c(FH7BS1),
    names_prefix = &quot;FH7BS1_&quot;
  )
## drop columns if all obs are missing
PFMB_SibWAD_or_OtherMemIssue &lt;- tmp[, colSums(!is.na(tmp)) &gt; 0]

cat(&quot;After duplicates handling - WRAPNo*VisNo is: &quot;,dupFixCheck(PFMB_SibWAD_or_OtherMemIssue,&quot;WRAPNo&quot;,&quot;VisNo&quot;),&quot;\n&quot;)</code></pre>
<pre><code>## After duplicates handling - WRAPNo*VisNo is:  265</code></pre>
<p><br></p>
</div>
<div id="sleep_mentalact_othdevice-1" class="section level3">
<h3>Sleep_MentalAct_OthDevice</h3>
<pre class="r"><code>result &lt;- lapply(names(Sleep_MentalAct_OthDevice), function(x) {
  cat(x, &quot; &quot;, length(unique(Sleep_MentalAct_OthDevice[[x]])), &quot;\n&quot;)
})</code></pre>
<pre><code>## wrapnum   37 
## WRAPNo   37 
## VisNo   8 
## oth_device_repeat_key   2 
## SA_OthDevFrq   5</code></pre>
<pre class="r"><code>cat(&quot;Before duplicates handling - WRAPNo*VisNo is: &quot;,dupFixCheck(Sleep_MentalAct_OthDevice,&quot;WRAPNo&quot;,&quot;VisNo&quot;),&quot;\n&quot;)</code></pre>
<pre><code>## Before duplicates handling - WRAPNo*VisNo is:  46</code></pre>
<pre class="r"><code>tmp &lt;- Sleep_MentalAct_OthDevice %&gt;%
  pivot_wider(
    id_cols = c(wrapnum, WRAPNo, VisNo),
    names_from = oth_device_repeat_key,
    values_from = c(SA_OthDevFrq),
    names_prefix = &quot;SA_OthDevFrq_&quot;
  )
## drop columns if all obs are missing
Sleep_MentalAct_OthDevice &lt;- tmp[, colSums(!is.na(tmp)) &gt; 0]

cat(&quot;After duplicates handling - WRAPNo*VisNo is: &quot;,dupFixCheck(Sleep_MentalAct_OthDevice,&quot;WRAPNo&quot;,&quot;VisNo&quot;),&quot;\n&quot;)</code></pre>
<pre><code>## After duplicates handling - WRAPNo*VisNo is:  46</code></pre>
<p><br></p>
</div>
<div id="womenquestions_bcp-1" class="section level3">
<h3>WomenQuestions_BCP</h3>
<pre class="r"><code>result &lt;- lapply(names(WomenQuestions_BCP), function(x) {
  cat(x, &quot; &quot;, length(unique(WomenQuestions_BCP[[x]])), &quot;\n&quot;)
})</code></pre>
<pre><code>## wrapnum   866 
## WRAPNo   866 
## VisNo   5 
## BCP_Repeat_Key   3 
## BCPName   289 
## BCPCur   3 
## BCPType   37 
## BCPAge   52 
## BCPLong   216 
## BCPSide   3</code></pre>
<pre class="r"><code>cat(&quot;Before duplicates handling - WRAPNo*VisNo is: &quot;,dupFixCheck(WomenQuestions_BCP,&quot;WRAPNo&quot;,&quot;VisNo&quot;),&quot;\n&quot;)</code></pre>
<pre><code>## Before duplicates handling - WRAPNo*VisNo is:  885</code></pre>
<pre class="r"><code>tmp &lt;- WomenQuestions_BCP %&gt;%
  pivot_wider(
    id_cols = c(wrapnum, WRAPNo, VisNo),
    names_from = BCP_Repeat_Key,
    values_from = c(BCPName:BCPSide),
    names_sep = &quot;_&quot;
  )
## drop columns if all obs are missing
WomenQuestions_BCP &lt;- tmp[, colSums(!is.na(tmp)) &gt; 0]

cat(&quot;After duplicates handling - WRAPNo*VisNo is: &quot;,dupFixCheck(WomenQuestions_BCP,&quot;WRAPNo&quot;,&quot;VisNo&quot;),&quot;\n&quot;)</code></pre>
<pre><code>## After duplicates handling - WRAPNo*VisNo is:  885</code></pre>
<p><br></p>
</div>
<div id="womenquestions_current_hormonetherapy-1"
class="section level3">
<h3>WomenQuestions_Current_HormoneTherapy</h3>
<pre class="r"><code>result &lt;- lapply(names(WomenQuestions_Current_HormoneTherapy), function(x) {
  cat(x, &quot; &quot;, length(unique(WomenQuestions_Current_HormoneTherapy[[x]])), &quot;\n&quot;)
})</code></pre>
<pre><code>## wrapnum   210 
## WRAPNo   210 
## VisNo   5 
## Cur_HT_Repeat_Key   3 
## CurMed   69 
## CurCur   3 
## CurType   20 
## CurAge   36 
## CurLong   73 
## CurSide   3</code></pre>
<pre class="r"><code>cat(&quot;Before duplicates handling - WRAPNo*VisNo is: &quot;,dupFixCheck(WomenQuestions_Current_HormoneTherapy,&quot;WRAPNo&quot;,&quot;VisNo&quot;),&quot;\n&quot;)</code></pre>
<pre><code>## Before duplicates handling - WRAPNo*VisNo is:  367</code></pre>
<pre class="r"><code>tmp &lt;- WomenQuestions_Current_HormoneTherapy %&gt;%
  pivot_wider(
    id_cols = c(wrapnum, WRAPNo, VisNo),
    names_from = Cur_HT_Repeat_Key,
    values_from = c(CurMed:CurSide),
    names_sep = &quot;_&quot;
  )
## drop columns if all obs are missing
WomenQuestions_Current_HormoneTherapy &lt;- tmp[, colSums(!is.na(tmp)) &gt; 0]

cat(&quot;After duplicates handling - WRAPNo*VisNo is: &quot;,dupFixCheck(WomenQuestions_Current_HormoneTherapy,&quot;WRAPNo&quot;,&quot;VisNo&quot;),&quot;\n&quot;)</code></pre>
<pre><code>## After duplicates handling - WRAPNo*VisNo is:  367</code></pre>
<p><br></p>
</div>
<div id="womenquestions_nonpresc_hormonetherapy-1"
class="section level3">
<h3>WomenQuestions_NonPresc_HormoneTherapy</h3>
<pre class="r"><code>result &lt;- lapply(names(WomenQuestions_NonPresc_HormoneTherapy), function(x) {
  cat(x, &quot; &quot;, length(unique(WomenQuestions_NonPresc_HormoneTherapy[[x]])), &quot;\n&quot;)
})</code></pre>
<pre><code>## wrapnum   198 
## WRAPNo   198 
## VisNo   5 
## NonPresc_HT_Repeat_Key   3 
## Horm   131 
## HormCur   4 
## HormType   13 
## HormAge   31 
## HormLong   97 
## HormSide   3</code></pre>
<pre class="r"><code>cat(&quot;Before duplicates handling - WRAPNo*VisNo is: &quot;,dupFixCheck(WomenQuestions_NonPresc_HormoneTherapy,&quot;WRAPNo&quot;,&quot;VisNo&quot;),&quot;\n&quot;)</code></pre>
<pre><code>## Before duplicates handling - WRAPNo*VisNo is:  239</code></pre>
<pre class="r"><code>tmp &lt;- WomenQuestions_NonPresc_HormoneTherapy %&gt;%
  pivot_wider(
    id_cols = c(wrapnum, WRAPNo, VisNo),
    names_from = NonPresc_HT_Repeat_Key,
    values_from = c(Horm:HormSide),
    names_sep = &quot;_&quot;
  )
## drop columns if all obs are missing
WomenQuestions_NonPresc_HormoneTherapy &lt;- tmp[, colSums(!is.na(tmp)) &gt; 0]

cat(&quot;After duplicates handling - WRAPNo*VisNo is: &quot;,dupFixCheck(WomenQuestions_NonPresc_HormoneTherapy,&quot;WRAPNo&quot;,&quot;VisNo&quot;),&quot;\n&quot;)</code></pre>
<pre><code>## After duplicates handling - WRAPNo*VisNo is:  239</code></pre>
<p><br></p>
</div>
<div id="womenquestions_past_hormonetherapy-1" class="section level3">
<h3>WomenQuestions_Past_HormoneTherapy</h3>
<pre class="r"><code>result &lt;- lapply(names(WomenQuestions_Past_HormoneTherapy), function(x) {
  cat(x, &quot; &quot;, length(unique(WomenQuestions_Past_HormoneTherapy[[x]])), &quot;\n&quot;)
})</code></pre>
<pre><code>## wrapnum   379 
## WRAPNo   379 
## VisNo   2 
## Past_HT_Repeat_Key   3 
## PastMed   154 
## PastCur   3 
## PastType   33 
## PastAge   40 
## PastLong   126 
## PastSide   3 
## PastStop   348</code></pre>
<pre class="r"><code>cat(&quot;Before duplicates handling - WRAPNo*VisNo is: &quot;,dupFixCheck(WomenQuestions_Past_HormoneTherapy,&quot;WRAPNo&quot;,&quot;VisNo&quot;),&quot;\n&quot;)</code></pre>
<pre><code>## Before duplicates handling - WRAPNo*VisNo is:  380</code></pre>
<pre class="r"><code>tmp &lt;- WomenQuestions_Past_HormoneTherapy %&gt;%
  pivot_wider(
    id_cols = c(wrapnum, WRAPNo, VisNo),
    names_from = Past_HT_Repeat_Key,
    values_from = c(PastMed:PastStop),
    names_sep = &quot;_&quot;
  )
## drop columns if all obs are missing
WomenQuestions_Past_HormoneTherapy &lt;- tmp[, colSums(!is.na(tmp)) &gt; 0]

cat(&quot;After duplicates handling - WRAPNo*VisNo is: &quot;,dupFixCheck(WomenQuestions_Past_HormoneTherapy,&quot;WRAPNo&quot;,&quot;VisNo&quot;),&quot;\n&quot;)</code></pre>
<pre><code>## After duplicates handling - WRAPNo*VisNo is:  380</code></pre>
<p><br></p>
</div>
<div id="womenquestions_serm-1" class="section level3">
<h3>WomenQuestions_SERM</h3>
<pre class="r"><code>result &lt;- lapply(names(WomenQuestions_SERM), function(x) {
  cat(x, &quot; &quot;, length(unique(WomenQuestions_SERM[[x]])), &quot;\n&quot;)
})</code></pre>
<pre><code>## wrapnum   58 
## WRAPNo   58 
## VisNo   5 
## SERM_Repeat_Key   3 
## SERM   19 
## SERMCur   4 
## SERMType   1 
## SERMAge   27 
## SERMLong   32 
## SERMSide   2</code></pre>
<pre class="r"><code>cat(&quot;Before duplicates handling - WRAPNo*VisNo is: &quot;,dupFixCheck(WomenQuestions_SERM,&quot;WRAPNo&quot;,&quot;VisNo&quot;),&quot;\n&quot;)</code></pre>
<pre><code>## Before duplicates handling - WRAPNo*VisNo is:  72</code></pre>
<pre class="r"><code>tmp &lt;- WomenQuestions_SERM %&gt;%
  pivot_wider(
    id_cols = c(wrapnum, WRAPNo, VisNo,SERMType),
    names_from = SERM_Repeat_Key,
    values_from = c(SERM:SERMSide,-SERMType),
    names_sep = &quot;_&quot;
  )
## drop columns if all obs are missing
WomenQuestions_SERM &lt;- tmp[, colSums(!is.na(tmp)) &gt; 0 | names(tmp) %in% c(&quot;SERMType&quot;)]

cat(&quot;After duplicates handling - WRAPNo*VisNo is: &quot;,dupFixCheck(WomenQuestions_SERM,&quot;WRAPNo&quot;,&quot;VisNo&quot;),&quot;\n&quot;)</code></pre>
<pre><code>## After duplicates handling - WRAPNo*VisNo is:  72</code></pre>
<p><br> <br></p>
</div>
</div>
</div>
<div id="common-columns-detection" class="section level1">
<h1>Common Columns Detection</h1>
<pre class="r"><code>commonColsDetect(df_names)</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-851b6b839467fd748453" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-851b6b839467fd748453">{"x":{"filter":"none","vertical":false,"data":[["465","666","324","754","261","308","869","885","887","274","295","1","4","5","724","758","797"],["Med_Lab_PDC","MIND_Diet_ScoredValues","IQCode","NeuropsychScores","IADL","IQCode","QuickDementia","QuickDementia","QuickDementia","IADL","Imaging_metadata","Abnormal_Factor_Flags_wide","APG","APG","Morbidity_Mortality","NeuropsychScores","PersFamMedBack"],["Med_Lab_NPDC","MIND_Diet","IADL","CompositeScoreCalcValues","CDR","CDR","CDR","IADL","IQCode","GenHealthDailyAct","fqryStatisticalData","Abnormal_Factor_Flags_long","Abnormal_Factor_Flags_long","Abnormal_Factor_Flags_wide","GenHealthDailyAct","Demographics","Demographics"],[35,21,8,8,7,7,7,7,7,4,4,3,3,3,3,3,3],["wrapnum, WRAPNo, VisNo, mg_1, mg_3, mg_2, mg_5, mg_4, mg_6, mg_8, mg_7, mg_15, mg_14, mg_13, mg_12, mg_11, mg_10, mg_9, mg_16, mg_18, mg_17, mg_20, mg_19, mg_30, mg_29, mg_27, mg_25, mg_24, mg_23, mg_22, mg_21, mg_31, mg_28, mg_26, mg_32","wrapnum, WRAPNo, VisNo, MINDScoredValue_GrnLfyVg, MINDScoredValue_OthrVg, MINDScoredValue_Berries, MINDScoredValue_RedMeat, MINDScoredValue_Fish, MINDScoredValue_Chicken, MINDScoredValue_Cheese, MINDScoredValue_Butter, MINDScoredValue_Beans, MINDScoredValue_WhlGrn, MINDScoredValue_Sweets, MINDScoredValue_Nuts, MINDScoredValue_FastFood, MINDScoredValue_Alcohol, MINDScoredValue_OliveOil, MIND_Straight_Sum, MIND_Extrapolated_Sum, MIND_NumberItemsMissing","wrapnum, WRAPNo, VisNo, Total, Returned_Days_From_Visit, estimated_questionnaire_days_after_baseline, RelPPT, RelLive","wrapnum, WRAPNo, VisNo, flucfl, drraw, stroocw, stroocwe, trlb","wrapnum, WRAPNo, VisNo, Returned_Days_From_Visit, estimated_questionnaire_days_after_baseline, RelPPT, RelLive","wrapnum, WRAPNo, VisNo, Returned_Days_From_Visit, estimated_questionnaire_days_after_baseline, RelPPT, RelLive","wrapnum, WRAPNo, VisNo, Returned_Days_From_Visit, estimated_questionnaire_days_after_baseline, RelPPT, RelLive","wrapnum, WRAPNo, VisNo, Returned_Days_From_Visit, estimated_questionnaire_days_after_baseline, RelPPT, RelLive","wrapnum, WRAPNo, VisNo, Returned_Days_From_Visit, estimated_questionnaire_days_after_baseline, RelPPT, RelLive","wrapnum, WRAPNo, VisNo, Money","wrapnum, Days_Since_Baseline, VisNo, WRAPNo","wrapnum, WRAPNo, Explanation","wrapnum, WRAPNo, Explanation","wrapnum, WRAPNo, Explanation","wrapnum, WRAPNo, MemProb","wrapnum, WRAPNo, interv","wrapnum, WRAPNo, gender"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>df1<\/th>\n      <th>df2<\/th>\n      <th>n_common<\/th>\n      <th>common_cols<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":3},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"df1","targets":1},{"name":"df2","targets":2},{"name":"n_common","targets":3},{"name":"common_cols","targets":4}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><br></p>
<div id="remove-explanation-column-from-all-datasets"
class="section level2">
<h2>Remove Explanation Column from all Datasets</h2>
<pre class="r"><code>## This column is recording the WRAP webseite link. We remove it to avoid duplicated cases in the merge step
for (df_name in df_names) {
  if (&quot;Explanation&quot; %in% names(get(df_name))) {
    assign(df_name, get(df_name)[ , !(names(get(df_name)) %in% &quot;Explanation&quot;) ])
  }
}</code></pre>
</div>
<div id="med_lab_pdc-med_lab_npdc" class="section level2">
<h2>Med_Lab_PDC | Med_Lab_NPDC</h2>
<pre class="r"><code>## the common columns between these two data framed are caused by the piovot_wider, mg is the Dose measurement
intersect(names(Med_Lab_PDC),names(Med_Lab_NPDC))</code></pre>
<pre><code>##  [1] &quot;wrapnum&quot; &quot;WRAPNo&quot;  &quot;VisNo&quot;   &quot;mg_1&quot;    &quot;mg_3&quot;    &quot;mg_2&quot;    &quot;mg_5&quot;   
##  [8] &quot;mg_4&quot;    &quot;mg_6&quot;    &quot;mg_8&quot;    &quot;mg_7&quot;    &quot;mg_15&quot;   &quot;mg_14&quot;   &quot;mg_13&quot;  
## [15] &quot;mg_12&quot;   &quot;mg_11&quot;   &quot;mg_10&quot;   &quot;mg_9&quot;    &quot;mg_16&quot;   &quot;mg_18&quot;   &quot;mg_17&quot;  
## [22] &quot;mg_20&quot;   &quot;mg_19&quot;   &quot;mg_30&quot;   &quot;mg_29&quot;   &quot;mg_27&quot;   &quot;mg_25&quot;   &quot;mg_24&quot;  
## [29] &quot;mg_23&quot;   &quot;mg_22&quot;   &quot;mg_21&quot;   &quot;mg_31&quot;   &quot;mg_28&quot;   &quot;mg_26&quot;   &quot;mg_32&quot;</code></pre>
<pre class="r"><code>## I decided to add prefix in each file
colnames(Med_Lab_PDC)[grepl(&quot;^mg_&quot;, colnames(Med_Lab_PDC))] &lt;- paste0(&quot;pdc_&quot;, colnames(Med_Lab_PDC)[grepl(&quot;^mg_&quot;, colnames(Med_Lab_PDC))])

colnames(Med_Lab_NPDC)[grepl(&quot;^mg_&quot;, colnames(Med_Lab_NPDC))] &lt;- paste0(&quot;npdc_&quot;, colnames(Med_Lab_NPDC)[grepl(&quot;^mg_&quot;, colnames(Med_Lab_NPDC))])</code></pre>
<pre class="r"><code>## print colnames to check the modification
print(colnames(Med_Lab_PDC))</code></pre>
<pre><code>##   [1] &quot;wrapnum&quot;     &quot;WRAPNo&quot;      &quot;VisNo&quot;       &quot;pdcd_1&quot;      &quot;pdcd_3&quot;     
##   [6] &quot;pdcd_2&quot;      &quot;pdcd_5&quot;      &quot;pdcd_4&quot;      &quot;pdcd_6&quot;      &quot;pdcd_8&quot;     
##  [11] &quot;pdcd_7&quot;      &quot;pdcd_15&quot;     &quot;pdcd_14&quot;     &quot;pdcd_13&quot;     &quot;pdcd_12&quot;    
##  [16] &quot;pdcd_11&quot;     &quot;pdcd_10&quot;     &quot;pdcd_9&quot;      &quot;pdcd_16&quot;     &quot;pdcd_18&quot;    
##  [21] &quot;pdcd_17&quot;     &quot;pdcd_20&quot;     &quot;pdcd_19&quot;     &quot;pdcd_30&quot;     &quot;pdcd_29&quot;    
##  [26] &quot;pdcd_27&quot;     &quot;pdcd_25&quot;     &quot;pdcd_24&quot;     &quot;pdcd_23&quot;     &quot;pdcd_22&quot;    
##  [31] &quot;pdcd_21&quot;     &quot;pdcd_31&quot;     &quot;pdcd_28&quot;     &quot;pdcd_26&quot;     &quot;pdcd_32&quot;    
##  [36] &quot;pdc_mg_1&quot;    &quot;pdc_mg_3&quot;    &quot;pdc_mg_2&quot;    &quot;pdc_mg_5&quot;    &quot;pdc_mg_4&quot;   
##  [41] &quot;pdc_mg_6&quot;    &quot;pdc_mg_8&quot;    &quot;pdc_mg_7&quot;    &quot;pdc_mg_15&quot;   &quot;pdc_mg_14&quot;  
##  [46] &quot;pdc_mg_13&quot;   &quot;pdc_mg_12&quot;   &quot;pdc_mg_11&quot;   &quot;pdc_mg_10&quot;   &quot;pdc_mg_9&quot;   
##  [51] &quot;pdc_mg_16&quot;   &quot;pdc_mg_18&quot;   &quot;pdc_mg_17&quot;   &quot;pdc_mg_20&quot;   &quot;pdc_mg_19&quot;  
##  [56] &quot;pdc_mg_30&quot;   &quot;pdc_mg_29&quot;   &quot;pdc_mg_27&quot;   &quot;pdc_mg_25&quot;   &quot;pdc_mg_24&quot;  
##  [61] &quot;pdc_mg_23&quot;   &quot;pdc_mg_22&quot;   &quot;pdc_mg_21&quot;   &quot;pdc_mg_31&quot;   &quot;pdc_mg_28&quot;  
##  [66] &quot;pdc_mg_26&quot;   &quot;pdc_mg_32&quot;   &quot;pdcf_1&quot;      &quot;pdcf_3&quot;      &quot;pdcf_2&quot;     
##  [71] &quot;pdcf_5&quot;      &quot;pdcf_4&quot;      &quot;pdcf_6&quot;      &quot;pdcf_8&quot;      &quot;pdcf_7&quot;     
##  [76] &quot;pdcf_15&quot;     &quot;pdcf_14&quot;     &quot;pdcf_13&quot;     &quot;pdcf_12&quot;     &quot;pdcf_11&quot;    
##  [81] &quot;pdcf_10&quot;     &quot;pdcf_9&quot;      &quot;pdcf_16&quot;     &quot;pdcf_18&quot;     &quot;pdcf_17&quot;    
##  [86] &quot;pdcf_20&quot;     &quot;pdcf_19&quot;     &quot;pdcf_30&quot;     &quot;pdcf_29&quot;     &quot;pdcf_27&quot;    
##  [91] &quot;pdcf_25&quot;     &quot;pdcf_24&quot;     &quot;pdcf_23&quot;     &quot;pdcf_22&quot;     &quot;pdcf_21&quot;    
##  [96] &quot;pdcf_31&quot;     &quot;pdcf_28&quot;     &quot;pdcf_26&quot;     &quot;pdcf_32&quot;     &quot;pdcname_1&quot;  
## [101] &quot;pdcname_3&quot;   &quot;pdcname_2&quot;   &quot;pdcname_5&quot;   &quot;pdcname_4&quot;   &quot;pdcname_6&quot;  
## [106] &quot;pdcname_8&quot;   &quot;pdcname_7&quot;   &quot;pdcname_15&quot;  &quot;pdcname_14&quot;  &quot;pdcname_13&quot; 
## [111] &quot;pdcname_12&quot;  &quot;pdcname_11&quot;  &quot;pdcname_10&quot;  &quot;pdcname_9&quot;   &quot;pdcname_16&quot; 
## [116] &quot;pdcname_18&quot;  &quot;pdcname_17&quot;  &quot;pdcname_20&quot;  &quot;pdcname_19&quot;  &quot;pdcname_30&quot; 
## [121] &quot;pdcname_29&quot;  &quot;pdcname_27&quot;  &quot;pdcname_25&quot;  &quot;pdcname_24&quot;  &quot;pdcname_23&quot; 
## [126] &quot;pdcname_22&quot;  &quot;pdcname_21&quot;  &quot;pdcname_31&quot;  &quot;pdcname_28&quot;  &quot;pdcname_26&quot; 
## [131] &quot;pdcname_32&quot;  &quot;pdcclass_1&quot;  &quot;pdcclass_3&quot;  &quot;pdcclass_2&quot;  &quot;pdcclass_5&quot; 
## [136] &quot;pdcclass_4&quot;  &quot;pdcclass_6&quot;  &quot;pdcclass_8&quot;  &quot;pdcclass_7&quot;  &quot;pdcclass_15&quot;
## [141] &quot;pdcclass_14&quot; &quot;pdcclass_13&quot; &quot;pdcclass_12&quot; &quot;pdcclass_11&quot; &quot;pdcclass_10&quot;
## [146] &quot;pdcclass_9&quot;  &quot;pdcclass_16&quot; &quot;pdcclass_18&quot; &quot;pdcclass_17&quot; &quot;pdcclass_20&quot;
## [151] &quot;pdcclass_19&quot; &quot;pdcclass_30&quot; &quot;pdcclass_29&quot; &quot;pdcclass_27&quot; &quot;pdcclass_25&quot;
## [156] &quot;pdcclass_24&quot; &quot;pdcclass_23&quot; &quot;pdcclass_22&quot; &quot;pdcclass_21&quot; &quot;pdcclass_31&quot;
## [161] &quot;pdcclass_28&quot; &quot;pdcclass_26&quot; &quot;pdcclass_32&quot;</code></pre>
<pre class="r"><code>print(colnames(Med_Lab_NPDC))</code></pre>
<pre><code>##   [1] &quot;wrapnum&quot;      &quot;WRAPNo&quot;       &quot;VisNo&quot;        &quot;npdcd_5&quot;      &quot;npdcd_4&quot;     
##   [6] &quot;npdcd_3&quot;      &quot;npdcd_2&quot;      &quot;npdcd_1&quot;      &quot;npdcd_7&quot;      &quot;npdcd_6&quot;     
##  [11] &quot;npdcd_8&quot;      &quot;npdcd_11&quot;     &quot;npdcd_10&quot;     &quot;npdcd_9&quot;      &quot;npdcd_13&quot;    
##  [16] &quot;npdcd_12&quot;     &quot;npdcd_15&quot;     &quot;npdcd_14&quot;     &quot;npdcd_22&quot;     &quot;npdcd_21&quot;    
##  [21] &quot;npdcd_20&quot;     &quot;npdcd_19&quot;     &quot;npdcd_18&quot;     &quot;npdcd_17&quot;     &quot;npdcd_16&quot;    
##  [26] &quot;npdcd_28&quot;     &quot;npdcd_27&quot;     &quot;npdcd_26&quot;     &quot;npdcd_25&quot;     &quot;npdcd_24&quot;    
##  [31] &quot;npdcd_23&quot;     &quot;npdcd_32&quot;     &quot;npdcd_31&quot;     &quot;npdcd_30&quot;     &quot;npdcd_29&quot;    
##  [36] &quot;npdcd_33&quot;     &quot;npdc_mg_5&quot;    &quot;npdc_mg_4&quot;    &quot;npdc_mg_3&quot;    &quot;npdc_mg_2&quot;   
##  [41] &quot;npdc_mg_1&quot;    &quot;npdc_mg_7&quot;    &quot;npdc_mg_6&quot;    &quot;npdc_mg_8&quot;    &quot;npdc_mg_11&quot;  
##  [46] &quot;npdc_mg_10&quot;   &quot;npdc_mg_9&quot;    &quot;npdc_mg_13&quot;   &quot;npdc_mg_12&quot;   &quot;npdc_mg_15&quot;  
##  [51] &quot;npdc_mg_14&quot;   &quot;npdc_mg_22&quot;   &quot;npdc_mg_21&quot;   &quot;npdc_mg_20&quot;   &quot;npdc_mg_19&quot;  
##  [56] &quot;npdc_mg_18&quot;   &quot;npdc_mg_17&quot;   &quot;npdc_mg_16&quot;   &quot;npdc_mg_28&quot;   &quot;npdc_mg_27&quot;  
##  [61] &quot;npdc_mg_26&quot;   &quot;npdc_mg_25&quot;   &quot;npdc_mg_24&quot;   &quot;npdc_mg_23&quot;   &quot;npdc_mg_32&quot;  
##  [66] &quot;npdc_mg_31&quot;   &quot;npdc_mg_30&quot;   &quot;npdc_mg_29&quot;   &quot;npdcf_5&quot;      &quot;npdcf_4&quot;     
##  [71] &quot;npdcf_3&quot;      &quot;npdcf_2&quot;      &quot;npdcf_1&quot;      &quot;npdcf_7&quot;      &quot;npdcf_6&quot;     
##  [76] &quot;npdcf_8&quot;      &quot;npdcf_11&quot;     &quot;npdcf_10&quot;     &quot;npdcf_9&quot;      &quot;npdcf_13&quot;    
##  [81] &quot;npdcf_12&quot;     &quot;npdcf_15&quot;     &quot;npdcf_14&quot;     &quot;npdcf_22&quot;     &quot;npdcf_21&quot;    
##  [86] &quot;npdcf_20&quot;     &quot;npdcf_19&quot;     &quot;npdcf_18&quot;     &quot;npdcf_17&quot;     &quot;npdcf_16&quot;    
##  [91] &quot;npdcf_28&quot;     &quot;npdcf_27&quot;     &quot;npdcf_26&quot;     &quot;npdcf_25&quot;     &quot;npdcf_24&quot;    
##  [96] &quot;npdcf_23&quot;     &quot;npdcf_32&quot;     &quot;npdcf_31&quot;     &quot;npdcf_30&quot;     &quot;npdcf_29&quot;    
## [101] &quot;npdcf_33&quot;     &quot;npdcname_5&quot;   &quot;npdcname_4&quot;   &quot;npdcname_3&quot;   &quot;npdcname_2&quot;  
## [106] &quot;npdcname_1&quot;   &quot;npdcname_7&quot;   &quot;npdcname_6&quot;   &quot;npdcname_8&quot;   &quot;npdcname_11&quot; 
## [111] &quot;npdcname_10&quot;  &quot;npdcname_9&quot;   &quot;npdcname_13&quot;  &quot;npdcname_12&quot;  &quot;npdcname_15&quot; 
## [116] &quot;npdcname_14&quot;  &quot;npdcname_22&quot;  &quot;npdcname_21&quot;  &quot;npdcname_20&quot;  &quot;npdcname_19&quot; 
## [121] &quot;npdcname_18&quot;  &quot;npdcname_17&quot;  &quot;npdcname_16&quot;  &quot;npdcname_28&quot;  &quot;npdcname_27&quot; 
## [126] &quot;npdcname_26&quot;  &quot;npdcname_25&quot;  &quot;npdcname_24&quot;  &quot;npdcname_23&quot;  &quot;npdcname_32&quot; 
## [131] &quot;npdcname_31&quot;  &quot;npdcname_30&quot;  &quot;npdcname_29&quot;  &quot;npdcname_33&quot;  &quot;npdcclass_5&quot; 
## [136] &quot;npdcclass_4&quot;  &quot;npdcclass_3&quot;  &quot;npdcclass_2&quot;  &quot;npdcclass_1&quot;  &quot;npdcclass_7&quot; 
## [141] &quot;npdcclass_6&quot;  &quot;npdcclass_8&quot;  &quot;npdcclass_11&quot; &quot;npdcclass_10&quot; &quot;npdcclass_9&quot; 
## [146] &quot;npdcclass_13&quot; &quot;npdcclass_12&quot; &quot;npdcclass_15&quot; &quot;npdcclass_14&quot; &quot;npdcclass_22&quot;
## [151] &quot;npdcclass_21&quot; &quot;npdcclass_20&quot; &quot;npdcclass_19&quot; &quot;npdcclass_18&quot; &quot;npdcclass_17&quot;
## [156] &quot;npdcclass_16&quot; &quot;npdcclass_28&quot; &quot;npdcclass_27&quot; &quot;npdcclass_26&quot; &quot;npdcclass_25&quot;
## [161] &quot;npdcclass_24&quot; &quot;npdcclass_23&quot; &quot;npdcclass_32&quot; &quot;npdcclass_31&quot; &quot;npdcclass_30&quot;
## [166] &quot;npdcclass_29&quot; &quot;npdcclass_33&quot;</code></pre>
<p><br></p>
</div>
<div id="iqcode-iadl-cdr-quickdementia" class="section level2">
<h2>IQCode | IADL | CDR | QuickDementia</h2>
<pre class="r"><code>intersect(names(IQCode),names(IADL))</code></pre>
<pre><code>## [1] &quot;wrapnum&quot;                                    
## [2] &quot;WRAPNo&quot;                                     
## [3] &quot;VisNo&quot;                                      
## [4] &quot;Total&quot;                                      
## [5] &quot;Returned_Days_From_Visit&quot;                   
## [6] &quot;estimated_questionnaire_days_after_baseline&quot;
## [7] &quot;RelPPT&quot;                                     
## [8] &quot;RelLive&quot;</code></pre>
<pre class="r"><code>intersect(names(CDR),names(IADL))</code></pre>
<pre><code>## [1] &quot;wrapnum&quot;                                    
## [2] &quot;WRAPNo&quot;                                     
## [3] &quot;VisNo&quot;                                      
## [4] &quot;estimated_questionnaire_days_after_baseline&quot;
## [5] &quot;Returned_Days_From_Visit&quot;                   
## [6] &quot;RelPPT&quot;                                     
## [7] &quot;RelLive&quot;</code></pre>
<pre class="r"><code>intersect(names(CDR),names(QuickDementia))</code></pre>
<pre><code>## [1] &quot;wrapnum&quot;                                    
## [2] &quot;WRAPNo&quot;                                     
## [3] &quot;VisNo&quot;                                      
## [4] &quot;estimated_questionnaire_days_after_baseline&quot;
## [5] &quot;Returned_Days_From_Visit&quot;                   
## [6] &quot;RelPPT&quot;                                     
## [7] &quot;RelLive&quot;</code></pre>
<pre class="r"><code>## common columns between IQCode and IADL
commnames_1 &lt;- &quot;Total&quot;

colnames(IQCode)[colnames(IQCode) %in% commnames_1] &lt;- paste0(colnames(IQCode)[colnames(IQCode) %in% commnames_1],&quot;_IQCode&quot;)
colnames(IADL)[colnames(IADL) %in% commnames_1] &lt;-  paste0(colnames(IADL)[colnames(IADL) %in% commnames_1],&quot;_IADL&quot;)


## common columns across all four datasets
commnames_2 &lt;- c(&quot;estimated_questionnaire_days_after_baseline&quot;,&quot;Returned_Days_From_Visit&quot;,&quot;RelPPT&quot;,&quot;RelLive&quot;)

colnames(IQCode)[colnames(IQCode) %in% commnames_2] &lt;- paste0(colnames(IQCode)[colnames(IQCode) %in% commnames_2],&quot;_IQCode&quot;)

colnames(IADL)[colnames(IADL) %in% commnames_2] &lt;-  paste0(colnames(IADL)[colnames(IADL) %in% commnames_2],&quot;_IADL&quot;)

colnames(QuickDementia)[colnames(QuickDementia) %in% commnames_2] &lt;-  paste0( colnames(QuickDementia)[colnames(QuickDementia) %in% commnames_2],&quot;_QuickDementia&quot;)

colnames(CDR)[colnames(CDR) %in% commnames_2] &lt;-  paste0(colnames(CDR)[colnames(CDR) %in% commnames_2],&quot;_CDR&quot;)</code></pre>
<pre class="r"><code>## rerun the Common Columns Detection code
commonColsDetect(df_names)</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-905d16f6afcb0164d939" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-905d16f6afcb0164d939">{"x":{"filter":"none","vertical":false,"data":[["666","754","274","295","724","758","797"],["MIND_Diet_ScoredValues","NeuropsychScores","IADL","Imaging_metadata","Morbidity_Mortality","NeuropsychScores","PersFamMedBack"],["MIND_Diet","CompositeScoreCalcValues","GenHealthDailyAct","fqryStatisticalData","GenHealthDailyAct","Demographics","Demographics"],[21,8,4,4,3,3,3],["wrapnum, WRAPNo, VisNo, MINDScoredValue_GrnLfyVg, MINDScoredValue_OthrVg, MINDScoredValue_Berries, MINDScoredValue_RedMeat, MINDScoredValue_Fish, MINDScoredValue_Chicken, MINDScoredValue_Cheese, MINDScoredValue_Butter, MINDScoredValue_Beans, MINDScoredValue_WhlGrn, MINDScoredValue_Sweets, MINDScoredValue_Nuts, MINDScoredValue_FastFood, MINDScoredValue_Alcohol, MINDScoredValue_OliveOil, MIND_Straight_Sum, MIND_Extrapolated_Sum, MIND_NumberItemsMissing","wrapnum, WRAPNo, VisNo, flucfl, drraw, stroocw, stroocwe, trlb","wrapnum, WRAPNo, VisNo, Money","wrapnum, Days_Since_Baseline, VisNo, WRAPNo","wrapnum, WRAPNo, MemProb","wrapnum, WRAPNo, interv","wrapnum, WRAPNo, gender"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>df1<\/th>\n      <th>df2<\/th>\n      <th>n_common<\/th>\n      <th>common_cols<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":3},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"df1","targets":1},{"name":"df2","targets":2},{"name":"n_common","targets":3},{"name":"common_cols","targets":4}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>## print colnames to check the modification
print(colnames(IQCode))</code></pre>
<pre><code>##  [1] &quot;wrapnum&quot;                                           
##  [2] &quot;WRAPNo&quot;                                            
##  [3] &quot;VisNo&quot;                                             
##  [4] &quot;FamDates&quot;                                          
##  [5] &quot;Recent&quot;                                            
##  [6] &quot;Convers&quot;                                           
##  [7] &quot;Address&quot;                                           
##  [8] &quot;DayMo&quot;                                             
##  [9] &quot;Kept&quot;                                              
## [10] &quot;Find&quot;                                              
## [11] &quot;Machines&quot;                                          
## [12] &quot;NewMach&quot;                                           
## [13] &quot;NewThings&quot;                                         
## [14] &quot;Story&quot;                                             
## [15] &quot;Decisions&quot;                                         
## [16] &quot;MonShop&quot;                                           
## [17] &quot;Finance&quot;                                           
## [18] &quot;MathProb&quot;                                          
## [19] &quot;Intell&quot;                                            
## [20] &quot;Total_IQCode&quot;                                      
## [21] &quot;Returned_Days_From_Visit_IQCode&quot;                   
## [22] &quot;estimated_questionnaire_days_after_baseline_IQCode&quot;
## [23] &quot;RelPPT_IQCode&quot;                                     
## [24] &quot;RelLive_IQCode&quot;</code></pre>
<pre class="r"><code>print(colnames(IADL))</code></pre>
<pre><code>##  [1] &quot;wrapnum&quot;                                         
##  [2] &quot;WRAPNo&quot;                                          
##  [3] &quot;VisNo&quot;                                           
##  [4] &quot;UsePhone&quot;                                        
##  [5] &quot;Walking&quot;                                         
##  [6] &quot;Shopping&quot;                                        
##  [7] &quot;Meals&quot;                                           
##  [8] &quot;HouseWk&quot;                                         
##  [9] &quot;Laundry&quot;                                         
## [10] &quot;TakeMeds&quot;                                        
## [11] &quot;Money&quot;                                           
## [12] &quot;Total_IADL&quot;                                      
## [13] &quot;Returned_Days_From_Visit_IADL&quot;                   
## [14] &quot;estimated_questionnaire_days_after_baseline_IADL&quot;
## [15] &quot;RelPPT_IADL&quot;                                     
## [16] &quot;RelLive_IADL&quot;</code></pre>
<pre class="r"><code>print(colnames(QuickDementia))</code></pre>
<pre><code>##  [1] &quot;wrapnum&quot;                                                  
##  [2] &quot;WRAPNo&quot;                                                   
##  [3] &quot;VisNo&quot;                                                    
##  [4] &quot;QD_MemRecall&quot;                                             
##  [5] &quot;QD_Orientation&quot;                                           
##  [6] &quot;QD_DMPSA&quot;                                                 
##  [7] &quot;QD_ActOutHome&quot;                                            
##  [8] &quot;QD_FuncHome&quot;                                              
##  [9] &quot;QD_Hygeine&quot;                                               
## [10] &quot;QD_Behavior&quot;                                              
## [11] &quot;QD_Language&quot;                                              
## [12] &quot;QD_Mood&quot;                                                  
## [13] &quot;QD_AttConc&quot;                                               
## [14] &quot;QD_CogSubTotal&quot;                                           
## [15] &quot;QD_BehSubTotal&quot;                                           
## [16] &quot;QD_TotalScore&quot;                                            
## [17] &quot;QD_SumOfBoxes&quot;                                            
## [18] &quot;Returned_Days_From_Visit_QuickDementia&quot;                   
## [19] &quot;estimated_questionnaire_days_after_baseline_QuickDementia&quot;
## [20] &quot;RelPPT_QuickDementia&quot;                                     
## [21] &quot;RelLive_QuickDementia&quot;                                    
## [22] &quot;QDRS_Rating&quot;</code></pre>
<pre class="r"><code>print(colnames(CDR))</code></pre>
<pre><code>##  [1] &quot;wrapnum&quot;                                        
##  [2] &quot;WRAPNo&quot;                                         
##  [3] &quot;VisNo&quot;                                          
##  [4] &quot;Memory&quot;                                         
##  [5] &quot;Orientation&quot;                                    
##  [6] &quot;JudgePS&quot;                                        
##  [7] &quot;CommAffairs&quot;                                    
##  [8] &quot;HomeHob&quot;                                        
##  [9] &quot;PersCare&quot;                                       
## [10] &quot;CDRRating&quot;                                      
## [11] &quot;SumOfBoxes&quot;                                     
## [12] &quot;estimated_questionnaire_days_after_baseline_CDR&quot;
## [13] &quot;Returned_Days_From_Visit_CDR&quot;                   
## [14] &quot;RelPPT_CDR&quot;                                     
## [15] &quot;RelLive_CDR&quot;</code></pre>
<p><br></p>
</div>
<div id="neuropsychscores-compositescorecalcvalues"
class="section level2">
<h2>NeuropsychScores | CompositeScoreCalcValues</h2>
<pre class="r"><code>intersect(names(NeuropsychScores),names(CompositeScoreCalcValues))</code></pre>
<pre><code>## [1] &quot;wrapnum&quot;  &quot;WRAPNo&quot;   &quot;VisNo&quot;    &quot;flucfl&quot;   &quot;drraw&quot;    &quot;stroocw&quot;  &quot;stroocwe&quot;
## [8] &quot;trlb&quot;</code></pre>
<pre class="r"><code>## add prefix to CompositeScoreCalcValues
tNames &lt;- c(&quot;flucfl&quot;,&quot;drraw&quot;,&quot;stroocw&quot;,&quot;stroocwe&quot;,&quot;trlb&quot;)
colnames(CompositeScoreCalcValues)[colnames(CompositeScoreCalcValues) %in% tNames] &lt;-  paste0(colnames(CompositeScoreCalcValues)[colnames(CompositeScoreCalcValues) %in% tNames],&quot;_CompositeScoreCalcValues&quot;)</code></pre>
<pre class="r"><code>## print colnames to check the modification
print(colnames(CompositeScoreCalcValues))</code></pre>
<pre><code>##  [1] &quot;wrapnum&quot;                           &quot;WRAPNo&quot;                           
##  [3] &quot;VisNo&quot;                             &quot;ttotal&quot;                           
##  [5] &quot;bvmttot&quot;                           &quot;wmsrtot&quot;                          
##  [7] &quot;drraw_CompositeScoreCalcValues&quot;    &quot;bvmtdel&quot;                          
##  [9] &quot;wmsrtot2&quot;                          &quot;trlb_CompositeScoreCalcValues&quot;    
## [11] &quot;stroocw_CompositeScoreCalcValues&quot;  &quot;waisrtot&quot;                         
## [13] &quot;animtotraw&quot;                        &quot;flucfl_CompositeScoreCalcValues&quot;  
## [15] &quot;mmsetot&quot;                           &quot;stroocwe_CompositeScoreCalcValues&quot;</code></pre>
<p><br></p>
</div>
<div id="iadl-genhealthdailyact" class="section level2">
<h2>IADL | GenHealthDailyAct</h2>
<pre class="r"><code>intersect(names(IADL),names(GenHealthDailyAct))</code></pre>
<pre><code>## [1] &quot;wrapnum&quot; &quot;WRAPNo&quot;  &quot;VisNo&quot;   &quot;Money&quot;</code></pre>
<pre class="r"><code>## add prefix to both datasets
colnames(IADL)[colnames(IADL) == &quot;Money&quot;] &lt;- &quot;Money_IADL&quot;
colnames(GenHealthDailyAct)[colnames(GenHealthDailyAct) == &quot;Money&quot;] &lt;- &quot;Money_GenHealthDailyAct&quot;</code></pre>
<pre class="r"><code>## print colnames to check the modification
print(colnames(IADL))</code></pre>
<pre><code>##  [1] &quot;wrapnum&quot;                                         
##  [2] &quot;WRAPNo&quot;                                          
##  [3] &quot;VisNo&quot;                                           
##  [4] &quot;UsePhone&quot;                                        
##  [5] &quot;Walking&quot;                                         
##  [6] &quot;Shopping&quot;                                        
##  [7] &quot;Meals&quot;                                           
##  [8] &quot;HouseWk&quot;                                         
##  [9] &quot;Laundry&quot;                                         
## [10] &quot;TakeMeds&quot;                                        
## [11] &quot;Money_IADL&quot;                                      
## [12] &quot;Total_IADL&quot;                                      
## [13] &quot;Returned_Days_From_Visit_IADL&quot;                   
## [14] &quot;estimated_questionnaire_days_after_baseline_IADL&quot;
## [15] &quot;RelPPT_IADL&quot;                                     
## [16] &quot;RelLive_IADL&quot;</code></pre>
<pre class="r"><code>print(colnames(GenHealthDailyAct))</code></pre>
<pre><code>##   [1] &quot;wrapnum&quot;                 &quot;WRAPNo&quot;                 
##   [3] &quot;VisNo&quot;                   &quot;CurHealth&quot;              
##   [5] &quot;Htft&quot;                    &quot;Htin&quot;                   
##   [7] &quot;Tall18ft&quot;                &quot;Tall18in&quot;               
##   [9] &quot;Wt18&quot;                    &quot;Wt35&quot;                   
##  [11] &quot;Wt50&quot;                    &quot;Wt50NA&quot;                 
##  [13] &quot;WtMax&quot;                   &quot;Wtlbs&quot;                  
##  [15] &quot;Smoke&quot;                   &quot;SmokeYrs&quot;               
##  [17] &quot;SmokeNo&quot;                 &quot;SmokePast&quot;              
##  [19] &quot;SmokePastNo&quot;             &quot;TobProd&quot;                
##  [21] &quot;LiveSmoke&quot;               &quot;LiveYrs&quot;                
##  [23] &quot;LivCur&quot;                  &quot;SmokeJob&quot;               
##  [25] &quot;JobYrs&quot;                  &quot;JobCur&quot;                 
##  [27] &quot;AlcPast&quot;                 &quot;AlcDays&quot;                
##  [29] &quot;AlcNo&quot;                   &quot;AlcProb&quot;                
##  [31] &quot;DrugProb&quot;                &quot;AlcDrugProb&quot;            
##  [33] &quot;Prob6&quot;                   &quot;Caffeine&quot;               
##  [35] &quot;MemProb&quot;                 &quot;ToldMem&quot;                
##  [37] &quot;MemRate&quot;                 &quot;memocclm&quot;               
##  [39] &quot;memoccyr&quot;                &quot;memoccfy&quot;               
##  [41] &quot;memoccty&quot;                &quot;freqtna&quot;                
##  [43] &quot;freqfac&quot;                 &quot;freqapp&quot;                
##  [45] &quot;freqitem&quot;                &quot;freqcho&quot;                
##  [47] &quot;freqdir&quot;                 &quot;freqphno&quot;               
##  [49] &quot;freqfph&quot;                 &quot;freqtel&quot;                
##  [51] &quot;freqcor&quot;                 &quot;freqpdat&quot;               
##  [53] &quot;freqwor&quot;                 &quot;freqbuy&quot;                
##  [55] &quot;freqtes&quot;                 &quot;freqdo&quot;                 
##  [57] &quot;freqconv&quot;                &quot;freqpubs&quot;               
##  [59] &quot;freqrep&quot;                 &quot;forgtna&quot;                
##  [61] &quot;forgfac&quot;                 &quot;forgapp&quot;                
##  [63] &quot;forgitem&quot;                &quot;forgcho&quot;                
##  [65] &quot;forgdir&quot;                 &quot;forgphno&quot;               
##  [67] &quot;forgfph&quot;                 &quot;forgtel&quot;                
##  [69] &quot;forgcor&quot;                 &quot;forgpdat&quot;               
##  [71] &quot;forgwor&quot;                 &quot;forgbuy&quot;                
##  [73] &quot;forgtes&quot;                 &quot;forgdo&quot;                 
##  [75] &quot;forgconv&quot;                &quot;forgpubs&quot;               
##  [77] &quot;forgrep&quot;                 &quot;retmem1&quot;                
##  [79] &quot;retmem5&quot;                 &quot;retmem10&quot;               
##  [81] &quot;retmem20&quot;                &quot;retmem18&quot;               
##  [83] &quot;mnemappt&quot;                &quot;mnemnote&quot;               
##  [85] &quot;mnemlist&quot;                &quot;mnemgrol&quot;               
##  [87] &quot;mnemsche&quot;                &quot;mnemrep&quot;                
##  [89] &quot;mnemasoc&quot;                &quot;mnemprom&quot;               
##  [91] &quot;PhysAct&quot;                 &quot;WalkOut&quot;                
##  [93] &quot;WalkMin&quot;                 &quot;WalkSp&quot;                 
##  [95] &quot;HardEx&quot;                  &quot;HardExTime&quot;             
##  [97] &quot;ModEx&quot;                   &quot;ModExTime&quot;              
##  [99] &quot;MildEx&quot;                  &quot;MildExTime&quot;             
## [101] &quot;Chores&quot;                  &quot;Yard&quot;                   
## [103] &quot;YardHrs&quot;                 &quot;Sitting&quot;                
## [105] &quot;ExTypical&quot;               &quot;TV&quot;                     
## [107] &quot;TVHrs&quot;                   &quot;Radio&quot;                  
## [109] &quot;RadioHrs&quot;                &quot;Paper&quot;                  
## [111] &quot;Mag&quot;                     &quot;Book&quot;                   
## [113] &quot;Games&quot;                   &quot;Museum&quot;                 
## [115] &quot;Plays&quot;                   &quot;Educ&quot;                   
## [117] &quot;Comp&quot;                    &quot;CompHrs&quot;                
## [119] &quot;SmartPhonTbltUse&quot;        &quot;SmartPhonTbltHrs&quot;       
## [121] &quot;KidClose&quot;                &quot;RelFr&quot;                  
## [123] &quot;LiveWith&quot;                &quot;LiveWithOth&quot;            
## [125] &quot;Pet&quot;                     &quot;Dog&quot;                    
## [127] &quot;Cat&quot;                     &quot;Bird&quot;                   
## [129] &quot;Fish&quot;                    &quot;OthPet&quot;                 
## [131] &quot;PetOth&quot;                  &quot;Help&quot;                   
## [133] &quot;HelpTime&quot;                &quot;HelpAD&quot;                 
## [135] &quot;HelpRel&quot;                 &quot;CareLiveWith&quot;           
## [137] &quot;CareLiveClose&quot;           &quot;CareFac&quot;                
## [139] &quot;CareLvl&quot;                 &quot;CarePrim&quot;               
## [141] &quot;CarePhone&quot;               &quot;CareSee&quot;                
## [143] &quot;CareShare&quot;               &quot;CareShareWho&quot;           
## [145] &quot;CarePleasant&quot;            &quot;Listen&quot;                 
## [147] &quot;Advice&quot;                  &quot;Doctor&quot;                 
## [149] &quot;GoodTime&quot;                &quot;Understand&quot;             
## [151] &quot;DailyCh&quot;                 &quot;Share&quot;                  
## [153] &quot;Fun&quot;                     &quot;Love&quot;                   
## [155] &quot;SpDie&quot;                   &quot;SpIll&quot;                  
## [157] &quot;FrDie&quot;                   &quot;Money_GenHealthDailyAct&quot;
## [159] &quot;SpDivorce&quot;               &quot;FrDivorce&quot;              
## [161] &quot;Conflict&quot;                &quot;Accid&quot;                  
## [163] &quot;LoseJob&quot;                 &quot;PetDie&quot;                 
## [165] &quot;MCIDEM&quot;                  &quot;MCIDEMTYPE&quot;             
## [167] &quot;SelfIllness&quot;             &quot;SleepHrs&quot;               
## [169] &quot;Church&quot;                  &quot;Comfort&quot;                
## [171] &quot;Meetings&quot;                &quot;PhAbused&quot;               
## [173] &quot;VAbused&quot;                 &quot;FTFSPPART&quot;              
## [175] &quot;FTFSPPART_ET&quot;            &quot;FTFSPPARTPL&quot;            
## [177] &quot;FTFFAM&quot;                  &quot;FTFFAM_ET&quot;              
## [179] &quot;FTFFAMPL&quot;                &quot;FTFFRIEND&quot;              
## [181] &quot;FTFFRIEND_ET&quot;            &quot;FTFFRIENDPL&quot;            
## [183] &quot;FTFCHURCH&quot;               &quot;FTFCHURCH_ET&quot;           
## [185] &quot;FTFCHURCHPL&quot;             &quot;FTFMEETINGS&quot;            
## [187] &quot;FTFMEETINGS_ET&quot;          &quot;FTFMEETINSSPL&quot;          
## [189] &quot;DAYSWORK&quot;                &quot;FTFWORK&quot;                
## [191] &quot;FTFWORK_ET&quot;              &quot;FTFWORKPL&quot;              
## [193] &quot;FTFSTRANGER&quot;             &quot;FTFSTRANGER_ET&quot;         
## [195] &quot;FTFSTRANGERPL&quot;           &quot;FTFOTHER&quot;               
## [197] &quot;FTFOTHER_TXT&quot;            &quot;FTFOTHERTIME&quot;           
## [199] &quot;FTFOTHERPL&quot;              &quot;VERBTEL&quot;                
## [201] &quot;VERBTEL_TXT&quot;             &quot;VERBTELPL&quot;              
## [203] &quot;WRITCONVIM&quot;              &quot;WRITCONVIM_TXT&quot;         
## [205] &quot;WRITCONVIMPL&quot;            &quot;WRITCONVEMAIL&quot;          
## [207] &quot;WRITCONVEMAIL_TXT&quot;       &quot;WRITCONVEMAILPL&quot;        
## [209] &quot;WRITLETTERS&quot;             &quot;WRITLETTERS_TXT&quot;        
## [211] &quot;HELPRELPL&quot;               &quot;xswkexer&quot;               
## [213] &quot;rateexer&quot;                &quot;HardHrWk&quot;               
## [215] &quot;ModHrWk&quot;                 &quot;MildHrWk&quot;               
## [217] &quot;WalkMet&quot;                 &quot;WalkHrWk&quot;               
## [219] &quot;MetHrWk&quot;</code></pre>
<p><br></p>
</div>
<div id="morbidity_mortality-genhealthdailyact" class="section level2">
<h2>Morbidity_Mortality | GenHealthDailyAct</h2>
<pre class="r"><code>intersect(names(Morbidity_Mortality),names(GenHealthDailyAct))</code></pre>
<pre><code>## [1] &quot;wrapnum&quot; &quot;WRAPNo&quot;  &quot;MemProb&quot;</code></pre>
<pre class="r"><code>colnames(Morbidity_Mortality)[colnames(Morbidity_Mortality) == &quot;MemProb&quot;] &lt;- &quot;MemProb_Morbidity_Mortality&quot;
colnames(GenHealthDailyAct)[colnames(GenHealthDailyAct) == &quot;MemProb&quot;] &lt;- &quot;MemProb_GenHealthDailyAct&quot;</code></pre>
<pre class="r"><code>## print colnames to check the modification
print(colnames(Morbidity_Mortality))</code></pre>
<pre><code>##  [1] &quot;wrapnum&quot;                     &quot;WRAPNo&quot;                     
##  [3] &quot;PTRel&quot;                       &quot;ageAtDeath&quot;                 
##  [5] &quot;death_days_after_baseline&quot;   &quot;ageAtDeath_Over89&quot;          
##  [7] &quot;CauseDeath&quot;                  &quot;BrainAutopsy&quot;               
##  [9] &quot;yearAutopsy&quot;                 &quot;yearADDiagnosis&quot;            
## [11] &quot;WhereAutopsy&quot;                &quot;AutopsyRes&quot;                 
## [13] &quot;MemProb_Morbidity_Mortality&quot; &quot;Confuse&quot;                    
## [15] &quot;Dementia&quot;                    &quot;MCI&quot;                        
## [17] &quot;WBDPTissueAvail&quot;</code></pre>
<pre class="r"><code>print(colnames(GenHealthDailyAct))</code></pre>
<pre><code>##   [1] &quot;wrapnum&quot;                   &quot;WRAPNo&quot;                   
##   [3] &quot;VisNo&quot;                     &quot;CurHealth&quot;                
##   [5] &quot;Htft&quot;                      &quot;Htin&quot;                     
##   [7] &quot;Tall18ft&quot;                  &quot;Tall18in&quot;                 
##   [9] &quot;Wt18&quot;                      &quot;Wt35&quot;                     
##  [11] &quot;Wt50&quot;                      &quot;Wt50NA&quot;                   
##  [13] &quot;WtMax&quot;                     &quot;Wtlbs&quot;                    
##  [15] &quot;Smoke&quot;                     &quot;SmokeYrs&quot;                 
##  [17] &quot;SmokeNo&quot;                   &quot;SmokePast&quot;                
##  [19] &quot;SmokePastNo&quot;               &quot;TobProd&quot;                  
##  [21] &quot;LiveSmoke&quot;                 &quot;LiveYrs&quot;                  
##  [23] &quot;LivCur&quot;                    &quot;SmokeJob&quot;                 
##  [25] &quot;JobYrs&quot;                    &quot;JobCur&quot;                   
##  [27] &quot;AlcPast&quot;                   &quot;AlcDays&quot;                  
##  [29] &quot;AlcNo&quot;                     &quot;AlcProb&quot;                  
##  [31] &quot;DrugProb&quot;                  &quot;AlcDrugProb&quot;              
##  [33] &quot;Prob6&quot;                     &quot;Caffeine&quot;                 
##  [35] &quot;MemProb_GenHealthDailyAct&quot; &quot;ToldMem&quot;                  
##  [37] &quot;MemRate&quot;                   &quot;memocclm&quot;                 
##  [39] &quot;memoccyr&quot;                  &quot;memoccfy&quot;                 
##  [41] &quot;memoccty&quot;                  &quot;freqtna&quot;                  
##  [43] &quot;freqfac&quot;                   &quot;freqapp&quot;                  
##  [45] &quot;freqitem&quot;                  &quot;freqcho&quot;                  
##  [47] &quot;freqdir&quot;                   &quot;freqphno&quot;                 
##  [49] &quot;freqfph&quot;                   &quot;freqtel&quot;                  
##  [51] &quot;freqcor&quot;                   &quot;freqpdat&quot;                 
##  [53] &quot;freqwor&quot;                   &quot;freqbuy&quot;                  
##  [55] &quot;freqtes&quot;                   &quot;freqdo&quot;                   
##  [57] &quot;freqconv&quot;                  &quot;freqpubs&quot;                 
##  [59] &quot;freqrep&quot;                   &quot;forgtna&quot;                  
##  [61] &quot;forgfac&quot;                   &quot;forgapp&quot;                  
##  [63] &quot;forgitem&quot;                  &quot;forgcho&quot;                  
##  [65] &quot;forgdir&quot;                   &quot;forgphno&quot;                 
##  [67] &quot;forgfph&quot;                   &quot;forgtel&quot;                  
##  [69] &quot;forgcor&quot;                   &quot;forgpdat&quot;                 
##  [71] &quot;forgwor&quot;                   &quot;forgbuy&quot;                  
##  [73] &quot;forgtes&quot;                   &quot;forgdo&quot;                   
##  [75] &quot;forgconv&quot;                  &quot;forgpubs&quot;                 
##  [77] &quot;forgrep&quot;                   &quot;retmem1&quot;                  
##  [79] &quot;retmem5&quot;                   &quot;retmem10&quot;                 
##  [81] &quot;retmem20&quot;                  &quot;retmem18&quot;                 
##  [83] &quot;mnemappt&quot;                  &quot;mnemnote&quot;                 
##  [85] &quot;mnemlist&quot;                  &quot;mnemgrol&quot;                 
##  [87] &quot;mnemsche&quot;                  &quot;mnemrep&quot;                  
##  [89] &quot;mnemasoc&quot;                  &quot;mnemprom&quot;                 
##  [91] &quot;PhysAct&quot;                   &quot;WalkOut&quot;                  
##  [93] &quot;WalkMin&quot;                   &quot;WalkSp&quot;                   
##  [95] &quot;HardEx&quot;                    &quot;HardExTime&quot;               
##  [97] &quot;ModEx&quot;                     &quot;ModExTime&quot;                
##  [99] &quot;MildEx&quot;                    &quot;MildExTime&quot;               
## [101] &quot;Chores&quot;                    &quot;Yard&quot;                     
## [103] &quot;YardHrs&quot;                   &quot;Sitting&quot;                  
## [105] &quot;ExTypical&quot;                 &quot;TV&quot;                       
## [107] &quot;TVHrs&quot;                     &quot;Radio&quot;                    
## [109] &quot;RadioHrs&quot;                  &quot;Paper&quot;                    
## [111] &quot;Mag&quot;                       &quot;Book&quot;                     
## [113] &quot;Games&quot;                     &quot;Museum&quot;                   
## [115] &quot;Plays&quot;                     &quot;Educ&quot;                     
## [117] &quot;Comp&quot;                      &quot;CompHrs&quot;                  
## [119] &quot;SmartPhonTbltUse&quot;          &quot;SmartPhonTbltHrs&quot;         
## [121] &quot;KidClose&quot;                  &quot;RelFr&quot;                    
## [123] &quot;LiveWith&quot;                  &quot;LiveWithOth&quot;              
## [125] &quot;Pet&quot;                       &quot;Dog&quot;                      
## [127] &quot;Cat&quot;                       &quot;Bird&quot;                     
## [129] &quot;Fish&quot;                      &quot;OthPet&quot;                   
## [131] &quot;PetOth&quot;                    &quot;Help&quot;                     
## [133] &quot;HelpTime&quot;                  &quot;HelpAD&quot;                   
## [135] &quot;HelpRel&quot;                   &quot;CareLiveWith&quot;             
## [137] &quot;CareLiveClose&quot;             &quot;CareFac&quot;                  
## [139] &quot;CareLvl&quot;                   &quot;CarePrim&quot;                 
## [141] &quot;CarePhone&quot;                 &quot;CareSee&quot;                  
## [143] &quot;CareShare&quot;                 &quot;CareShareWho&quot;             
## [145] &quot;CarePleasant&quot;              &quot;Listen&quot;                   
## [147] &quot;Advice&quot;                    &quot;Doctor&quot;                   
## [149] &quot;GoodTime&quot;                  &quot;Understand&quot;               
## [151] &quot;DailyCh&quot;                   &quot;Share&quot;                    
## [153] &quot;Fun&quot;                       &quot;Love&quot;                     
## [155] &quot;SpDie&quot;                     &quot;SpIll&quot;                    
## [157] &quot;FrDie&quot;                     &quot;Money_GenHealthDailyAct&quot;  
## [159] &quot;SpDivorce&quot;                 &quot;FrDivorce&quot;                
## [161] &quot;Conflict&quot;                  &quot;Accid&quot;                    
## [163] &quot;LoseJob&quot;                   &quot;PetDie&quot;                   
## [165] &quot;MCIDEM&quot;                    &quot;MCIDEMTYPE&quot;               
## [167] &quot;SelfIllness&quot;               &quot;SleepHrs&quot;                 
## [169] &quot;Church&quot;                    &quot;Comfort&quot;                  
## [171] &quot;Meetings&quot;                  &quot;PhAbused&quot;                 
## [173] &quot;VAbused&quot;                   &quot;FTFSPPART&quot;                
## [175] &quot;FTFSPPART_ET&quot;              &quot;FTFSPPARTPL&quot;              
## [177] &quot;FTFFAM&quot;                    &quot;FTFFAM_ET&quot;                
## [179] &quot;FTFFAMPL&quot;                  &quot;FTFFRIEND&quot;                
## [181] &quot;FTFFRIEND_ET&quot;              &quot;FTFFRIENDPL&quot;              
## [183] &quot;FTFCHURCH&quot;                 &quot;FTFCHURCH_ET&quot;             
## [185] &quot;FTFCHURCHPL&quot;               &quot;FTFMEETINGS&quot;              
## [187] &quot;FTFMEETINGS_ET&quot;            &quot;FTFMEETINSSPL&quot;            
## [189] &quot;DAYSWORK&quot;                  &quot;FTFWORK&quot;                  
## [191] &quot;FTFWORK_ET&quot;                &quot;FTFWORKPL&quot;                
## [193] &quot;FTFSTRANGER&quot;               &quot;FTFSTRANGER_ET&quot;           
## [195] &quot;FTFSTRANGERPL&quot;             &quot;FTFOTHER&quot;                 
## [197] &quot;FTFOTHER_TXT&quot;              &quot;FTFOTHERTIME&quot;             
## [199] &quot;FTFOTHERPL&quot;                &quot;VERBTEL&quot;                  
## [201] &quot;VERBTEL_TXT&quot;               &quot;VERBTELPL&quot;                
## [203] &quot;WRITCONVIM&quot;                &quot;WRITCONVIM_TXT&quot;           
## [205] &quot;WRITCONVIMPL&quot;              &quot;WRITCONVEMAIL&quot;            
## [207] &quot;WRITCONVEMAIL_TXT&quot;         &quot;WRITCONVEMAILPL&quot;          
## [209] &quot;WRITLETTERS&quot;               &quot;WRITLETTERS_TXT&quot;          
## [211] &quot;HELPRELPL&quot;                 &quot;xswkexer&quot;                 
## [213] &quot;rateexer&quot;                  &quot;HardHrWk&quot;                 
## [215] &quot;ModHrWk&quot;                   &quot;MildHrWk&quot;                 
## [217] &quot;WalkMet&quot;                   &quot;WalkHrWk&quot;                 
## [219] &quot;MetHrWk&quot;</code></pre>
<p><br></p>
</div>
<div id="neuropsychscores-demographics" class="section level2">
<h2>NeuropsychScores | Demographics</h2>
<pre class="r"><code>intersect(names(NeuropsychScores),names(Demographics))</code></pre>
<pre><code>## [1] &quot;wrapnum&quot; &quot;WRAPNo&quot;  &quot;interv&quot;</code></pre>
<pre class="r"><code>colnames(NeuropsychScores)[colnames(NeuropsychScores) == &quot;interv&quot;] &lt;- &quot;interv_NeuropsychScores&quot;
colnames(Demographics)[colnames(Demographics) == &quot;interv&quot;] &lt;- &quot;interv_Demographics&quot;</code></pre>
<pre class="r"><code>## print colnames to check the modification
print(colnames(NeuropsychScores))</code></pre>
<pre><code>##   [1] &quot;wrapnum&quot;                 &quot;WRAPNo&quot;                 
##   [3] &quot;VisNo&quot;                   &quot;curage&quot;                 
##   [5] &quot;iqvocr&quot;                  &quot;iqvocs&quot;                 
##   [7] &quot;iqbdr&quot;                   &quot;iqbds&quot;                  
##   [9] &quot;iqsimr&quot;                  &quot;iqsims&quot;                 
##  [11] &quot;iqmrr&quot;                   &quot;iqmrs&quot;                  
##  [13] &quot;iq4verbr&quot;                &quot;iq4verbs&quot;               
##  [15] &quot;iq4verbp&quot;                &quot;iq4perfr&quot;               
##  [17] &quot;iq4perfs&quot;                &quot;iq4perfp&quot;               
##  [19] &quot;iq4fullr&quot;                &quot;iq4fulls&quot;               
##  [21] &quot;iq4fullp&quot;                &quot;iq2fullr&quot;               
##  [23] &quot;iq2fulls&quot;                &quot;iq2fullp&quot;               
##  [25] &quot;iqdspf&quot;                  &quot;iqdspb&quot;                 
##  [27] &quot;iqdspr&quot;                  &quot;iqdsps&quot;                 
##  [29] &quot;iqlnsr&quot;                  &quot;iqlnss&quot;                 
##  [31] &quot;readraw&quot;                 &quot;readstn&quot;                
##  [33] &quot;flucfl&quot;                  &quot;flutot&quot;                 
##  [35] &quot;bntspon&quot;                 &quot;bntstim&quot;                
##  [37] &quot;bntstimc&quot;                &quot;bntphon&quot;                
##  [39] &quot;bntphoc&quot;                 &quot;bntrecu&quot;                
##  [41] &quot;bntreco&quot;                 &quot;bnttot&quot;                 
##  [43] &quot;bnteng&quot;                  &quot;t1raw&quot;                  
##  [45] &quot;t2raw&quot;                   &quot;t3raw&quot;                  
##  [47] &quot;t4raw&quot;                   &quot;t5raw&quot;                  
##  [49] &quot;tTotal&quot;                  &quot;lbraw&quot;                  
##  [51] &quot;t6raw&quot;                   &quot;drraw&quot;                  
##  [53] &quot;recraw&quot;                  &quot;errraw&quot;                 
##  [55] &quot;avltPresRaw&quot;             &quot;avltIntrRaw&quot;            
##  [57] &quot;strooc&quot;                  &quot;strooce&quot;                
##  [59] &quot;stroocw&quot;                 &quot;stroocwe&quot;               
##  [61] &quot;jolobent&quot;                &quot;joloplus&quot;               
##  [63] &quot;trla&quot;                    &quot;trlaerr&quot;                
##  [65] &quot;trlass&quot;                  &quot;trlats&quot;                 
##  [67] &quot;trlb&quot;                    &quot;trlberr&quot;                
##  [69] &quot;trlbss&quot;                  &quot;trlbts&quot;                 
##  [71] &quot;wmfacir&quot;                 &quot;wmfacis&quot;                
##  [73] &quot;wmfacdr&quot;                 &quot;wmfacds&quot;                
##  [75] &quot;wcscat&quot;                  &quot;wcspr&quot;                  
##  [77] &quot;wcsterr&quot;                 &quot;wcsbrk&quot;                 
##  [79] &quot;cdtts&quot;                   &quot;cestyp&quot;                 
##  [81] &quot;cesdts&quot;                  &quot;bvmt1r&quot;                 
##  [83] &quot;bvmt2r&quot;                  &quot;bvmt3r&quot;                 
##  [85] &quot;bvmtTot&quot;                 &quot;bvmtDel&quot;                
##  [87] &quot;bvmtRec&quot;                 &quot;bvmtFA&quot;                 
##  [89] &quot;mmseTot&quot;                 &quot;wmsrAr&quot;                 
##  [91] &quot;wmsrBr&quot;                  &quot;wmsrTot&quot;                
##  [93] &quot;wmsrAr2&quot;                 &quot;wmsrBr2&quot;                
##  [95] &quot;wmsrTot2&quot;                &quot;waisrTot&quot;               
##  [97] &quot;waisrSca&quot;                &quot;animTotRaw&quot;             
##  [99] &quot;animTotRep&quot;              &quot;animTotErr&quot;             
## [101] &quot;cflTotRep&quot;               &quot;cflTotErr&quot;              
## [103] &quot;LangSampComp&quot;            &quot;minttot&quot;                
## [105] &quot;mintspon&quot;                &quot;mintstim&quot;               
## [107] &quot;mintstimc&quot;               &quot;mintphon&quot;               
## [109] &quot;mintphonc&quot;               &quot;interv_NeuropsychScores&quot;
## [111] &quot;NoTesting&quot;               &quot;TestNotComp&quot;            
## [113] &quot;iqarr&quot;                   &quot;iqars&quot;                  
## [115] &quot;iqwmemr&quot;                 &quot;iqwmems&quot;                
## [117] &quot;iqwmemp&quot;                 &quot;fluper&quot;                 
## [119] &quot;flueng&quot;                  &quot;lotraw&quot;                 
## [121] &quot;lass&quot;                    &quot;lotss&quot;                  
## [123] &quot;lbss&quot;                    &quot;t6ss&quot;                   
## [125] &quot;drss&quot;                    &quot;recss&quot;                  
## [127] &quot;stprraw&quot;                 &quot;ltprraw&quot;                
## [129] &quot;stprss&quot;                  &quot;ltprss&quot;                 
## [131] &quot;avlteff&quot;                 &quot;avltdr&quot;                 
## [133] &quot;avltpr&quot;                  &quot;wmfacpr&quot;                
## [135] &quot;wmfacps&quot;                 &quot;cesd_call&quot;              
## [137] &quot;bvmt1p&quot;                  &quot;bvmt2p&quot;                 
## [139] &quot;bvmt3p&quot;                  &quot;bvmtTotp&quot;               
## [141] &quot;bvmtDelp&quot;                &quot;bvmtRecp&quot;               
## [143] &quot;bvmtFAp&quot;                 &quot;wmsrp&quot;                  
## [145] &quot;wmsrp2&quot;                  &quot;animSemMean&quot;            
## [147] &quot;animSemSwi&quot;              &quot;animPhoMean&quot;            
## [149] &quot;animPhoSwi&quot;              &quot;cflTotRaw&quot;              
## [151] &quot;cflSemMean&quot;              &quot;cflSemSwi&quot;              
## [153] &quot;cflPhoMean&quot;              &quot;cflPhoSwi&quot;              
## [155] &quot;StroopC_Cor&quot;             &quot;StroopCW_Cor&quot;</code></pre>
<pre class="r"><code>print(colnames(Demographics))</code></pre>
<pre><code>##  [1] &quot;wrapnum&quot;                &quot;WRAPNo&quot;                 &quot;gender&quot;                
##  [4] &quot;race1&quot;                  &quot;race2&quot;                  &quot;race3&quot;                 
##  [7] &quot;raceoth&quot;                &quot;race_native_tribe&quot;      &quot;hispanic_or_latino&quot;    
## [10] &quot;other_ethnicity&quot;        &quot;BioKids&quot;                &quot;hand&quot;                  
## [13] &quot;frstlang&quot;               &quot;speclang&quot;               &quot;moloss&quot;                
## [16] &quot;moepisz&quot;                &quot;mostroke&quot;               &quot;moms&quot;                  
## [19] &quot;mopark&quot;                 &quot;momen&quot;                  &quot;mohdache&quot;              
## [22] &quot;moneuro&quot;                &quot;motneuro&quot;               &quot;specmoth&quot;              
## [25] &quot;faloss&quot;                 &quot;faepisz&quot;                &quot;fastroke&quot;              
## [28] &quot;fams&quot;                   &quot;fapark&quot;                 &quot;famen&quot;                 
## [31] &quot;fahdache&quot;               &quot;faneuro&quot;                &quot;foneuro&quot;               
## [34] &quot;specfath&quot;               &quot;famdown&quot;                &quot;numsis&quot;                
## [37] &quot;numbro&quot;                 &quot;ldschool&quot;               &quot;add&quot;                   
## [40] &quot;adhd&quot;                   &quot;ld&quot;                     &quot;otherld&quot;               
## [43] &quot;specld&quot;                 &quot;LearnProb&quot;              &quot;RelLearnProb&quot;          
## [46] &quot;Farm&quot;                   &quot;FarmYrs&quot;                &quot;Color&quot;                 
## [49] &quot;Grade&quot;                  &quot;EducYrs&quot;                &quot;OC_Imputed_grade&quot;      
## [52] &quot;OC_Reporting_EducYears&quot; &quot;DxLearnDis&quot;             &quot;DxADHD&quot;                
## [55] &quot;DxAttDD&quot;                &quot;DxOth&quot;                  &quot;DxOthTxt&quot;              
## [58] &quot;interv_Demographics&quot;    &quot;nochild&quot;                &quot;FH6Bro&quot;                
## [61] &quot;FH6Sis&quot;                 &quot;interest&quot;               &quot;parntres&quot;              
## [64] &quot;Lang&quot;                   &quot;LangOth&quot;</code></pre>
<p><br></p>
</div>
<div id="persfammedback-demographics" class="section level2">
<h2>PersFamMedBack | Demographics</h2>
<pre class="r"><code>intersect(names(PersFamMedBack),names(Demographics))</code></pre>
<pre><code>## [1] &quot;wrapnum&quot; &quot;WRAPNo&quot;  &quot;gender&quot;</code></pre>
<pre class="r"><code>## remove geneder column from PersFamMedBack
PersFamMedBack &lt;- subset(PersFamMedBack, select = -gender)</code></pre>
<pre class="r"><code>## print colnames to check the modification
print(colnames(PersFamMedBack))</code></pre>
<pre><code>##  [1] &quot;wrapnum&quot;                &quot;WRAPNo&quot;                 &quot;VisNo&quot;                 
##  [4] &quot;CurJob&quot;                 &quot;RetireAge&quot;              &quot;RetireAge_Over89&quot;      
##  [7] &quot;ChangeJob&quot;              &quot;Volunt&quot;                 &quot;VolHrs&quot;                
## [10] &quot;MarStat&quot;                &quot;AddSchool&quot;              &quot;Income&quot;                
## [13] &quot;FH1&quot;                    &quot;FH1DK&quot;                  &quot;FH1MD&quot;                 
## [16] &quot;FH1AD_Mo&quot;               &quot;FH2&quot;                    &quot;FH1AD_Fa&quot;              
## [19] &quot;FH1_MO_LewyBodyDem_dx&quot;  &quot;FH1_FA_LewyBodyDem_dx&quot;  &quot;FH1_MO_Parkinsons_dx&quot;  
## [22] &quot;FH1_FA_Parkinsons_dx&quot;   &quot;FH1_MO_Parkinsonism_dx&quot; &quot;FH1_FA_Parkinsonism_dx&quot;
## [25] &quot;FHMoLiving&quot;             &quot;FH3&quot;                    &quot;FHFaLiving&quot;            
## [28] &quot;FH4MP&quot;                  &quot;FH4MS&quot;                  &quot;FH4MB&quot;                 
## [31] &quot;FH5FP&quot;                  &quot;FH5FS&quot;                  &quot;FH5FB&quot;                 
## [34] &quot;HR60M&quot;                  &quot;HR60F&quot;                  &quot;FH7&quot;                   
## [37] &quot;FH8&quot;                    &quot;modement_W1&quot;            &quot;fadement_W1&quot;           
## [40] &quot;gender_other&quot;           &quot;gender_identity&quot;        &quot;gender_identity_other&quot; 
## [43] &quot;prefix&quot;                 &quot;employ&quot;                 &quot;ses&quot;                   
## [46] &quot;jobyrs1&quot;                &quot;jobyrs2&quot;                &quot;jobyrs3&quot;</code></pre>
<p><br></p>
</div>
<div id="imaging_metadata-fqrystatisticaldata" class="section level2">
<h2>Imaging_metadata | fqryStatisticalData</h2>
<pre class="r"><code>intersect(names(Imaging_metadata),names(fqryStatisticalData))</code></pre>
<pre><code>## [1] &quot;wrapnum&quot;             &quot;Days_Since_Baseline&quot; &quot;VisNo&quot;              
## [4] &quot;WRAPNo&quot;</code></pre>
<pre class="r"><code>colnames(Imaging_metadata)[colnames(Imaging_metadata) == &quot;Days_Since_Baseline&quot;] &lt;- &quot;Days_Since_Baseline_Imaging&quot;</code></pre>
<pre class="r"><code>## print colnames to check the modification
print(colnames(Imaging_metadata))</code></pre>
<pre><code>##  [1] &quot;wrapnum&quot;                     &quot;imaging_visit&quot;              
##  [3] &quot;age&quot;                         &quot;Days_Since_Baseline_Imaging&quot;
##  [5] &quot;Modality&quot;                    &quot;Manufacturer&quot;               
##  [7] &quot;ManufacturersModelName&quot;      &quot;Anaesthesia&quot;                
##  [9] &quot;BodyPart&quot;                    &quot;BodyWeight&quot;                 
## [11] &quot;BodyWeightUnits&quot;             &quot;TracerName&quot;                 
## [13] &quot;TracerIsotope&quot;               &quot;InstitutionName&quot;            
## [15] &quot;DeviceSerialNumber&quot;          &quot;InjectedRadioactivity&quot;      
## [17] &quot;InjectedRadioactivityUnits&quot;  &quot;ScanStart&quot;                  
## [19] &quot;ScanStartUnits&quot;              &quot;InjectionStart&quot;             
## [21] &quot;InjectionStartUnits&quot;         &quot;ModeOfAdministration&quot;       
## [23] &quot;FrameTimesStart&quot;             &quot;FrameTimesStartUnits&quot;       
## [25] &quot;FrameTimesEnd&quot;               &quot;FrameTimesEndUnits&quot;         
## [27] &quot;FrameDuration&quot;               &quot;FrameDurationUnits&quot;         
## [29] &quot;DecayCorrectionFactor&quot;       &quot;ImageDecayCorrected&quot;        
## [31] &quot;ImageDecayCorrectionTime&quot;    &quot;AcquisitionMode&quot;            
## [33] &quot;ReconMethodName&quot;             &quot;ReconMethodParameterLabels&quot; 
## [35] &quot;ReconMethodParameterUnits&quot;   &quot;ReconMethodParameterValues&quot; 
## [37] &quot;AttenuationCorrection&quot;       &quot;MethodImplementationVersion&quot;
## [39] &quot;ReconFilterType&quot;             &quot;ReconFilterSize&quot;            
## [41] &quot;ReconFilterSizeUnits&quot;        &quot;SoftwareVersion&quot;            
## [43] &quot;LONIImageID&quot;                 &quot;CorrectedImage&quot;             
## [45] &quot;VisNo&quot;                       &quot;WRAPNo&quot;</code></pre>
<p><br> <br></p>
</div>
</div>
<div id="merge-sub-datasets" class="section level1">
<h1>Merge Sub-Datasets</h1>
<p>Exclude the MIND_Diet_ScoreValues as this exactly the same as the
MIND_Diet data frame.</p>
<pre class="r"><code>remove &lt;- c(&quot;MIND_Diet_ScoredValues&quot;) ## this dataset is exactly same as MIND_Diet
df_names2 &lt;- setdiff(df_names,remove)</code></pre>
<p><br></p>
<div id="group-sub-datasets" class="section level2">
<h2>Group Sub-Datasets</h2>
<p>Group all sub datasets into two: one with WRAPNo+VisNo combo; one
without VisNo</p>
<pre class="r"><code>df_names2_wVisNo &lt;- c()
df_names2_woVisNo &lt;- c()

#Loop over dataframe names
for (df_name in df_names2) {
  df_obj &lt;- get(df_name)  # get the dataframe
  if (&quot;VisNo&quot; %in% colnames(df_obj)) {
    df_names2_wVisNo &lt;- c(df_names2_wVisNo, df_name)
  } else {
    df_names2_woVisNo &lt;- c(df_names2_woVisNo, df_name)
  }
}

length(df_names2_wVisNo) ## 47</code></pre>
<pre><code>## [1] 47</code></pre>
<pre class="r"><code>length(df_names2_woVisNo) ## 10</code></pre>
<pre><code>## [1] 10</code></pre>
<pre class="r"><code>print(df_names2_wVisNo)</code></pre>
<pre><code>##  [1] &quot;Abnormal_Factor_Flags_long&quot;            
##  [2] &quot;AnimalNaming_ItemLevel&quot;                
##  [3] &quot;AVLTPositionScores&quot;                    
##  [4] &quot;BehavioralObservationChecklist&quot;        
##  [5] &quot;CDR&quot;                                   
##  [6] &quot;CES_D_Scale&quot;                           
##  [7] &quot;CHAMPS&quot;                                
##  [8] &quot;CogStateData&quot;                          
##  [9] &quot;CompositeScoreCalcValues&quot;              
## [10] &quot;CompositeScores&quot;                       
## [11] &quot;ConsensusConference&quot;                   
## [12] &quot;DailyActivity&quot;                         
## [13] &quot;FactorScores&quot;                          
## [14] &quot;fqryStatisticalData&quot;                   
## [15] &quot;GenHealthDailyAct&quot;                     
## [16] &quot;HeadInjuries&quot;                          
## [17] &quot;HeadInjurySub&quot;                         
## [18] &quot;IADL&quot;                                  
## [19] &quot;Imaging_metadata&quot;                      
## [20] &quot;IQCode&quot;                                
## [21] &quot;LearningDifficulties&quot;                  
## [22] &quot;LIBRA&quot;                                 
## [23] &quot;Med_Lab&quot;                               
## [24] &quot;Med_Lab_NPDC&quot;                          
## [25] &quot;Med_Lab_PDC&quot;                           
## [26] &quot;MedHistory&quot;                            
## [27] &quot;MedHistory_Admit&quot;                      
## [28] &quot;MedHistory_SurgBF&quot;                     
## [29] &quot;MedHistory_SurgS&quot;                      
## [30] &quot;MIND_Diet&quot;                             
## [31] &quot;MOCA&quot;                                  
## [32] &quot;NeuropsychScores&quot;                      
## [33] &quot;PersFamMedBack&quot;                        
## [34] &quot;PFMB_SibWAD_or_OtherMemIssue&quot;          
## [35] &quot;QuickDementia&quot;                         
## [36] &quot;RobustNormZScores&quot;                     
## [37] &quot;Sleep_MentalAct&quot;                       
## [38] &quot;Sleep_MentalAct_OthDevice&quot;             
## [39] &quot;SubjectiveHearing&quot;                     
## [40] &quot;TICSM&quot;                                 
## [41] &quot;VisitVitals&quot;                           
## [42] &quot;WomenQuestions&quot;                        
## [43] &quot;WomenQuestions_BCP&quot;                    
## [44] &quot;WomenQuestions_Current_HormoneTherapy&quot; 
## [45] &quot;WomenQuestions_NonPresc_HormoneTherapy&quot;
## [46] &quot;WomenQuestions_Past_HormoneTherapy&quot;    
## [47] &quot;WomenQuestions_SERM&quot;</code></pre>
<pre class="r"><code>print(df_names2_woVisNo)</code></pre>
<pre><code>##  [1] &quot;Abnormal_Factor_Flags_wide&quot; &quot;APG&quot;                       
##  [3] &quot;CA_Genome&quot;                  &quot;CERAD&quot;                     
##  [5] &quot;Demographics&quot;               &quot;Gene_DataCollection&quot;       
##  [7] &quot;Morbidity_Mortality&quot;        &quot;WorkHistory&quot;               
##  [9] &quot;WorkHistory_Family&quot;         &quot;WRAP_to_DBID&quot;</code></pre>
<p><br></p>
</div>
<div id="merge-sub-datasets-without-visno" class="section level2">
<h2>Merge Sub-Datasets without VisNo</h2>
<pre class="r"><code>## Merge all cross-sectional dfs - those with WRAPNo/wrapnum only

merged_data_cross &lt;- get(df_names2_woVisNo[1])

cat(&quot;Info for&quot;, df_names2_woVisNo[1], &quot;:\n&quot;) </code></pre>
<pre><code>## Info for Abnormal_Factor_Flags_wide :</code></pre>
<pre class="r"><code>info(get(df_names2_woVisNo[1]), &quot;WRAPNo&quot;)</code></pre>
<pre><code>## #obs:1815, cols:4, inds:1815</code></pre>
<pre class="r"><code>for (i in 2:length(df_names2_woVisNo)) { 
  merged_data_cross &lt;- merge_with_info(merged_data_cross, df_names2_woVisNo[i], merge_cols = c(&quot;wrapnum&quot;, &quot;WRAPNo&quot;), info_col = &quot;WRAPNo&quot;) 
}</code></pre>
<pre><code>## Info for  APG :
## #obs:1638, cols:5, inds:1638 
## 
## Info for merged dataset:
## #obs:1815, cols:7, inds:1815 
## =======================================================================
## Info for  CA_Genome :
## #obs:1405, cols:4, inds:1405 
## 
## Info for merged dataset:
## #obs:1815, cols:9, inds:1815 
## =======================================================================
## Info for  CERAD :
## #obs:372, cols:19, inds:372 
## 
## Info for merged dataset:
## #obs:1815, cols:26, inds:1815 
## =======================================================================
## Info for  Demographics :
## #obs:1811, cols:65, inds:1811 
## 
## Info for merged dataset:
## #obs:1815, cols:89, inds:1815 
## =======================================================================
## Info for  Gene_DataCollection :
## #obs:1018, cols:14, inds:1018 
## 
## Info for merged dataset:
## #obs:1815, cols:101, inds:1815 
## =======================================================================
## Info for  Morbidity_Mortality :
## #obs:94, cols:17, inds:94 
## 
## Info for merged dataset:
## #obs:1815, cols:116, inds:1815 
## =======================================================================
## Info for  WorkHistory :
## #obs:1685, cols:44, inds:1685 
## 
## Info for merged dataset:
## #obs:1815, cols:158, inds:1815 
## =======================================================================
## Info for  WorkHistory_Family :
## #obs:1242, cols:16, inds:1242 
## 
## Info for merged dataset:
## #obs:1815, cols:172, inds:1815 
## =======================================================================
## Info for  WRAP_to_DBID :
## #obs:1717, cols:3, inds:1717 
## 
## Info for merged dataset:
## #obs:1816, cols:173, inds:1816 
## =======================================================================</code></pre>
<pre class="r"><code>info(merged_data_cross, &quot;WRAPNo&quot;) #obs:1816, cols:173, inds:1816</code></pre>
<pre><code>## #obs:1816, cols:173, inds:1816</code></pre>
<p><br></p>
</div>
<div id="merge-sub-datasets-with-visno" class="section level2">
<h2>Merge Sub-Datasets with VisNo</h2>
<pre class="r"><code>## Merge all longitudinal dfs - those with WRAPNo/wrapnum and VisNo
merged_data_long &lt;- get(df_names2_wVisNo[1]) 

cat(&quot;Info for&quot;, df_names2_wVisNo[1], &quot;:\n&quot;) </code></pre>
<pre><code>## Info for Abnormal_Factor_Flags_long :</code></pre>
<pre class="r"><code>info(get(df_names2_wVisNo[1]), &quot;WRAPNo&quot;)</code></pre>
<pre><code>## #obs:8212, cols:5, inds:1814</code></pre>
<pre class="r"><code>for (i in 2:length(df_names2_wVisNo)) {
  merged_data_long &lt;- merge_with_info(
    merged_data_long,
    df_names2_wVisNo[i],
    merge_cols = c(&quot;wrapnum&quot;, &quot;WRAPNo&quot;, &quot;VisNo&quot;),
    info_col   = &quot;WRAPNo&quot;
  )
}</code></pre>
<pre><code>## Info for  AnimalNaming_ItemLevel :
## #obs:1096, cols:53, inds:944 
## 
## Info for merged dataset:
## #obs:8212, cols:55, inds:1814 
## =======================================================================
## Info for  AVLTPositionScores :
## #obs:7946, cols:123, inds:1795 
## 
## Info for merged dataset:
## #obs:8212, cols:175, inds:1814 
## =======================================================================
## Info for  BehavioralObservationChecklist :
## #obs:2601, cols:84, inds:1322 
## 
## Info for merged dataset:
## #obs:8212, cols:256, inds:1814 
## =======================================================================
## Info for  CDR :
## #obs:1361, cols:15, inds:886 
## 
## Info for merged dataset:
## #obs:8212, cols:268, inds:1814 
## =======================================================================
## Info for  CES_D_Scale :
## #obs:6566, cols:24, inds:1590 
## 
## Info for merged dataset:
## #obs:8212, cols:289, inds:1814 
## =======================================================================
## Info for  CHAMPS :
## #obs:1888, cols:126, inds:1181 
## 
## Info for merged dataset:
## #obs:8212, cols:412, inds:1814 
## =======================================================================
## Info for  CogStateData :
## #obs:2283, cols:227, inds:1224 
## 
## Info for merged dataset:
## #obs:8212, cols:636, inds:1814 
## =======================================================================
## Info for  CompositeScoreCalcValues :
## #obs:7996, cols:16, inds:1799 
## 
## Info for merged dataset:
## #obs:8212, cols:649, inds:1814 
## =======================================================================
## Info for  CompositeScores :
## #obs:7996, cols:31, inds:1799 
## 
## Info for merged dataset:
## #obs:8212, cols:677, inds:1814 
## =======================================================================
## Info for  ConsensusConference :
## #obs:8176, cols:95, inds:1808 
## 
## Info for merged dataset:
## #obs:8212, cols:769, inds:1814 
## =======================================================================
## Info for  DailyActivity :
## #obs:1263, cols:84, inds:1263 
## 
## Info for merged dataset:
## #obs:8212, cols:850, inds:1814 
## =======================================================================
## Info for  FactorScores :
## #obs:7996, cols:13, inds:1799 
## 
## Info for merged dataset:
## #obs:8212, cols:860, inds:1814 
## =======================================================================
## Info for  fqryStatisticalData :
## #obs:8212, cols:9, inds:1814 
## 
## Info for merged dataset:
## #obs:8212, cols:866, inds:1814 
## =======================================================================
## Info for  GenHealthDailyAct :
## #obs:8168, cols:219, inds:1796 
## 
## Info for merged dataset:
## #obs:8212, cols:1082, inds:1814 
## =======================================================================
## Info for  HeadInjuries :
## #obs:551, cols:4, inds:468 
## 
## Info for merged dataset:
## #obs:8212, cols:1083, inds:1814 
## =======================================================================
## Info for  HeadInjurySub :
## #obs:578, cols:83, inds:489 
## 
## Info for merged dataset:
## #obs:8212, cols:1163, inds:1814 
## =======================================================================
## Info for  IADL :
## #obs:5913, cols:16, inds:1496 
## 
## Info for merged dataset:
## #obs:8212, cols:1176, inds:1814 
## =======================================================================
## Info for  Imaging_metadata :
## #obs:3443, cols:46, inds:818 
## 
## Info for merged dataset:
## #obs:8902, cols:1219, inds:1816 
## =======================================================================
## Info for  IQCode :
## #obs:5896, cols:24, inds:1497 
## 
## Info for merged dataset:
## #obs:8902, cols:1240, inds:1816 
## =======================================================================
## Info for  LearningDifficulties :
## #obs:213, cols:26, inds:204 
## 
## Info for merged dataset:
## #obs:8902, cols:1263, inds:1816 
## =======================================================================
## Info for  LIBRA :
## #obs:8173, cols:17, inds:1818 
## 
## Info for merged dataset:
## #obs:8981, cols:1277, inds:1836 
## =======================================================================
## Info for  Med_Lab :
## #obs:8094, cols:30, inds:1797 
## 
## Info for merged dataset:
## #obs:8981, cols:1304, inds:1836 
## =======================================================================
## Info for  Med_Lab_NPDC :
## #obs:6882, cols:167, inds:1631 
## 
## Info for merged dataset:
## #obs:8981, cols:1468, inds:1836 
## =======================================================================
## Info for  Med_Lab_PDC :
## #obs:6298, cols:163, inds:1577 
## 
## Info for merged dataset:
## #obs:8981, cols:1628, inds:1836 
## =======================================================================
## Info for  MedHistory :
## #obs:8205, cols:70, inds:1810 
## 
## Info for merged dataset:
## #obs:8981, cols:1695, inds:1836 
## =======================================================================
## Info for  MedHistory_Admit :
## #obs:782, cols:11, inds:603 
## 
## Info for merged dataset:
## #obs:8981, cols:1703, inds:1836 
## =======================================================================
## Info for  MedHistory_SurgBF :
## #obs:1115, cols:10, inds:1115 
## 
## Info for merged dataset:
## #obs:8981, cols:1710, inds:1836 
## =======================================================================
## Info for  MedHistory_SurgS :
## #obs:2503, cols:10, inds:1116 
## 
## Info for merged dataset:
## #obs:8981, cols:1717, inds:1836 
## =======================================================================
## Info for  MIND_Diet :
## #obs:3621, cols:21, inds:1416 
## 
## Info for merged dataset:
## #obs:8981, cols:1735, inds:1836 
## =======================================================================
## Info for  MOCA :
## #obs:316, cols:65, inds:309 
## 
## Info for merged dataset:
## #obs:8981, cols:1797, inds:1836 
## =======================================================================
## Info for  NeuropsychScores :
## #obs:7996, cols:156, inds:1799 
## 
## Info for merged dataset:
## #obs:8981, cols:1950, inds:1836 
## =======================================================================
## Info for  PersFamMedBack :
## #obs:8176, cols:48, inds:1802 
## 
## Info for merged dataset:
## #obs:8981, cols:1995, inds:1836 
## =======================================================================
## Info for  PFMB_SibWAD_or_OtherMemIssue :
## #obs:265, cols:6, inds:175 
## 
## Info for merged dataset:
## #obs:8981, cols:1998, inds:1836 
## =======================================================================
## Info for  QuickDementia :
## #obs:3006, cols:22, inds:1303 
## 
## Info for merged dataset:
## #obs:8981, cols:2017, inds:1836 
## =======================================================================
## Info for  RobustNormZScores :
## #obs:7996, cols:21, inds:1799 
## 
## Info for merged dataset:
## #obs:8981, cols:2035, inds:1836 
## =======================================================================
## Info for  Sleep_MentalAct :
## #obs:4208, cols:63, inds:1462 
## 
## Info for merged dataset:
## #obs:8981, cols:2095, inds:1836 
## =======================================================================
## Info for  Sleep_MentalAct_OthDevice :
## #obs:46, cols:5, inds:37 
## 
## Info for merged dataset:
## #obs:8981, cols:2097, inds:1836 
## =======================================================================
## Info for  SubjectiveHearing :
## #obs:2732, cols:8, inds:1312 
## 
## Info for merged dataset:
## #obs:8981, cols:2102, inds:1836 
## =======================================================================
## Info for  TICSM :
## #obs:65, cols:44, inds:64 
## 
## Info for merged dataset:
## #obs:8982, cols:2143, inds:1836 
## =======================================================================
## Info for  VisitVitals :
## #obs:8205, cols:33, inds:1810 
## 
## Info for merged dataset:
## #obs:8982, cols:2173, inds:1836 
## =======================================================================
## Info for  WomenQuestions :
## #obs:5544, cols:67, inds:1249 
## 
## Info for merged dataset:
## #obs:8982, cols:2237, inds:1836 
## =======================================================================
## Info for  WomenQuestions_BCP :
## #obs:885, cols:21, inds:866 
## 
## Info for merged dataset:
## #obs:8982, cols:2255, inds:1836 
## =======================================================================
## Info for  WomenQuestions_Current_HormoneTherapy :
## #obs:367, cols:21, inds:210 
## 
## Info for merged dataset:
## #obs:8982, cols:2273, inds:1836 
## =======================================================================
## Info for  WomenQuestions_NonPresc_HormoneTherapy :
## #obs:239, cols:21, inds:198 
## 
## Info for merged dataset:
## #obs:8982, cols:2291, inds:1836 
## =======================================================================
## Info for  WomenQuestions_Past_HormoneTherapy :
## #obs:380, cols:24, inds:379 
## 
## Info for merged dataset:
## #obs:8982, cols:2312, inds:1836 
## =======================================================================
## Info for  WomenQuestions_SERM :
## #obs:72, cols:19, inds:58 
## 
## Info for merged dataset:
## #obs:8982, cols:2328, inds:1836 
## =======================================================================</code></pre>
<pre class="r"><code>info(merged_data_long, &quot;WRAPNo&quot;) #obs:8982, cols:2328, inds:1836</code></pre>
<pre><code>## #obs:8982, cols:2328, inds:1836</code></pre>
<pre class="r"><code>check_dups_long &lt;- merged_data_long %&gt;% dplyr::group_by(wrapnum, WRAPNo, VisNo) %&gt;% dplyr::filter(n() &gt; 1) %&gt;% dplyr::ungroup()

nrow(check_dups_long) #obs: 0 - good</code></pre>
<pre><code>## [1] 0</code></pre>
<p><br></p>
</div>
<div id="merge-all" class="section level2">
<h2>Merge All</h2>
<pre class="r"><code>#Merge cross-sectional and longitudinal together
merged_data_all &lt;- merge(merged_data_cross, merged_data_long, by = c(&quot;wrapnum&quot;, &quot;WRAPNo&quot;), all = T)
info(merged_data_all, &quot;WRAPNo&quot;) #obs:8984, cols:2499, inds:1838 - looks like we are up 2 inds/2 obs from longitudinal and up 22 inds from cross</code></pre>
<pre><code>## #obs:8984, cols:2499, inds:1838</code></pre>
<p><br></p>
</div>
<div id="final-merge-check" class="section level2">
<h2>Final Merge Check</h2>
<pre class="r"><code>#look more into the increase to make sure it makes sense
in_cross_not_long &lt;- anti_join(merged_data_cross, merged_data_long, by = &quot;WRAPNo&quot;) #confirmed nothing like them in long
info(in_cross_not_long, &quot;WRAPNo&quot;)</code></pre>
<pre><code>## #obs:2, cols:173, inds:2</code></pre>
<pre class="r"><code>in_long_not_cross &lt;- anti_join(merged_data_long, merged_data_cross, by = &quot;WRAPNo&quot;) #confirmed nothing like them in cross
info(in_long_not_cross, &quot;WRAPNo&quot;)</code></pre>
<pre><code>## #obs:81, cols:2328, inds:22</code></pre>
<pre class="r"><code>#realistically the 2+22 will probably be dropped from analyses for not having important info (age/birth/sex/diagnosis), but ok to leave in for now</code></pre>
<p><br></p>
</div>
<div id="save-finalized-merged-dataset" class="section level2">
<h2>Save Finalized Merged Dataset</h2>
<pre class="r"><code>## save rds
saveRDS(merged_data_all,paste0(out_directory,&quot;Build_WRAP.rds&quot;))</code></pre>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
